// primeng/components/tabmenu/tabmenu.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/tabmenu/tabmenu';
import * as import2 from '@angular/common';
import * as import3 from '@angular/router';
import * as import4 from 'primeng/components/dom/domhandler';
class TabMenuModuleInjector extends import0.ɵNgModuleInjector<import1.TabMenuModule> {
  _CommonModule_0:import2.CommonModule;
  _RouterModule_1:import3.RouterModule;
  _TabMenuModule_2:import1.TabMenuModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.TabMenuModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._RouterModule_1 = new import3.RouterModule(this.parent.get(import3.ɵa,(null as any)),this.parent.get(import3.Router,(null as any)));
    this._TabMenuModule_2 = new import1.TabMenuModule();
    return this._TabMenuModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.RouterModule)) { return this._RouterModule_1; }
    if ((token === import1.TabMenuModule)) { return this._TabMenuModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const TabMenuModuleNgFactory:import0.NgModuleFactory<import1.TabMenuModule> = new import0.NgModuleFactory<any>(TabMenuModuleInjector,import1.TabMenuModule);
const styles_TabMenu:any[] = ([] as any[]);
export const RenderType_TabMenu:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_TabMenu,
  data: {}
}
);
function View_TabMenu_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'a',[[
        'class',
        'ui-menuitem-link ui-corner-all'
      ]
    ],[
      [
        8,
        'href',
        4
      ]
      ,
      [
        1,
        'target',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.itemClick($event,(<any>v.parent).context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-icon fa'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    const currVal_2:any = 'ui-menuitem-icon fa';
    const currVal_3:any = (<any>v.parent).context.$implicit.icon;
    ck(v,3,0,currVal_2,currVal_3);
  },(ck,v) => {
    const currVal_0:any = ((<any>v.parent).context.$implicit.url || '#');
    const currVal_1:any = (<any>v.parent).context.$implicit.target;
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_4:any = (<any>v.parent).context.$implicit.label;
    ck(v,6,0,currVal_4);
  });
}
function View_TabMenu_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),12,'a',[[
        'class',
        'ui-menuitem-link ui-corner-all'
      ]
    ],[
      [
        1,
        'target',
        0
      ]
      ,
      [
        1,
        'target',
        0
      ]
      ,
      [
        8,
        'href',
        4
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).onClick($event.button,$event.ctrlKey,$event.metaKey)) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>co.itemClick($event,(<any>v.parent).context.$implicit)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
      import0.ɵdid(335872,[[
        2,
        4
      ]
    ],0,import3.RouterLinkWithHref,[
      import3.Router,
      import3.ActivatedRoute,
      import2.LocationStrategy
    ]
      ,{routerLink: [
        0,
        'routerLink'
      ]
    },(null as any)),
    import0.ɵdid(860160,(null as any),2,import3.RouterLinkActive,[
      import3.Router,
      import0.ElementRef,
      import0.Renderer,
      import0.ChangeDetectorRef
    ]
    ,{
      routerLinkActiveOptions: [
        0,
        'routerLinkActiveOptions'
      ]
      ,
      routerLinkActive: [
        1,
        'routerLinkActive'
      ]

    }
    ,(null as any)),
    import0.ɵqud(301989888,1,{links: 1}),
    import0.ɵqud(301989888,2,{linksWithHrefs: 1}),
    import0.ɵpod(['exact']),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-icon fa'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    const currVal_3:any = (<any>v.parent).context.$implicit.routerLink;
    ck(v,1,0,currVal_3);
    const currVal_4:any = ((<any>v.parent).context.$implicit.routerLinkActiveOptions || ck(v,5,0,false));
    const currVal_5:any = 'ui-state-active';
    ck(v,2,0,currVal_4,currVal_5);
    const currVal_6:any = 'ui-menuitem-icon fa';
    const currVal_7:any = (<any>v.parent).context.$implicit.icon;
    ck(v,8,0,currVal_6,currVal_7);
  },(ck,v) => {
    const currVal_0:any = (<any>v.parent).context.$implicit.target;
    const currVal_1:any = import0.ɵnov(v,1).target;
    const currVal_2:any = import0.ɵnov(v,1).href;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
    const currVal_8:any = (<any>v.parent).context.$implicit.label;
    ck(v,11,0,currVal_8);
  });
}
function View_TabMenu_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),9,'li',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-tabmenuitem ui-state-default ui-corner-top',
      'ui-state-disabled',
      'ui-tabmenuitem-hasicon',
      'ui-state-active'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TabMenu_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TabMenu_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,true,v.context.$implicit.disabled,v.context.$implicit.icon,(co.activeItem == v.context.$implicit));
    ck(v,1,0,currVal_0);
    const currVal_1:boolean = !v.context.$implicit.routerLink;
    ck(v,5,0,currVal_1);
    const currVal_2:any = v.context.$implicit.routerLink;
    ck(v,8,0,currVal_2);
  },(null as any));
}
export function View_TabMenu_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),9,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ul',[
      [
        'class',
        'ui-tabmenu-nav ui-helper-reset ui-helper-clearfix ui-widget-header ui-corner-all'
      ]
      ,
      [
        'role',
        'tablist'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TabMenu_1)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.TabMenu = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-tabmenu ui-widget ui-widget-content ui-corner-all';
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_3:any = co.model;
    ck(v,8,0,currVal_3);
  },(null as any));
}
function View_TabMenu_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-tabMenu',([] as any[]),(null as any),(null as any),(null as any),View_TabMenu_0,RenderType_TabMenu)),
    import0.ɵdid(57344,(null as any),0,import1.TabMenu,([] as any[]),(null as any),(null as any)),
    import0.ɵprd(2304,(null as any),import4.DomHandler,import4.DomHandler,([] as any[]))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const TabMenuNgFactory:import0.ComponentFactory<import1.TabMenu> = import0.ɵccf('p-tabMenu',import1.TabMenu,View_TabMenu_Host_0,{
  model: 'model',
  activeItem: 'activeItem',
  popup: 'popup',
  style: 'style',
  styleClass: 'styleClass'
}
,{},([] as any[]));
