// primeng/components/common/shared.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/common/shared';
import * as import2 from '@angular/common';
class SharedModuleInjector extends import0.ɵNgModuleInjector<import1.SharedModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import1.SharedModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.SharedModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import1.SharedModule();
    return this._SharedModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.SharedModule)) { return this._SharedModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const SharedModuleNgFactory:import0.NgModuleFactory<import1.SharedModule> = new import0.NgModuleFactory<any>(SharedModuleInjector,import1.SharedModule);
const styles_Header:any[] = ([] as any[]);
export const RenderType_Header:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Header,
  data: {}
}
);
export function View_Header_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[import0.ɵncd((null as any),0)],(null as any),(null as any));
}
function View_Header_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-header',([] as any[]),(null as any),(null as any),(null as any),View_Header_0,RenderType_Header)),
    import0.ɵdid(24576,(null as any),0,import1.Header,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const HeaderNgFactory:import0.ComponentFactory<import1.Header> = import0.ɵccf('p-header',import1.Header,View_Header_Host_0,{},{},['*']);
const styles_Footer:any[] = ([] as any[]);
export const RenderType_Footer:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Footer,
  data: {}
}
);
export function View_Footer_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[import0.ɵncd((null as any),0)],(null as any),(null as any));
}
function View_Footer_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-footer',([] as any[]),(null as any),(null as any),(null as any),View_Footer_0,RenderType_Footer)),
    import0.ɵdid(24576,(null as any),0,import1.Footer,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const FooterNgFactory:import0.ComponentFactory<import1.Footer> = import0.ɵccf('p-footer',import1.Footer,View_Footer_Host_0,{},{},['*']);
const styles_Column:any[] = ([] as any[]);
export const RenderType_Column:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Column,
  data: {}
}
);
export function View_Column_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_Column_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'p-column',([] as any[]),(null as any),(null as any),(null as any),View_Column_0,RenderType_Column)),
    import0.ɵdid(548864,(null as any),2,import1.Column,([] as any[]),(null as any),(null as any)),
    import0.ɵqud(301989888,1,{templates: 1}),
    import0.ɵqud(167772160,2,{template: 0})
  ]
  ,(null as any),(null as any));
}
export const ColumnNgFactory:import0.ComponentFactory<import1.Column> = import0.ɵccf('p-column',import1.Column,View_Column_Host_0,{
  field: 'field',
  colId: 'colId',
  sortField: 'sortField',
  header: 'header',
  footer: 'footer',
  sortable: 'sortable',
  editable: 'editable',
  filter: 'filter',
  filterMatchMode: 'filterMatchMode',
  filterType: 'filterType',
  rowspan: 'rowspan',
  colspan: 'colspan',
  style: 'style',
  styleClass: 'styleClass',
  hidden: 'hidden',
  expander: 'expander',
  selectionMode: 'selectionMode',
  filterPlaceholder: 'filterPlaceholder',
  filterMaxlength: 'filterMaxlength',
  frozen: 'frozen'
}
,{sortFunction: 'sortFunction'},([] as any[]));
const styles_ColumnHeaderTemplateLoader:any[] = ([] as any[]);
export const RenderType_ColumnHeaderTemplateLoader:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ColumnHeaderTemplateLoader,
  data: {}
}
);
export function View_ColumnHeaderTemplateLoader_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_ColumnHeaderTemplateLoader_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-columnHeaderTemplateLoader',([] as any[]),(null as any),(null as any),(null as any),View_ColumnHeaderTemplateLoader_0,RenderType_ColumnHeaderTemplateLoader)),
    import0.ɵdid(122880,(null as any),0,import1.ColumnHeaderTemplateLoader,[import0.ViewContainerRef],(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const ColumnHeaderTemplateLoaderNgFactory:import0.ComponentFactory<import1.ColumnHeaderTemplateLoader> = import0.ɵccf('p-columnHeaderTemplateLoader',import1.ColumnHeaderTemplateLoader,View_ColumnHeaderTemplateLoader_Host_0,{column: 'column'},{},([] as any[]));
const styles_ColumnBodyTemplateLoader:any[] = ([] as any[]);
export const RenderType_ColumnBodyTemplateLoader:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ColumnBodyTemplateLoader,
  data: {}
}
);
export function View_ColumnBodyTemplateLoader_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_ColumnBodyTemplateLoader_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-columnBodyTemplateLoader',([] as any[]),(null as any),(null as any),(null as any),View_ColumnBodyTemplateLoader_0,RenderType_ColumnBodyTemplateLoader)),
    import0.ɵdid(385024,(null as any),0,import1.ColumnBodyTemplateLoader,[import0.ViewContainerRef],(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const ColumnBodyTemplateLoaderNgFactory:import0.ComponentFactory<import1.ColumnBodyTemplateLoader> = import0.ɵccf('p-columnBodyTemplateLoader',import1.ColumnBodyTemplateLoader,View_ColumnBodyTemplateLoader_Host_0,{
  column: 'column',
  rowData: 'rowData',
  rowIndex: 'rowIndex'
}
,{},([] as any[]));
const styles_ColumnFooterTemplateLoader:any[] = ([] as any[]);
export const RenderType_ColumnFooterTemplateLoader:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ColumnFooterTemplateLoader,
  data: {}
}
);
export function View_ColumnFooterTemplateLoader_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_ColumnFooterTemplateLoader_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-columnFooterTemplateLoader',([] as any[]),(null as any),(null as any),(null as any),View_ColumnFooterTemplateLoader_0,RenderType_ColumnFooterTemplateLoader)),
    import0.ɵdid(122880,(null as any),0,import1.ColumnFooterTemplateLoader,[import0.ViewContainerRef],(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const ColumnFooterTemplateLoaderNgFactory:import0.ComponentFactory<import1.ColumnFooterTemplateLoader> = import0.ɵccf('p-columnFooterTemplateLoader',import1.ColumnFooterTemplateLoader,View_ColumnFooterTemplateLoader_Host_0,{column: 'column'},{},([] as any[]));
const styles_ColumnFilterTemplateLoader:any[] = ([] as any[]);
export const RenderType_ColumnFilterTemplateLoader:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ColumnFilterTemplateLoader,
  data: {}
}
);
export function View_ColumnFilterTemplateLoader_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_ColumnFilterTemplateLoader_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-columnFilterTemplateLoader',([] as any[]),(null as any),(null as any),(null as any),View_ColumnFilterTemplateLoader_0,RenderType_ColumnFilterTemplateLoader)),
    import0.ɵdid(122880,(null as any),0,import1.ColumnFilterTemplateLoader,[import0.ViewContainerRef],(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const ColumnFilterTemplateLoaderNgFactory:import0.ComponentFactory<import1.ColumnFilterTemplateLoader> = import0.ɵccf('p-columnFilterTemplateLoader',import1.ColumnFilterTemplateLoader,View_ColumnFilterTemplateLoader_Host_0,{column: 'column'},{},([] as any[]));
const styles_TemplateLoader:any[] = ([] as any[]);
export const RenderType_TemplateLoader:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_TemplateLoader,
  data: {}
}
);
export function View_TemplateLoader_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_TemplateLoader_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-templateLoader',([] as any[]),(null as any),(null as any),(null as any),View_TemplateLoader_0,RenderType_TemplateLoader)),
    import0.ɵdid(122880,(null as any),0,import1.TemplateLoader,[import0.ViewContainerRef],(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const TemplateLoaderNgFactory:import0.ComponentFactory<import1.TemplateLoader> = import0.ɵccf('p-templateLoader',import1.TemplateLoader,View_TemplateLoader_Host_0,{
  template: 'template',
  data: 'data'
}
,{},([] as any[]));
const styles_Row:any[] = ([] as any[]);
export const RenderType_Row:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Row,
  data: {}
}
);
export function View_Row_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_Row_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-row',([] as any[]),(null as any),(null as any),(null as any),View_Row_0,RenderType_Row)),
    import0.ɵdid(24576,(null as any),1,import1.Row,([] as any[]),(null as any),(null as any)),
    import0.ɵqud(301989888,1,{columns: 1})
  ]
  ,(null as any),(null as any));
}
export const RowNgFactory:import0.ComponentFactory<import1.Row> = import0.ɵccf('p-row',import1.Row,View_Row_Host_0,{},{},([] as any[]));
const styles_HeaderColumnGroup:any[] = ([] as any[]);
export const RenderType_HeaderColumnGroup:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_HeaderColumnGroup,
  data: {}
}
);
export function View_HeaderColumnGroup_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_HeaderColumnGroup_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-headerColumnGroup',([] as any[]),(null as any),(null as any),(null as any),View_HeaderColumnGroup_0,RenderType_HeaderColumnGroup)),
    import0.ɵdid(24576,(null as any),1,import1.HeaderColumnGroup,([] as any[]),(null as any),(null as any)),
    import0.ɵqud(301989888,1,{rows: 1})
  ]
  ,(null as any),(null as any));
}
export const HeaderColumnGroupNgFactory:import0.ComponentFactory<import1.HeaderColumnGroup> = import0.ɵccf('p-headerColumnGroup',import1.HeaderColumnGroup,View_HeaderColumnGroup_Host_0,{},{},([] as any[]));
const styles_FooterColumnGroup:any[] = ([] as any[]);
export const RenderType_FooterColumnGroup:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_FooterColumnGroup,
  data: {}
}
);
export function View_FooterColumnGroup_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_FooterColumnGroup_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-footerColumnGroup',([] as any[]),(null as any),(null as any),(null as any),View_FooterColumnGroup_0,RenderType_FooterColumnGroup)),
    import0.ɵdid(24576,(null as any),1,import1.FooterColumnGroup,([] as any[]),(null as any),(null as any)),
    import0.ɵqud(301989888,1,{rows: 1})
  ]
  ,(null as any),(null as any));
}
export const FooterColumnGroupNgFactory:import0.ComponentFactory<import1.FooterColumnGroup> = import0.ɵccf('p-footerColumnGroup',import1.FooterColumnGroup,View_FooterColumnGroup_Host_0,{},{},([] as any[]));
const styles_ColumnEditorTemplateLoader:any[] = ([] as any[]);
export const RenderType_ColumnEditorTemplateLoader:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ColumnEditorTemplateLoader,
  data: {}
}
);
export function View_ColumnEditorTemplateLoader_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_ColumnEditorTemplateLoader_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-columnEditorTemplateLoader',([] as any[]),(null as any),(null as any),(null as any),View_ColumnEditorTemplateLoader_0,RenderType_ColumnEditorTemplateLoader)),
    import0.ɵdid(122880,(null as any),0,import1.ColumnEditorTemplateLoader,[import0.ViewContainerRef],(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const ColumnEditorTemplateLoaderNgFactory:import0.ComponentFactory<import1.ColumnEditorTemplateLoader> = import0.ɵccf('p-columnEditorTemplateLoader',import1.ColumnEditorTemplateLoader,View_ColumnEditorTemplateLoader_Host_0,{
  column: 'column',
  rowData: 'rowData',
  rowIndex: 'rowIndex'
}
,{},([] as any[]));
