// primeng/components/spinner/spinner.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/spinner/spinner';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/inputtext/inputtext';
import * as import4 from 'primeng/components/dom/domhandler';
import * as import5 from '@angular/forms';
class SpinnerModuleInjector extends import0.ɵNgModuleInjector<import1.SpinnerModule> {
  _CommonModule_0:import2.CommonModule;
  _InputTextModule_1:import3.InputTextModule;
  _SpinnerModule_2:import1.SpinnerModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.SpinnerModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._InputTextModule_1 = new import3.InputTextModule();
    this._SpinnerModule_2 = new import1.SpinnerModule();
    return this._SpinnerModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.InputTextModule)) { return this._InputTextModule_1; }
    if ((token === import1.SpinnerModule)) { return this._SpinnerModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const SpinnerModuleNgFactory:import0.NgModuleFactory<import1.SpinnerModule> = new import0.NgModuleFactory<any>(SpinnerModuleInjector,import1.SpinnerModule);
const styles_Spinner:any[] = ([] as any[]);
export const RenderType_Spinner:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Spinner,
  data: {}
}
);
export function View_Spinner_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),17,'span',[[
        'class',
        'ui-spinner ui-widget ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,[[
        'in',
        1
      ]
      ],(null as any),0,'input',[[
        'class',
        'ui-inputtext ui-widget ui-state-default ui-corner-all'
      ]
    ],[
      [
        1,
        'type',
        0
      ]
      ,
      [
        1,
        'id',
        0
      ]
      ,
      [
        8,
        'value',
        0
      ]
      ,
      [
        1,
        'size',
        0
      ]
      ,
      [
        1,
        'maxlength',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        1,
        'placeholder',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        8,
        'readOnly',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'keydown'
      ]
      ,
      [
        (null as any),
        'keyup'
      ]
      ,
      [
        (null as any),
        'keypress'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'change'
      ]
      ,
      [
        (null as any),
        'focus'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Spinner = v.component;
      if (('keydown' === en)) {
        const pd_0:any = ((<any>co.onInputKeydown($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('keyup' === en)) {
        const pd_1:any = ((<any>co.onInput($event,import0.ɵnov(v,3).value)) !== false);
        ad = (pd_1 && ad);
      }
      if (('keypress' === en)) {
        const pd_2:any = ((<any>co.onInputKeyPress($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('blur' === en)) {
        const pd_3:any = ((<any>co.onInputBlur($event)) !== false);
        ad = (pd_3 && ad);
      }
      if (('change' === en)) {
        const pd_4:any = ((<any>co.handleChange($event)) !== false);
        ad = (pd_4 && ad);
      }
      if (('focus' === en)) {
        const pd_5:any = ((<any>co.onFocus()) !== false);
        ad = (pd_5 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'button',[[
        'type',
        'button'
      ]
      ],[[
        8,
        'disabled',
        0
      ]
    ],[
      [
        (null as any),
        'mouseleave'
      ]
      ,
      [
        (null as any),
        'mousedown'
      ]
      ,
      [
        (null as any),
        'mouseup'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Spinner = v.component;
      if (('mouseleave' === en)) {
        const pd_0:any = ((<any>co.onUpButtonMouseleave($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mousedown' === en)) {
        const pd_1:any = ((<any>co.onUpButtonMousedown($event,import0.ɵnov(v,3))) !== false);
        ad = (pd_1 && ad);
      }
      if (('mouseup' === en)) {
        const pd_2:any = ((<any>co.onUpButtonMouseup($event)) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-spinner-button ui-spinner-up ui-corner-tr ui-button ui-widget ui-state-default',
      'ui-state-disabled'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-caret-up ui-clickable'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'button',[[
        'type',
        'button'
      ]
      ],[[
        8,
        'disabled',
        0
      ]
    ],[
      [
        (null as any),
        'mouseleave'
      ]
      ,
      [
        (null as any),
        'mousedown'
      ]
      ,
      [
        (null as any),
        'mouseup'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Spinner = v.component;
      if (('mouseleave' === en)) {
        const pd_0:any = ((<any>co.onDownButtonMouseleave($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mousedown' === en)) {
        const pd_1:any = ((<any>co.onDownButtonMousedown($event,import0.ɵnov(v,3))) !== false);
        ad = (pd_1 && ad);
      }
      if (('mouseup' === en)) {
        const pd_2:any = ((<any>co.onDownButtonMouseup($event)) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-spinner-button ui-spinner-down ui-corner-br ui-button ui-widget ui-state-default',
      'ui-state-disabled'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-caret-down ui-clickable'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Spinner = v.component;
    const currVal_10:any = ck(v,7,0,true,co.disabled);
    ck(v,6,0,currVal_10);
    const currVal_12:any = ck(v,14,0,true,co.disabled);
    ck(v,13,0,currVal_12);
  },(ck,v) => {
    var co:import1.Spinner = v.component;
    const currVal_0:any = co.type;
    const currVal_1:any = co.inputId;
    const currVal_2:any = co.valueAsString;
    const currVal_3:any = co.size;
    const currVal_4:any = co.maxlength;
    const currVal_5:any = co.tabindex;
    const currVal_6:any = co.placeholder;
    const currVal_7:any = co.disabled;
    const currVal_8:any = co.readonly;
    ck(v,3,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8);
    const currVal_9:any = co.disabled;
    ck(v,5,0,currVal_9);
    const currVal_11:any = co.disabled;
    ck(v,12,0,currVal_11);
  });
}
function View_Spinner_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'p-spinner',([] as any[]),[
      [
        2,
        'ui-inputwrapper-filled',
        (null as any)
      ]
      ,
      [
        2,
        'ui-inputwrapper-focus',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_Spinner_0,RenderType_Spinner)),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵdid(57344,(null as any),0,import1.Spinner,[
      import0.ElementRef,
      import4.DomHandler
    ]
    ,(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import5.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.Spinner])
  ]
  ,(ck,v) => {
    ck(v,2,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2).filled;
    const currVal_1:any = import0.ɵnov(v,2).focus;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const SpinnerNgFactory:import0.ComponentFactory<import1.Spinner> = import0.ɵccf('p-spinner',import1.Spinner,View_Spinner_Host_0,{
  step: 'step',
  min: 'min',
  max: 'max',
  maxlength: 'maxlength',
  size: 'size',
  placeholder: 'placeholder',
  inputId: 'inputId',
  disabled: 'disabled',
  readonly: 'readonly',
  decimalSeparator: 'decimalSeparator',
  thousandSeparator: 'thousandSeparator',
  tabindex: 'tabindex',
  formatInput: 'formatInput',
  type: 'type'
}
,{
  onChange: 'onChange',
  onBlur: 'onBlur'
}
,([] as any[]));
