// primeng/components/tree/tree.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/tree/tree';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from 'primeng/components/common/treedragdropservice';
class TreeModuleInjector extends import0.ɵNgModuleInjector<import1.TreeModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _TreeModule_2:import1.TreeModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.TreeModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._TreeModule_2 = new import1.TreeModule();
    return this._TreeModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import1.TreeModule)) { return this._TreeModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const TreeModuleNgFactory:import0.NgModuleFactory<import1.TreeModule> = new import0.NgModuleFactory<any>(TreeModuleInjector,import1.TreeModule);
const styles_Tree:any[] = ([] as any[]);
export const RenderType_Tree:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Tree,
  data: {}
}
);
function View_Tree_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-treeNode',([] as any[]),(null as any),(null as any),(null as any),View_UITreeNode_0,RenderType_UITreeNode)),
    import0.ɵdid(57344,(null as any),0,import1.UITreeNode,[import1.Tree],{
      node: [
        0,
        'node'
      ]
      ,
      index: [
        1,
        'index'
      ]
      ,
      firstChild: [
        2,
        'firstChild'
      ]
      ,
      lastChild: [
        3,
        'lastChild'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit;
    const currVal_1:any = v.context.index;
    const currVal_2:any = v.context.first;
    const currVal_3:any = v.context.last;
    ck(v,1,0,currVal_0,currVal_1,currVal_2,currVal_3);
  },(null as any));
}
function View_Tree_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'div',([] as any[]),(null as any),[
      [
        (null as any),
        'drop'
      ]
      ,
      [
        (null as any),
        'dragover'
      ]
      ,
      [
        (null as any),
        'dragenter'
      ]
      ,
      [
        (null as any),
        'dragleave'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('drop' === en)) {
        const pd_0:any = ((<any>co.onDrop($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('dragover' === en)) {
        const pd_1:any = ((<any>co.onDragOver($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('dragenter' === en)) {
        const pd_2:any = ((<any>co.onDragEnter($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('dragleave' === en)) {
        const pd_3:any = ((<any>co.onDragLeave($event)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-tree ui-widget ui-widget-content ui-corner-all',
      'ui-tree-selectable',
      'ui-treenode-dragover'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ul',[[
        'class',
        'ui-tree-container'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Tree_2)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,2,0,true,co.selectionMode,co.dragHover);
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_3:any = co.value;
    ck(v,8,0,currVal_3);
  },(null as any));
}
function View_Tree_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'table',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-treeNode',([] as any[]),(null as any),(null as any),(null as any),View_UITreeNode_0,RenderType_UITreeNode)),
    import0.ɵdid(57344,(null as any),0,import1.UITreeNode,[import1.Tree],{
      node: [
        0,
        'node'
      ]
      ,
      root: [
        1,
        'root'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.value[0];
    const currVal_1:any = true;
    ck(v,3,0,currVal_0,currVal_1);
  },(null as any));
}
function View_Tree_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-tree ui-tree-horizontal ui-widget ui-widget-content ui-corner-all',
      'ui-tree-selectable'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Tree_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,2,0,true,co.selectionMode);
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_3:any = (co.value && co.value[0]);
    ck(v,6,0,currVal_3);
  },(null as any));
}
export function View_Tree_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Tree_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Tree_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Tree = v.component;
    const currVal_0:boolean = !co.horizontal;
    ck(v,2,0,currVal_0);
    const currVal_1:any = co.horizontal;
    ck(v,5,0,currVal_1);
  },(null as any));
}
function View_Tree_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-tree',([] as any[]),(null as any),(null as any),(null as any),View_Tree_0,RenderType_Tree)),
      import0.ɵdid(647168,(null as any),1,import1.Tree,[[
        2,
        import4.TreeDragDropService
      ]
    ],(null as any),(null as any)),
    import0.ɵqud(301989888,1,{templates: 1})
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const TreeNgFactory:import0.ComponentFactory<import1.Tree> = import0.ɵccf('p-tree',import1.Tree,View_Tree_Host_0,{
  value: 'value',
  selectionMode: 'selectionMode',
  selection: 'selection',
  style: 'style',
  styleClass: 'styleClass',
  contextMenu: 'contextMenu',
  layout: 'layout',
  draggableScope: 'draggableScope',
  droppableScope: 'droppableScope',
  draggableNodes: 'draggableNodes',
  droppableNodes: 'droppableNodes',
  metaKeySelection: 'metaKeySelection',
  propagateSelectionUp: 'propagateSelectionUp',
  propagateSelectionDown: 'propagateSelectionDown'
}
,{
  selectionChange: 'selectionChange',
  onNodeSelect: 'onNodeSelect',
  onNodeUnselect: 'onNodeUnselect',
  onNodeExpand: 'onNodeExpand',
  onNodeCollapse: 'onNodeCollapse',
  onNodeContextMenuSelect: 'onNodeContextMenuSelect',
  onNodeDrop: 'onNodeDrop'
}
,([] as any[]));
const styles_UITreeNode:any[] = ([] as any[]);
export const RenderType_UITreeNode:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_UITreeNode,
  data: {}
}
);
function View_UITreeNode_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'li',[[
        'class',
        'ui-treenode-droppoint'
      ]
    ],(null as any),[
      [
        (null as any),
        'drop'
      ]
      ,
      [
        (null as any),
        'dragover'
      ]
      ,
      [
        (null as any),
        'dragenter'
      ]
      ,
      [
        (null as any),
        'dragleave'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('drop' === en)) {
        const pd_0:any = ((<any>co.onDropPoint($event,(0 - 1))) !== false);
        ad = (pd_0 && ad);
      }
      if (('dragover' === en)) {
        const pd_1:any = ((<any>co.onDropPointDragOver($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('dragenter' === en)) {
        const pd_2:any = ((<any>co.onDropPointDragEnter($event,(0 - 1))) !== false);
        ad = (pd_2 && ad);
      }
      if (('dragleave' === en)) {
        const pd_3:any = ((<any>co.onDropPointDragLeave($event)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-treenode-droppoint-active ui-state-highlight'])
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-treenode-droppoint';
    const currVal_1:any = ck(v,2,0,co.draghoverPrev);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_UITreeNode_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',[[
        'class',
        'ui-chkbox'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'ui-chkbox-box ui-widget ui-corner-all ui-state-default'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-chkbox-icon ui-clickable fa'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'fa-check',
      'fa-minus'
    ]
    )
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-chkbox-icon ui-clickable fa';
    const currVal_1:any = ck(v,5,0,co.isSelected(),co.node.partialSelected);
    ck(v,4,0,currVal_0,currVal_1);
  },(null as any));
}
function View_UITreeNode_5(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',([] as any[]),[[
        8,
        'className',
        0
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.getIcon();
    ck(v,0,0,currVal_0);
  });
}
function View_UITreeNode_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.node.label;
    ck(v,1,0,currVal_0);
  });
}
function View_UITreeNode_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-treeNodeTemplateLoader',([] as any[]),(null as any),(null as any),(null as any),View_TreeNodeTemplateLoader_0,RenderType_TreeNodeTemplateLoader)),
    import0.ɵdid(122880,(null as any),0,import1.TreeNodeTemplateLoader,[import0.ViewContainerRef],{
      node: [
        0,
        'node'
      ]
      ,
      template: [
        1,
        'template'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.node;
    const currVal_1:any = co.tree.getTemplateForNode(co.node);
    ck(v,3,0,currVal_0,currVal_1);
  },(null as any));
}
function View_UITreeNode_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-treeNode',([] as any[]),(null as any),(null as any),(null as any),View_UITreeNode_0,RenderType_UITreeNode)),
    import0.ɵdid(57344,(null as any),0,import1.UITreeNode,[import1.Tree],{
      node: [
        0,
        'node'
      ]
      ,
      parentNode: [
        1,
        'parentNode'
      ]
      ,
      index: [
        2,
        'index'
      ]
      ,
      firstChild: [
        3,
        'firstChild'
      ]
      ,
      lastChild: [
        4,
        'lastChild'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = v.context.$implicit;
    const currVal_1:any = co.node;
    const currVal_2:any = v.context.index;
    const currVal_3:any = v.context.first;
    const currVal_4:any = v.context.last;
    ck(v,1,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  },(null as any));
}
function View_UITreeNode_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ul',[
      [
        'class',
        'ui-treenode-children'
      ]
      ,
      [
        'style',
        'display: none;'
      ]

    ]
      ,[[
        4,
        'display',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_9)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.node.children;
    ck(v,3,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.node.expanded? 'block': 'none');
    ck(v,0,0,currVal_0);
  });
}
function View_UITreeNode_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),29,'li',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-treenode-leaf']),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),21,'div',[[
        'class',
        'ui-treenode-content'
      ]
      ],[[
        8,
        'draggable',
        0
      ]
    ],[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'contextmenu'
      ]
      ,
      [
        (null as any),
        'touchend'
      ]
      ,
      [
        (null as any),
        'drop'
      ]
      ,
      [
        (null as any),
        'dragover'
      ]
      ,
      [
        (null as any),
        'dragenter'
      ]
      ,
      [
        (null as any),
        'dragleave'
      ]
      ,
      [
        (null as any),
        'dragstart'
      ]
      ,
      [
        (null as any),
        'dragend'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onNodeClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('contextmenu' === en)) {
        const pd_1:any = ((<any>co.onNodeRightClick($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('touchend' === en)) {
        const pd_2:any = ((<any>co.onNodeTouchEnd()) !== false);
        ad = (pd_2 && ad);
      }
      if (('drop' === en)) {
        const pd_3:any = ((<any>co.onDropNode($event)) !== false);
        ad = (pd_3 && ad);
      }
      if (('dragover' === en)) {
        const pd_4:any = ((<any>co.onDropNodeDragOver($event)) !== false);
        ad = (pd_4 && ad);
      }
      if (('dragenter' === en)) {
        const pd_5:any = ((<any>co.onDropNodeDragEnter($event)) !== false);
        ad = (pd_5 && ad);
      }
      if (('dragleave' === en)) {
        const pd_6:any = ((<any>co.onDropNodeDragLeave($event)) !== false);
        ad = (pd_6 && ad);
      }
      if (('dragstart' === en)) {
        const pd_7:any = ((<any>co.onDragStart($event)) !== false);
        ad = (pd_7 && ad);
      }
      if (('dragend' === en)) {
        const pd_8:any = ((<any>co.onDragStop($event)) !== false);
        ad = (pd_8 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-treenode-selectable',
      'ui-treenode-dragover',
      'ui-treenode-content-selected'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-tree-toggler  fa fa-fw'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggle($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'fa-caret-right',
      'fa-caret-down'
    ]
    ),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),9,'span',[[
        'class',
        'ui-treenode-label ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-highlight']),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_7)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_8)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = import0.ɵinlineInterpolate(1,'ui-treenode ',co.node.styleClass,'');
    const currVal_1:any = ck(v,2,0,co.isLeaf());
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_3:any = 'ui-treenode-content';
    const currVal_4:any = ck(v,6,0,(co.tree.selectionMode && (co.node.selectable !== false)),co.draghoverNode,co.isSelected());
    ck(v,5,0,currVal_3,currVal_4);
    const currVal_5:any = 'ui-tree-toggler  fa fa-fw';
    const currVal_6:any = ck(v,10,0,!co.node.expanded,co.node.expanded);
    ck(v,9,0,currVal_5,currVal_6);
    const currVal_7:any = (co.tree.selectionMode == 'checkbox');
    ck(v,12,0,currVal_7);
    const currVal_8:any = ((co.node.icon || co.node.expandedIcon) || co.node.collapsedIcon);
    ck(v,14,0,currVal_8);
    const currVal_9:any = 'ui-treenode-label ui-corner-all';
    const currVal_10:any = ck(v,17,0,co.isSelected());
    ck(v,16,0,currVal_9,currVal_10);
    const currVal_11:boolean = !co.tree.getTemplateForNode(co.node);
    ck(v,20,0,currVal_11);
    const currVal_12:any = co.tree.getTemplateForNode(co.node);
    ck(v,23,0,currVal_12);
    const currVal_13:any = (co.node.children && co.node.expanded);
    ck(v,28,0,currVal_13);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co.tree.draggableNodes;
    ck(v,4,0,currVal_2);
  });
}
function View_UITreeNode_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'li',[[
        'class',
        'ui-treenode-droppoint'
      ]
    ],(null as any),[
      [
        (null as any),
        'drop'
      ]
      ,
      [
        (null as any),
        'dragover'
      ]
      ,
      [
        (null as any),
        'dragenter'
      ]
      ,
      [
        (null as any),
        'dragleave'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('drop' === en)) {
        const pd_0:any = ((<any>co.onDropPoint($event,1)) !== false);
        ad = (pd_0 && ad);
      }
      if (('dragover' === en)) {
        const pd_1:any = ((<any>co.onDropPointDragOver($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('dragenter' === en)) {
        const pd_2:any = ((<any>co.onDropPointDragEnter($event,1)) !== false);
        ad = (pd_2 && ad);
      }
      if (('dragleave' === en)) {
        const pd_3:any = ((<any>co.onDropPointDragLeave($event)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-treenode-droppoint-active ui-state-highlight'])
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-treenode-droppoint';
    const currVal_1:any = ck(v,2,0,co.draghoverNext);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_UITreeNode_12(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),21,'td',[[
        'class',
        'ui-treenode-connector'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),18,'table',[[
        'class',
        'ui-treenode-connector-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),15,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'td',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['ui-treenode-connector-line']),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'td',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['ui-treenode-connector-line']),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,10,0,!co.firstChild);
    ck(v,9,0,currVal_0);
    const currVal_1:any = ck(v,17,0,!co.lastChild);
    ck(v,16,0,currVal_1);
  },(null as any));
}
function View_UITreeNode_13(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-tree-toggler fa fa-fw'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggle($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'fa-plus',
      'fa-minus'
    ]
    )
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-tree-toggler fa fa-fw';
    const currVal_1:any = ck(v,2,0,!co.node.expanded,co.node.expanded);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_UITreeNode_14(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',([] as any[]),[[
        8,
        'className',
        0
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.getIcon();
    ck(v,0,0,currVal_0);
  });
}
function View_UITreeNode_15(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.node.label;
    ck(v,1,0,currVal_0);
  });
}
function View_UITreeNode_16(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                            '])),
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-treeNodeTemplateLoader',([] as any[]),(null as any),(null as any),(null as any),View_TreeNodeTemplateLoader_0,RenderType_TreeNodeTemplateLoader)),
    import0.ɵdid(122880,(null as any),0,import1.TreeNodeTemplateLoader,[import0.ViewContainerRef],{
      node: [
        0,
        'node'
      ]
      ,
      template: [
        1,
        'template'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.node;
    const currVal_1:any = co.tree.getTemplateForNode(co.node);
    ck(v,3,0,currVal_0,currVal_1);
  },(null as any));
}
function View_UITreeNode_18(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-treeNode',([] as any[]),(null as any),(null as any),(null as any),View_UITreeNode_0,RenderType_UITreeNode)),
    import0.ɵdid(57344,(null as any),0,import1.UITreeNode,[import1.Tree],{
      node: [
        0,
        'node'
      ]
      ,
      firstChild: [
        1,
        'firstChild'
      ]
      ,
      lastChild: [
        2,
        'lastChild'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit;
    const currVal_1:any = v.context.first;
    const currVal_2:any = v.context.last;
    ck(v,1,0,currVal_0,currVal_1,currVal_2);
  },(null as any));
}
function View_UITreeNode_17(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'td',[[
        'class',
        'ui-treenode-children-container'
      ]
      ],[[
        4,
        'display',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'ui-treenode-children'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_18)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.node.children;
    ck(v,5,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.node.expanded? 'table-cell': 'none');
    ck(v,0,0,currVal_0);
  });
}
function View_UITreeNode_11(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),36,'table',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),33,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),30,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_12)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),21,'td',[[
        'class',
        'ui-treenode'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-treenode-collapsed']),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),16,'div',[[
        'class',
        'ui-treenode-content ui-state-default ui-corner-all'
      ]
    ],(null as any),[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'contextmenu'
      ]
      ,
      [
        (null as any),
        'touchend'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onNodeClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('contextmenu' === en)) {
        const pd_1:any = ((<any>co.onNodeRightClick($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('touchend' === en)) {
        const pd_2:any = ((<any>co.onNodeTouchEnd()) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-treenode-selectable',
      'ui-state-highlight'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_13)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_14)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'span',[[
        'class',
        'ui-treenode-label ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_15)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_16)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_17)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:boolean = !co.root;
    ck(v,7,0,currVal_1);
    const currVal_2:any = 'ui-treenode';
    const currVal_3:any = ck(v,11,0,!co.node.expanded);
    ck(v,10,0,currVal_2,currVal_3);
    const currVal_4:any = 'ui-treenode-content ui-state-default ui-corner-all';
    const currVal_5:any = ck(v,15,0,co.tree.selectionMode,co.isSelected());
    ck(v,14,0,currVal_4,currVal_5);
    const currVal_6:boolean = !co.isLeaf();
    ck(v,18,0,currVal_6);
    const currVal_7:any = ((co.node.icon || co.node.expandedIcon) || co.node.collapsedIcon);
    ck(v,20,0,currVal_7);
    const currVal_8:boolean = !co.tree.getTemplateForNode(co.node);
    ck(v,24,0,currVal_8);
    const currVal_9:any = co.tree.getTemplateForNode(co.node);
    ck(v,27,0,currVal_9);
    const currVal_10:any = (co.node.children && co.node.expanded);
    ck(v,33,0,currVal_10);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.node.styleClass;
    ck(v,0,0,currVal_0);
  });
}
function View_UITreeNode_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_10)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_11)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.tree.droppableNodes;
    ck(v,2,0,currVal_0);
    const currVal_1:boolean = !co.tree.horizontal;
    ck(v,5,0,currVal_1);
    const currVal_2:any = (co.tree.droppableNodes && co.lastChild);
    ck(v,8,0,currVal_2);
    const currVal_3:any = co.tree.horizontal;
    ck(v,11,0,currVal_3);
  },(null as any));
}
export function View_UITreeNode_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeNode_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.UITreeNode = v.component;
    const currVal_0:any = co.node;
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_UITreeNode_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-treeNode',([] as any[]),(null as any),(null as any),(null as any),View_UITreeNode_0,RenderType_UITreeNode)),
    import0.ɵdid(57344,(null as any),0,import1.UITreeNode,[import1.Tree],(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const UITreeNodeNgFactory:import0.ComponentFactory<import1.UITreeNode> = import0.ɵccf('p-treeNode',import1.UITreeNode,View_UITreeNode_Host_0,{
  node: 'node',
  parentNode: 'parentNode',
  root: 'root',
  index: 'index',
  firstChild: 'firstChild',
  lastChild: 'lastChild'
}
,{},([] as any[]));
const styles_TreeNodeTemplateLoader:any[] = ([] as any[]);
export const RenderType_TreeNodeTemplateLoader:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_TreeNodeTemplateLoader,
  data: {}
}
);
export function View_TreeNodeTemplateLoader_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_TreeNodeTemplateLoader_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-treeNodeTemplateLoader',([] as any[]),(null as any),(null as any),(null as any),View_TreeNodeTemplateLoader_0,RenderType_TreeNodeTemplateLoader)),
    import0.ɵdid(122880,(null as any),0,import1.TreeNodeTemplateLoader,[import0.ViewContainerRef],(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const TreeNodeTemplateLoaderNgFactory:import0.ComponentFactory<import1.TreeNodeTemplateLoader> = import0.ɵccf('p-treeNodeTemplateLoader',import1.TreeNodeTemplateLoader,View_TreeNodeTemplateLoader_Host_0,{
  node: 'node',
  template: 'template'
}
,{},([] as any[]));
