// primeng/components/orderlist/orderlist.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/orderlist/orderlist';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/button/button';
import * as import4 from 'primeng/components/common/shared';
import * as import5 from 'primeng/components/dom/domhandler';
import * as import6 from 'primeng/components/utils/objectutils';
class OrderListModuleInjector extends import0.ɵNgModuleInjector<import1.OrderListModule> {
  _CommonModule_0:import2.CommonModule;
  _ButtonModule_1:import3.ButtonModule;
  _SharedModule_2:import4.SharedModule;
  _OrderListModule_3:import1.OrderListModule;
  __NgLocalization_4:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_4():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_4 == null)) { (this.__NgLocalization_4 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_4;
  }
  createInternal():import1.OrderListModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ButtonModule_1 = new import3.ButtonModule();
    this._SharedModule_2 = new import4.SharedModule();
    this._OrderListModule_3 = new import1.OrderListModule();
    return this._OrderListModule_3;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.ButtonModule)) { return this._ButtonModule_1; }
    if ((token === import4.SharedModule)) { return this._SharedModule_2; }
    if ((token === import1.OrderListModule)) { return this._OrderListModule_3; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_4; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const OrderListModuleNgFactory:import0.NgModuleFactory<import1.OrderListModule> = new import0.NgModuleFactory<any>(OrderListModuleInjector,import1.OrderListModule);
const styles_OrderList:any[] = ([] as any[]);
export const RenderType_OrderList:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_OrderList,
  data: {}
}
);
function View_OrderList_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'ui-orderlist-caption ui-widget-header ui-corner-top'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.header;
    ck(v,1,0,currVal_0);
  });
}
function View_OrderList_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',[[
        'class',
        'ui-orderlist-filter-container ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'input',[
      [
        'class',
        'ui-inputtext ui-widget ui-state-default ui-corner-all'
      ]
      ,
      [
        'role',
        'textbox'
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
    ,[
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        1,
        'placeholder',
        0
      ]

    ]
      ,[[
        (null as any),
        'keyup'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('keyup' === en)) {
        const pd_0:any = ((<any>co.onFilterKeyup($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-search'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.disabled;
    const currVal_1:any = co.filterPlaceholder;
    ck(v,2,0,currVal_0,currVal_1);
  });
}
function View_OrderList_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'li',[[
        'class',
        'ui-orderlist-droppoint'
      ]
    ],(null as any),[
      [
        (null as any),
        'dragover'
      ]
      ,
      [
        (null as any),
        'drop'
      ]
      ,
      [
        (null as any),
        'dragleave'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('dragover' === en)) {
        const pd_0:any = ((<any>co.onDragOver($event,(<any>v.parent).context.index)) !== false);
        ad = (pd_0 && ad);
      }
      if (('drop' === en)) {
        const pd_1:any = ((<any>co.onDrop($event,(<any>v.parent).context.index)) !== false);
        ad = (pd_1 && ad);
      }
      if (('dragleave' === en)) {
        const pd_2:any = ((<any>co.onDragLeave($event)) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-highlight'])
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-orderlist-droppoint';
    const currVal_1:any = ck(v,2,0,((<any>v.parent).context.index === co.dragOverItemIndex));
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_OrderList_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_OrderList_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'li',[[
        'class',
        'ui-orderlist-droppoint'
      ]
    ],(null as any),[
      [
        (null as any),
        'dragover'
      ]
      ,
      [
        (null as any),
        'drop'
      ]
      ,
      [
        (null as any),
        'dragleave'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('dragover' === en)) {
        const pd_0:any = ((<any>co.onDragOver($event,((<any>v.parent).context.index + 1))) !== false);
        ad = (pd_0 && ad);
      }
      if (('drop' === en)) {
        const pd_1:any = ((<any>co.onDrop($event,((<any>v.parent).context.index + 1))) !== false);
        ad = (pd_1 && ad);
      }
      if (('dragleave' === en)) {
        const pd_2:any = ((<any>co.onDragLeave($event)) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-highlight'])
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-orderlist-droppoint';
    const currVal_1:any = ck(v,2,0,(((<any>v.parent).context.index + 1) === co.dragOverItemIndex));
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_OrderList_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_OrderList_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'li',[[
        'class',
        'ui-orderlist-item'
      ]
    ],[
      [
        4,
        'display',
        (null as any)
      ]
      ,
      [
        8,
        'draggable',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'touchend'
      ]
      ,
      [
        (null as any),
        'dragstart'
      ]
      ,
      [
        (null as any),
        'dragend'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onItemClick($event,v.context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      if (('touchend' === en)) {
        const pd_1:any = ((<any>co.onItemTouchEnd($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('dragstart' === en)) {
        const pd_2:any = ((<any>co.onDragStart($event,v.context.index)) !== false);
        ad = (pd_2 && ad);
      }
      if (('dragend' === en)) {
        const pd_3:any = ((<any>co.onDragEnd($event)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-highlight']),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_OrderList_5)),
    import0.ɵdid(106496,(null as any),0,import4.TemplateWrapper,[import0.ViewContainerRef],{
      item: [
        0,
        'item'
      ]
      ,
      templateRef: [
        1,
        'templateRef'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_OrderList_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.dragdrop && co.isItemVisible(v.context.$implicit));
    ck(v,2,0,currVal_0);
    const currVal_3:any = 'ui-orderlist-item';
    const currVal_4:any = ck(v,6,0,co.isSelected(v.context.$implicit));
    ck(v,5,0,currVal_3,currVal_4);
    const currVal_5:any = v.context.$implicit;
    const currVal_6:any = co.itemTemplate;
    ck(v,9,0,currVal_5,currVal_6);
    const currVal_7:any = (co.dragdrop && v.context.last);
    ck(v,13,0,currVal_7);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = (co.isItemVisible(v.context.$implicit)? 'block': 'none');
    const currVal_2:any = co.dragdrop;
    ck(v,4,0,currVal_1,currVal_2);
  });
}
export function View_OrderList_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{listViewChild: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),39,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-orderlist ui-widget',
      'ui-orderlist-responsive'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),17,'div',[[
        'class',
        'ui-orderlist-controls'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-up'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.OrderList = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveUp($event,import0.ɵnov(v,34))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-double-up'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.OrderList = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveTop($event,import0.ɵnov(v,34))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-down'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.OrderList = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveDown($event,import0.ɵnov(v,34))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-double-down'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.OrderList = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveBottom($event,import0.ɵnov(v,34))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),14,'div',[[
        'class',
        'ui-orderlist-list-container'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_OrderList_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_OrderList_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'listelement',
        1
      ]

    ]
      ,(null as any),5,'ul',[[
        'class',
        'ui-widget-content ui-orderlist-list ui-corner-bottom'
      ]
      ],(null as any),[[
        (null as any),
        'dragover'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.OrderList = v.component;
      if (('dragover' === en)) {
        const pd_0:any = ((<any>co.onListMouseMove($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_OrderList_3)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.OrderList = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,4,0,true,co.responsive);
    ck(v,3,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,5,0,currVal_2);
    const currVal_3:any = 'fa-angle-up';
    ck(v,11,0,currVal_3);
    const currVal_4:any = 'fa-angle-double-up';
    ck(v,15,0,currVal_4);
    const currVal_5:any = 'fa-angle-down';
    ck(v,19,0,currVal_5);
    const currVal_6:any = 'fa-angle-double-down';
    ck(v,23,0,currVal_6);
    const currVal_7:any = co.header;
    ck(v,29,0,currVal_7);
    const currVal_8:any = co.filterBy;
    ck(v,32,0,currVal_8);
    const currVal_9:any = co.listStyle;
    ck(v,35,0,currVal_9);
    const currVal_10:any = co.value;
    ck(v,38,0,currVal_10);
  },(null as any));
}
function View_OrderList_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'p-orderList',([] as any[]),(null as any),(null as any),(null as any),View_OrderList_0,RenderType_OrderList)),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵprd(256,(null as any),import6.ObjectUtils,import6.ObjectUtils,([] as any[])),
    import0.ɵdid(6840320,(null as any),1,import1.OrderList,[
      import0.ElementRef,
      import5.DomHandler,
      import6.ObjectUtils
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(301989888,1,{templates: 1})
  ]
  ,(null as any),(null as any));
}
export const OrderListNgFactory:import0.ComponentFactory<import1.OrderList> = import0.ɵccf('p-orderList',import1.OrderList,View_OrderList_Host_0,{
  header: 'header',
  style: 'style',
  styleClass: 'styleClass',
  listStyle: 'listStyle',
  responsive: 'responsive',
  filterBy: 'filterBy',
  filterPlaceholder: 'filterPlaceholder',
  metaKeySelection: 'metaKeySelection',
  dragdrop: 'dragdrop',
  dragdropScope: 'dragdropScope',
  value: 'value'
}
,{
  onReorder: 'onReorder',
  onSelectionChange: 'onSelectionChange',
  onFilterEvent: 'onFilterEvent'
}
,([] as any[]));
