// primeng/components/confirmdialog/confirmdialog.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/confirmdialog/confirmdialog';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/button/button';
import * as import4 from 'primeng/components/common/shared';
import * as import5 from 'primeng/components/dom/domhandler';
import * as import6 from 'primeng/components/common/confirmationservice';
class ConfirmDialogModuleInjector extends import0.ɵNgModuleInjector<import1.ConfirmDialogModule> {
  _CommonModule_0:import2.CommonModule;
  _ButtonModule_1:import3.ButtonModule;
  _SharedModule_2:import4.SharedModule;
  _ConfirmDialogModule_3:import1.ConfirmDialogModule;
  __NgLocalization_4:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_4():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_4 == null)) { (this.__NgLocalization_4 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_4;
  }
  createInternal():import1.ConfirmDialogModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ButtonModule_1 = new import3.ButtonModule();
    this._SharedModule_2 = new import4.SharedModule();
    this._ConfirmDialogModule_3 = new import1.ConfirmDialogModule();
    return this._ConfirmDialogModule_3;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.ButtonModule)) { return this._ButtonModule_1; }
    if ((token === import4.SharedModule)) { return this._SharedModule_2; }
    if ((token === import1.ConfirmDialogModule)) { return this._ConfirmDialogModule_3; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_4; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const ConfirmDialogModuleNgFactory:import0.NgModuleFactory<import1.ConfirmDialogModule> = new import0.NgModuleFactory<any>(ConfirmDialogModuleInjector,import1.ConfirmDialogModule);
const styles_ConfirmDialog:any[] = ([] as any[]);
export const RenderType_ConfirmDialog:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ConfirmDialog,
      data: {'animation': [{
        name: 'dialogState',
        definitions: [
          {
            type: 0,
            name: 'hidden',
            styles: {
              type: 6,
              styles: {opacity: 0}
            }

          }
          ,
          {
            type: 0,
            name: 'visible',
            styles: {
              type: 6,
              styles: {opacity: 1}
            }

          }
          ,
          {
            type: 1,
            expr: 'visible => hidden',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms ease-in'
            }

          }
          ,
          {
            type: 1,
            expr: 'hidden => visible',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms ease-out'
            }

          }

        ]

      }
  ]}
}
);
function View_ConfirmDialog_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-dialog-title'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.header;
    ck(v,1,0,currVal_0);
  });
}
function View_ConfirmDialog_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'a',[
      [
        'href',
        '#'
      ]
      ,
      [
        'role',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.close($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['ui-dialog-titlebar-icon ui-dialog-titlebar-close ui-corner-all']),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-fw fa-close'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,true);
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_ConfirmDialog_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-dialog-footer ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(null as any));
}
function View_ConfirmDialog_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.accept()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
    ,{
      label: [
        0,
        'label'
      ]
      ,
      icon: [
        1,
        'icon'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.acceptLabel;
    const currVal_1:any = co.acceptIcon;
    ck(v,2,0,currVal_0,currVal_1);
  },(null as any));
}
function View_ConfirmDialog_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.reject()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
    ,{
      label: [
        0,
        'label'
      ]
      ,
      icon: [
        1,
        'icon'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.rejectLabel;
    const currVal_1:any = co.rejectIcon;
    ck(v,2,0,currVal_0,currVal_1);
  },(null as any));
}
function View_ConfirmDialog_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'ui-dialog-footer ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ConfirmDialog_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ConfirmDialog_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.acceptVisible;
    ck(v,3,0,currVal_0);
    const currVal_1:any = co.rejectVisible;
    ck(v,6,0,currVal_1);
  },(null as any));
}
export function View_ConfirmDialog_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),26,'div',([] as any[]),[
      [
        4,
        'display',
        (null as any)
      ]
      ,
      [
        4,
        'width',
        'px'
      ]
      ,
      [
        4,
        'height',
        'px'
      ]
      ,
      [
        24,
        '@dialogState',
        0
      ]

    ]
      ,[[
        (null as any),
        'mousedown'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.ConfirmDialog = v.component;
      if (('mousedown' === en)) {
        const pd_0:any = ((<any>co.moveOnTop()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-dialog ui-confirmdialog ui-widget ui-widget-content ui-corner-all ui-shadow',
      'ui-dialog-rtl'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'ui-dialog-titlebar ui-widget-header ui-helper-clearfix ui-corner-top'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ConfirmDialog_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ConfirmDialog_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',[[
        'class',
        'ui-dialog-content ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'i',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'ui-confirmdialog-message'
      ]
      ],[[
        8,
        'innerHTML',
        1
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ConfirmDialog_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ConfirmDialog_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.ConfirmDialog = v.component;
    const currVal_4:any = ck(v,3,0,true,co.rtl);
    ck(v,2,0,currVal_4);
    const currVal_5:any = co.header;
    ck(v,8,0,currVal_5);
    const currVal_6:any = co.closable;
    ck(v,11,0,currVal_6);
    const currVal_7:any = co.icon;
    const currVal_8:any = 'fa';
    ck(v,17,0,currVal_7,currVal_8);
    const currVal_10:any = co.footer;
    ck(v,23,0,currVal_10);
    const currVal_11:boolean = !co.footer;
    ck(v,26,0,currVal_11);
  },(ck,v) => {
    var co:import1.ConfirmDialog = v.component;
    const currVal_0:any = (co.visible? 'block': 'none');
    const currVal_1:any = co.width;
    const currVal_2:any = co.height;
    const currVal_3:any = (co.visible? 'visible': 'hidden');
    ck(v,1,0,currVal_0,currVal_1,currVal_2,currVal_3);
    const currVal_9:any = co.message;
    ck(v,19,0,currVal_9);
  });
}
function View_ConfirmDialog_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'p-confirmDialog',([] as any[]),(null as any),(null as any),(null as any),View_ConfirmDialog_0,RenderType_ConfirmDialog)),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(6381568,(null as any),1,import1.ConfirmDialog,[
      import0.ElementRef,
      import5.DomHandler,
      import0.Renderer2,
      import6.ConfirmationService
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(167772160,1,{footer: 0})
  ]
  ,(null as any),(null as any));
}
export const ConfirmDialogNgFactory:import0.ComponentFactory<import1.ConfirmDialog> = import0.ɵccf('p-confirmDialog',import1.ConfirmDialog,View_ConfirmDialog_Host_0,{
  header: 'header',
  icon: 'icon',
  message: 'message',
  acceptIcon: 'acceptIcon',
  acceptLabel: 'acceptLabel',
  acceptVisible: 'acceptVisible',
  rejectIcon: 'rejectIcon',
  rejectLabel: 'rejectLabel',
  rejectVisible: 'rejectVisible',
  width: 'width',
  height: 'height',
  closeOnEscape: 'closeOnEscape',
  rtl: 'rtl',
  closable: 'closable',
  responsive: 'responsive',
  appendTo: 'appendTo',
  key: 'key',
  visible: 'visible'
}
,{},['p-footer']);
