// primeng/components/gmap/gmap.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/gmap/gmap';
import * as import2 from '@angular/common';
class GMapModuleInjector extends import0.ɵNgModuleInjector<import1.GMapModule> {
  _CommonModule_0:import2.CommonModule;
  _GMapModule_1:import1.GMapModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.GMapModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._GMapModule_1 = new import1.GMapModule();
    return this._GMapModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.GMapModule)) { return this._GMapModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const GMapModuleNgFactory:import0.NgModuleFactory<import1.GMapModule> = new import0.NgModuleFactory<any>(GMapModuleInjector,import1.GMapModule);
const styles_GMap:any[] = ([] as any[]);
export const RenderType_GMap:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_GMap,
  data: {}
}
);
export function View_GMap_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:import1.GMap = v.component;
    const currVal_1:any = co.style;
    ck(v,1,0,currVal_1);
  },(ck,v) => {
    var co:import1.GMap = v.component;
    const currVal_0:any = co.styleClass;
    ck(v,0,0,currVal_0);
  });
}
function View_GMap_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-gmap',([] as any[]),(null as any),(null as any),(null as any),View_GMap_0,RenderType_GMap)),
    import0.ɵdid(4349952,(null as any),0,import1.GMap,[
      import0.ElementRef,
      import0.IterableDiffers,
      import0.ChangeDetectorRef,
      import0.NgZone
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const GMapNgFactory:import0.ComponentFactory<import1.GMap> = import0.ɵccf('p-gmap',import1.GMap,View_GMap_Host_0,{
  style: 'style',
  styleClass: 'styleClass',
  options: 'options',
  overlays: 'overlays'
}
,{
  onMapClick: 'onMapClick',
  onOverlayClick: 'onOverlayClick',
  onOverlayDragStart: 'onOverlayDragStart',
  onOverlayDrag: 'onOverlayDrag',
  onOverlayDragEnd: 'onOverlayDragEnd',
  onMapReady: 'onMapReady',
  onMapDragEnd: 'onMapDragEnd',
  onZoomChanged: 'onZoomChanged'
}
,([] as any[]));
