// primeng/components/treetable/treetable.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/treetable/treetable';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from '../common/shared.ngfactory';
import * as import5 from 'primeng/components/dom/domhandler';
class TreeTableModuleInjector extends import0.ɵNgModuleInjector<import1.TreeTableModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _TreeTableModule_2:import1.TreeTableModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.TreeTableModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._TreeTableModule_2 = new import1.TreeTableModule();
    return this._TreeTableModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import1.TreeTableModule)) { return this._TreeTableModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const TreeTableModuleNgFactory:import0.NgModuleFactory<import1.TreeTableModule> = new import0.NgModuleFactory<any>(TreeTableModuleInjector,import1.TreeTableModule);
const styles_TreeTable:any[] = ([] as any[]);
export const RenderType_TreeTable:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_TreeTable,
  data: {}
}
);
function View_TreeTable_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-treetable-header ui-widget-header'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(null as any));
}
function View_TreeTable_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-column-title'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = (<any>v.parent).context.$implicit.header;
    ck(v,1,0,currVal_0);
  });
}
function View_TreeTable_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'span',[[
        'class',
        'ui-column-title'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-columnHeaderTemplateLoader',([] as any[]),(null as any),(null as any),(null as any),import4.View_ColumnHeaderTemplateLoader_0,import4.RenderType_ColumnHeaderTemplateLoader)),
      import0.ɵdid(122880,(null as any),0,import3.ColumnHeaderTemplateLoader,[import0.ViewContainerRef],{column: [
        0,
        'column'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = (<any>v.parent).context.$implicit;
    ck(v,3,0,currVal_0);
  },(null as any));
}
function View_TreeTable_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,[[
        'headerCell',
        1
      ]
    ],(null as any),9,'th',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TreeTable_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TreeTable_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.styleClass;
    const currVal_1:any = 'ui-state-default ui-unselectable-text';
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = v.context.$implicit.style;
    ck(v,2,0,currVal_2);
    const currVal_3:boolean = !v.context.$implicit.headerTemplate;
    ck(v,5,0,currVal_3);
    const currVal_4:any = v.context.$implicit.headerTemplate;
    ck(v,8,0,currVal_4);
  },(null as any));
}
function View_TreeTable_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-column-footer'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = (<any>v.parent).context.$implicit.footer;
    ck(v,1,0,currVal_0);
  });
}
function View_TreeTable_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'span',[[
        'class',
        'ui-column-footer'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-columnFooterTemplateLoader',([] as any[]),(null as any),(null as any),(null as any),import4.View_ColumnFooterTemplateLoader_0,import4.RenderType_ColumnFooterTemplateLoader)),
      import0.ɵdid(122880,(null as any),0,import3.ColumnFooterTemplateLoader,[import0.ViewContainerRef],{column: [
        0,
        'column'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = (<any>v.parent).context.$implicit;
    ck(v,3,0,currVal_0);
  },(null as any));
}
function View_TreeTable_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'td',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-default']),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TreeTable_7)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TreeTable_8)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.styleClass;
    const currVal_1:any = ck(v,2,0,true);
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = v.context.$implicit.style;
    ck(v,3,0,currVal_2);
    const currVal_3:boolean = !v.context.$implicit.footerTemplate;
    ck(v,6,0,currVal_3);
    const currVal_4:any = v.context.$implicit.footerTemplate;
    ck(v,9,0,currVal_4);
  },(null as any));
}
function View_TreeTable_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tfoot',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TreeTable_6)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.columns;
    ck(v,5,0,currVal_0);
  },(null as any));
}
function View_TreeTable_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'tbody',[
      [
        'class',
        'ui-treetable-data ui-widget-content'
      ]
      ,
      [
        'pTreeRow',
        ''
      ]

    ]
    ,(null as any),(null as any),(null as any),View_UITreeRow_0,RenderType_UITreeRow)),
    import0.ɵdid(57344,(null as any),0,import1.UITreeRow,[import1.TreeTable],{
      node: [
        0,
        'node'
      ]
      ,
      level: [
        1,
        'level'
      ]
      ,
      labelExpand: [
        2,
        'labelExpand'
      ]
      ,
      labelCollapse: [
        3,
        'labelCollapse'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = v.context.$implicit;
    const currVal_1:any = 0;
    const currVal_2:any = co.labelExpand;
    const currVal_3:any = co.labelCollapse;
    ck(v,1,0,currVal_0,currVal_1,currVal_2,currVal_3);
  },(null as any));
}
function View_TreeTable_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-treetable-footer ui-widget-header'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),1),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(null as any));
}
export function View_TreeTable_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{tableViewChild: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),31,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TreeTable_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),20,'div',[[
        'class',
        'ui-treetable-tablewrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'tbl',
        1
      ]

    ]
      ,(null as any),17,'table',[[
        'class',
        'ui-widget-content'
      ]
      ],[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'thead',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tr',[[
        'class',
        'ui-state-default'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TreeTable_2)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TreeTable_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TreeTable_9)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            \n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TreeTable_10)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.TreeTable = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-treetable ui-widget';
    ck(v,3,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,4,0,currVal_2);
    const currVal_3:any = co.header;
    ck(v,7,0,currVal_3);
    const currVal_5:any = co.tableStyle;
    ck(v,12,0,currVal_5);
    const currVal_6:any = co.columns;
    ck(v,19,0,currVal_6);
    const currVal_7:any = co.hasFooter();
    ck(v,24,0,currVal_7);
    const currVal_8:any = co.value;
    ck(v,27,0,currVal_8);
    const currVal_9:any = co.footer;
    ck(v,32,0,currVal_9);
  },(ck,v) => {
    var co:import1.TreeTable = v.component;
    const currVal_4:any = co.tableStyleClass;
    ck(v,11,0,currVal_4);
  });
}
function View_TreeTable_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'p-treeTable',([] as any[]),(null as any),(null as any),(null as any),View_TreeTable_0,RenderType_TreeTable)),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(548864,(null as any),3,import1.TreeTable,[
      import0.ElementRef,
      import5.DomHandler,
      import0.ChangeDetectorRef,
      import0.Renderer2
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(167772160,1,{header: 0}),
    import0.ɵqud(167772160,2,{footer: 0}),
    import0.ɵqud(301989888,3,{cols: 1})
  ]
  ,(null as any),(null as any));
}
export const TreeTableNgFactory:import0.ComponentFactory<import1.TreeTable> = import0.ɵccf('p-treeTable',import1.TreeTable,View_TreeTable_Host_0,{
  value: 'value',
  selectionMode: 'selectionMode',
  selection: 'selection',
  style: 'style',
  styleClass: 'styleClass',
  labelExpand: 'labelExpand',
  labelCollapse: 'labelCollapse',
  metaKeySelection: 'metaKeySelection',
  contextMenu: 'contextMenu',
  toggleColumnIndex: 'toggleColumnIndex',
  tableStyle: 'tableStyle',
  tableStyleClass: 'tableStyleClass',
  collapsedIcon: 'collapsedIcon',
  expandedIcon: 'expandedIcon'
}
,{
  onRowDblclick: 'onRowDblclick',
  selectionChange: 'selectionChange',
  onNodeSelect: 'onNodeSelect',
  onNodeUnselect: 'onNodeUnselect',
  onNodeExpand: 'onNodeExpand',
  onNodeCollapse: 'onNodeCollapse',
  onContextMenuSelect: 'onContextMenuSelect'
}
,[
  'p-header',
  'p-footer'
]
);
const styles_UITreeRow:any[] = ([] as any[]);
export const RenderType_UITreeRow:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_UITreeRow,
  data: {}
}
);
function View_UITreeRow_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'a',[
      [
        'class',
        'ui-treetable-toggler fa fa-fw ui-clickable'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,[[
        8,
        'title',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggle($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'margin-left',
      'visibility'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = 'ui-treetable-toggler fa fa-fw ui-clickable';
    const currVal_2:any = (co.node.expanded? co.treeTable.expandedIcon: co.treeTable.collapsedIcon);
    ck(v,1,0,currVal_1,currVal_2);
    const currVal_3:any = ck(v,3,0,((co.level * 16) + 'px'),(co.isLeaf()? 'hidden': 'visible'));
    ck(v,2,0,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.node.expanded? co.labelCollapse: co.labelExpand);
    ck(v,0,0,currVal_0);
  });
}
function View_UITreeRow_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',[[
        'class',
        'ui-chkbox ui-treetable-checkbox'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'ui-chkbox-box ui-widget ui-corner-all ui-state-default'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-chkbox-icon ui-clickable fa'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'fa-check',
      'fa-minus'
    ]
    )
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-chkbox-icon ui-clickable fa';
    const currVal_1:any = ck(v,5,0,co.isSelected(),co.node.partialSelected);
    ck(v,4,0,currVal_0,currVal_1);
  },(null as any));
}
function View_UITreeRow_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.resolveFieldData(co.node.data,(<any>v.parent).context.$implicit.field);
    ck(v,1,0,currVal_0);
  });
}
function View_UITreeRow_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-columnBodyTemplateLoader',([] as any[]),(null as any),(null as any),(null as any),import4.View_ColumnBodyTemplateLoader_0,import4.RenderType_ColumnBodyTemplateLoader)),
    import0.ɵdid(385024,(null as any),0,import3.ColumnBodyTemplateLoader,[import0.ViewContainerRef],{
      column: [
        0,
        'column'
      ]
      ,
      rowData: [
        1,
        'rowData'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (<any>v.parent).context.$implicit;
    const currVal_1:any = co.node;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_UITreeRow_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),13,'td',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'dblclick'
      ]
      ,
      [
        (null as any),
        'touchend'
      ]
      ,
      [
        (null as any),
        'contextmenu'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onRowClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('dblclick' === en)) {
        const pd_1:any = ((<any>co.rowDblClick($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('touchend' === en)) {
        const pd_2:any = ((<any>co.onRowTouchEnd()) !== false);
        ad = (pd_2 && ad);
      }
      if (('contextmenu' === en)) {
        const pd_3:any = ((<any>co.onRowRightClick($event)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeRow_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeRow_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeRow_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeRow_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = v.context.$implicit.style;
    ck(v,1,0,currVal_1);
    const currVal_2:any = (v.context.index == co.treeTable.toggleColumnIndex);
    ck(v,4,0,currVal_2);
    const currVal_3:any = ((co.treeTable.selectionMode == 'checkbox') && (v.context.index == 0));
    ck(v,7,0,currVal_3);
    const currVal_4:boolean = !v.context.$implicit.template;
    ck(v,9,0,currVal_4);
    const currVal_5:any = v.context.$implicit.template;
    ck(v,12,0,currVal_5);
  },(ck,v) => {
    const currVal_0:any = v.context.$implicit.styleClass;
    ck(v,0,0,currVal_0);
  });
}
function View_UITreeRow_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'tbody',[[
        'pTreeRow',
        ''
      ]
    ],(null as any),(null as any),(null as any),View_UITreeRow_0,RenderType_UITreeRow)),
    import0.ɵdid(57344,(null as any),0,import1.UITreeRow,[import1.TreeTable],{
      node: [
        0,
        'node'
      ]
      ,
      parentNode: [
        1,
        'parentNode'
      ]
      ,
      level: [
        2,
        'level'
      ]
      ,
      labelExpand: [
        3,
        'labelExpand'
      ]
      ,
      labelCollapse: [
        4,
        'labelCollapse'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = v.context.$implicit;
    const currVal_1:any = co.node;
    const currVal_2:any = (co.level + 1);
    const currVal_3:any = co.labelExpand;
    const currVal_4:any = co.labelCollapse;
    ck(v,1,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  },(null as any));
}
function View_UITreeRow_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),11,'div',[
      [
        'class',
        'ui-treetable-row'
      ]
      ,
      [
        'style',
        'display:table-row'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),8,'td',[[
        'class',
        'ui-treetable-child-table-container'
      ]
      ],[[
        1,
        'colspan',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'table',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeRow_7)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co.treeTable.tableStyle;
    ck(v,5,0,currVal_2);
    const currVal_3:any = co.node.children;
    ck(v,8,0,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.treeTable.columns.length;
    ck(v,2,0,currVal_0);
    const currVal_1:any = co.treeTable.tableStyleClass;
    ck(v,4,0,currVal_1);
  });
}
export function View_UITreeRow_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',[[
        'class',
        'ui-treetable-row'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-state-highlight',
      'ui-treetable-row-selectable'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeRow_1)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_UITreeRow_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.UITreeRow = v.component;
    const currVal_0:any = 'ui-treetable-row';
    const currVal_1:any = ck(v,3,0,co.isSelected(),(co.treeTable.selectionMode && (co.node.selectable !== false)));
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.treeTable.columns;
    ck(v,6,0,currVal_2);
    const currVal_3:any = (co.node.children && co.node.expanded);
    ck(v,10,0,currVal_3);
  },(null as any));
}
function View_UITreeRow_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'pTreeRow',
        ''
      ]
    ],(null as any),(null as any),(null as any),View_UITreeRow_0,RenderType_UITreeRow)),
    import0.ɵdid(57344,(null as any),0,import1.UITreeRow,[import1.TreeTable],(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const UITreeRowNgFactory:import0.ComponentFactory<import1.UITreeRow> = import0.ɵccf('[pTreeRow]',import1.UITreeRow,View_UITreeRow_Host_0,{
  node: 'node',
  parentNode: 'parentNode',
  level: 'level',
  labelExpand: 'labelExpand',
  labelCollapse: 'labelCollapse'
}
,{},([] as any[]));
