// primeng/components/slidemenu/slidemenu.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/slidemenu/slidemenu';
import * as import2 from '@angular/common';
import * as import3 from '@angular/router';
import * as import4 from 'primeng/components/dom/domhandler';
class SlideMenuModuleInjector extends import0.ɵNgModuleInjector<import1.SlideMenuModule> {
  _CommonModule_0:import2.CommonModule;
  _RouterModule_1:import3.RouterModule;
  _SlideMenuModule_2:import1.SlideMenuModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.SlideMenuModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._RouterModule_1 = new import3.RouterModule(this.parent.get(import3.ɵa,(null as any)),this.parent.get(import3.Router,(null as any)));
    this._SlideMenuModule_2 = new import1.SlideMenuModule();
    return this._SlideMenuModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.RouterModule)) { return this._RouterModule_1; }
    if ((token === import1.SlideMenuModule)) { return this._SlideMenuModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const SlideMenuModuleNgFactory:import0.NgModuleFactory<import1.SlideMenuModule> = new import0.NgModuleFactory<any>(SlideMenuModuleInjector,import1.SlideMenuModule);
const styles_SlideMenu:any[] = ([] as any[]);
export const RenderType_SlideMenu:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_SlideMenu,
  data: {}
}
);
export function View_SlideMenu_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{containerViewChild: 0}),
    import0.ɵqud(201326592,2,{backwardViewChild: 0}),
    import0.ɵqud(201326592,3,{slideMenuContentViewChild: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'container',
        1
      ]

    ]
      ,(null as any),20,'div',([] as any[]),(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.SlideMenu = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-menu ui-slidemenu ui-widget ui-widget-content ui-corner-all',
      'ui-menu-dynamic ui-shadow'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),14,'div',[[
        'class',
        'ui-slidemenu-wrapper'
      ]
      ],[[
        4,
        'height',
        'px'
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,[
      [
        3,
        0
      ]
      ,
      [
        'slideMenuContent',
        1
      ]

    ]
      ,(null as any),4,'div',[[
        'class',
        'ui-slidemenu-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-slideMenuSub',[[
        'root',
        'root'
      ]
    ],(null as any),(null as any),(null as any),View_SlideMenuSub_0,RenderType_SlideMenuSub)),
    import0.ɵdid(90112,(null as any),0,import1.SlideMenuSub,[import1.SlideMenu],{
      item: [
        0,
        'item'
      ]
      ,
      root: [
        1,
        'root'
      ]
      ,
      menuWidth: [
        2,
        'menuWidth'
      ]
      ,
      effectDuration: [
        3,
        'effectDuration'
      ]
      ,
      easing: [
        4,
        'easing'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'backward',
        1
      ]

    ]
      ,(null as any),5,'div',[[
        'class',
        'ui-slidemenu-backward ui-widget-header ui-corner-all'
      ]
      ],[[
        4,
        'display',
        (null as any)
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.SlideMenu = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.goBack()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-fw fa-caret-left'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.SlideMenu = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,6,0,true,co.popup);
    ck(v,5,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,7,0,currVal_2);
    const currVal_4:any = co.model;
    const currVal_5:any = 'root';
    const currVal_6:any = co.menuWidth;
    const currVal_7:any = co.effectDuration;
    const currVal_8:any = co.easing;
    ck(v,14,0,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8);
  },(ck,v) => {
    var co:import1.SlideMenu = v.component;
    const currVal_3:any = co.viewportHeight;
    ck(v,9,0,currVal_3);
    const currVal_9:any = (co.left? 'block': 'none');
    ck(v,17,0,currVal_9);
    const currVal_10:any = co.backLabel;
    ck(v,21,0,currVal_10);
  });
}
function View_SlideMenu_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-slideMenu',([] as any[]),(null as any),(null as any),(null as any),View_SlideMenu_0,RenderType_SlideMenu)),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵdid(2187264,(null as any),0,import1.SlideMenu,[
      import0.ElementRef,
      import4.DomHandler,
      import0.Renderer2
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const SlideMenuNgFactory:import0.ComponentFactory<import1.SlideMenu> = import0.ɵccf('p-slideMenu',import1.SlideMenu,View_SlideMenu_Host_0,{
  model: 'model',
  popup: 'popup',
  style: 'style',
  styleClass: 'styleClass',
  menuWidth: 'menuWidth',
  viewportHeight: 'viewportHeight',
  effectDuration: 'effectDuration',
  easing: 'easing',
  backLabel: 'backLabel'
}
,{},([] as any[]));
const styles_SlideMenuSub:any[] = ([] as any[]);
export const RenderType_SlideMenuSub:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_SlideMenuSub,
  data: {}
}
);
function View_SlideMenuSub_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'li',[[
        'class',
        'ui-menu-separator ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(null as any),(null as any));
}
function View_SlideMenuSub_5(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'ui-submenu-icon fa fa-fw fa-caret-right'
      ]
  ],(null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),(null as any));
}
function View_SlideMenuSub_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-icon fa fa-fw'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-menuitem-icon fa fa-fw';
    const currVal_1:any = (<any>(<any>(<any>v.parent).parent).parent).context.$implicit.icon;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_SlideMenuSub_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),12,'a',[[
        'class',
        'ui-menuitem-link ui-corner-all'
      ]
    ],[
      [
        8,
        'href',
        4
      ]
      ,
      [
        1,
        'target',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.itemClick($event,(<any>(<any>v.parent).parent).context.$implicit,import0.ɵnov((<any>v.parent),0))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-menuitem-link-parent',
      'ui-state-disabled'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SlideMenuSub_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SlideMenuSub_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    const currVal_2:any = 'ui-menuitem-link ui-corner-all';
    const currVal_3:any = ck(v,2,0,(<any>(<any>v.parent).parent).context.$implicit.items,(<any>(<any>v.parent).parent).context.$implicit.disabled);
    ck(v,1,0,currVal_2,currVal_3);
    const currVal_4:any = (<any>(<any>v.parent).parent).context.$implicit.items;
    ck(v,5,0,currVal_4);
    const currVal_5:any = (<any>(<any>v.parent).parent).context.$implicit.icon;
    ck(v,8,0,currVal_5);
  },(ck,v) => {
    const currVal_0:any = ((<any>(<any>v.parent).parent).context.$implicit.url || '#');
    const currVal_1:any = (<any>(<any>v.parent).parent).context.$implicit.target;
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_6:any = (<any>(<any>v.parent).parent).context.$implicit.label;
    ck(v,11,0,currVal_6);
  });
}
function View_SlideMenuSub_8(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'ui-submenu-icon fa fa-fw fa-caret-right'
      ]
  ],(null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),(null as any));
}
function View_SlideMenuSub_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-icon fa fa-fw'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-menuitem-icon fa fa-fw';
    const currVal_1:any = (<any>(<any>(<any>v.parent).parent).parent).context.$implicit.icon;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_SlideMenuSub_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),17,'a',[[
        'class',
        'ui-menuitem-link ui-corner-all'
      ]
    ],[
      [
        8,
        'href',
        4
      ]
      ,
      [
        1,
        'target',
        0
      ]
      ,
      [
        1,
        'target',
        0
      ]
      ,
      [
        8,
        'href',
        4
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,3).onClick($event.button,$event.ctrlKey,$event.metaKey)) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>co.itemClick($event,(<any>(<any>v.parent).parent).context.$implicit,import0.ɵnov((<any>v.parent),0))) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-menuitem-link-parent',
      'ui-state-disabled'
    ]
    ),
      import0.ɵdid(335872,[[
        2,
        4
      ]
    ],0,import3.RouterLinkWithHref,[
      import3.Router,
      import3.ActivatedRoute,
      import2.LocationStrategy
    ]
      ,{routerLink: [
        0,
        'routerLink'
      ]
    },(null as any)),
    import0.ɵdid(860160,(null as any),2,import3.RouterLinkActive,[
      import3.Router,
      import0.ElementRef,
      import0.Renderer,
      import0.ChangeDetectorRef
    ]
    ,{
      routerLinkActiveOptions: [
        0,
        'routerLinkActiveOptions'
      ]
      ,
      routerLinkActive: [
        1,
        'routerLinkActive'
      ]

    }
    ,(null as any)),
    import0.ɵqud(301989888,1,{links: 1}),
    import0.ɵqud(301989888,2,{linksWithHrefs: 1}),
    import0.ɵpod(['exact']),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SlideMenuSub_8)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SlideMenuSub_9)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    const currVal_4:any = 'ui-menuitem-link ui-corner-all';
    const currVal_5:any = ck(v,2,0,(<any>(<any>v.parent).parent).context.$implicit.items,(<any>(<any>v.parent).parent).context.$implicit.disabled);
    ck(v,1,0,currVal_4,currVal_5);
    const currVal_6:any = (<any>(<any>v.parent).parent).context.$implicit.routerLink;
    ck(v,3,0,currVal_6);
    const currVal_7:any = ((<any>(<any>v.parent).parent).context.$implicit.routerLinkActiveOptions || ck(v,7,0,false));
    const currVal_8:any = 'ui-state-active';
    ck(v,4,0,currVal_7,currVal_8);
    const currVal_9:any = (<any>(<any>v.parent).parent).context.$implicit.items;
    ck(v,10,0,currVal_9);
    const currVal_10:any = (<any>(<any>v.parent).parent).context.$implicit.icon;
    ck(v,13,0,currVal_10);
  },(ck,v) => {
    const currVal_0:any = ((<any>(<any>v.parent).parent).context.$implicit.url || '#');
    const currVal_1:any = (<any>(<any>v.parent).parent).context.$implicit.target;
    const currVal_2:any = import0.ɵnov(v,3).target;
    const currVal_3:any = import0.ɵnov(v,3).href;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
    const currVal_11:any = (<any>(<any>v.parent).parent).context.$implicit.label;
    ck(v,16,0,currVal_11);
  });
}
function View_SlideMenuSub_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-slideMenuSub',[[
        'class',
        'ui-submenu'
      ]
    ],(null as any),(null as any),(null as any),View_SlideMenuSub_0,RenderType_SlideMenuSub)),
    import0.ɵdid(90112,(null as any),0,import1.SlideMenuSub,[import1.SlideMenu],{
      item: [
        0,
        'item'
      ]
      ,
      menuWidth: [
        1,
        'menuWidth'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (<any>(<any>v.parent).parent).context.$implicit;
    const currVal_1:any = co.menuWidth;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_SlideMenuSub_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,[[
        'listitem',
        1
      ]
    ],(null as any),12,'li',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-menuitem ui-widget ui-corner-all',
      'ui-menu-parent',
      'ui-slidemenuitem-active'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SlideMenuSub_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SlideMenuSub_7)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SlideMenuSub_10)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,true,(<any>v.parent).context.$implicit.items,(import0.ɵnov(v,0) == co.activeItem));
    ck(v,1,0,currVal_0);
    const currVal_1:boolean = !(<any>v.parent).context.$implicit.routerLink;
    ck(v,5,0,currVal_1);
    const currVal_2:any = (<any>v.parent).context.$implicit.routerLink;
    ck(v,8,0,currVal_2);
    const currVal_3:any = (<any>v.parent).context.$implicit.items;
    ck(v,11,0,currVal_3);
  },(null as any));
}
function View_SlideMenuSub_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SlideMenuSub_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SlideMenuSub_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.separator;
    ck(v,2,0,currVal_0);
    const currVal_1:boolean = !v.context.$implicit.separator;
    ck(v,4,0,currVal_1);
  },(null as any));
}
export function View_SlideMenuSub_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'ul',[[
        'class',
        'ui-menu-list'
      ]
    ],[
      [
        4,
        'width',
        'px'
      ]
      ,
      [
        4,
        'left',
        'px'
      ]
      ,
      [
        4,
        'transitionProperty',
        (null as any)
      ]
      ,
      [
        4,
        'transitionDuration',
        (null as any)
      ]
      ,
      [
        4,
        'transitionTimingFunction',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-helper-reset ui-menu-rootlist',
      'ui-widget-content ui-corner-all ui-helper-clearfix ui-menu-child'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SlideMenuSub_1)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.SlideMenuSub = v.component;
    const currVal_5:any = 'ui-menu-list';
    const currVal_6:any = ck(v,3,0,co.root,!co.root);
    ck(v,2,0,currVal_5,currVal_6);
    const currVal_7:any = (co.root? co.item: co.item.items);
    ck(v,6,0,currVal_7);
  },(ck,v) => {
    var co:import1.SlideMenuSub = v.component;
    const currVal_0:any = co.menuWidth;
    const currVal_1:any = (co.root? co.slideMenu.left: co.slideMenu.menuWidth);
    const currVal_2:any = (co.root? 'left': 'none');
    const currVal_3:any = (co.effectDuration + 'ms');
    const currVal_4:any = co.easing;
    ck(v,1,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
function View_SlideMenuSub_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-slideMenuSub',([] as any[]),(null as any),(null as any),(null as any),View_SlideMenuSub_0,RenderType_SlideMenuSub)),
    import0.ɵdid(90112,(null as any),0,import1.SlideMenuSub,[import1.SlideMenu],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const SlideMenuSubNgFactory:import0.ComponentFactory<import1.SlideMenuSub> = import0.ɵccf('p-slideMenuSub',import1.SlideMenuSub,View_SlideMenuSub_Host_0,{
  item: 'item',
  root: 'root',
  backLabel: 'backLabel',
  menuWidth: 'menuWidth',
  effectDuration: 'effectDuration',
  easing: 'easing'
}
,{},([] as any[]));
