// primeng/components/lightbox/lightbox.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/lightbox/lightbox';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/dom/domhandler';
class LightboxModuleInjector extends import0.ɵNgModuleInjector<import1.LightboxModule> {
  _CommonModule_0:import2.CommonModule;
  _LightboxModule_1:import1.LightboxModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.LightboxModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._LightboxModule_1 = new import1.LightboxModule();
    return this._LightboxModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.LightboxModule)) { return this._LightboxModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const LightboxModuleNgFactory:import0.NgModuleFactory<import1.LightboxModule> = new import0.NgModuleFactory<any>(LightboxModuleInjector,import1.LightboxModule);
const styles_Lightbox:any[] = ([] as any[]);
export const RenderType_Lightbox:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Lightbox,
  data: {}
}
);
function View_Lightbox_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'a',([] as any[]),[[
        8,
        'href',
        4
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onImageClick($event,v.context.$implicit,v.context.index,import0.ɵnov((<any>(<any>v.parent).parent),18))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'img',([] as any[]),[
      [
        8,
        'src',
        4
      ]
      ,
      [
        8,
        'title',
        0
      ]
      ,
      [
        8,
        'alt',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = v.context.$implicit.source;
    ck(v,0,0,currVal_0);
    const currVal_1:any = v.context.$implicit.thumbnail;
    const currVal_2:any = v.context.$implicit.title;
    const currVal_3:any = v.context.$implicit.alt;
    ck(v,2,0,currVal_1,currVal_2,currVal_3);
  });
}
function View_Lightbox_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Lightbox_2)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.style;
    ck(v,1,0,currVal_1);
    const currVal_2:any = co.images;
    ck(v,4,0,currVal_2);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.styleClass;
    ck(v,0,0,currVal_0);
  });
}
function View_Lightbox_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'span',([] as any[]),[[
        8,
        'className',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onLinkClick($event,import0.ɵnov((<any>v.parent),18))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.style;
    ck(v,1,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.styleClass;
    ck(v,0,0,currVal_0);
  });
}
export function View_Lightbox_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Lightbox_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Lightbox_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),33,'div',[[
        'class',
        'ui-lightbox ui-widget ui-helper-hidden ui-corner-all ui-shadow'
      ]
    ],[
      [
        4,
        'display',
        (null as any)
      ]
      ,
      [
        4,
        'zIndex',
        (null as any)
      ]
      ,
      [
        4,
        'transitionProperty',
        (null as any)
      ]
      ,
      [
        4,
        'transitionDuration',
        (null as any)
      ]
      ,
      [
        4,
        'transitionTimingFunction',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Lightbox = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>(co.preventDocumentClickListener = true)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-lightbox-loading']),
    (l()(),import0.ɵted((null as any),['\n           '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),18,'div',[[
        'class',
        'ui-lightbox-content-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n              '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'a',[[
        'class',
        'ui-state-default ui-lightbox-nav-left ui-corner-right'
      ]
      ],[[
        4,
        'zIndex',
        (null as any)
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Lightbox = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.prev(import0.ɵnov(v,20))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-helper-hidden']),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-fw fa-caret-left'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n              '])),
    (l()(),import0.ɵeld(0,[
      [
        'content',
        1
      ]
      ,
      [
        'content',
        1
      ]

    ]
      ,(null as any),5,'div',[[
        'class',
        'ui-lightbox-content ui-corner-all'
      ]
    ],[
      [
        4,
        'transitionProperty',
        (null as any)
      ]
      ,
      [
        4,
        'transitionDuration',
        (null as any)
      ]
      ,
      [
        4,
        'transitionTimingFunction',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,[[
        'img',
        1
      ]
      ],(null as any),0,'img',[[
        'style',
        'display:none'
      ]
      ],[[
        8,
        'src',
        4
      ]
      ],[[
        (null as any),
        'load'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Lightbox = v.component;
      if (('load' === en)) {
        const pd_0:any = ((<any>co.onImageLoad($event,import0.ɵnov(v,18))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),1),
    (l()(),import0.ɵted((null as any),['\n              '])),
    (l()(),import0.ɵted((null as any),['\n              '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'a',[[
        'class',
        'ui-state-default ui-lightbox-nav-right ui-corner-left ui-helper-hidden'
      ]
      ],[[
        4,
        'zIndex',
        (null as any)
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Lightbox = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.next(import0.ɵnov(v,20))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-helper-hidden']),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-fw fa-caret-right'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n           '])),
    (l()(),import0.ɵted((null as any),['\n           '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),8,'div',[[
        'class',
        'ui-lightbox-caption ui-widget-header'
      ]
      ],[[
        4,
        'display',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n              '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-lightbox-caption-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'a',[
      [
        'class',
        'ui-lightbox-close ui-corner-all'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Lightbox = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.hide($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-fw fa-close'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n              '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'style',
        'clear:both'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n           '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Lightbox = v.component;
    const currVal_0:any = (co.type == 'image');
    ck(v,2,0,currVal_0);
    const currVal_1:any = (co.type == 'content');
    ck(v,5,0,currVal_1);
    const currVal_7:any = 'ui-lightbox ui-widget ui-helper-hidden ui-corner-all ui-shadow';
    const currVal_8:any = ck(v,9,0,co.loading);
    ck(v,8,0,currVal_7,currVal_8);
    const currVal_10:any = 'ui-state-default ui-lightbox-nav-left ui-corner-right';
    const currVal_11:any = ck(v,15,0,!co.leftVisible);
    ck(v,14,0,currVal_10,currVal_11);
    const currVal_17:any = 'ui-state-default ui-lightbox-nav-right ui-corner-left ui-helper-hidden';
    const currVal_18:any = ck(v,27,0,!co.rightVisible);
    ck(v,26,0,currVal_17,currVal_18);
  },(ck,v) => {
    var co:import1.Lightbox = v.component;
    const currVal_2:any = (co.visible? 'block': 'none');
    const currVal_3:any = co.zindex;
    const currVal_4:any = 'all';
    const currVal_5:any = co.effectDuration;
    const currVal_6:any = co.easing;
    ck(v,7,0,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6);
    const currVal_9:any = (co.zindex + 1);
    ck(v,13,0,currVal_9);
    const currVal_12:any = 'width,height';
    const currVal_13:any = co.effectDuration;
    const currVal_14:any = co.easing;
    ck(v,18,0,currVal_12,currVal_13,currVal_14);
    const currVal_15:any = (co.currentImage? (co.currentImage.source || ''): '');
    ck(v,20,0,currVal_15);
    const currVal_16:any = (co.zindex + 1);
    ck(v,25,0,currVal_16);
    const currVal_19:any = (co.captionText? 'block': 'none');
    ck(v,31,0,currVal_19);
    const currVal_20:any = co.captionText;
    ck(v,34,0,currVal_20);
  });
}
function View_Lightbox_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-lightbox',([] as any[]),(null as any),(null as any),(null as any),View_Lightbox_0,RenderType_Lightbox)),
    import0.ɵprd(256,(null as any),import3.DomHandler,import3.DomHandler,([] as any[])),
    import0.ɵdid(2187264,(null as any),0,import1.Lightbox,[
      import0.ElementRef,
      import3.DomHandler,
      import0.Renderer2
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const LightboxNgFactory:import0.ComponentFactory<import1.Lightbox> = import0.ɵccf('p-lightbox',import1.Lightbox,View_Lightbox_Host_0,{
  images: 'images',
  type: 'type',
  style: 'style',
  styleClass: 'styleClass',
  appendTo: 'appendTo',
  easing: 'easing',
  effectDuration: 'effectDuration'
}
,{},[
  'a',
  '*'
]
);
