// @ng-bootstrap/ng-bootstrap/tabset/tabset.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@ng-bootstrap/ng-bootstrap/tabset/tabset';
import * as import3 from '@ng-bootstrap/ng-bootstrap/tabset/tabset-config';
const styles_NgbTabset:any[] = ([] as any[]);
export const RenderType_NgbTabset:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NgbTabset,
  data: {}
}
);
function View_NgbTabset_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_NgbTabset_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'li',[[
        'class',
        'nav-item'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'a',[
      [
        'class',
        'nav-link'
      ]
      ,
      [
        'href',
        ''
      ]
      ,
      [
        'role',
        'tab'
      ]

    ]
    ,[
      [
        8,
        'id',
        0
      ]
      ,
      [
        2,
        'active',
        (null as any)
      ]
      ,
      [
        2,
        'disabled',
        (null as any)
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        1,
        'aria-controls',
        0
      ]
      ,
      [
        1,
        'aria-expanded',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>!!co.select(v.context.$implicit.id)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n          ',
      ''
    ]
    )),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTabset_2)),
      import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(ck,v) => {
    const currVal_8:any = ((v.context.$implicit.titleTpl == null)? (null as any): v.context.$implicit.titleTpl.templateRef);
    ck(v,5,0,currVal_8);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = v.context.$implicit.id;
    const currVal_1:any = (v.context.$implicit.id === co.activeId);
    const currVal_2:any = v.context.$implicit.disabled;
    const currVal_3:any = (v.context.$implicit.disabled? '-1': (undefined as any));
    const currVal_4:any = ((!co.destroyOnHide || (v.context.$implicit.id === co.activeId))? (v.context.$implicit.id + '-panel'): (null as any));
    const currVal_5:any = (v.context.$implicit.id === co.activeId);
    const currVal_6:any = v.context.$implicit.disabled;
    ck(v,2,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6);
    const currVal_7:any = v.context.$implicit.title;
    ck(v,3,0,currVal_7);
  });
}
function View_NgbTabset_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_NgbTabset_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'role',
        'tabpanel'
      ]
    ],[
      [
        8,
        'className',
        0
      ]
      ,
      [
        1,
        'aria-labelledby',
        0
      ]
      ,
      [
        8,
        'id',
        0
      ]
      ,
      [
        1,
        'aria-expanded',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTabset_5)),
      import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    const currVal_4:any = (<any>v.parent).context.$implicit.contentTpl.templateRef;
    ck(v,3,0,currVal_4);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = import0.ɵinlineInterpolate(1,'tab-pane ',(((<any>v.parent).context.$implicit.id === co.activeId)? 'active': (null as any)),'');
    const currVal_1:any = (<any>v.parent).context.$implicit.id;
    const currVal_2:any = import0.ɵinlineInterpolate(1,'',(<any>v.parent).context.$implicit.id,'-panel');
    const currVal_3:any = ((<any>v.parent).context.$implicit.id === co.activeId);
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
function View_NgbTabset_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTabset_4)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:boolean = (!co.destroyOnHide || (v.context.$implicit.id === co.activeId));
    ck(v,2,0,currVal_0);
  },(null as any));
}
export function View_NgbTabset_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ul',[[
        'role',
        'tablist'
      ]
      ],[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTabset_1)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'tab-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTabset_3)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import2.NgbTabset = v.component;
    const currVal_1:any = co.tabs;
    ck(v,4,0,currVal_1);
    const currVal_2:any = co.tabs;
    ck(v,10,0,currVal_2);
  },(ck,v) => {
    var co:import2.NgbTabset = v.component;
    const currVal_0:any = (('nav nav-' + co.type) + ((co.orientation == 'horizontal')? (' ' + co.justifyClass): ' flex-column'));
    ck(v,1,0,currVal_0);
  });
}
function View_NgbTabset_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'ngb-tabset',([] as any[]),(null as any),(null as any),(null as any),View_NgbTabset_0,RenderType_NgbTabset)),
    import0.ɵdid(1073152,(null as any),1,import2.NgbTabset,[import3.NgbTabsetConfig],(null as any),(null as any)),
    import0.ɵqud(301989888,1,{tabs: 1})
  ]
  ,(null as any),(null as any));
}
export const NgbTabsetNgFactory:import0.ComponentFactory<import2.NgbTabset> = import0.ɵccf('ngb-tabset',import2.NgbTabset,View_NgbTabset_Host_0,{
  activeId: 'activeId',
  destroyOnHide: 'destroyOnHide',
  justify: 'justify',
  orientation: 'orientation',
  type: 'type'
}
,{tabChange: 'tabChange'},([] as any[]));
