// primeng/components/chips/chips.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/chips/chips';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/inputtext/inputtext';
import * as import4 from 'primeng/components/common/shared';
import * as import5 from 'primeng/components/dom/domhandler';
import * as import6 from '@angular/forms';
class ChipsModuleInjector extends import0.ɵNgModuleInjector<import1.ChipsModule> {
  _CommonModule_0:import2.CommonModule;
  _InputTextModule_1:import3.InputTextModule;
  _SharedModule_2:import4.SharedModule;
  _ChipsModule_3:import1.ChipsModule;
  __NgLocalization_4:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_4():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_4 == null)) { (this.__NgLocalization_4 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_4;
  }
  createInternal():import1.ChipsModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._InputTextModule_1 = new import3.InputTextModule();
    this._SharedModule_2 = new import4.SharedModule();
    this._ChipsModule_3 = new import1.ChipsModule();
    return this._ChipsModule_3;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.InputTextModule)) { return this._InputTextModule_1; }
    if ((token === import4.SharedModule)) { return this._SharedModule_2; }
    if ((token === import1.ChipsModule)) { return this._ChipsModule_3; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_4; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const ChipsModuleNgFactory:import0.NgModuleFactory<import1.ChipsModule> = new import0.NgModuleFactory<any>(ChipsModuleInjector,import1.ChipsModule);
const styles_Chips:any[] = ([] as any[]);
export const RenderType_Chips:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Chips,
  data: {}
}
);
function View_Chips_2(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'ui-chips-token-icon fa fa-fw fa-close'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.removeItem($event,(<any>v.parent).context.index)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
  },(null as any),(null as any)))],(null as any),(null as any));
}
function View_Chips_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-chips-token-label'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.field? co.resolveFieldData((<any>v.parent).context.$implicit,co.field): (<any>v.parent).context.$implicit);
    ck(v,1,0,currVal_0);
  });
}
function View_Chips_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_Chips_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Chips_5)),
    import0.ɵdid(106496,(null as any),0,import4.TemplateWrapper,[import0.ViewContainerRef],{
      item: [
        0,
        'item'
      ]
      ,
      templateRef: [
        1,
        'templateRef'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (<any>v.parent).context.$implicit;
    const currVal_1:any = co.itemTemplate;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_Chips_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,[[
        'token',
        1
      ]
      ],(null as any),10,'li',[[
        'class',
        'ui-chips-token ui-state-highlight ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Chips_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Chips_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Chips_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:boolean = !co.disabled;
    ck(v,3,0,currVal_0);
    const currVal_1:boolean = !co.itemTemplate;
    ck(v,6,0,currVal_1);
    const currVal_2:any = co.itemTemplate;
    ck(v,9,0,currVal_2);
  },(null as any));
}
export function View_Chips_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),17,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),12,'ul',([] as any[]),(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,14).focus()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-inputtext ui-state-default ui-corner-all',
      'ui-state-focus',
      'ui-state-disabled'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Chips_1)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'li',[[
        'class',
        'ui-chips-input-token'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,[[
        'inputtext',
        1
      ]
      ],(null as any),1,'input',[[
        'type',
        'text'
      ]
    ],[
      [
        1,
        'id',
        0
      ]
      ,
      [
        1,
        'placeholder',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        8,
        'className',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'keydown'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Chips = v.component;
      if (('keydown' === en)) {
        const pd_0:any = ((<any>co.onKeydown($event,import0.ɵnov(v,14))) !== false);
        ad = (pd_0 && ad);
      }
      if (('focus' === en)) {
        const pd_1:any = ((<any>co.onFocus()) !== false);
        ad = (pd_1 && ad);
      }
      if (('blur' === en)) {
        const pd_2:any = ((<any>co.onBlur()) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Chips = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-chips ui-widget';
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_3:any = ck(v,7,0,true,co.focus,co.disabled);
    ck(v,6,0,currVal_3);
    const currVal_4:any = co.value;
    ck(v,10,0,currVal_4);
    const currVal_11:any = co.inputStyle;
    ck(v,15,0,currVal_11);
  },(ck,v) => {
    var co:import1.Chips = v.component;
    const currVal_5:any = co.inputId;
    const currVal_6:any = co.placeholder;
    const currVal_7:any = co.tabindex;
    const currVal_8:any = (co.maxedOut || co.disabled);
    const currVal_9:any = co.disabled;
    const currVal_10:any = co.inputStyleClass;
    ck(v,14,0,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9,currVal_10);
  });
}
function View_Chips_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'p-chips',([] as any[]),(null as any),(null as any),(null as any),View_Chips_0,RenderType_Chips)),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(548864,(null as any),1,import1.Chips,[
      import0.ElementRef,
      import5.DomHandler
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(301989888,1,{templates: 1}),
    import0.ɵprd(2560,(null as any),import6.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.Chips])
  ]
  ,(null as any),(null as any));
}
export const ChipsNgFactory:import0.ComponentFactory<import1.Chips> = import0.ɵccf('p-chips',import1.Chips,View_Chips_Host_0,{
  style: 'style',
  styleClass: 'styleClass',
  disabled: 'disabled',
  field: 'field',
  placeholder: 'placeholder',
  max: 'max',
  tabindex: 'tabindex',
  inputId: 'inputId',
  allowDuplicate: 'allowDuplicate',
  inputStyle: 'inputStyle',
  inputStyleClass: 'inputStyleClass',
  addOnTab: 'addOnTab'
}
,{
  onAdd: 'onAdd',
  onRemove: 'onRemove'
}
,([] as any[]));
