// primeng/components/colorpicker/colorpicker.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/colorpicker/colorpicker';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/dom/domhandler';
import * as import4 from '@angular/forms';
class ColorPickerModuleInjector extends import0.ɵNgModuleInjector<import1.ColorPickerModule> {
  _CommonModule_0:import2.CommonModule;
  _ColorPickerModule_1:import1.ColorPickerModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.ColorPickerModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ColorPickerModule_1 = new import1.ColorPickerModule();
    return this._ColorPickerModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.ColorPickerModule)) { return this._ColorPickerModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const ColorPickerModuleNgFactory:import0.NgModuleFactory<import1.ColorPickerModule> = new import0.NgModuleFactory<any>(ColorPickerModuleInjector,import1.ColorPickerModule);
const styles_ColorPicker:any[] = ([] as any[]);
export const RenderType_ColorPicker:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ColorPicker,
      data: {'animation': [{
        name: 'panelState',
        definitions: [
          {
            type: 0,
            name: 'hidden',
            styles: {
              type: 6,
              styles: {opacity: 0}
            }

          }
          ,
          {
            type: 0,
            name: 'visible',
            styles: {
              type: 6,
              styles: {opacity: 1}
            }

          }
          ,
          {
            type: 1,
            expr: 'visible => hidden',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms ease-in'
            }

          }
          ,
          {
            type: 1,
            expr: 'hidden => visible',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms ease-out'
            }

          }

        ]

      }
  ]}
}
);
function View_ColorPicker_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,[
      [
        6,
        0
      ]
      ,
      [
        'input',
        1
      ]

    ]
    ,(null as any),2,'input',[
      [
        'class',
        'ui-colorpicker-preview ui-inputtext ui-state-default ui-corner-all'
      ]
      ,
      [
        'readonly',
        'readonly'
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
    ,[
      [
        1,
        'id',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        4,
        'backgroundColor',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('focus' === en)) {
        const pd_0:any = ((<any>co.onInputFocus()) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>co.onInputClick()) !== false);
        ad = (pd_1 && ad);
      }
      if (('keydown' === en)) {
        const pd_2:any = ((<any>co.onInputKeydown($event)) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled'])
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_4:any = 'ui-colorpicker-preview ui-inputtext ui-state-default ui-corner-all';
    const currVal_5:any = ck(v,2,0,co.disabled);
    ck(v,1,0,currVal_4,currVal_5);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.inputId;
    const currVal_1:any = co.tabindex;
    const currVal_2:any = co.disabled;
    const currVal_3:any = co.inputBgColor;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
export function View_ColorPicker_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{panelViewChild: 0}),
    import0.ɵqud(201326592,2,{colorSelectorViewChild: 0}),
    import0.ɵqud(201326592,3,{colorHandleViewChild: 0}),
    import0.ɵqud(201326592,4,{hueViewChild: 0}),
    import0.ɵqud(201326592,5,{hueHandleViewChild: 0}),
    import0.ɵqud(335544320,6,{inputViewChild: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),28,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-colorpicker ui-widget',
      'ui-colorpicker-overlay',
      'ui-colorpicker-dragging'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ColorPicker_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'panel',
        1
      ]

    ]
    ,(null as any),19,'div',([] as any[]),[
      [
        24,
        '@panelState',
        0
      ]
      ,
      [
        4,
        'display',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.ColorPicker = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onPanelClick()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-colorpicker-panel ui-corner-all',
      'ui-colorpicker-overlay-panel ui-shadow',
      'ui-state-disabled'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),14,'div',[[
        'class',
        'ui-colorpicker-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'colorSelector',
        1
      ]

    ]
      ,(null as any),6,'div',[[
        'class',
        'ui-colorpicker-color-selector'
      ]
      ],(null as any),[[
        (null as any),
        'mousedown'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.ColorPicker = v.component;
      if (('mousedown' === en)) {
        const pd_0:any = ((<any>co.onColorMousedown($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-colorpicker-color'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,[
      [
        3,
        0
      ]
      ,
      [
        'colorHandle',
        1
      ]

    ]
      ,(null as any),0,'div',[[
        'class',
        'ui-colorpicker-color-handle'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,[
      [
        4,
        0
      ]
      ,
      [
        'hue',
        1
      ]

    ]
      ,(null as any),3,'div',[[
        'class',
        'ui-colorpicker-hue'
      ]
      ],(null as any),[[
        (null as any),
        'mousedown'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.ColorPicker = v.component;
      if (('mousedown' === en)) {
        const pd_0:any = ((<any>co.onHueMousedown($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,[
      [
        5,
        0
      ]
      ,
      [
        'hueHandle',
        1
      ]

    ]
      ,(null as any),0,'div',[[
        'class',
        'ui-colorpicker-hue-handle'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.ColorPicker = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,9,0,true,!co.inline,(co.colorDragging || co.hueDragging));
    ck(v,8,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,10,0,currVal_2);
    const currVal_3:boolean = !co.inline;
    ck(v,13,0,currVal_3);
    const currVal_6:any = ck(v,17,0,true,!co.inline,co.disabled);
    ck(v,16,0,currVal_6);
  },(ck,v) => {
    var co:import1.ColorPicker = v.component;
    const currVal_4:any = (co.inline? 'visible': (co.panelVisible? 'visible': 'hidden'));
    const currVal_5:any = (co.inline? 'block': (co.panelVisible? 'block': 'none'));
    ck(v,15,0,currVal_4,currVal_5);
  });
}
function View_ColorPicker_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'p-colorPicker',([] as any[]),(null as any),(null as any),(null as any),View_ColorPicker_0,RenderType_ColorPicker)),
    import0.ɵprd(256,(null as any),import3.DomHandler,import3.DomHandler,([] as any[])),
    import0.ɵdid(4284416,(null as any),0,import1.ColorPicker,[
      import0.ElementRef,
      import3.DomHandler,
      import0.Renderer2,
      import0.ChangeDetectorRef
    ]
    ,(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import4.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.ColorPicker])
  ]
  ,(null as any),(null as any));
}
export const ColorPickerNgFactory:import0.ComponentFactory<import1.ColorPicker> = import0.ɵccf('p-colorPicker',import1.ColorPicker,View_ColorPicker_Host_0,{
  style: 'style',
  styleClass: 'styleClass',
  inline: 'inline',
  format: 'format',
  appendTo: 'appendTo',
  disabled: 'disabled',
  tabindex: 'tabindex',
  inputId: 'inputId'
}
,{onChange: 'onChange'},([] as any[]));
