// primeng/components/inplace/inplace.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/inplace/inplace';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/button/button';
import * as import4 from 'primeng/components/dom/domhandler';
class InplaceModuleInjector extends import0.ɵNgModuleInjector<import1.InplaceModule> {
  _CommonModule_0:import2.CommonModule;
  _ButtonModule_1:import3.ButtonModule;
  _InplaceModule_2:import1.InplaceModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.InplaceModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ButtonModule_1 = new import3.ButtonModule();
    this._InplaceModule_2 = new import1.InplaceModule();
    return this._InplaceModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.ButtonModule)) { return this._ButtonModule_1; }
    if ((token === import1.InplaceModule)) { return this._InplaceModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const InplaceModuleNgFactory:import0.NgModuleFactory<import1.InplaceModule> = new import0.NgModuleFactory<any>(InplaceModuleInjector,import1.InplaceModule);
const styles_Inplace:any[] = ([] as any[]);
export const RenderType_Inplace:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Inplace,
  data: {}
}
);
function View_Inplace_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',[[
        'class',
        'ui-inplace-display'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.activate($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled']),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-inplace-display';
    const currVal_1:any = ck(v,2,0,co.disabled);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_Inplace_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-close'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.deactivate($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import4.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'fa-close';
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_Inplace_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',[[
        'class',
        'ui-inplace-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),1),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Inplace_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.closable;
    ck(v,5,0,currVal_0);
  },(null as any));
}
export function View_Inplace_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),9,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Inplace_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Inplace_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Inplace = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-inplace ui-widget';
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_3:boolean = !co.active;
    ck(v,6,0,currVal_3);
    const currVal_4:any = co.active;
    ck(v,9,0,currVal_4);
  },(null as any));
}
function View_Inplace_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-inplace',([] as any[]),(null as any),(null as any),(null as any),View_Inplace_0,RenderType_Inplace)),
    import0.ɵdid(24576,(null as any),0,import1.Inplace,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const InplaceNgFactory:import0.ComponentFactory<import1.Inplace> = import0.ɵccf('p-inplace',import1.Inplace,View_Inplace_Host_0,{
  active: 'active',
  closable: 'closable',
  disabled: 'disabled',
  style: 'style',
  styleClass: 'styleClass'
}
,{
  onActivate: 'onActivate',
  onDeactivate: 'onDeactivate'
}
,[
  '[pInplaceDisplay]',
  '[pInplaceContent]'
]
);
const styles_InplaceDisplay:any[] = ([] as any[]);
export const RenderType_InplaceDisplay:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_InplaceDisplay,
  data: {}
}
);
export function View_InplaceDisplay_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[import0.ɵncd((null as any),0)],(null as any),(null as any));
}
function View_InplaceDisplay_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-inplaceDisplay',([] as any[]),(null as any),(null as any),(null as any),View_InplaceDisplay_0,RenderType_InplaceDisplay)),
    import0.ɵdid(24576,(null as any),0,import1.InplaceDisplay,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const InplaceDisplayNgFactory:import0.ComponentFactory<import1.InplaceDisplay> = import0.ɵccf('p-inplaceDisplay',import1.InplaceDisplay,View_InplaceDisplay_Host_0,{},{},['*']);
const styles_InplaceContent:any[] = ([] as any[]);
export const RenderType_InplaceContent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_InplaceContent,
  data: {}
}
);
export function View_InplaceContent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[import0.ɵncd((null as any),0)],(null as any),(null as any));
}
function View_InplaceContent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-inplaceContent',([] as any[]),(null as any),(null as any),(null as any),View_InplaceContent_0,RenderType_InplaceContent)),
    import0.ɵdid(24576,(null as any),0,import1.InplaceContent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const InplaceContentNgFactory:import0.ComponentFactory<import1.InplaceContent> = import0.ɵccf('p-inplaceContent',import1.InplaceContent,View_InplaceContent_Host_0,{},{},['*']);
