// primeng/components/radiobutton/radiobutton.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/radiobutton/radiobutton';
import * as import2 from '@angular/common';
import * as import3 from '@angular/forms';
class RadioButtonModuleInjector extends import0.ɵNgModuleInjector<import1.RadioButtonModule> {
  _CommonModule_0:import2.CommonModule;
  _RadioButtonModule_1:import1.RadioButtonModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.RadioButtonModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._RadioButtonModule_1 = new import1.RadioButtonModule();
    return this._RadioButtonModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.RadioButtonModule)) { return this._RadioButtonModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const RadioButtonModuleNgFactory:import0.NgModuleFactory<import1.RadioButtonModule> = new import0.NgModuleFactory<any>(RadioButtonModuleInjector,import1.RadioButtonModule);
const styles_RadioButton:any[] = ([] as any[]);
export const RenderType_RadioButton:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_RadioButton,
  data: {}
}
);
function View_RadioButton_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'label',[[
        'class',
        'ui-radiobutton-label'
      ]
      ],[[
        1,
        'for',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.select()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.inputId;
    ck(v,0,0,currVal_0);
    const currVal_1:any = co.label;
    ck(v,1,0,currVal_1);
  });
}
export function View_RadioButton_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{inputViewChild: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),17,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-helper-hidden-accessible'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'rb',
        1
      ]

    ]
      ,(null as any),0,'input',[[
        'type',
        'radio'
      ]
    ],[
      [
        1,
        'id',
        0
      ]
      ,
      [
        1,
        'name',
        0
      ]
      ,
      [
        1,
        'value',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        8,
        'checked',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'change'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.RadioButton = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.onChange($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('focus' === en)) {
        const pd_1:any = ((<any>co.onFocus($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('blur' === en)) {
        const pd_2:any = ((<any>co.onBlur($event)) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',([] as any[]),(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.RadioButton = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.handleClick()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-radiobutton-box ui-widget ui-state-default',
      'ui-state-active',
      'ui-state-disabled',
      'ui-state-focus'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-radiobutton-icon ui-clickable'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['fa fa-circle']),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_RadioButton_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.RadioButton = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-radiobutton ui-widget';
    ck(v,3,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,4,0,currVal_2);
    const currVal_8:any = ck(v,13,0,true,import0.ɵnov(v,8).checked,co.disabled,co.focused);
    ck(v,12,0,currVal_8);
    const currVal_9:any = 'ui-radiobutton-icon ui-clickable';
    const currVal_10:any = ck(v,17,0,import0.ɵnov(v,8).checked);
    ck(v,16,0,currVal_9,currVal_10);
    const currVal_11:any = co.label;
    ck(v,22,0,currVal_11);
  },(ck,v) => {
    var co:import1.RadioButton = v.component;
    const currVal_3:any = co.inputId;
    const currVal_4:any = co.name;
    const currVal_5:any = co.value;
    const currVal_6:any = co.tabindex;
    const currVal_7:any = co.checked;
    ck(v,8,0,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7);
  });
}
function View_RadioButton_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-radioButton',([] as any[]),(null as any),(null as any),(null as any),View_RadioButton_0,RenderType_RadioButton)),
    import0.ɵdid(24576,(null as any),0,import1.RadioButton,[import0.ChangeDetectorRef],(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import3.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.RadioButton])
  ]
  ,(null as any),(null as any));
}
export const RadioButtonNgFactory:import0.ComponentFactory<import1.RadioButton> = import0.ɵccf('p-radioButton',import1.RadioButton,View_RadioButton_Host_0,{
  value: 'value',
  name: 'name',
  disabled: 'disabled',
  label: 'label',
  tabindex: 'tabindex',
  inputId: 'inputId',
  style: 'style',
  styleClass: 'styleClass'
}
,{onClick: 'onClick'},([] as any[]));
