// primeng/components/terminal/terminal.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/terminal/terminal';
import * as import2 from '@angular/common';
import * as import3 from '@angular/forms';
import * as import4 from 'primeng/components/dom/domhandler';
class TerminalModuleInjector extends import0.ɵNgModuleInjector<import1.TerminalModule> {
  _CommonModule_0:import2.CommonModule;
  _ɵba_1:import3.ɵba;
  _FormsModule_2:import3.FormsModule;
  _TerminalModule_3:import1.TerminalModule;
  __NgLocalization_4:import2.NgLocaleLocalization;
  __ɵi_5:import3.ɵi;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_4():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_4 == null)) { (this.__NgLocalization_4 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_4;
  }
  get _ɵi_5():import3.ɵi {
    if ((this.__ɵi_5 == null)) { (this.__ɵi_5 = new import3.ɵi()); }
    return this.__ɵi_5;
  }
  createInternal():import1.TerminalModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ɵba_1 = new import3.ɵba();
    this._FormsModule_2 = new import3.FormsModule();
    this._TerminalModule_3 = new import1.TerminalModule();
    return this._TerminalModule_3;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.ɵba)) { return this._ɵba_1; }
    if ((token === import3.FormsModule)) { return this._FormsModule_2; }
    if ((token === import1.TerminalModule)) { return this._TerminalModule_3; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_4; }
    if ((token === import3.ɵi)) { return this._ɵi_5; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const TerminalModuleNgFactory:import0.NgModuleFactory<import1.TerminalModule> = new import0.NgModuleFactory<any>(TerminalModuleInjector,import1.TerminalModule);
const styles_Terminal:any[] = ([] as any[]);
export const RenderType_Terminal:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Terminal,
  data: {}
}
);
function View_Terminal_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.welcomeMessage;
    ck(v,1,0,currVal_0);
  });
}
function View_Terminal_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-terminal-command'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.prompt;
    ck(v,3,0,currVal_0);
    const currVal_1:any = v.context.$implicit.text;
    ck(v,6,0,currVal_1);
    const currVal_2:any = v.context.$implicit.response;
    ck(v,9,0,currVal_2);
  });
}
export function View_Terminal_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),25,'div',([] as any[]),(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Terminal = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.focus(import0.ɵnov(v,19))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Terminal_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'ui-terminal-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Terminal_2)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),11,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-terminal-content-prompt'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,[[
        'in',
        1
      ]
    ],(null as any),5,'input',[
      [
        'autocomplete',
        'off'
      ]
      ,
      [
        'autofocus',
        ''
      ]
      ,
      [
        'class',
        'ui-terminal-input'
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
    ,[
      [
        2,
        'ng-untouched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-touched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pristine',
        (null as any)
      ]
      ,
      [
        2,
        'ng-dirty',
        (null as any)
      ]
      ,
      [
        2,
        'ng-valid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-invalid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pending',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'ngModelChange'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]
      ,
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'compositionstart'
      ]
      ,
      [
        (null as any),
        'compositionend'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Terminal = v.component;
      if (('input' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,20)._handleInput($event.target.value)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,20).onTouched()) !== false);
        ad = (pd_1 && ad);
      }
      if (('compositionstart' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,20)._compositionStart()) !== false);
        ad = (pd_2 && ad);
      }
      if (('compositionend' === en)) {
        const pd_3:any = ((<any>import0.ɵnov(v,20)._compositionEnd($event.target.value)) !== false);
        ad = (pd_3 && ad);
      }
      if (('ngModelChange' === en)) {
        const pd_4:any = ((<any>(co.command = $event)) !== false);
        ad = (pd_4 && ad);
      }
      if (('keydown' === en)) {
        const pd_5:any = ((<any>co.handleCommand($event)) !== false);
        ad = (pd_5 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(8192,(null as any),0,import3.DefaultValueAccessor,[
      import0.Renderer,
      import0.ElementRef,
      [
        2,
        import3.COMPOSITION_BUFFER_MODE
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵprd(512,(null as any),import3.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import3.DefaultValueAccessor]),
    import0.ɵdid(335872,(null as any),0,import3.NgModel,[
      [
        8,
        (null as any)
      ]
      ,
      [
        8,
        (null as any)
      ]
      ,
      [
        8,
        (null as any)
      ]
      ,
      [
        2,
        import3.NG_VALUE_ACCESSOR
      ]

    ]
      ,{model: [
        0,
        'model'
      ]
    },{update: 'ngModelChange'}),
    import0.ɵprd(1024,(null as any),import3.NgControl,(null as any),[import3.NgModel]),
    import0.ɵdid(8192,(null as any),0,import3.NgControlStatus,[import3.NgControl],(null as any),(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Terminal = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-terminal ui-widget ui-widget-content ui-corner-all';
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_3:any = co.welcomeMessage;
    ck(v,6,0,currVal_3);
    const currVal_4:any = co.commands;
    ck(v,11,0,currVal_4);
    const currVal_13:any = co.command;
    ck(v,22,0,currVal_13);
  },(ck,v) => {
    var co:import1.Terminal = v.component;
    const currVal_5:any = co.prompt;
    ck(v,17,0,currVal_5);
    const currVal_6:any = import0.ɵnov(v,24).ngClassUntouched;
    const currVal_7:any = import0.ɵnov(v,24).ngClassTouched;
    const currVal_8:any = import0.ɵnov(v,24).ngClassPristine;
    const currVal_9:any = import0.ɵnov(v,24).ngClassDirty;
    const currVal_10:any = import0.ɵnov(v,24).ngClassValid;
    const currVal_11:any = import0.ɵnov(v,24).ngClassInvalid;
    const currVal_12:any = import0.ɵnov(v,24).ngClassPending;
    ck(v,19,0,currVal_6,currVal_7,currVal_8,currVal_9,currVal_10,currVal_11,currVal_12);
  });
}
function View_Terminal_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-terminal',([] as any[]),(null as any),(null as any),(null as any),View_Terminal_0,RenderType_Terminal)),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵdid(6316032,(null as any),0,import1.Terminal,[
      import0.ElementRef,
      import4.DomHandler
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const TerminalNgFactory:import0.ComponentFactory<import1.Terminal> = import0.ɵccf('p-terminal',import1.Terminal,View_Terminal_Host_0,{
  welcomeMessage: 'welcomeMessage',
  prompt: 'prompt',
  style: 'style',
  styleClass: 'styleClass',
  response: 'response'
}
,{
  responseChange: 'responseChange',
  handler: 'handler'
}
,([] as any[]));
