// primeng/components/toolbar/toolbar.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/toolbar/toolbar';
import * as import2 from '@angular/common';
class ToolbarModuleInjector extends import0.ɵNgModuleInjector<import1.ToolbarModule> {
  _CommonModule_0:import2.CommonModule;
  _ToolbarModule_1:import1.ToolbarModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.ToolbarModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ToolbarModule_1 = new import1.ToolbarModule();
    return this._ToolbarModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.ToolbarModule)) { return this._ToolbarModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const ToolbarModuleNgFactory:import0.NgModuleFactory<import1.ToolbarModule> = new import0.NgModuleFactory<any>(ToolbarModuleInjector,import1.ToolbarModule);
const styles_Toolbar:any[] = ([] as any[]);
export const RenderType_Toolbar:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Toolbar,
  data: {}
}
);
export function View_Toolbar_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Toolbar = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-toolbar ui-widget ui-widget-header ui-corner-all ui-helper-clearfix';
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
  },(null as any));
}
function View_Toolbar_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-toolbar',([] as any[]),(null as any),(null as any),(null as any),View_Toolbar_0,RenderType_Toolbar)),
    import0.ɵdid(24576,(null as any),0,import1.Toolbar,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ToolbarNgFactory:import0.ComponentFactory<import1.Toolbar> = import0.ɵccf('p-toolbar',import1.Toolbar,View_Toolbar_Host_0,{
  style: 'style',
  styleClass: 'styleClass'
}
,{},['*']);
