// primeng/components/fileupload/fileupload.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/fileupload/fileupload';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from 'primeng/components/button/button';
import * as import5 from 'primeng/components/progressbar/progressbar';
import * as import6 from 'primeng/components/messages/messages';
import * as import7 from 'primeng/components/dom/domhandler';
import * as import8 from '../progressbar/progressbar.ngfactory';
import * as import9 from '../common/shared.ngfactory';
import * as import10 from '../messages/messages.ngfactory';
import * as import11 from '@angular/platform-browser';
class FileUploadModuleInjector extends import0.ɵNgModuleInjector<import1.FileUploadModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _ButtonModule_2:import4.ButtonModule;
  _ProgressBarModule_3:import5.ProgressBarModule;
  _MessagesModule_4:import6.MessagesModule;
  _FileUploadModule_5:import1.FileUploadModule;
  __NgLocalization_6:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_6():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_6 == null)) { (this.__NgLocalization_6 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_6;
  }
  createInternal():import1.FileUploadModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._ButtonModule_2 = new import4.ButtonModule();
    this._ProgressBarModule_3 = new import5.ProgressBarModule();
    this._MessagesModule_4 = new import6.MessagesModule();
    this._FileUploadModule_5 = new import1.FileUploadModule();
    return this._FileUploadModule_5;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import4.ButtonModule)) { return this._ButtonModule_2; }
    if ((token === import5.ProgressBarModule)) { return this._ProgressBarModule_3; }
    if ((token === import6.MessagesModule)) { return this._MessagesModule_4; }
    if ((token === import1.FileUploadModule)) { return this._FileUploadModule_5; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_6; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const FileUploadModuleNgFactory:import0.NgModuleFactory<import1.FileUploadModule> = new import0.NgModuleFactory<any>(FileUploadModuleInjector,import1.FileUploadModule);
const styles_FileUpload:any[] = ([] as any[]);
export const RenderType_FileUpload:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_FileUpload,
  data: {}
}
);
function View_FileUpload_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-upload'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,[[
        8,
        'disabled',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.upload()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import7.DomHandler,import7.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import4.Button,[
      import0.ElementRef,
      import7.DomHandler
    ]
    ,{
      label: [
        0,
        'label'
      ]
      ,
      icon: [
        1,
        'icon'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.uploadLabel;
    const currVal_2:any = 'fa-upload';
    ck(v,2,0,currVal_1,currVal_2);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:boolean = !co.hasFiles();
    ck(v,0,0,currVal_0);
  });
}
function View_FileUpload_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-close'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,[[
        8,
        'disabled',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.clear()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import7.DomHandler,import7.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import4.Button,[
      import0.ElementRef,
      import7.DomHandler
    ]
    ,{
      label: [
        0,
        'label'
      ]
      ,
      icon: [
        1,
        'icon'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.cancelLabel;
    const currVal_2:any = 'fa-close';
    ck(v,2,0,currVal_1,currVal_2);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:boolean = !co.hasFiles();
    ck(v,0,0,currVal_0);
  });
}
function View_FileUpload_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-progressBar',([] as any[]),(null as any),(null as any),(null as any),import8.View_ProgressBar_0,import8.RenderType_ProgressBar)),
    import0.ɵdid(24576,(null as any),0,import5.ProgressBar,([] as any[]),{
      value: [
        0,
        'value'
      ]
      ,
      showValue: [
        1,
        'showValue'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.progress;
    const currVal_1:any = false;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_FileUpload_8(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'img',([] as any[]),[
      [
        8,
        'src',
        4
      ]
      ,
      [
        8,
        'width',
        0
      ]

    ]
  ,(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (<any>v.parent).context.$implicit.objectURL;
    const currVal_1:any = co.previewWidth;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_FileUpload_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),16,'div',[[
        'class',
        'ui-fileupload-row'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FileUpload_8)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-close'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.remove($event,v.context.index)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import7.DomHandler,import7.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import4.Button,[
      import0.ElementRef,
      import7.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.isImage(v.context.$implicit);
    ck(v,4,0,currVal_0);
    const currVal_3:any = 'fa-close';
    ck(v,15,0,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = v.context.$implicit.name;
    ck(v,7,0,currVal_1);
    const currVal_2:any = co.formatSize(v.context.$implicit.size);
    ck(v,10,0,currVal_2);
  });
}
function View_FileUpload_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FileUpload_7)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.files;
    ck(v,3,0,currVal_0);
  },(null as any));
}
function View_FileUpload_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_FileUpload_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FileUpload_10)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
    ,{
      ngForOf: [
        0,
        'ngForOf'
      ]
      ,
      ngForTemplate: [
        1,
        'ngForTemplate'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.files;
    const currVal_1:any = co.fileTemplate;
    ck(v,3,0,currVal_0,currVal_1);
  },(null as any));
}
function View_FileUpload_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'ui-fileupload-files'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FileUpload_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FileUpload_9)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:boolean = !co.fileTemplate;
    ck(v,3,0,currVal_0);
    const currVal_1:any = co.fileTemplate;
    ck(v,6,0,currVal_1);
  },(null as any));
}
function View_FileUpload_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),41,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),19,'div',[[
        'class',
        'ui-fileupload-buttonbar ui-widget-header ui-corner-top'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'span',[
      [
        'class',
        'ui-fileupload-choose'
      ]
      ,
      [
        'icon',
        'fa-plus'
      ]
      ,
      [
        'pButton',
        ''
      ]

    ]
      ,[[
        1,
        'disabled',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-fileupload-choose-selected',
      'ui-state-focus'
    ]
    ),
    import0.ɵprd(256,(null as any),import7.DomHandler,import7.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import4.Button,[
      import0.ElementRef,
      import7.DomHandler
    ]
    ,{
      label: [
        0,
        'label'
      ]
      ,
      icon: [
        1,
        'icon'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),[' \n                    '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'fileinput',
        1
      ]

    ]
      ,(null as any),0,'input',[[
        'type',
        'file'
      ]
    ],[
      [
        8,
        'multiple',
        0
      ]
      ,
      [
        8,
        'accept',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'change'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.onFileSelect($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('focus' === en)) {
        const pd_1:any = ((<any>co.onFocus()) !== false);
        ad = (pd_1 && ad);
      }
      if (('blur' === en)) {
        const pd_2:any = ((<any>co.onBlur()) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FileUpload_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FileUpload_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            \n                '])),
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-templateLoader',([] as any[]),(null as any),(null as any),(null as any),import9.View_TemplateLoader_0,import9.RenderType_TemplateLoader)),
      import0.ɵdid(122880,(null as any),0,import3.TemplateLoader,[import0.ViewContainerRef],{template: [
        0,
        'template'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),15,'div',([] as any[]),(null as any),[
      [
        (null as any),
        'dragenter'
      ]
      ,
      [
        (null as any),
        'dragover'
      ]
      ,
      [
        (null as any),
        'dragleave'
      ]
      ,
      [
        (null as any),
        'drop'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('dragenter' === en)) {
        const pd_0:any = ((<any>co.onDragEnter($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('dragover' === en)) {
        const pd_1:any = ((<any>co.onDragOver($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('dragleave' === en)) {
        const pd_2:any = ((<any>co.onDragLeave($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('drop' === en)) {
        const pd_3:any = ((<any>co.onDrop($event)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-fileupload-content ui-widget-content ui-corner-bottom',
      'ui-fileupload-highlight'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FileUpload_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                \n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-messages',([] as any[]),(null as any),(null as any),(null as any),import10.View_Messages_0,import10.RenderType_Messages)),
      import0.ɵdid(24576,(null as any),0,import6.Messages,([] as any[]),{value: [
        0,
        'value'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                \n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FileUpload_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-templateLoader',([] as any[]),(null as any),(null as any),(null as any),import9.View_TemplateLoader_0,import9.RenderType_TemplateLoader)),
      import0.ɵdid(122880,(null as any),0,import3.TemplateLoader,[import0.ViewContainerRef],{template: [
        0,
        'template'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-fileupload ui-widget';
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,2,0,currVal_2);
    const currVal_4:any = 'ui-fileupload-choose';
    const currVal_5:any = ck(v,8,0,co.hasFiles(),co.focus);
    ck(v,7,0,currVal_4,currVal_5);
    const currVal_6:any = co.chooseLabel;
    const currVal_7:any = 'fa-plus';
    ck(v,10,0,currVal_6,currVal_7);
    const currVal_11:boolean = (!co.auto && co.showUploadButton);
    ck(v,16,0,currVal_11);
    const currVal_12:boolean = (!co.auto && co.showCancelButton);
    ck(v,19,0,currVal_12);
    const currVal_13:any = co.toolbarTemplate;
    ck(v,22,0,currVal_13);
    const currVal_14:any = ck(v,27,0,true,co.dragHighlight);
    ck(v,26,0,currVal_14);
    const currVal_15:any = co.hasFiles();
    ck(v,30,0,currVal_15);
    const currVal_16:any = co.msgs;
    ck(v,33,0,currVal_16);
    const currVal_17:any = co.hasFiles();
    ck(v,36,0,currVal_17);
    const currVal_18:any = co.contentTemplate;
    ck(v,39,0,currVal_18);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = co.disabled;
    ck(v,6,0,currVal_3);
    const currVal_8:any = co.multiple;
    const currVal_9:any = co.accept;
    const currVal_10:any = co.disabled;
    ck(v,12,0,currVal_8,currVal_9,currVal_10);
  });
}
function View_FileUpload_12(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'input',[[
        'type',
        'file'
      ]
    ],[
      [
        8,
        'accept',
        0
      ]
      ,
      [
        8,
        'multiple',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'change'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.onFileSelect($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('focus' === en)) {
        const pd_1:any = ((<any>co.onFocus()) !== false);
        ad = (pd_1 && ad);
      }
      if (('blur' === en)) {
        const pd_2:any = ((<any>co.onBlur()) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
  },(null as any),(null as any)))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.accept;
    const currVal_1:any = co.multiple;
    const currVal_2:any = co.disabled;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
function View_FileUpload_11(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),13,'span',[[
        'class',
        'ui-button ui-fileupload-choose ui-widget ui-state-default ui-corner-all ui-button-text-icon-left'
      ]
      ],(null as any),[[
        (null as any),
        'mouseup'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('mouseup' === en)) {
        const pd_0:any = ((<any>co.onSimpleUploaderClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-fileupload-choose-selected',
      'ui-state-focus'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-button-icon-left fa'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'fa-plus',
      'fa-upload'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-button-text ui-clickable'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FileUpload_12)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-button ui-fileupload-choose ui-widget ui-state-default ui-corner-all ui-button-text-icon-left';
    const currVal_1:any = ck(v,2,0,co.hasFiles(),co.focus);
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = 'ui-button-icon-left fa';
    const currVal_3:any = ck(v,6,0,(!co.hasFiles() || co.auto),(co.hasFiles() && !co.auto));
    ck(v,5,0,currVal_2,currVal_3);
    const currVal_5:boolean = !co.hasFiles();
    ck(v,12,0,currVal_5);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_4:any = (co.auto? co.chooseLabel: (co.hasFiles()? co.files[0].name: co.chooseLabel));
    ck(v,9,0,currVal_4);
  });
}
export function View_FileUpload_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(335544320,1,{fileinput: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FileUpload_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FileUpload_11)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.FileUpload = v.component;
    const currVal_0:any = (co.mode === 'advanced');
    ck(v,3,0,currVal_0);
    const currVal_1:any = (co.mode === 'basic');
    ck(v,6,0,currVal_1);
  },(null as any));
}
function View_FileUpload_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-fileUpload',([] as any[]),(null as any),(null as any),(null as any),View_FileUpload_0,RenderType_FileUpload)),
    import0.ɵdid(581632,(null as any),1,import1.FileUpload,[import11.DomSanitizer],(null as any),(null as any)),
    import0.ɵqud(301989888,1,{templates: 1})
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const FileUploadNgFactory:import0.ComponentFactory<import1.FileUpload> = import0.ɵccf('p-fileUpload',import1.FileUpload,View_FileUpload_Host_0,{
  name: 'name',
  url: 'url',
  method: 'method',
  multiple: 'multiple',
  accept: 'accept',
  disabled: 'disabled',
  auto: 'auto',
  withCredentials: 'withCredentials',
  maxFileSize: 'maxFileSize',
  invalidFileSizeMessageSummary: 'invalidFileSizeMessageSummary',
  invalidFileSizeMessageDetail: 'invalidFileSizeMessageDetail',
  invalidFileTypeMessageSummary: 'invalidFileTypeMessageSummary',
  invalidFileTypeMessageDetail: 'invalidFileTypeMessageDetail',
  style: 'style',
  styleClass: 'styleClass',
  previewWidth: 'previewWidth',
  chooseLabel: 'chooseLabel',
  uploadLabel: 'uploadLabel',
  cancelLabel: 'cancelLabel',
  showUploadButton: 'showUploadButton',
  showCancelButton: 'showCancelButton',
  mode: 'mode',
  customUpload: 'customUpload',
  files: 'files'
}
,{
  onBeforeUpload: 'onBeforeUpload',
  onBeforeSend: 'onBeforeSend',
  onUpload: 'onUpload',
  onError: 'onError',
  onClear: 'onClear',
  onRemove: 'onRemove',
  onSelect: 'onSelect',
  onProgress: 'onProgress',
  uploadHandler: 'uploadHandler'
}
,([] as any[]));
