// primeng/components/tristatecheckbox/tristatecheckbox.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/tristatecheckbox/tristatecheckbox';
import * as import2 from '@angular/common';
import * as import3 from '@angular/forms';
class TriStateCheckboxModuleInjector extends import0.ɵNgModuleInjector<import1.TriStateCheckboxModule> {
  _CommonModule_0:import2.CommonModule;
  _TriStateCheckboxModule_1:import1.TriStateCheckboxModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.TriStateCheckboxModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._TriStateCheckboxModule_1 = new import1.TriStateCheckboxModule();
    return this._TriStateCheckboxModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.TriStateCheckboxModule)) { return this._TriStateCheckboxModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const TriStateCheckboxModuleNgFactory:import0.NgModuleFactory<import1.TriStateCheckboxModule> = new import0.NgModuleFactory<any>(TriStateCheckboxModuleInjector,import1.TriStateCheckboxModule);
const styles_TriStateCheckbox:any[] = ([] as any[]);
export const RenderType_TriStateCheckbox:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_TriStateCheckbox,
  data: {}
}
);
export function View_TriStateCheckbox_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),17,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-helper-hidden-accessible'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,[[
        'input',
        1
      ]
    ],(null as any),0,'input',[
      [
        'readonly',
        ''
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
    ,[
      [
        1,
        'id',
        0
      ]
      ,
      [
        8,
        'name',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'keyup'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.TriStateCheckbox = v.component;
      if (('keyup' === en)) {
        const pd_0:any = ((<any>co.onKeyup($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('keydown' === en)) {
        const pd_1:any = ((<any>co.onKeydown($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('focus' === en)) {
        const pd_2:any = ((<any>co.onFocus()) !== false);
        ad = (pd_2 && ad);
      }
      if (('blur' === en)) {
        const pd_3:any = ((<any>co.onBlur()) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'ui-chkbox-box ui-widget ui-corner-all ui-state-default'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.TriStateCheckbox = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onClick($event,import0.ɵnov(v,7))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-state-active',
      'ui-state-disabled',
      'ui-state-focus'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-chkbox-icon fa ui-clickable'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'fa-check',
      'fa-close'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.TriStateCheckbox = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-chkbox ui-tristatechkbox ui-widget';
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_7:any = 'ui-chkbox-box ui-widget ui-corner-all ui-state-default';
    const currVal_8:any = ck(v,12,0,(co.value != (null as any)),co.disabled,co.focus);
    ck(v,11,0,currVal_7,currVal_8);
    const currVal_9:any = 'ui-chkbox-icon fa ui-clickable';
    const currVal_10:any = ck(v,16,0,(co.value == true),(co.value == false));
    ck(v,15,0,currVal_9,currVal_10);
  },(ck,v) => {
    var co:import1.TriStateCheckbox = v.component;
    const currVal_3:any = co.inputId;
    const currVal_4:any = co.name;
    const currVal_5:any = co.tabindex;
    const currVal_6:any = co.disabled;
    ck(v,7,0,currVal_3,currVal_4,currVal_5,currVal_6);
  });
}
function View_TriStateCheckbox_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-triStateCheckbox',([] as any[]),(null as any),(null as any),(null as any),View_TriStateCheckbox_0,RenderType_TriStateCheckbox)),
    import0.ɵdid(24576,(null as any),0,import1.TriStateCheckbox,[import0.ChangeDetectorRef],(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import3.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.TriStateCheckbox])
  ]
  ,(null as any),(null as any));
}
export const TriStateCheckboxNgFactory:import0.ComponentFactory<import1.TriStateCheckbox> = import0.ɵccf('p-triStateCheckbox',import1.TriStateCheckbox,View_TriStateCheckbox_Host_0,{
  disabled: 'disabled',
  name: 'name',
  tabindex: 'tabindex',
  inputId: 'inputId',
  style: 'style',
  styleClass: 'styleClass'
}
,{onChange: 'onChange'},([] as any[]));
