// primeng/components/togglebutton/togglebutton.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/togglebutton/togglebutton';
import * as import2 from '@angular/common';
import * as import3 from '@angular/forms';
class ToggleButtonModuleInjector extends import0.ɵNgModuleInjector<import1.ToggleButtonModule> {
  _CommonModule_0:import2.CommonModule;
  _ToggleButtonModule_1:import1.ToggleButtonModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.ToggleButtonModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ToggleButtonModule_1 = new import1.ToggleButtonModule();
    return this._ToggleButtonModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.ToggleButtonModule)) { return this._ToggleButtonModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const ToggleButtonModuleNgFactory:import0.NgModuleFactory<import1.ToggleButtonModule> = new import0.NgModuleFactory<any>(ToggleButtonModuleInjector,import1.ToggleButtonModule);
const styles_ToggleButton:any[] = ([] as any[]);
export const RenderType_ToggleButton:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ToggleButton,
  data: {}
}
);
function View_ToggleButton_1(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',([] as any[]),[[
        8,
        'className',
        0
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.getIconClass();
    ck(v,0,0,currVal_0);
  });
}
export function View_ToggleButton_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{checkboxViewChild: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),15,'div',([] as any[]),(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.ToggleButton = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggle($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-button ui-togglebutton ui-widget ui-state-default ui-corner-all',
      'ui-button-text-only',
      'ui-button-text-icon-left',
      'ui-button-icon-only',
      'ui-state-active',
      'ui-state-focus',
      'ui-state-disabled'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-helper-hidden-accessible'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'checkbox',
        1
      ]

    ]
      ,(null as any),0,'input',[[
        'type',
        'checkbox'
      ]
    ],[
      [
        1,
        'id',
        0
      ]
      ,
      [
        8,
        'checked',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.ToggleButton = v.component;
      if (('focus' === en)) {
        const pd_0:any = ((<any>co.onFocus()) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>co.onBlur()) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ToggleButton_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-button-text ui-unselectable-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.ToggleButton = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,4,0,true,(!co.onIcon && !co.offIcon),(((co.onIcon && co.offIcon) && co.hasOnLabel) && co.hasOffLabel),(((co.onIcon && co.offIcon) && !co.hasOnLabel) && !co.hasOffLabel),co.checked,co.focus,co.disabled);
    ck(v,3,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,5,0,currVal_2);
    const currVal_6:any = (co.onIcon || co.offIcon);
    ck(v,13,0,currVal_6);
  },(ck,v) => {
    var co:import1.ToggleButton = v.component;
    const currVal_3:any = co.inputId;
    const currVal_4:any = co.checked;
    const currVal_5:any = co.tabindex;
    ck(v,9,0,currVal_3,currVal_4,currVal_5);
    const currVal_7:any = (co.checked? (co.hasOnLabel? co.onLabel: 'ui-btn'): (co.hasOffLabel? co.offLabel: 'ui-btn'));
    ck(v,16,0,currVal_7);
  });
}
function View_ToggleButton_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-toggleButton',([] as any[]),(null as any),(null as any),(null as any),View_ToggleButton_0,RenderType_ToggleButton)),
    import0.ɵdid(2121728,(null as any),0,import1.ToggleButton,([] as any[]),(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import3.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.ToggleButton])
  ]
  ,(null as any),(null as any));
}
export const ToggleButtonNgFactory:import0.ComponentFactory<import1.ToggleButton> = import0.ɵccf('p-toggleButton',import1.ToggleButton,View_ToggleButton_Host_0,{
  onLabel: 'onLabel',
  offLabel: 'offLabel',
  onIcon: 'onIcon',
  offIcon: 'offIcon',
  disabled: 'disabled',
  style: 'style',
  styleClass: 'styleClass',
  inputId: 'inputId',
  tabindex: 'tabindex'
}
,{onChange: 'onChange'},([] as any[]));
