// @swimlane/ngx-datatable/release/components/footer/footer.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from './pager.component.ngfactory';
import * as import3 from '@swimlane/ngx-datatable/release/components/footer/pager.component';
import * as import4 from '@swimlane/ngx-datatable/release/components/footer/footer.component';
const styles_DataTableFooterComponent:any[] = ([] as any[]);
export const RenderType_DataTableFooterComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DataTableFooterComponent,
  data: {}
}
);
function View_DataTableFooterComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n      ']))],(null as any),(null as any));
}
function View_DataTableFooterComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_DataTableFooterComponent_2)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'rowCount',
      'pageSize',
      'selectedCount',
      'curPage',
      'offset'
    ]
    ),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.footerTemplate.template;
    const currVal_1:any = ck(v,2,0,co.rowCount,co.pageSize,co.selectedCount,co.curPage,co.offset);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_DataTableFooterComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n          ',
      ' ',
      ' / \n        '
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.selectedCount.toLocaleString();
    const currVal_1:any = co.selectedMessage;
    ck(v,1,0,currVal_0,currVal_1);
  });
}
function View_DataTableFooterComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'page-count'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataTableFooterComponent_4)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[
      '\n        ',
      ' ',
      '\n      '
    ]
    ))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.selectedMessage;
    ck(v,3,0,currVal_0);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.rowCount.toLocaleString();
    const currVal_2:any = co.totalMessage;
    ck(v,4,0,currVal_1,currVal_2);
  });
}
function View_DataTableFooterComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'datatable-pager',[[
        'class',
        'datatable-pager'
      ]
      ],[[
        8,
        'hidden',
        0
      ]
      ],[[
        (null as any),
        'change'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.page.emit($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import2.View_DataTablePagerComponent_0,import2.RenderType_DataTablePagerComponent)),
    import0.ɵdid(24576,(null as any),0,import3.DataTablePagerComponent,([] as any[]),{
      pagerLeftArrowIcon: [
        0,
        'pagerLeftArrowIcon'
      ]
      ,
      pagerRightArrowIcon: [
        1,
        'pagerRightArrowIcon'
      ]
      ,
      pagerPreviousIcon: [
        2,
        'pagerPreviousIcon'
      ]
      ,
      pagerNextIcon: [
        3,
        'pagerNextIcon'
      ]
      ,
      size: [
        4,
        'size'
      ]
      ,
      count: [
        5,
        'count'
      ]
      ,
      page: [
        6,
        'page'
      ]

    }
    ,{change: 'change'}),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.pagerLeftArrowIcon;
    const currVal_2:any = co.pagerRightArrowIcon;
    const currVal_3:any = co.pagerPreviousIcon;
    const currVal_4:any = co.pagerNextIcon;
    const currVal_5:any = co.pageSize;
    const currVal_6:any = co.rowCount;
    const currVal_7:any = co.curPage;
    ck(v,1,0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:boolean = !co.isVisible;
    ck(v,0,0,currVal_0);
  });
}
export function View_DataTableFooterComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),12,'div',[[
        'class',
        'datatable-footer-inner'
      ]
      ],[[
        4,
        'height',
        'px'
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['selected-count']),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataTableFooterComponent_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataTableFooterComponent_3)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataTableFooterComponent_5)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import4.DataTableFooterComponent = v.component;
    const currVal_1:any = 'datatable-footer-inner';
    const currVal_2:any = ck(v,3,0,co.selectedMessage);
    ck(v,2,0,currVal_1,currVal_2);
    const currVal_3:any = co.footerTemplate;
    ck(v,6,0,currVal_3);
    const currVal_4:boolean = !co.footerTemplate;
    ck(v,9,0,currVal_4);
    const currVal_5:boolean = !co.footerTemplate;
    ck(v,12,0,currVal_5);
  },(ck,v) => {
    var co:import4.DataTableFooterComponent = v.component;
    const currVal_0:any = co.footerHeight;
    ck(v,1,0,currVal_0);
  });
}
function View_DataTableFooterComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'datatable-footer',[[
        'class',
        'datatable-footer'
      ]
    ],(null as any),(null as any),(null as any),View_DataTableFooterComponent_0,RenderType_DataTableFooterComponent)),
    import0.ɵdid(24576,(null as any),0,import4.DataTableFooterComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const DataTableFooterComponentNgFactory:import0.ComponentFactory<import4.DataTableFooterComponent> = import0.ɵccf('datatable-footer',import4.DataTableFooterComponent,View_DataTableFooterComponent_Host_0,{
  footerHeight: 'footerHeight',
  rowCount: 'rowCount',
  pageSize: 'pageSize',
  offset: 'offset',
  pagerLeftArrowIcon: 'pagerLeftArrowIcon',
  pagerRightArrowIcon: 'pagerRightArrowIcon',
  pagerPreviousIcon: 'pagerPreviousIcon',
  pagerNextIcon: 'pagerNextIcon',
  totalMessage: 'totalMessage',
  footerTemplate: 'footerTemplate',
  selectedCount: 'selectedCount',
  selectedMessage: 'selectedMessage'
}
,{page: 'page'},([] as any[]));
