// @angular/cdk/table/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/cdk/table';
import * as import2 from '@angular/common';
class CdkTableModuleInjector extends import0.ɵNgModuleInjector<import1.CdkTableModule> {
  _CommonModule_0:import2.CommonModule;
  _CdkTableModule_1:import1.CdkTableModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.CdkTableModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._CdkTableModule_1 = new import1.CdkTableModule();
    return this._CdkTableModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.CdkTableModule)) { return this._CdkTableModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const CdkTableModuleNgFactory:import0.NgModuleFactory<import1.CdkTableModule> = new import0.NgModuleFactory<any>(CdkTableModuleInjector,import1.CdkTableModule);
const styles_CdkTable:any[] = ([] as any[]);
export const RenderType_CdkTable:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_CdkTable,
  data: {}
}
);
export function View_CdkTable_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    import0.ɵqud(201326592,1,{_rowPlaceholder: 0}),
    import0.ɵqud(201326592,2,{_headerRowPlaceholder: 0}),
    (l()(),import0.ɵted((null as any),['\n  '])),
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      import0.ɵdid(8192,[[
        2,
        4
      ]
    ],0,import1.HeaderRowPlaceholder,[import0.ViewContainerRef],(null as any),(null as any)),
    (l()(),import0.ɵted((null as any),['\n  '])),
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      import0.ɵdid(8192,[[
        1,
        4
      ]
    ],0,import1.RowPlaceholder,[import0.ViewContainerRef],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
function View_CdkTable_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'cdk-table',[[
        'class',
        'cdk-table'
      ]
    ],(null as any),(null as any),(null as any),View_CdkTable_0,RenderType_CdkTable)),
    import0.ɵdid(1695744,(null as any),3,import1.CdkTable,[
      import0.IterableDiffers,
      import0.ChangeDetectorRef,
      import0.ElementRef,
      import0.Renderer2,
      [
        8,
        (null as any)
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵqud(301989888,1,{_columnDefinitions: 1}),
    import0.ɵqud(167772160,2,{_headerDefinition: 0}),
    import0.ɵqud(301989888,3,{_rowDefinitions: 1})
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const CdkTableNgFactory:import0.ComponentFactory<import1.CdkTable<any>> = import0.ɵccf('cdk-table',import1.CdkTable,View_CdkTable_Host_0,{
  trackBy: 'trackBy',
  dataSource: 'dataSource'
}
,{},([] as any[]));
const styles_CdkRow:any[] = ([] as any[]);
export const RenderType_CdkRow:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_CdkRow,
  data: {}
}
);
export function View_CdkRow_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(8192,(null as any),0,import1.CdkCellOutlet,[import0.ViewContainerRef],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
function View_CdkRow_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'cdk-row',[
      [
        'class',
        'cdk-row'
      ]
      ,
      [
        'role',
        'row'
      ]

    ]
    ,(null as any),(null as any),(null as any),View_CdkRow_0,RenderType_CdkRow)),
    import0.ɵdid(24576,(null as any),0,import1.CdkRow,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const CdkRowNgFactory:import0.ComponentFactory<import1.CdkRow> = import0.ɵccf('cdk-row',import1.CdkRow,View_CdkRow_Host_0,{},{},([] as any[]));
const styles_CdkHeaderRow:any[] = ([] as any[]);
export const RenderType_CdkHeaderRow:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_CdkHeaderRow,
  data: {}
}
);
export function View_CdkHeaderRow_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(8192,(null as any),0,import1.CdkCellOutlet,[import0.ViewContainerRef],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
function View_CdkHeaderRow_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'cdk-header-row',[
      [
        'class',
        'cdk-header-row'
      ]
      ,
      [
        'role',
        'row'
      ]

    ]
    ,(null as any),(null as any),(null as any),View_CdkHeaderRow_0,RenderType_CdkHeaderRow)),
    import0.ɵdid(24576,(null as any),0,import1.CdkHeaderRow,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const CdkHeaderRowNgFactory:import0.ComponentFactory<import1.CdkHeaderRow> = import0.ɵccf('cdk-header-row',import1.CdkHeaderRow,View_CdkHeaderRow_Host_0,{},{},([] as any[]));
