// primeng/components/chart/chart.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/chart/chart';
import * as import2 from '@angular/common';
class ChartModuleInjector extends import0.ɵNgModuleInjector<import1.ChartModule> {
  _CommonModule_0:import2.CommonModule;
  _ChartModule_1:import1.ChartModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.ChartModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ChartModule_1 = new import1.ChartModule();
    return this._ChartModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.ChartModule)) { return this._ChartModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const ChartModuleNgFactory:import0.NgModuleFactory<import1.ChartModule> = new import0.NgModuleFactory<any>(ChartModuleInjector,import1.ChartModule);
const styles_UIChart:any[] = ([] as any[]);
export const RenderType_UIChart:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_UIChart,
  data: {}
}
);
export function View_UIChart_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'canvas',([] as any[]),[
      [
        1,
        'width',
        0
      ]
      ,
      [
        1,
        'height',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.UIChart = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onCanvasClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(null as any),(ck,v) => {
    var co:import1.UIChart = v.component;
    const currVal_0:any = co.width;
    const currVal_1:any = co.height;
    ck(v,3,0,currVal_0,currVal_1);
  });
}
function View_UIChart_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-chart',([] as any[]),(null as any),(null as any),(null as any),View_UIChart_0,RenderType_UIChart)),
    import0.ɵdid(2187264,(null as any),0,import1.UIChart,[import0.ElementRef],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const UIChartNgFactory:import0.ComponentFactory<import1.UIChart> = import0.ɵccf('p-chart',import1.UIChart,View_UIChart_Host_0,{
  type: 'type',
  options: 'options',
  width: 'width',
  height: 'height',
  data: 'data'
}
,{onDataSelect: 'onDataSelect'},([] as any[]));
