// @ng-bootstrap/ng-bootstrap/typeahead/typeahead-window.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './highlight.ngfactory';
import * as import2 from '@ng-bootstrap/ng-bootstrap/typeahead/highlight';
import * as import3 from '@angular/common';
import * as import4 from '@ng-bootstrap/ng-bootstrap/typeahead/typeahead-window';
const styles_NgbTypeaheadWindow:any[] = ([] as any[]);
export const RenderType_NgbTypeaheadWindow:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NgbTypeaheadWindow,
  data: {}
}
);
function View_NgbTypeaheadWindow_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ngb-highlight',([] as any[]),(null as any),(null as any),(null as any),import1.View_NgbHighlight_0,import1.RenderType_NgbHighlight)),
    import0.ɵdid(286720,(null as any),0,import2.NgbHighlight,([] as any[]),{
      result: [
        0,
        'result'
      ]
      ,
      term: [
        1,
        'term'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.formatter(v.context.result);
    const currVal_1:any = v.context.term;
    ck(v,2,0,currVal_0,currVal_1);
  },(null as any));
}
function View_NgbTypeaheadWindow_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_NgbTypeaheadWindow_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'button',[
      [
        'class',
        'dropdown-item'
      ]
      ,
      [
        'role',
        'option'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,[
      [
        8,
        'id',
        0
      ]
      ,
      [
        2,
        'active',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'mouseenter'
      ]
      ,
      [
        (null as any),
        'click'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('mouseenter' === en)) {
        const pd_0:any = ((<any>co.markActive(v.context.index)) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>co.select(v.context.$implicit)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_NgbTypeaheadWindow_3)),
    import0.ɵdid(270336,(null as any),0,import3.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutletContext: [
        0,
        'ngTemplateOutletContext'
      ]
      ,
      ngTemplateOutlet: [
        1,
        'ngTemplateOutlet'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'result',
      'term',
      'formatter'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = ck(v,5,0,v.context.$implicit,co.term,co.formatter);
    const currVal_3:any = (co.resultTemplate || import0.ɵnov((<any>v.parent),1));
    ck(v,4,0,currVal_2,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ((co.id + '-') + v.context.index);
    const currVal_1:any = (v.context.index === co.activeIdx);
    ck(v,1,0,currVal_0,currVal_1);
  });
}
export function View_NgbTypeaheadWindow_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵand(0,[[
        'rt',
        2
      ]
    ],(null as any),0,(null as any),View_NgbTypeaheadWindow_1)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTypeaheadWindow_2)),
    import0.ɵdid(401408,(null as any),0,import3.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import4.NgbTypeaheadWindow = v.component;
    const currVal_0:any = co.results;
    ck(v,4,0,currVal_0);
  },(null as any));
}
function View_NgbTypeaheadWindow_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ngb-typeahead-window',[
      [
        'class',
        'dropdown-menu'
      ]
      ,
      [
        'role',
        'listbox'
      ]
      ,
      [
        'style',
        'display: block'
      ]

    ]
      ,[[
        8,
        'id',
        0
      ]
    ],(null as any),(null as any),View_NgbTypeaheadWindow_0,RenderType_NgbTypeaheadWindow)),
    import0.ɵdid(57344,(null as any),0,import4.NgbTypeaheadWindow,([] as any[]),(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).id;
    ck(v,0,0,currVal_0);
  });
}
export const NgbTypeaheadWindowNgFactory:import0.ComponentFactory<import4.NgbTypeaheadWindow> = import0.ɵccf('ngb-typeahead-window',import4.NgbTypeaheadWindow,View_NgbTypeaheadWindow_Host_0,{
  id: 'id',
  focusFirst: 'focusFirst',
  results: 'results',
  term: 'term',
  formatter: 'formatter',
  resultTemplate: 'resultTemplate'
}
,{
  selectEvent: 'select',
  activeChangeEvent: 'activeChange'
}
,([] as any[]));
