// primeng/components/growl/growl.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/growl/growl';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/dom/domhandler';
class GrowlModuleInjector extends import0.ɵNgModuleInjector<import1.GrowlModule> {
  _CommonModule_0:import2.CommonModule;
  _GrowlModule_1:import1.GrowlModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.GrowlModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._GrowlModule_1 = new import1.GrowlModule();
    return this._GrowlModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.GrowlModule)) { return this._GrowlModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const GrowlModuleNgFactory:import0.NgModuleFactory<import1.GrowlModule> = new import0.NgModuleFactory<any>(GrowlModuleInjector,import1.GrowlModule);
const styles_Growl:any[] = ([] as any[]);
export const RenderType_Growl:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Growl,
  data: {}
}
);
function View_Growl_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,[[
        'msgel',
        1
      ]
    ],(null as any),22,'div',[
      [
        'aria-live',
        'polite'
      ]
      ,
      [
        'class',
        'ui-growl-item-container ui-state-highlight ui-corner-all ui-shadow'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onMessageClick(v.context.index)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-growl-message-info',
      'ui-growl-message-warn',
      'ui-growl-message-error',
      'ui-growl-message-success'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),17,'div',[[
        'class',
        'ui-growl-item'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                     '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'ui-growl-icon-close fa fa-close'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.remove(v.context.index,import0.ɵnov(v,0))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                     '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-growl-image fa fa-2x'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'fa-info-circle',
      'fa-exclamation-circle',
      'fa-close',
      'fa-check'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                     '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',[[
        'class',
        'ui-growl-message'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-growl-title'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'p',([] as any[]),[[
        8,
        'innerHTML',
        1
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                     '])),
    (l()(),import0.ɵted((null as any),['\n                     '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'style',
        'clear: both;'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-growl-item-container ui-state-highlight ui-corner-all ui-shadow';
    const currVal_1:any = ck(v,2,0,(v.context.$implicit.severity == 'info'),(v.context.$implicit.severity == 'warn'),(v.context.$implicit.severity == 'error'),(v.context.$implicit.severity == 'success'));
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = 'ui-growl-image fa fa-2x';
    const currVal_3:any = ck(v,10,0,(v.context.$implicit.severity == 'info'),(v.context.$implicit.severity == 'warn'),(v.context.$implicit.severity == 'error'),(v.context.$implicit.severity == 'success'));
    ck(v,9,0,currVal_2,currVal_3);
  },(ck,v) => {
    const currVal_4:any = v.context.$implicit.summary;
    ck(v,15,0,currVal_4);
    const currVal_5:any = v.context.$implicit.detail;
    ck(v,17,0,currVal_5);
  });
}
export function View_Growl_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{containerViewChild: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'container',
        1
      ]

    ]
      ,(null as any),6,'div',([] as any[]),[[
        4,
        'zIndex',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Growl_1)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Growl = v.component;
    const currVal_1:any = co.styleClass;
    const currVal_2:any = 'ui-growl ui-widget';
    ck(v,3,0,currVal_1,currVal_2);
    const currVal_3:any = co.style;
    ck(v,4,0,currVal_3);
    const currVal_4:any = co.value;
    ck(v,7,0,currVal_4);
  },(ck,v) => {
    var co:import1.Growl = v.component;
    const currVal_0:any = co.zIndex;
    ck(v,2,0,currVal_0);
  });
}
function View_Growl_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-growl',([] as any[]),(null as any),(null as any),(null as any),View_Growl_0,RenderType_Growl)),
    import0.ɵprd(256,(null as any),import3.DomHandler,import3.DomHandler,([] as any[])),
    import0.ɵdid(2318336,(null as any),0,import1.Growl,[
      import0.ElementRef,
      import3.DomHandler,
      import0.IterableDiffers
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,2,0);
  },(null as any));
}
export const GrowlNgFactory:import0.ComponentFactory<import1.Growl> = import0.ɵccf('p-growl',import1.Growl,View_Growl_Host_0,{
  sticky: 'sticky',
  life: 'life',
  style: 'style',
  styleClass: 'styleClass',
  immutable: 'immutable',
  value: 'value'
}
,{
  onClick: 'onClick',
  onClose: 'onClose',
  valueChange: 'valueChange'
}
,([] as any[]));
