// primeng/components/fieldset/fieldset.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/fieldset/fieldset';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/common/shared';
class FieldsetModuleInjector extends import0.ɵNgModuleInjector<import1.FieldsetModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _FieldsetModule_2:import1.FieldsetModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.FieldsetModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._FieldsetModule_2 = new import1.FieldsetModule();
    return this._FieldsetModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import1.FieldsetModule)) { return this._FieldsetModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const FieldsetModuleNgFactory:import0.NgModuleFactory<import1.FieldsetModule> = new import0.NgModuleFactory<any>(FieldsetModuleInjector,import1.FieldsetModule);
const styles_Fieldset:any[] = ([] as any[]);
export const RenderType_Fieldset:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Fieldset,
      data: {'animation': [{
        name: 'fieldsetContent',
        definitions: [
          {
            type: 0,
            name: 'hidden',
            styles: {
              type: 6,
              styles: {height: '0px'}
            }

          }
          ,
          {
            type: 0,
            name: 'visible',
            styles: {
              type: 6,
              styles: {height: '*'}
            }

          }
          ,
          {
            type: 1,
            expr: 'visible => hidden',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms cubic-bezier(0.86, 0, 0.07, 1)'
            }

          }
          ,
          {
            type: 1,
            expr: 'hidden => visible',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms cubic-bezier(0.86, 0, 0.07, 1)'
            }

          }

        ]

      }
  ]}
}
);
function View_Fieldset_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-fieldset-toggler fa fa-w'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'fa-minus',
      'fa-plus'
    ]
    )
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-fieldset-toggler fa fa-w';
    const currVal_1:any = ck(v,2,0,!co.collapsed,co.collapsed);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
export function View_Fieldset_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),22,'fieldset',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-fieldset ui-widget ui-widget-content ui-corner-all',
      'ui-fieldset-toggleable'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'legend',[[
        'class',
        'ui-fieldset-legend ui-corner-all ui-state-default ui-unselectable-text'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Fieldset = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggle($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Fieldset_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[
      '\n                ',
      '\n                '
    ]
    )),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),8,'div',[[
        'class',
        'ui-fieldset-content-wrapper'
      ]
      ],[[
        24,
        '@fieldsetContent',
        0
      ]
      ],[[
        (null as any),
        '@fieldsetContent.done'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Fieldset = v.component;
      if (('@fieldsetContent.done' === en)) {
        const pd_0:any = ((<any>co.onToggleDone($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-fieldset-content-wrapper-overflown']),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-fieldset-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    import0.ɵncd((null as any),1),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Fieldset = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,3,0,true,co.toggleable);
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,4,0,currVal_2);
    const currVal_3:any = co.toggleable;
    ck(v,9,0,currVal_3);
    const currVal_6:any = 'ui-fieldset-content-wrapper';
    const currVal_7:any = ck(v,16,0,(co.collapsed || co.animating));
    ck(v,15,0,currVal_6,currVal_7);
  },(ck,v) => {
    var co:import1.Fieldset = v.component;
    const currVal_4:any = co.legend;
    ck(v,10,0,currVal_4);
    const currVal_5:any = (co.collapsed? 'hidden': 'visible');
    ck(v,14,0,currVal_5);
  });
}
function View_Fieldset_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-fieldset',([] as any[]),(null as any),(null as any),(null as any),View_Fieldset_0,RenderType_Fieldset)),
    import0.ɵdid(24576,(null as any),0,import1.Fieldset,[import0.ElementRef],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const FieldsetNgFactory:import0.ComponentFactory<import1.Fieldset> = import0.ɵccf('p-fieldset',import1.Fieldset,View_Fieldset_Host_0,{
  legend: 'legend',
  toggleable: 'toggleable',
  collapsed: 'collapsed',
  style: 'style',
  styleClass: 'styleClass'
}
,{
  onBeforeToggle: 'onBeforeToggle',
  onAfterToggle: 'onAfterToggle'
}
,[
  'p-header',
  '*'
]
);
