// primeng/components/autocomplete/autocomplete.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/autocomplete/autocomplete';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/inputtext/inputtext';
import * as import4 from 'primeng/components/button/button';
import * as import5 from 'primeng/components/common/shared';
import * as import6 from 'primeng/components/dom/domhandler';
import * as import7 from 'primeng/components/utils/objectutils';
import * as import8 from '@angular/forms';
class AutoCompleteModuleInjector extends import0.ɵNgModuleInjector<import1.AutoCompleteModule> {
  _CommonModule_0:import2.CommonModule;
  _InputTextModule_1:import3.InputTextModule;
  _ButtonModule_2:import4.ButtonModule;
  _SharedModule_3:import5.SharedModule;
  _AutoCompleteModule_4:import1.AutoCompleteModule;
  __NgLocalization_5:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_5():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_5 == null)) { (this.__NgLocalization_5 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_5;
  }
  createInternal():import1.AutoCompleteModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._InputTextModule_1 = new import3.InputTextModule();
    this._ButtonModule_2 = new import4.ButtonModule();
    this._SharedModule_3 = new import5.SharedModule();
    this._AutoCompleteModule_4 = new import1.AutoCompleteModule();
    return this._AutoCompleteModule_4;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.InputTextModule)) { return this._InputTextModule_1; }
    if ((token === import4.ButtonModule)) { return this._ButtonModule_2; }
    if ((token === import5.SharedModule)) { return this._SharedModule_3; }
    if ((token === import1.AutoCompleteModule)) { return this._AutoCompleteModule_4; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_5; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const AutoCompleteModuleNgFactory:import0.NgModuleFactory<import1.AutoCompleteModule> = new import0.NgModuleFactory<any>(AutoCompleteModuleInjector,import1.AutoCompleteModule);
const styles_AutoComplete:any[] = ([] as any[]);
export const RenderType_AutoComplete:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_AutoComplete,
  data: {}
}
);
function View_AutoComplete_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'in',
        1
      ]

    ]
      ,(null as any),2,'input',[[
        'autocomplete',
        'off'
      ]
    ],[
      [
        1,
        'type',
        0
      ]
      ,
      [
        1,
        'id',
        0
      ]
      ,
      [
        8,
        'value',
        0
      ]
      ,
      [
        1,
        'placeholder',
        0
      ]
      ,
      [
        1,
        'size',
        0
      ]
      ,
      [
        1,
        'maxlength',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        8,
        'readOnly',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]
      ,
      [
        (null as any),
        'keyup'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onInputClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('input' === en)) {
        const pd_1:any = ((<any>co.onInput($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('keydown' === en)) {
        const pd_2:any = ((<any>co.onKeydown($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('keyup' === en)) {
        const pd_3:any = ((<any>co.onKeyup($event)) !== false);
        ad = (pd_3 && ad);
      }
      if (('focus' === en)) {
        const pd_4:any = ((<any>co.onInputFocus($event)) !== false);
        ad = (pd_4 && ad);
      }
      if (('blur' === en)) {
        const pd_5:any = ((<any>co.onInputBlur($event)) !== false);
        ad = (pd_5 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_9:any = co.inputStyleClass;
    const currVal_10:any = 'ui-inputtext ui-widget ui-state-default ui-corner-all ui-autocomplete-input';
    ck(v,1,0,currVal_9,currVal_10);
    const currVal_11:any = co.inputStyle;
    ck(v,2,0,currVal_11);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.type;
    const currVal_1:any = co.inputId;
    const currVal_2:any = (co.value? (co.field? (co.objectUtils.resolveFieldData(co.value,co.field) || ''): co.value): (null as any));
    const currVal_3:any = co.placeholder;
    const currVal_4:any = co.size;
    const currVal_5:any = co.maxlength;
    const currVal_6:any = co.tabindex;
    const currVal_7:any = co.readonly;
    const currVal_8:any = co.disabled;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8);
  });
}
function View_AutoComplete_4(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'ui-autocomplete-token-icon fa fa-fw fa-close'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.removeItem(import0.ɵnov((<any>v.parent),0))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
  },(null as any),(null as any)))],(null as any),(null as any));
}
function View_AutoComplete_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-autocomplete-token-label'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.field? (<any>v.parent).context.$implicit[co.field]: (<any>v.parent).context.$implicit);
    ck(v,1,0,currVal_0);
  });
}
function View_AutoComplete_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_AutoComplete_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoComplete_7)),
    import0.ɵdid(106496,(null as any),0,import5.TemplateWrapper,[import0.ViewContainerRef],{
      item: [
        0,
        'item'
      ]
      ,
      templateRef: [
        1,
        'templateRef'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (<any>v.parent).context.$implicit;
    const currVal_1:any = co.selectedItemTemplate;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_AutoComplete_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,[[
        'token',
        1
      ]
      ],(null as any),10,'li',[[
        'class',
        'ui-autocomplete-token ui-state-highlight ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoComplete_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoComplete_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoComplete_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:boolean = !co.disabled;
    ck(v,3,0,currVal_0);
    const currVal_1:boolean = !co.selectedItemTemplate;
    ck(v,6,0,currVal_1);
    const currVal_2:any = co.selectedItemTemplate;
    ck(v,9,0,currVal_2);
  },(null as any));
}
function View_AutoComplete_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,[
      [
        4,
        0
      ]
      ,
      [
        'multiContainer',
        1
      ]

    ]
      ,(null as any),12,'ul',[[
        'class',
        'ui-autocomplete-multiple-container ui-widget ui-inputtext ui-state-default ui-corner-all'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,9).focus()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-state-disabled',
      'ui-state-focus'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoComplete_3)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'li',[[
        'class',
        'ui-autocomplete-input-token'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'multiIn',
        1
      ]

    ]
      ,(null as any),1,'input',[[
        'autocomplete',
        'off'
      ]
    ],[
      [
        1,
        'type',
        0
      ]
      ,
      [
        1,
        'id',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        1,
        'placeholder',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        8,
        'className',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]
      ,
      [
        (null as any),
        'keyup'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('input' === en)) {
        const pd_0:any = ((<any>co.onInput($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>co.onInputClick($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('keydown' === en)) {
        const pd_2:any = ((<any>co.onKeydown($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('keyup' === en)) {
        const pd_3:any = ((<any>co.onKeyup($event)) !== false);
        ad = (pd_3 && ad);
      }
      if (('focus' === en)) {
        const pd_4:any = ((<any>co.onInputFocus($event)) !== false);
        ad = (pd_4 && ad);
      }
      if (('blur' === en)) {
        const pd_5:any = ((<any>co.onInputBlur($event)) !== false);
        ad = (pd_5 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-autocomplete-multiple-container ui-widget ui-inputtext ui-state-default ui-corner-all';
    const currVal_1:any = ck(v,2,0,co.disabled,co.focus);
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = co.value;
    ck(v,5,0,currVal_2);
    const currVal_9:any = co.inputStyle;
    ck(v,10,0,currVal_9);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = co.type;
    const currVal_4:any = co.inputId;
    const currVal_5:any = co.disabled;
    const currVal_6:any = ((co.value && co.value.length)? (null as any): co.placeholder);
    const currVal_7:any = co.tabindex;
    const currVal_8:any = co.inputStyleClass;
    ck(v,9,0,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8);
  });
}
function View_AutoComplete_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'class',
        'ui-autocomplete-dropdown'
      ]
      ,
      [
        'icon',
        'fa-fw fa-caret-down'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,[[
        8,
        'disabled',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.handleDropdownClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import6.DomHandler,import6.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import4.Button,[
      import0.ElementRef,
      import6.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    const currVal_1:any = 'fa-fw fa-caret-down';
    ck(v,2,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.disabled;
    ck(v,0,0,currVal_0);
  });
}
function View_AutoComplete_11(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.field? (<any>v.parent).context.$implicit[co.field]: (<any>v.parent).context.$implicit);
    ck(v,1,0,currVal_0);
  });
}
function View_AutoComplete_13(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_AutoComplete_12(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoComplete_13)),
    import0.ɵdid(106496,(null as any),0,import5.TemplateWrapper,[import0.ViewContainerRef],{
      item: [
        0,
        'item'
      ]
      ,
      index: [
        1,
        'index'
      ]
      ,
      templateRef: [
        2,
        'templateRef'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (<any>v.parent).context.$implicit;
    const currVal_1:any = (<any>v.parent).context.index;
    const currVal_2:any = co.itemTemplate;
    ck(v,1,0,currVal_0,currVal_1,currVal_2);
  },(null as any));
}
function View_AutoComplete_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),9,'li',([] as any[]),(null as any),[
      [
        (null as any),
        'mouseenter'
      ]
      ,
      [
        (null as any),
        'mouseleave'
      ]
      ,
      [
        (null as any),
        'click'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('mouseenter' === en)) {
        const pd_0:any = ((<any>(co.highlightOption = v.context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mouseleave' === en)) {
        const pd_1:any = ((<any>(co.highlightOption = (null as any))) !== false);
        ad = (pd_1 && ad);
      }
      if (('click' === en)) {
        const pd_2:any = ((<any>co.selectItem(v.context.$implicit)) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-autocomplete-list-item ui-corner-all',
      'ui-state-highlight'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoComplete_11)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoComplete_12)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,true,(co.highlightOption == v.context.$implicit));
    ck(v,1,0,currVal_0);
    const currVal_1:boolean = !co.itemTemplate;
    ck(v,5,0,currVal_1);
    const currVal_2:any = co.itemTemplate;
    ck(v,8,0,currVal_2);
  },(null as any));
}
function View_AutoComplete_14(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'li',[[
        'class',
        'ui-autocomplete-list-item ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.emptyMessage;
    ck(v,1,0,currVal_0);
  });
}
function View_AutoComplete_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'ul',[[
        'class',
        'ui-autocomplete-items ui-autocomplete-list ui-widget-content ui-widget ui-corner-all ui-helper-reset'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoComplete_10)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoComplete_14)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.suggestions;
    ck(v,3,0,currVal_0);
    const currVal_1:any = (co.noResults && co.emptyMessage);
    ck(v,6,0,currVal_1);
  },(null as any));
}
export function View_AutoComplete_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(335544320,1,{inputEL: 0}),
    import0.ɵqud(335544320,2,{multiInputEL: 0}),
    import0.ɵqud(201326592,3,{panelEL: 0}),
    import0.ɵqud(335544320,4,{multiContainerEL: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),17,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-autocomplete ui-widget',
      'ui-autocomplete-dd',
      'ui-autocomplete-multiple'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoComplete_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoComplete_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoComplete_8)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,[
      [
        3,
        0
      ]
      ,
      [
        'panel',
        1
      ]

    ]
      ,(null as any),4,'div',[[
        'class',
        'ui-autocomplete-panel ui-widget-content ui-corner-all ui-shadow'
      ]
    ],[
      [
        4,
        'display',
        (null as any)
      ]
      ,
      [
        4,
        'width',
        (null as any)
      ]
      ,
      [
        4,
        'max-height',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoComplete_9)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.AutoComplete = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,7,0,true,co.dropdown,co.multiple);
    ck(v,6,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,8,0,currVal_2);
    const currVal_3:boolean = !co.multiple;
    ck(v,11,0,currVal_3);
    const currVal_4:any = co.multiple;
    ck(v,13,0,currVal_4);
    const currVal_5:any = co.dropdown;
    ck(v,15,0,currVal_5);
    const currVal_9:any = co.panelVisible;
    ck(v,20,0,currVal_9);
  },(ck,v) => {
    var co:import1.AutoComplete = v.component;
    const currVal_6:any = (co.panelVisible? 'block': 'none');
    const currVal_7:any = (co.appendTo? 'auto': '100%');
    const currVal_8:any = co.scrollHeight;
    ck(v,17,0,currVal_6,currVal_7,currVal_8);
  });
}
function View_AutoComplete_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'p-autoComplete',([] as any[]),[
      [
        2,
        'ui-inputwrapper-filled',
        (null as any)
      ]
      ,
      [
        2,
        'ui-inputwrapper-focus',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_AutoComplete_0,RenderType_AutoComplete)),
    import0.ɵprd(256,(null as any),import6.DomHandler,import6.DomHandler,([] as any[])),
    import0.ɵprd(256,(null as any),import7.ObjectUtils,import7.ObjectUtils,([] as any[])),
    import0.ɵdid(7036928,(null as any),1,import1.AutoComplete,[
      import0.ElementRef,
      import6.DomHandler,
      import0.Renderer2,
      import7.ObjectUtils,
      import0.ChangeDetectorRef,
      import0.IterableDiffers
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(301989888,1,{templates: 1}),
    import0.ɵprd(2560,(null as any),import8.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.AutoComplete])
  ]
  ,(ck,v) => {
    ck(v,3,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,3).filled;
    const currVal_1:any = import0.ɵnov(v,3).focus;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const AutoCompleteNgFactory:import0.ComponentFactory<import1.AutoComplete> = import0.ɵccf('p-autoComplete',import1.AutoComplete,View_AutoComplete_Host_0,{
  minLength: 'minLength',
  delay: 'delay',
  style: 'style',
  styleClass: 'styleClass',
  inputStyle: 'inputStyle',
  inputId: 'inputId',
  inputStyleClass: 'inputStyleClass',
  placeholder: 'placeholder',
  readonly: 'readonly',
  disabled: 'disabled',
  maxlength: 'maxlength',
  size: 'size',
  appendTo: 'appendTo',
  autoHighlight: 'autoHighlight',
  type: 'type',
  field: 'field',
  scrollHeight: 'scrollHeight',
  dropdown: 'dropdown',
  multiple: 'multiple',
  tabindex: 'tabindex',
  dataKey: 'dataKey',
  emptyMessage: 'emptyMessage',
  immutable: 'immutable',
  suggestions: 'suggestions'
}
,{
  completeMethod: 'completeMethod',
  onSelect: 'onSelect',
  onUnselect: 'onUnselect',
  onFocus: 'onFocus',
  onBlur: 'onBlur',
  onDropdownClick: 'onDropdownClick',
  onClear: 'onClear',
  onKeyUp: 'onKeyUp'
}
,([] as any[]));
