// primeng/components/inputmask/inputmask.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/inputmask/inputmask';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/inputtext/inputtext';
import * as import4 from 'primeng/components/dom/domhandler';
import * as import5 from '@angular/forms';
class InputMaskModuleInjector extends import0.ɵNgModuleInjector<import1.InputMaskModule> {
  _CommonModule_0:import2.CommonModule;
  _InputTextModule_1:import3.InputTextModule;
  _InputMaskModule_2:import1.InputMaskModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.InputMaskModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._InputTextModule_1 = new import3.InputTextModule();
    this._InputMaskModule_2 = new import1.InputMaskModule();
    return this._InputMaskModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.InputTextModule)) { return this._InputTextModule_1; }
    if ((token === import1.InputMaskModule)) { return this._InputMaskModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const InputMaskModuleNgFactory:import0.NgModuleFactory<import1.InputMaskModule> = new import0.NgModuleFactory<any>(InputMaskModuleInjector,import1.InputMaskModule);
const styles_InputMask:any[] = ([] as any[]);
export const RenderType_InputMask:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_InputMask,
  data: {}
}
);
export function View_InputMask_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{inputViewChild: 0}),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'input',
        1
      ]

    ]
      ,(null as any),3,'input',[[
        'pInputText',
        ''
      ]
    ],[
      [
        1,
        'id',
        0
      ]
      ,
      [
        1,
        'type',
        0
      ]
      ,
      [
        1,
        'name',
        0
      ]
      ,
      [
        1,
        'placeholder',
        0
      ]
      ,
      [
        1,
        'size',
        0
      ]
      ,
      [
        1,
        'maxlength',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        8,
        'readOnly',
        0
      ]
      ,
      [
        2,
        'ui-inputtext',
        (null as any)
      ]
      ,
      [
        2,
        'ui-corner-all',
        (null as any)
      ]
      ,
      [
        2,
        'ui-state-default',
        (null as any)
      ]
      ,
      [
        2,
        'ui-widget',
        (null as any)
      ]
      ,
      [
        2,
        'ui-state-filled',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]
      ,
      [
        (null as any),
        'keypress'
      ]
      ,
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'paste'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.InputMask = v.component;
      if (('input' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,4).onInput($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('focus' === en)) {
        const pd_1:any = ((<any>co.onInputFocus($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('blur' === en)) {
        const pd_2:any = ((<any>co.onInputBlur($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('keydown' === en)) {
        const pd_3:any = ((<any>co.onKeyDown($event)) !== false);
        ad = (pd_3 && ad);
      }
      if (('keypress' === en)) {
        const pd_4:any = ((<any>co.onKeyPress($event)) !== false);
        ad = (pd_4 && ad);
      }
      if (('input' === en)) {
        const pd_5:any = ((<any>co.onInput($event)) !== false);
        ad = (pd_5 && ad);
      }
      if (('paste' === en)) {
        const pd_6:any = ((<any>co.handleInputChange($event)) !== false);
        ad = (pd_6 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵdid(139264,(null as any),0,import3.InputText,[import0.ElementRef],(null as any),(null as any))
  ]
  ,(ck,v) => {
    var co:import1.InputMask = v.component;
    const currVal_14:any = co.styleClass;
    ck(v,2,0,currVal_14);
    const currVal_15:any = co.style;
    ck(v,3,0,currVal_15);
    ck(v,4,0);
  },(ck,v) => {
    var co:import1.InputMask = v.component;
    const currVal_0:any = co.inputId;
    const currVal_1:any = co.type;
    const currVal_2:any = co.name;
    const currVal_3:any = co.placeholder;
    const currVal_4:any = co.size;
    const currVal_5:any = co.maxlength;
    const currVal_6:any = co.tabindex;
    const currVal_7:any = co.disabled;
    const currVal_8:any = co.readonly;
    const currVal_9:any = true;
    const currVal_10:any = true;
    const currVal_11:any = true;
    const currVal_12:any = true;
    const currVal_13:any = import0.ɵnov(v,4).filled;
    ck(v,1,1,[
      currVal_0,
      currVal_1,
      currVal_2,
      currVal_3,
      currVal_4,
      currVal_5,
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10,
      currVal_11,
      currVal_12,
      currVal_13
    ]
    );
  });
}
function View_InputMask_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'p-inputMask',([] as any[]),[
      [
        2,
        'ui-inputwrapper-filled',
        (null as any)
      ]
      ,
      [
        2,
        'ui-inputwrapper-focus',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_InputMask_0,RenderType_InputMask)),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵdid(122880,(null as any),0,import1.InputMask,[
      import0.ElementRef,
      import4.DomHandler
    ]
    ,(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import5.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.InputMask])
  ]
  ,(ck,v) => {
    ck(v,2,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2).filled;
    const currVal_1:any = import0.ɵnov(v,2).focus;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const InputMaskNgFactory:import0.ComponentFactory<import1.InputMask> = import0.ɵccf('p-inputMask',import1.InputMask,View_InputMask_Host_0,{
  type: 'type',
  slotChar: 'slotChar',
  autoClear: 'autoClear',
  style: 'style',
  inputId: 'inputId',
  styleClass: 'styleClass',
  placeholder: 'placeholder',
  size: 'size',
  maxlength: 'maxlength',
  tabindex: 'tabindex',
  disabled: 'disabled',
  readonly: 'readonly',
  unmask: 'unmask',
  name: 'name',
  mask: 'mask'
}
,{
  onComplete: 'onComplete',
  onFocus: 'onFocus',
  onBlur: 'onBlur'
}
,([] as any[]));
