// primeng/components/megamenu/megamenu.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/megamenu/megamenu';
import * as import2 from '@angular/common';
import * as import3 from '@angular/router';
import * as import4 from 'primeng/components/dom/domhandler';
class MegaMenuModuleInjector extends import0.ɵNgModuleInjector<import1.MegaMenuModule> {
  _CommonModule_0:import2.CommonModule;
  _RouterModule_1:import3.RouterModule;
  _MegaMenuModule_2:import1.MegaMenuModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.MegaMenuModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._RouterModule_1 = new import3.RouterModule(this.parent.get(import3.ɵa,(null as any)),this.parent.get(import3.Router,(null as any)));
    this._MegaMenuModule_2 = new import1.MegaMenuModule();
    return this._MegaMenuModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.RouterModule)) { return this._RouterModule_1; }
    if ((token === import1.MegaMenuModule)) { return this._MegaMenuModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const MegaMenuModuleNgFactory:import0.NgModuleFactory<import1.MegaMenuModule> = new import0.NgModuleFactory<any>(MegaMenuModuleInjector,import1.MegaMenuModule);
const styles_MegaMenu:any[] = ([] as any[]);
export const RenderType_MegaMenu:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_MegaMenu,
  data: {}
}
);
function View_MegaMenu_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'li',[[
        'class',
        'ui-menu-separator ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(null as any),(null as any));
}
function View_MegaMenu_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'li',[[
        'class',
        'ui-menu-separator ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                                    ']))
  ]
  ,(null as any),(null as any));
}
function View_MegaMenu_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-icon fa fa-fw'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-menuitem-icon fa fa-fw';
    const currVal_1:any = (<any>(<any>(<any>v.parent).parent).parent).context.$implicit.icon;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_MegaMenu_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),9,'a',[[
        'class',
        'ui-menuitem-link ui-corner-all'
      ]
    ],[
      [
        8,
        'href',
        4
      ]
      ,
      [
        1,
        'target',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.itemClick($event,(<any>(<any>v.parent).parent).context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled']),
    (l()(),import0.ɵted((null as any),['\n                                                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MegaMenu_10)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                                                        ']))
  ]
  ,(ck,v) => {
    const currVal_2:any = 'ui-menuitem-link ui-corner-all';
    const currVal_3:any = ck(v,2,0,(<any>(<any>v.parent).parent).context.$implicit.disabled);
    ck(v,1,0,currVal_2,currVal_3);
    const currVal_4:any = (<any>(<any>v.parent).parent).context.$implicit.icon;
    ck(v,5,0,currVal_4);
  },(ck,v) => {
    const currVal_0:any = ((<any>(<any>v.parent).parent).context.$implicit.url || '#');
    const currVal_1:any = (<any>(<any>v.parent).parent).context.$implicit.target;
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_5:any = (<any>(<any>v.parent).parent).context.$implicit.label;
    ck(v,8,0,currVal_5);
  });
}
function View_MegaMenu_12(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-icon fa fa-fw'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-menuitem-icon fa fa-fw';
    const currVal_1:any = (<any>(<any>(<any>v.parent).parent).parent).context.$implicit.icon;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_MegaMenu_11(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),14,'a',[[
        'class',
        'ui-menuitem-link ui-corner-all'
      ]
    ],[
      [
        1,
        'target',
        0
      ]
      ,
      [
        1,
        'target',
        0
      ]
      ,
      [
        8,
        'href',
        4
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,3).onClick($event.button,$event.ctrlKey,$event.metaKey)) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>co.itemClick($event,(<any>(<any>v.parent).parent).context.$implicit)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled']),
      import0.ɵdid(335872,[[
        2,
        4
      ]
    ],0,import3.RouterLinkWithHref,[
      import3.Router,
      import3.ActivatedRoute,
      import2.LocationStrategy
    ]
      ,{routerLink: [
        0,
        'routerLink'
      ]
    },(null as any)),
    import0.ɵdid(860160,(null as any),2,import3.RouterLinkActive,[
      import3.Router,
      import0.ElementRef,
      import0.Renderer,
      import0.ChangeDetectorRef
    ]
    ,{
      routerLinkActiveOptions: [
        0,
        'routerLinkActiveOptions'
      ]
      ,
      routerLinkActive: [
        1,
        'routerLinkActive'
      ]

    }
    ,(null as any)),
    import0.ɵqud(301989888,1,{links: 1}),
    import0.ɵqud(301989888,2,{linksWithHrefs: 1}),
    import0.ɵpod(['exact']),
    (l()(),import0.ɵted((null as any),['\n                                                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MegaMenu_12)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                                                        ']))
  ]
  ,(ck,v) => {
    const currVal_3:any = 'ui-menuitem-link ui-corner-all';
    const currVal_4:any = ck(v,2,0,(<any>(<any>v.parent).parent).context.$implicit.disabled);
    ck(v,1,0,currVal_3,currVal_4);
    const currVal_5:any = (<any>(<any>v.parent).parent).context.$implicit.routerLink;
    ck(v,3,0,currVal_5);
    const currVal_6:any = ((<any>(<any>v.parent).parent).context.$implicit.routerLinkActiveOptions || ck(v,7,0,false));
    const currVal_7:any = 'ui-state-active';
    ck(v,4,0,currVal_6,currVal_7);
    const currVal_8:any = (<any>(<any>v.parent).parent).context.$implicit.icon;
    ck(v,10,0,currVal_8);
  },(ck,v) => {
    const currVal_0:any = (<any>(<any>v.parent).parent).context.$implicit.target;
    const currVal_1:any = import0.ɵnov(v,3).target;
    const currVal_2:any = import0.ɵnov(v,3).href;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
    const currVal_9:any = (<any>(<any>v.parent).parent).context.$implicit.label;
    ck(v,13,0,currVal_9);
  });
}
function View_MegaMenu_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'li',[[
        'class',
        'ui-menuitem ui-widget ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MegaMenu_9)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MegaMenu_11)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                                    ']))
  ]
  ,(ck,v) => {
    const currVal_0:boolean = !(<any>v.parent).context.$implicit.routerLink;
    ck(v,3,0,currVal_0);
    const currVal_1:any = (<any>v.parent).context.$implicit.routerLink;
    ck(v,6,0,currVal_1);
  },(null as any));
}
function View_MegaMenu_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MegaMenu_7)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MegaMenu_8)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                                ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.separator;
    ck(v,2,0,currVal_0);
    const currVal_1:boolean = !v.context.$implicit.separator;
    ck(v,4,0,currVal_1);
  },(null as any));
}
function View_MegaMenu_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),8,'ul',[[
        'class',
        'ui-menu-list ui-helper-reset'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'li',[[
        'class',
        'ui-widget-header ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'h3',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MegaMenu_6)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                            '])),
    (l()(),import0.ɵted((null as any),['\n                                        ']))
  ]
  ,(ck,v) => {
    const currVal_1:any = v.context.$implicit.items;
    ck(v,8,0,currVal_1);
  },(ck,v) => {
    const currVal_0:any = v.context.$implicit.label;
    ck(v,5,0,currVal_0);
  });
}
function View_MegaMenu_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MegaMenu_5)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(ck,v) => {
    const currVal_1:any = v.context.$implicit;
    ck(v,4,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.getColumnClass((<any>(<any>v.parent).parent).context.$implicit);
    ck(v,1,0,currVal_0);
  });
}
function View_MegaMenu_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,[[
        'item',
        1
      ]
    ],(null as any),27,'li',([] as any[]),(null as any),[
      [
        (null as any),
        'mouseenter'
      ]
      ,
      [
        (null as any),
        'mouseleave'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('mouseenter' === en)) {
        const pd_0:any = ((<any>co.onItemMouseEnter($event,import0.ɵnov(v,0),(<any>v.parent).context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mouseleave' === en)) {
        const pd_1:any = ((<any>co.onItemMouseLeave($event,import0.ɵnov(v,0))) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-menuitem ui-widget ui-corner-all',
      'ui-menu-parent',
      'ui-menuitem-active'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),13,'a',[[
        'class',
        'ui-menuitem-link ui-corner-all ui-submenu-link'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled']),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-icon fa fa-fw'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-submenu-icon fa fa-fw'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'fa-caret-down',
      'fa-caret-right'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'ui-megamenu-panel ui-widget-content ui-menu-list ui-corner-all ui-helper-clearfix ui-menu-child ui-shadow'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'ui-g'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MegaMenu_4)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,true,(<any>v.parent).context.$implicit.items,(import0.ɵnov(v,0) == co.activeItem));
    ck(v,1,0,currVal_0);
    const currVal_1:any = 'ui-menuitem-link ui-corner-all ui-submenu-link';
    const currVal_2:any = ck(v,6,0,(<any>v.parent).context.$implicit.disabled);
    ck(v,5,0,currVal_1,currVal_2);
    const currVal_3:any = 'ui-menuitem-icon fa fa-fw';
    const currVal_4:any = (<any>v.parent).context.$implicit.icon;
    ck(v,9,0,currVal_3,currVal_4);
    const currVal_6:any = 'ui-submenu-icon fa fa-fw';
    const currVal_7:any = ck(v,16,0,(co.orientation == 'horizontal'),(co.orientation == 'vertical'));
    ck(v,15,0,currVal_6,currVal_7);
    const currVal_8:any = (<any>v.parent).context.$implicit.items;
    ck(v,24,0,currVal_8);
  },(ck,v) => {
    const currVal_5:any = (<any>v.parent).context.$implicit.label;
    ck(v,12,0,currVal_5);
  });
}
function View_MegaMenu_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MegaMenu_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MegaMenu_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.separator;
    ck(v,2,0,currVal_0);
    const currVal_1:boolean = !v.context.$implicit.separator;
    ck(v,4,0,currVal_1);
  },(null as any));
}
function View_MegaMenu_13(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'li',[[
        'class',
        'ui-menuitem ui-menuitem-custom ui-widget ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(null as any),(null as any));
}
export function View_MegaMenu_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),13,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-menu ui-menubar ui-megamenu ui-widget ui-widget-content ui-corner-all ui-helper-clearfix',
      'ui-megamenu-vertical'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'ul',[[
        'class',
        'ui-menu-list ui-helper-reset ui-menubar-root-list'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MegaMenu_1)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MegaMenu_13)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.MegaMenu = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,3,0,true,(co.orientation == 'vertical'));
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,4,0,currVal_2);
    const currVal_3:any = co.model;
    ck(v,9,0,currVal_3);
    const currVal_4:any = (co.orientation === 'horizontal');
    ck(v,12,0,currVal_4);
  },(null as any));
}
function View_MegaMenu_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-megaMenu',([] as any[]),(null as any),(null as any),(null as any),View_MegaMenu_0,RenderType_MegaMenu)),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵdid(24576,(null as any),0,import1.MegaMenu,[
      import0.ElementRef,
      import4.DomHandler,
      import0.Renderer2
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const MegaMenuNgFactory:import0.ComponentFactory<import1.MegaMenu> = import0.ɵccf('p-megaMenu',import1.MegaMenu,View_MegaMenu_Host_0,{
  model: 'model',
  style: 'style',
  styleClass: 'styleClass',
  orientation: 'orientation'
}
,{},['*']);
