// primeng/components/tabview/tabview.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/tabview/tabview';
import * as import2 from '@angular/common';
class TabViewModuleInjector extends import0.ɵNgModuleInjector<import1.TabViewModule> {
  _CommonModule_0:import2.CommonModule;
  _TabViewModule_1:import1.TabViewModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.TabViewModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._TabViewModule_1 = new import1.TabViewModule();
    return this._TabViewModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.TabViewModule)) { return this._TabViewModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const TabViewModuleNgFactory:import0.NgModuleFactory<import1.TabViewModule> = new import0.NgModuleFactory<any>(TabViewModuleInjector,import1.TabViewModule);
const styles_TabView:any[] = ([] as any[]);
export const RenderType_TabView:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_TabView,
  data: {}
}
);
function View_TabView_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ul',[
      [
        'p-tabViewNav',
        ''
      ]
      ,
      [
        'role',
        'tablist'
      ]

    ]
    ,[
      [
        2,
        'ui-tabview-nav',
        (null as any)
      ]
      ,
      [
        2,
        'ui-helper-reset',
        (null as any)
      ]
      ,
      [
        2,
        'ui-helper-clearfix',
        (null as any)
      ]
      ,
      [
        2,
        'ui-widget-header',
        (null as any)
      ]
      ,
      [
        2,
        'ui-corner-all',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'onTabClick'
      ]
      ,
      [
        (null as any),
        'onTabCloseClick'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('onTabClick' === en)) {
        const pd_0:any = ((<any>co.open($event.originalEvent,$event.tab)) !== false);
        ad = (pd_0 && ad);
      }
      if (('onTabCloseClick' === en)) {
        const pd_1:any = ((<any>co.close($event.originalEvent,$event.tab)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },View_TabViewNav_0,RenderType_TabViewNav)),
    import0.ɵdid(24576,(null as any),0,import1.TabViewNav,([] as any[]),{
      tabs: [
        0,
        'tabs'
      ]
      ,
      orientation: [
        1,
        'orientation'
      ]

    }
    ,{
      onTabClick: 'onTabClick',
      onTabCloseClick: 'onTabCloseClick'
    }
    )
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_5:any = co.tabs;
    const currVal_6:any = co.orientation;
    ck(v,1,0,currVal_5,currVal_6);
  },(ck,v) => {
    const currVal_0:any = true;
    const currVal_1:any = true;
    const currVal_2:any = true;
    const currVal_3:any = true;
    const currVal_4:any = true;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
function View_TabView_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ul',[
      [
        'p-tabViewNav',
        ''
      ]
      ,
      [
        'role',
        'tablist'
      ]

    ]
    ,[
      [
        2,
        'ui-tabview-nav',
        (null as any)
      ]
      ,
      [
        2,
        'ui-helper-reset',
        (null as any)
      ]
      ,
      [
        2,
        'ui-helper-clearfix',
        (null as any)
      ]
      ,
      [
        2,
        'ui-widget-header',
        (null as any)
      ]
      ,
      [
        2,
        'ui-corner-all',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'onTabClick'
      ]
      ,
      [
        (null as any),
        'onTabCloseClick'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('onTabClick' === en)) {
        const pd_0:any = ((<any>co.open($event.originalEvent,$event.tab)) !== false);
        ad = (pd_0 && ad);
      }
      if (('onTabCloseClick' === en)) {
        const pd_1:any = ((<any>co.close($event.originalEvent,$event.tab)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },View_TabViewNav_0,RenderType_TabViewNav)),
    import0.ɵdid(24576,(null as any),0,import1.TabViewNav,([] as any[]),{
      tabs: [
        0,
        'tabs'
      ]
      ,
      orientation: [
        1,
        'orientation'
      ]

    }
    ,{
      onTabClick: 'onTabClick',
      onTabCloseClick: 'onTabCloseClick'
    }
    )
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_5:any = co.tabs;
    const currVal_6:any = co.orientation;
    ck(v,1,0,currVal_5,currVal_6);
  },(ck,v) => {
    const currVal_0:any = true;
    const currVal_1:any = true;
    const currVal_2:any = true;
    const currVal_3:any = true;
    const currVal_4:any = true;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
export function View_TabView_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),14,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TabView_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-tabview-panels'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TabView_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.TabView = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ('ui-tabview ui-widget ui-widget-content ui-corner-all ui-tabview-' + co.orientation);
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_3:any = (co.orientation != 'bottom');
    ck(v,6,0,currVal_3);
    const currVal_4:any = (co.orientation == 'bottom');
    ck(v,14,0,currVal_4);
  },(null as any));
}
function View_TabView_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-tabView',([] as any[]),(null as any),(null as any),(null as any),View_TabView_0,RenderType_TabView)),
    import0.ɵdid(548864,(null as any),1,import1.TabView,[import0.ElementRef],(null as any),(null as any)),
    import0.ɵqud(301989888,1,{tabPanels: 1})
  ]
  ,(null as any),(null as any));
}
export const TabViewNgFactory:import0.ComponentFactory<import1.TabView> = import0.ɵccf('p-tabView',import1.TabView,View_TabView_Host_0,{
  orientation: 'orientation',
  style: 'style',
  styleClass: 'styleClass',
  controlClose: 'controlClose',
  lazy: 'lazy',
  activeIndex: 'activeIndex'
}
,{
  onChange: 'onChange',
  onClose: 'onClose'
}
,['*']);
const styles_TabPanel:any[] = ([] as any[]);
export const RenderType_TabPanel:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_TabPanel,
  data: {}
}
);
function View_TabPanel_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[
      [
        'class',
        'ui-tabview-panel ui-widget-content'
      ]
      ,
      [
        'role',
        'tabpanel'
      ]

    ]
    ,[
      [
        4,
        'display',
        (null as any)
      ]
      ,
      [
        1,
        'aria-hidden',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.selected? 'block': 'none');
    const currVal_1:boolean = !co.selected;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export function View_TabPanel_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TabPanel_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.TabPanel = v.component;
    const currVal_0:any = (co.closed? false: (co.lazy? co.selected: true));
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_TabPanel_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-tabPanel',([] as any[]),(null as any),(null as any),(null as any),View_TabPanel_0,RenderType_TabPanel)),
    import0.ɵdid(24576,(null as any),0,import1.TabPanel,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const TabPanelNgFactory:import0.ComponentFactory<import1.TabPanel> = import0.ɵccf('p-tabPanel',import1.TabPanel,View_TabPanel_Host_0,{
  header: 'header',
  selected: 'selected',
  disabled: 'disabled',
  closable: 'closable',
  headerStyle: 'headerStyle',
  headerStyleClass: 'headerStyleClass',
  leftIcon: 'leftIcon',
  rightIcon: 'rightIcon'
}
,{},['*']);
const styles_TabViewNav:any[] = ([] as any[]);
export const RenderType_TabViewNav:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_TabViewNav,
  data: {}
}
);
function View_TabViewNav_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-tabview-left-icon fa'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-tabview-left-icon fa';
    const currVal_1:any = (<any>(<any>v.parent).parent).context.$implicit.leftIcon;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_TabViewNav_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-tabview-right-icon fa'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-tabview-right-icon fa';
    const currVal_1:any = (<any>(<any>v.parent).parent).context.$implicit.rightIcon;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_TabViewNav_5(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'ui-tabview-close fa fa-close'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.clickClose($event,(<any>(<any>v.parent).parent).context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
  },(null as any),(null as any)))],(null as any),(null as any));
}
function View_TabViewNav_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),19,'li',[[
        'role',
        'tab'
      ]
    ],[
      [
        1,
        'aria-expanded',
        0
      ]
      ,
      [
        1,
        'aria-selected',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.clickTab($event,(<any>v.parent).context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-tabview-selected ui-state-active',
      'ui-state-disabled'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'a',[[
        'href',
        '#'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TabViewNav_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-tabview-title'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TabViewNav_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TabViewNav_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co.getDefaultHeaderClass((<any>v.parent).context.$implicit);
    const currVal_3:any = ck(v,2,0,(<any>v.parent).context.$implicit.selected,(<any>v.parent).context.$implicit.disabled);
    ck(v,1,0,currVal_2,currVal_3);
    const currVal_4:any = (<any>v.parent).context.$implicit.headerStyle;
    ck(v,3,0,currVal_4);
    const currVal_5:any = (<any>v.parent).context.$implicit.leftIcon;
    ck(v,8,0,currVal_5);
    const currVal_7:any = (<any>v.parent).context.$implicit.rightIcon;
    ck(v,14,0,currVal_7);
    const currVal_8:any = (<any>v.parent).context.$implicit.closable;
    ck(v,18,0,currVal_8);
  },(ck,v) => {
    const currVal_0:any = (<any>v.parent).context.$implicit.selected;
    const currVal_1:any = (<any>v.parent).context.$implicit.selected;
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_6:any = (<any>v.parent).context.$implicit.header;
    ck(v,11,0,currVal_6);
  });
}
function View_TabViewNav_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TabViewNav_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    const currVal_0:boolean = !v.context.$implicit.closed;
    ck(v,2,0,currVal_0);
  },(null as any));
}
export function View_TabViewNav_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TabViewNav_1)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.TabViewNav = v.component;
    const currVal_0:any = co.tabs;
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_TabViewNav_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'p-tabViewNav',
        ''
      ]
    ],[
      [
        2,
        'ui-tabview-nav',
        (null as any)
      ]
      ,
      [
        2,
        'ui-helper-reset',
        (null as any)
      ]
      ,
      [
        2,
        'ui-helper-clearfix',
        (null as any)
      ]
      ,
      [
        2,
        'ui-widget-header',
        (null as any)
      ]
      ,
      [
        2,
        'ui-corner-all',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_TabViewNav_0,RenderType_TabViewNav)),
    import0.ɵdid(24576,(null as any),0,import1.TabViewNav,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = true;
    const currVal_1:any = true;
    const currVal_2:any = true;
    const currVal_3:any = true;
    const currVal_4:any = true;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
export const TabViewNavNgFactory:import0.ComponentFactory<import1.TabViewNav> = import0.ɵccf('[p-tabViewNav]',import1.TabViewNav,View_TabViewNav_Host_0,{
  tabs: 'tabs',
  orientation: 'orientation'
}
,{
  onTabClick: 'onTabClick',
  onTabCloseClick: 'onTabCloseClick'
}
,([] as any[]));
