// primeng/components/dialog/dialog.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/dialog/dialog';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from 'primeng/components/dom/domhandler';
class DialogModuleInjector extends import0.ɵNgModuleInjector<import1.DialogModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _DialogModule_2:import1.DialogModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.DialogModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._DialogModule_2 = new import1.DialogModule();
    return this._DialogModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import1.DialogModule)) { return this._DialogModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const DialogModuleNgFactory:import0.NgModuleFactory<import1.DialogModule> = new import0.NgModuleFactory<any>(DialogModuleInjector,import1.DialogModule);
const styles_Dialog:any[] = ([] as any[]);
export const RenderType_Dialog:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Dialog,
      data: {'animation': [{
        name: 'dialogState',
        definitions: [
          {
            type: 0,
            name: 'hidden',
            styles: {
              type: 6,
              styles: {opacity: 0}
            }

          }
          ,
          {
            type: 0,
            name: 'visible',
            styles: {
              type: 6,
              styles: {opacity: 1}
            }

          }
          ,
          {
            type: 1,
            expr: 'visible => hidden',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms ease-in'
            }

          }
          ,
          {
            type: 1,
            expr: 'hidden => visible',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms ease-out'
            }

          }

        ]

      }
  ]}
}
);
function View_Dialog_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-dialog-title'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.header;
    ck(v,1,0,currVal_0);
  });
}
function View_Dialog_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'span',[[
        'class',
        'ui-dialog-title'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(null as any),(null as any));
}
function View_Dialog_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'a',[
      [
        'href',
        '#'
      ]
      ,
      [
        'role',
        'button'
      ]

    ]
    ,(null as any),[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'mousedown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.close($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mousedown' === en)) {
        const pd_1:any = ((<any>co.onCloseMouseDown($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['ui-dialog-titlebar-icon ui-dialog-titlebar-close ui-corner-all']),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-fw fa-close'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,true);
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_Dialog_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'titlebar',
        1
      ]

    ]
      ,(null as any),10,'div',[[
        'class',
        'ui-dialog-titlebar ui-widget-header ui-helper-clearfix ui-corner-top'
      ]
    ],(null as any),[
      [
        (null as any),
        'mousedown'
      ]
      ,
      [
        (null as any),
        'mouseup'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('mousedown' === en)) {
        const pd_0:any = ((<any>co.initDrag($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mouseup' === en)) {
        const pd_1:any = ((<any>co.endDrag($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dialog_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dialog_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dialog_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.header;
    ck(v,3,0,currVal_0);
    const currVal_1:any = co.headerFacet;
    ck(v,6,0,currVal_1);
    const currVal_2:any = co.closable;
    ck(v,9,0,currVal_2);
  },(null as any));
}
function View_Dialog_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-dialog-footer ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),2),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(null as any));
}
function View_Dialog_6(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[
      [
        'class',
        'ui-resizable-handle ui-resizable-se ui-icon ui-icon-gripsmall-diagonal-se'
      ]
      ,
      [
        'style',
        'z-index: 90;'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'mousedown'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('mousedown' === en)) {
        const pd_0:any = ((<any>co.initResize($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
  },(null as any),(null as any)))],(null as any),(null as any));
}
export function View_Dialog_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{containerViewChild: 0}),
    import0.ɵqud(335544320,2,{headerViewChild: 0}),
    import0.ɵqud(201326592,3,{contentViewChild: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'container',
        1
      ]

    ]
    ,(null as any),19,'div',([] as any[]),[
      [
        4,
        'display',
        (null as any)
      ]
      ,
      [
        4,
        'width',
        'px'
      ]
      ,
      [
        4,
        'height',
        'px'
      ]
      ,
      [
        4,
        'minWidth',
        'px'
      ]
      ,
      [
        24,
        '@dialogState',
        0
      ]

    ]
      ,[[
        (null as any),
        'mousedown'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Dialog = v.component;
      if (('mousedown' === en)) {
        const pd_0:any = ((<any>co.moveOnTop()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-dialog ui-widget ui-widget-content ui-corner-all ui-shadow',
      'ui-dialog-rtl',
      'ui-dialog-draggable'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dialog_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,[
      [
        3,
        0
      ]
      ,
      [
        'content',
        1
      ]

    ]
      ,(null as any),4,'div',[[
        'class',
        'ui-dialog-content ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),1),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dialog_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dialog_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Dialog = v.component;
    const currVal_5:any = co.styleClass;
    const currVal_6:any = ck(v,6,0,true,co.rtl,co.draggable);
    ck(v,5,0,currVal_5,currVal_6);
    const currVal_7:any = co.style;
    ck(v,7,0,currVal_7);
    const currVal_8:any = co.showHeader;
    ck(v,10,0,currVal_8);
    const currVal_9:any = co.contentStyle;
    ck(v,13,0,currVal_9);
    const currVal_10:any = co.footerFacet;
    ck(v,19,0,currVal_10);
    const currVal_11:any = co.resizable;
    ck(v,22,0,currVal_11);
  },(ck,v) => {
    var co:import1.Dialog = v.component;
    const currVal_0:any = (co.visible? 'block': 'none');
    const currVal_1:any = co.width;
    const currVal_2:any = co.height;
    const currVal_3:any = co.minWidth;
    const currVal_4:any = (co.visible? 'visible': 'hidden');
    ck(v,4,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
function View_Dialog_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'p-dialog',([] as any[]),(null as any),(null as any),(null as any),View_Dialog_0,RenderType_Dialog)),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵdid(6381568,(null as any),2,import1.Dialog,[
      import0.ElementRef,
      import4.DomHandler,
      import0.Renderer2
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(167772160,1,{headerFacet: 0}),
    import0.ɵqud(167772160,2,{footerFacet: 0})
  ]
  ,(null as any),(null as any));
}
export const DialogNgFactory:import0.ComponentFactory<import1.Dialog> = import0.ɵccf('p-dialog',import1.Dialog,View_Dialog_Host_0,{
  header: 'header',
  draggable: 'draggable',
  resizable: 'resizable',
  minWidth: 'minWidth',
  minHeight: 'minHeight',
  width: 'width',
  height: 'height',
  positionLeft: 'positionLeft',
  positionTop: 'positionTop',
  contentStyle: 'contentStyle',
  modal: 'modal',
  closeOnEscape: 'closeOnEscape',
  dismissableMask: 'dismissableMask',
  rtl: 'rtl',
  closable: 'closable',
  responsive: 'responsive',
  appendTo: 'appendTo',
  style: 'style',
  styleClass: 'styleClass',
  showHeader: 'showHeader',
  breakpoint: 'breakpoint',
  blockScroll: 'blockScroll',
  visible: 'visible'
}
,{
  onShow: 'onShow',
  onHide: 'onHide',
  visibleChange: 'visibleChange'
}
,[
  'p-header',
  '*',
  'p-footer'
]
);
