// primeng/components/captcha/captcha.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/captcha/captcha';
import * as import2 from '@angular/common';
class CaptchaModuleInjector extends import0.ɵNgModuleInjector<import1.CaptchaModule> {
  _CommonModule_0:import2.CommonModule;
  _CaptchaModule_1:import1.CaptchaModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.CaptchaModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._CaptchaModule_1 = new import1.CaptchaModule();
    return this._CaptchaModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.CaptchaModule)) { return this._CaptchaModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const CaptchaModuleNgFactory:import0.NgModuleFactory<import1.CaptchaModule> = new import0.NgModuleFactory<any>(CaptchaModuleInjector,import1.CaptchaModule);
const styles_Captcha:any[] = ([] as any[]);
export const RenderType_Captcha:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Captcha,
  data: {}
}
);
export function View_Captcha_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),(null as any));
}
function View_Captcha_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-captcha',([] as any[]),(null as any),(null as any),(null as any),View_Captcha_0,RenderType_Captcha)),
    import0.ɵdid(2187264,(null as any),0,import1.Captcha,[
      import0.ElementRef,
      import0.NgZone
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const CaptchaNgFactory:import0.ComponentFactory<import1.Captcha> = import0.ɵccf('p-captcha',import1.Captcha,View_Captcha_Host_0,{
  siteKey: 'siteKey',
  theme: 'theme',
  type: 'type',
  size: 'size',
  tabindex: 'tabindex',
  language: 'language',
  initCallback: 'initCallback'
}
,{
  onResponse: 'onResponse',
  onExpire: 'onExpire'
}
,([] as any[]));
