// primeng/components/schedule/schedule.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/schedule/schedule';
import * as import2 from '@angular/common';
class ScheduleModuleInjector extends import0.ɵNgModuleInjector<import1.ScheduleModule> {
  _CommonModule_0:import2.CommonModule;
  _ScheduleModule_1:import1.ScheduleModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.ScheduleModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ScheduleModule_1 = new import1.ScheduleModule();
    return this._ScheduleModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.ScheduleModule)) { return this._ScheduleModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const ScheduleModuleNgFactory:import0.NgModuleFactory<import1.ScheduleModule> = new import0.NgModuleFactory<any>(ScheduleModuleInjector,import1.ScheduleModule);
const styles_Schedule:any[] = ([] as any[]);
export const RenderType_Schedule:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Schedule,
  data: {}
}
);
export function View_Schedule_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:import1.Schedule = v.component;
    const currVal_1:any = co.style;
    ck(v,1,0,currVal_1);
  },(ck,v) => {
    var co:import1.Schedule = v.component;
    const currVal_0:any = co.styleClass;
    ck(v,0,0,currVal_0);
  });
}
function View_Schedule_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-schedule',([] as any[]),(null as any),(null as any),(null as any),View_Schedule_0,RenderType_Schedule)),
    import0.ɵdid(4710400,(null as any),0,import1.Schedule,[
      import0.ElementRef,
      import0.IterableDiffers
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const ScheduleNgFactory:import0.ComponentFactory<import1.Schedule> = import0.ɵccf('p-schedule',import1.Schedule,View_Schedule_Host_0,{
  events: 'events',
  header: 'header',
  style: 'style',
  styleClass: 'styleClass',
  rtl: 'rtl',
  weekends: 'weekends',
  hiddenDays: 'hiddenDays',
  fixedWeekCount: 'fixedWeekCount',
  weekNumbers: 'weekNumbers',
  businessHours: 'businessHours',
  height: 'height',
  contentHeight: 'contentHeight',
  aspectRatio: 'aspectRatio',
  eventLimit: 'eventLimit',
  defaultDate: 'defaultDate',
  editable: 'editable',
  droppable: 'droppable',
  eventStartEditable: 'eventStartEditable',
  eventDurationEditable: 'eventDurationEditable',
  defaultView: 'defaultView',
  allDaySlot: 'allDaySlot',
  allDayText: 'allDayText',
  slotDuration: 'slotDuration',
  slotLabelInterval: 'slotLabelInterval',
  snapDuration: 'snapDuration',
  scrollTime: 'scrollTime',
  minTime: 'minTime',
  maxTime: 'maxTime',
  slotEventOverlap: 'slotEventOverlap',
  nowIndicator: 'nowIndicator',
  dragRevertDuration: 'dragRevertDuration',
  dragOpacity: 'dragOpacity',
  dragScroll: 'dragScroll',
  eventOverlap: 'eventOverlap',
  eventConstraint: 'eventConstraint',
  locale: 'locale',
  timezone: 'timezone',
  timeFormat: 'timeFormat',
  eventRender: 'eventRender',
  dayRender: 'dayRender',
  options: 'options'
}
,{
  onDayClick: 'onDayClick',
  onDrop: 'onDrop',
  onEventClick: 'onEventClick',
  onEventMouseover: 'onEventMouseover',
  onEventMouseout: 'onEventMouseout',
  onEventDragStart: 'onEventDragStart',
  onEventDragStop: 'onEventDragStop',
  onEventDrop: 'onEventDrop',
  onEventResizeStart: 'onEventResizeStart',
  onEventResizeStop: 'onEventResizeStop',
  onEventResize: 'onEventResize',
  onViewRender: 'onViewRender',
  onViewDestroy: 'onViewDestroy'
}
,([] as any[]));
