// primeng/components/organizationchart/organizationchart.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/organizationchart/organizationchart';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from 'primeng/components/dom/domhandler';
class OrganizationChartModuleInjector extends import0.ɵNgModuleInjector<import1.OrganizationChartModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _OrganizationChartModule_2:import1.OrganizationChartModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.OrganizationChartModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._OrganizationChartModule_2 = new import1.OrganizationChartModule();
    return this._OrganizationChartModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import1.OrganizationChartModule)) { return this._OrganizationChartModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const OrganizationChartModuleNgFactory:import0.NgModuleFactory<import1.OrganizationChartModule> = new import0.NgModuleFactory<any>(OrganizationChartModuleInjector,import1.OrganizationChartModule);
const styles_OrganizationChart:any[] = ([] as any[]);
export const RenderType_OrganizationChart:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_OrganizationChart,
  data: {}
}
);
export function View_OrganizationChart_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'table',[
      [
        'class',
        'ui-organizationchart-table'
      ]
      ,
      [
        'pOrganizationChartNode',
        ''
      ]

    ]
    ,(null as any),(null as any),(null as any),View_OrganizationChartNode_0,RenderType_OrganizationChartNode)),
      import0.ɵdid(24576,(null as any),0,import1.OrganizationChartNode,[import1.OrganizationChart],{node: [
        0,
        'node'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.OrganizationChart = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-organizationchart ui-widget';
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_3:any = co.root;
    ck(v,6,0,currVal_3);
  },(null as any));
}
function View_OrganizationChart_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'p-organizationChart',([] as any[]),(null as any),(null as any),(null as any),View_OrganizationChart_0,RenderType_OrganizationChart)),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵdid(548864,(null as any),1,import1.OrganizationChart,[
      import0.ElementRef,
      import4.DomHandler
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(301989888,1,{templates: 1})
  ]
  ,(null as any),(null as any));
}
export const OrganizationChartNgFactory:import0.ComponentFactory<import1.OrganizationChart> = import0.ɵccf('p-organizationChart',import1.OrganizationChart,View_OrganizationChart_Host_0,{
  value: 'value',
  style: 'style',
  styleClass: 'styleClass',
  selectionMode: 'selectionMode',
  selection: 'selection'
}
,{
  selectionChange: 'selectionChange',
  onNodeSelect: 'onNodeSelect',
  onNodeUnselect: 'onNodeUnselect'
}
,([] as any[]));
const styles_OrganizationChartNode:any[] = ([] as any[]);
export const RenderType_OrganizationChartNode:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_OrganizationChartNode,
      data: {'animation': [{
        name: 'childState',
        definitions: [
          {
            type: 0,
            name: 'in',
            styles: {
              type: 6,
              styles: {opacity: 1}
            }

          }
          ,
          {
            type: 1,
            expr: 'void => *',
            animation: [
              {
                type: 6,
                styles: {opacity: 0}
              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: 150
              }

            ]

          }
          ,
          {
            type: 1,
            expr: '* => void',
              animation: [{
                type: 4,
                styles: {
                  type: 6,
                  styles: {opacity: 0}
                }
                ,
                timings: 150
              }
            ]
          }

        ]

      }
  ]}
}
);
function View_OrganizationChartNode_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.node.label;
    ck(v,1,0,currVal_0);
  });
}
function View_OrganizationChartNode_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-organizationChartNodeTemplateLoader',([] as any[]),(null as any),(null as any),(null as any),View_OrganizationChartNodeTemplateLoader_0,RenderType_OrganizationChartNodeTemplateLoader)),
    import0.ɵdid(122880,(null as any),0,import1.OrganizationChartNodeTemplateLoader,[import0.ViewContainerRef],{
      node: [
        0,
        'node'
      ]
      ,
      template: [
        1,
        'template'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.node;
    const currVal_1:any = co.chart.getTemplateForNode(co.node);
    ck(v,3,0,currVal_0,currVal_1);
  },(null as any));
}
function View_OrganizationChartNode_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'a',[
      [
        'class',
        'ui-node-toggler'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggleNode($event,co.node)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'i',[[
        'class',
        'fa ui-node-toggler-icon'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'fa-chevron-down',
      'fa-chevron-up'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'fa ui-node-toggler-icon';
    const currVal_1:any = ck(v,4,0,co.node.expanded,!co.node.expanded);
    ck(v,3,0,currVal_0,currVal_1);
  },(null as any));
}
function View_OrganizationChartNode_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'td',[[
        'class',
        'ui-organizationchart-line-left'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-organizationchart-line-top']),
    (l()(),import0.ɵted((null as any),[' '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'td',[[
        'class',
        'ui-organizationchart-line-right'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-organizationchart-line-top']),
    (l()(),import0.ɵted((null as any),[' '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-organizationchart-line-left';
    const currVal_1:any = ck(v,3,0,!v.context.first);
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = 'ui-organizationchart-line-right';
    const currVal_3:any = ck(v,8,0,!v.context.last);
    ck(v,7,0,currVal_2,currVal_3);
  },(null as any));
}
function View_OrganizationChartNode_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'td',[[
        'colspan',
        '2'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'table',[
      [
        'class',
        'ui-organizationchart-table'
      ]
      ,
      [
        'pOrganizationChartNode',
        ''
      ]

    ]
    ,(null as any),(null as any),(null as any),View_OrganizationChartNode_0,RenderType_OrganizationChartNode)),
      import0.ɵdid(24576,(null as any),0,import1.OrganizationChartNode,[import1.OrganizationChart],{node: [
        0,
        'node'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit;
    ck(v,3,0,currVal_0);
  },(null as any));
}
export function View_OrganizationChartNode_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),18,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),15,'td',([] as any[]),[[
        1,
        'colspan',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),12,'div',([] as any[]),(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.OrganizationChartNode = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onNodeClick($event,co.node)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-organizationchart-selectable-node',
      'ui-state-highlight'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_OrganizationChartNode_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_OrganizationChartNode_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_OrganizationChartNode_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'tr',[[
        'class',
        'ui-organizationchart-lines'
      ]
    ],[
      [
        4,
        'visibility',
        (null as any)
      ]
      ,
      [
        24,
        '@childState',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'td',([] as any[]),[[
        1,
        'colspan',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'ui-organizationchart-line-down'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tr',[[
        'class',
        'ui-organizationchart-lines'
      ]
    ],[
      [
        4,
        'visibility',
        (null as any)
      ]
      ,
      [
        24,
        '@childState',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_OrganizationChartNode_4)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tr',[[
        'class',
        'ui-organizationchart-nodes'
      ]
    ],[
      [
        4,
        'visibility',
        (null as any)
      ]
      ,
      [
        24,
        '@childState',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_OrganizationChartNode_5)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.OrganizationChartNode = v.component;
    const currVal_1:any = import0.ɵinlineInterpolate(1,'ui-organizationchart-node-content ui-widget-content ui-corner-all ',co.node.styleClass,'');
    const currVal_2:any = ck(v,7,0,(co.chart.selectionMode && (co.node.selectable !== false)),co.isSelected());
    ck(v,6,0,currVal_1,currVal_2);
    const currVal_3:boolean = !co.chart.getTemplateForNode(co.node);
    ck(v,10,0,currVal_3);
    const currVal_4:any = co.chart.getTemplateForNode(co.node);
    ck(v,13,0,currVal_4);
    const currVal_5:boolean = !co.leaf;
    ck(v,16,0,currVal_5);
    const currVal_11:any = co.node.children;
    ck(v,32,0,currVal_11);
    const currVal_14:any = co.node.children;
    ck(v,38,0,currVal_14);
  },(ck,v) => {
    var co:import1.OrganizationChartNode = v.component;
    const currVal_0:any = co.colspan;
    ck(v,3,0,currVal_0);
    const currVal_6:any = ((!co.leaf && co.node.expanded)? 'inherit': 'hidden');
    const currVal_7:any = 'in';
    ck(v,21,0,currVal_6,currVal_7);
    const currVal_8:any = co.colspan;
    ck(v,23,0,currVal_8);
    const currVal_9:any = ((!co.leaf && co.node.expanded)? 'inherit': 'hidden');
    const currVal_10:any = 'in';
    ck(v,29,0,currVal_9,currVal_10);
    const currVal_12:any = ((!co.leaf && co.node.expanded)? 'inherit': 'hidden');
    const currVal_13:any = 'in';
    ck(v,35,0,currVal_12,currVal_13);
  });
}
function View_OrganizationChartNode_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'pOrganizationChartNode',
        ''
      ]
    ],(null as any),(null as any),(null as any),View_OrganizationChartNode_0,RenderType_OrganizationChartNode)),
    import0.ɵdid(24576,(null as any),0,import1.OrganizationChartNode,[import1.OrganizationChart],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const OrganizationChartNodeNgFactory:import0.ComponentFactory<import1.OrganizationChartNode> = import0.ɵccf('[pOrganizationChartNode]',import1.OrganizationChartNode,View_OrganizationChartNode_Host_0,{
  node: 'node',
  root: 'root',
  first: 'first',
  last: 'last'
}
,{},([] as any[]));
const styles_OrganizationChartNodeTemplateLoader:any[] = ([] as any[]);
export const RenderType_OrganizationChartNodeTemplateLoader:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_OrganizationChartNodeTemplateLoader,
  data: {}
}
);
export function View_OrganizationChartNodeTemplateLoader_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_OrganizationChartNodeTemplateLoader_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,'p-organizationChartNodeTemplateLoader',([] as any[]),(null as any),(null as any),(null as any),View_OrganizationChartNodeTemplateLoader_0,RenderType_OrganizationChartNodeTemplateLoader)),
    import0.ɵdid(122880,(null as any),0,import1.OrganizationChartNodeTemplateLoader,[import0.ViewContainerRef],(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const OrganizationChartNodeTemplateLoaderNgFactory:import0.ComponentFactory<import1.OrganizationChartNodeTemplateLoader> = import0.ɵccf('p-organizationChartNodeTemplateLoader',import1.OrganizationChartNodeTemplateLoader,View_OrganizationChartNodeTemplateLoader_Host_0,{
  node: 'node',
  template: 'template'
}
,{},([] as any[]));
