// primeng/components/paginator/paginator.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/paginator/paginator';
import * as import2 from '@angular/common';
class PaginatorModuleInjector extends import0.ɵNgModuleInjector<import1.PaginatorModule> {
  _CommonModule_0:import2.CommonModule;
  _PaginatorModule_1:import1.PaginatorModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.PaginatorModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._PaginatorModule_1 = new import1.PaginatorModule();
    return this._PaginatorModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.PaginatorModule)) { return this._PaginatorModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const PaginatorModuleNgFactory:import0.NgModuleFactory<import1.PaginatorModule> = new import0.NgModuleFactory<any>(PaginatorModuleInjector,import1.PaginatorModule);
const styles_Paginator:any[] = ([] as any[]);
export const RenderType_Paginator:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Paginator,
  data: {}
}
);
function View_Paginator_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'a',[
      [
        'class',
        'ui-paginator-page ui-paginator-element ui-state-default ui-corner-all'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.changePage((v.context.$implicit - 1),$event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-active']),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-paginator-page ui-paginator-element ui-state-default ui-corner-all';
    const currVal_1:any = ck(v,2,0,((v.context.$implicit - 1) == co.getPage()));
    ck(v,1,0,currVal_0,currVal_1);
  },(ck,v) => {
    const currVal_2:any = v.context.$implicit;
    ck(v,3,0,currVal_2);
  });
}
function View_Paginator_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'option',([] as any[]),[
      [
        8,
        'value',
        0
      ]
      ,
      [
        8,
        'selected',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = v.context.$implicit;
    const currVal_1:any = (co.rows == v.context.$implicit);
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_2:any = v.context.$implicit;
    ck(v,1,0,currVal_2);
  });
}
function View_Paginator_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'select',[[
        'class',
        'ui-paginator-rpp-options ui-widget ui-state-default'
      ]
      ],(null as any),[[
        (null as any),
        'change'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.onRppChange($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Paginator_4)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.rowsPerPageOptions;
    ck(v,3,0,currVal_0);
  },(null as any));
}
function View_Paginator_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),40,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'a',[
      [
        'class',
        'ui-paginator-first ui-paginator-element ui-state-default ui-corner-all'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,[[
        8,
        'tabIndex',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.changePageToFirst($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled']),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-step-backward'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'a',[
      [
        'class',
        'ui-paginator-prev ui-paginator-element ui-state-default ui-corner-all'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,[[
        8,
        'tabIndex',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.changePageToPrev($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled']),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-backward'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'span',[[
        'class',
        'ui-paginator-pages'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Paginator_2)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'a',[
      [
        'class',
        'ui-paginator-next ui-paginator-element ui-state-default ui-corner-all'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,[[
        8,
        'tabIndex',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.changePageToNext($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled']),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-forward'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'a',[
      [
        'class',
        'ui-paginator-last ui-paginator-element ui-state-default ui-corner-all'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,[[
        8,
        'tabIndex',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.changePageToLast($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled']),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-step-forward'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Paginator_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-paginator ui-widget ui-widget-header ui-unselectable-text';
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,2,0,currVal_2);
    const currVal_4:any = 'ui-paginator-first ui-paginator-element ui-state-default ui-corner-all';
    const currVal_5:any = ck(v,6,0,co.isFirstPage());
    ck(v,5,0,currVal_4,currVal_5);
    const currVal_7:any = 'ui-paginator-prev ui-paginator-element ui-state-default ui-corner-all';
    const currVal_8:any = ck(v,13,0,co.isFirstPage());
    ck(v,12,0,currVal_7,currVal_8);
    const currVal_9:any = co.pageLinks;
    ck(v,21,0,currVal_9);
    const currVal_11:any = 'ui-paginator-next ui-paginator-element ui-state-default ui-corner-all';
    const currVal_12:any = ck(v,26,0,co.isLastPage());
    ck(v,25,0,currVal_11,currVal_12);
    const currVal_14:any = 'ui-paginator-last ui-paginator-element ui-state-default ui-corner-all';
    const currVal_15:any = ck(v,33,0,co.isLastPage());
    ck(v,32,0,currVal_14,currVal_15);
    const currVal_16:any = co.rowsPerPageOptions;
    ck(v,39,0,currVal_16);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = (co.isFirstPage()? (0 - 1): (null as any));
    ck(v,4,0,currVal_3);
    const currVal_6:any = (co.isFirstPage()? (0 - 1): (null as any));
    ck(v,11,0,currVal_6);
    const currVal_10:any = (co.isLastPage()? (0 - 1): (null as any));
    ck(v,24,0,currVal_10);
    const currVal_13:any = (co.isLastPage()? (0 - 1): (null as any));
    ck(v,31,0,currVal_13);
  });
}
export function View_Paginator_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Paginator_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Paginator = v.component;
    const currVal_0:any = (co.alwaysShow? true: (co.pageLinks && (co.pageLinks.length > 1)));
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_Paginator_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-paginator',([] as any[]),(null as any),(null as any),(null as any),View_Paginator_0,RenderType_Paginator)),
    import0.ɵdid(24576,(null as any),0,import1.Paginator,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const PaginatorNgFactory:import0.ComponentFactory<import1.Paginator> = import0.ɵccf('p-paginator',import1.Paginator,View_Paginator_Host_0,{
  pageLinkSize: 'pageLinkSize',
  style: 'style',
  styleClass: 'styleClass',
  rowsPerPageOptions: 'rowsPerPageOptions',
  alwaysShow: 'alwaysShow',
  totalRecords: 'totalRecords',
  first: 'first',
  rows: 'rows'
}
,{onPageChange: 'onPageChange'},([] as any[]));
