// primeng/components/panel/panel.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/panel/panel';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/common/shared';
class PanelModuleInjector extends import0.ɵNgModuleInjector<import1.PanelModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _PanelModule_2:import1.PanelModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.PanelModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._PanelModule_2 = new import1.PanelModule();
    return this._PanelModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import1.PanelModule)) { return this._PanelModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const PanelModuleNgFactory:import0.NgModuleFactory<import1.PanelModule> = new import0.NgModuleFactory<any>(PanelModuleInjector,import1.PanelModule);
const styles_Panel:any[] = ([] as any[]);
export const RenderType_Panel:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Panel,
      data: {'animation': [{
        name: 'panelContent',
        definitions: [
          {
            type: 0,
            name: 'hidden',
            styles: {
              type: 6,
              styles: {height: '0'}
            }

          }
          ,
          {
            type: 0,
            name: 'visible',
            styles: {
              type: 6,
              styles: {height: '*'}
            }

          }
          ,
          {
            type: 1,
            expr: 'visible <=> hidden',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms cubic-bezier(0.86, 0, 0.07, 1)'
            }

          }

        ]

      }
  ]}
}
);
function View_Panel_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-panel-title'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.header;
    ck(v,1,0,currVal_0);
  });
}
function View_Panel_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'a',[
      [
        'class',
        'ui-panel-titlebar-icon ui-panel-titlebar-toggler ui-corner-all ui-state-default'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggle($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.collapsed? ('fa fa-fw ' + co.expandIcon): ('fa fa-fw ' + co.collapseIcon));
    ck(v,2,0,currVal_0);
  });
}
function View_Panel_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-panel-footer ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    import0.ɵncd((null as any),2),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(null as any),(null as any));
}
export function View_Panel_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),27,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),9,'div',[[
        'class',
        'ui-panel-titlebar ui-widget-header ui-helper-clearfix ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Panel_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Panel_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'div',[[
        'class',
        'ui-panel-content-wrapper'
      ]
      ],[[
        24,
        '@panelContent',
        0
      ]
      ],[[
        (null as any),
        '@panelContent.done'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Panel = v.component;
      if (('@panelContent.done' === en)) {
        const pd_0:any = ((<any>co.onToggleDone($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-panel-content-wrapper-overflown']),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-panel-content ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    import0.ɵncd((null as any),1),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n                \n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Panel_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Panel = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-panel ui-widget ui-widget-content ui-corner-all';
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_3:any = co.header;
    ck(v,8,0,currVal_3);
    const currVal_4:any = co.toggleable;
    ck(v,13,0,currVal_4);
    const currVal_6:any = 'ui-panel-content-wrapper';
    const currVal_7:any = ck(v,18,0,(co.collapsed || co.animating));
    ck(v,17,0,currVal_6,currVal_7);
    const currVal_8:any = co.footerFacet;
    ck(v,26,0,currVal_8);
  },(ck,v) => {
    var co:import1.Panel = v.component;
    const currVal_5:any = (co.collapsed? 'hidden': 'visible');
    ck(v,16,0,currVal_5);
  });
}
function View_Panel_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-panel',([] as any[]),(null as any),(null as any),(null as any),View_Panel_0,RenderType_Panel)),
    import0.ɵdid(24576,(null as any),1,import1.Panel,[import0.ElementRef],(null as any),(null as any)),
    import0.ɵqud(167772160,1,{footerFacet: 0})
  ]
  ,(null as any),(null as any));
}
export const PanelNgFactory:import0.ComponentFactory<import1.Panel> = import0.ɵccf('p-panel',import1.Panel,View_Panel_Host_0,{
  toggleable: 'toggleable',
  header: 'header',
  collapsed: 'collapsed',
  style: 'style',
  styleClass: 'styleClass',
  expandIcon: 'expandIcon',
  collapseIcon: 'collapseIcon'
}
,{
  collapsedChange: 'collapsedChange',
  onBeforeToggle: 'onBeforeToggle',
  onAfterToggle: 'onAfterToggle'
}
,[
  'p-header',
  '*',
  'p-footer'
]
);
