// primeng/components/progressbar/progressbar.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/progressbar/progressbar';
import * as import2 from '@angular/common';
class ProgressBarModuleInjector extends import0.ɵNgModuleInjector<import1.ProgressBarModule> {
  _CommonModule_0:import2.CommonModule;
  _ProgressBarModule_1:import1.ProgressBarModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.ProgressBarModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ProgressBarModule_1 = new import1.ProgressBarModule();
    return this._ProgressBarModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.ProgressBarModule)) { return this._ProgressBarModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const ProgressBarModuleNgFactory:import0.NgModuleFactory<import1.ProgressBarModule> = new import0.NgModuleFactory<any>(ProgressBarModuleInjector,import1.ProgressBarModule);
const styles_ProgressBar:any[] = ([] as any[]);
export const RenderType_ProgressBar:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ProgressBar,
  data: {}
}
);
function View_ProgressBar_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'ui-progressbar-label'
      ]
      ],[[
        4,
        'display',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.value? 'block': 'none');
    ck(v,0,0,currVal_0);
    const currVal_1:any = co.value;
    const currVal_2:any = co.unit;
    ck(v,1,0,currVal_1,currVal_2);
  });
}
export function View_ProgressBar_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',[
      [
        'aria-valuemax',
        '100'
      ]
      ,
      [
        'aria-valuemin',
        '0'
      ]
      ,
      [
        'class',
        'ui-progressbar ui-widget ui-widget-content ui-corner-all'
      ]
      ,
      [
        'role',
        'progressbar'
      ]

    ]
      ,[[
        1,
        'aria-valuenow',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[
      [
        'class',
        'ui-progressbar-value ui-progressbar-value-animate ui-widget-header ui-corner-all'
      ]
      ,
      [
        'style',
        'display:block'
      ]

    ]
      ,[[
        4,
        'width',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ProgressBar_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.ProgressBar = v.component;
    const currVal_2:any = co.showValue;
    ck(v,6,0,currVal_2);
  },(ck,v) => {
    var co:import1.ProgressBar = v.component;
    const currVal_0:any = co.value;
    ck(v,1,0,currVal_0);
    const currVal_1:any = (co.value + '%');
    ck(v,3,0,currVal_1);
  });
}
function View_ProgressBar_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-progressBar',([] as any[]),(null as any),(null as any),(null as any),View_ProgressBar_0,RenderType_ProgressBar)),
    import0.ɵdid(24576,(null as any),0,import1.ProgressBar,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ProgressBarNgFactory:import0.ComponentFactory<import1.ProgressBar> = import0.ɵccf('p-progressBar',import1.ProgressBar,View_ProgressBar_Host_0,{
  value: 'value',
  showValue: 'showValue',
  unit: 'unit'
}
,{},([] as any[]));
