// @swimlane/ngx-datatable/release/components/datatable.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from './datatable.component.css.ngstyle';
import * as import1 from '@angular/core';
import * as import2 from './header/header.component.ngfactory';
import * as import3 from '@swimlane/ngx-datatable/release/components/header/header.component';
import * as import4 from './footer/footer.component.ngfactory';
import * as import5 from '@swimlane/ngx-datatable/release/components/footer/footer.component';
import * as import6 from '@swimlane/ngx-datatable/release/components/datatable.component';
import * as import7 from '@swimlane/ngx-datatable/release/directives/visibility.directive';
import * as import8 from '@angular/common';
import * as import9 from './body/body.component.ngfactory';
import * as import10 from '@swimlane/ngx-datatable/release/components/body/body.component';
import * as import11 from '@swimlane/ngx-datatable/release/services/scrollbar-helper.service';
const styles_DatatableComponent:any[] = [import0.styles];
export const RenderType_DatatableComponent:import1.RendererType2 = import1.ɵcrt({
  encapsulation: 2,
  styles: styles_DatatableComponent,
  data: {}
}
);
function View_DatatableComponent_1(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
      (l()(),import1.ɵeld(0,(null as any),(null as any),2,'datatable-header',[[
        'class',
        'datatable-header'
      ]
    ],[
      [
        4,
        'height',
        (null as any)
      ]
      ,
      [
        4,
        'width',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'sort'
      ]
      ,
      [
        (null as any),
        'resize'
      ]
      ,
      [
        (null as any),
        'reorder'
      ]
      ,
      [
        (null as any),
        'select'
      ]
      ,
      [
        (null as any),
        'columnContextmenu'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('sort' === en)) {
        const pd_0:any = ((<any>co.onColumnSort($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('resize' === en)) {
        const pd_1:any = ((<any>co.onColumnResize($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('reorder' === en)) {
        const pd_2:any = ((<any>co.onColumnReorder($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('select' === en)) {
        const pd_3:any = ((<any>co.onHeaderSelect($event)) !== false);
        ad = (pd_3 && ad);
      }
      if (('columnContextmenu' === en)) {
        const pd_4:any = ((<any>co.onColumnContextmenu($event)) !== false);
        ad = (pd_4 && ad);
      }
      return ad;
    },import2.View_DataTableHeaderComponent_0,import2.RenderType_DataTableHeaderComponent)),
    import1.ɵdid(24576,(null as any),0,import3.DataTableHeaderComponent,([] as any[]),{
      sortAscendingIcon: [
        0,
        'sortAscendingIcon'
      ]
      ,
      sortDescendingIcon: [
        1,
        'sortDescendingIcon'
      ]
      ,
      scrollbarH: [
        2,
        'scrollbarH'
      ]
      ,
      innerWidth: [
        3,
        'innerWidth'
      ]
      ,
      offsetX: [
        4,
        'offsetX'
      ]
      ,
      sorts: [
        5,
        'sorts'
      ]
      ,
      sortType: [
        6,
        'sortType'
      ]
      ,
      allRowsSelected: [
        7,
        'allRowsSelected'
      ]
      ,
      selectionType: [
        8,
        'selectionType'
      ]
      ,
      reorderable: [
        9,
        'reorderable'
      ]
      ,
      headerHeight: [
        10,
        'headerHeight'
      ]
      ,
      columns: [
        11,
        'columns'
      ]

    }
    ,{
      sort: 'sort',
      reorder: 'reorder',
      resize: 'resize',
      select: 'select',
      columnContextmenu: 'columnContextmenu'
    }
    ),
    (l()(),import1.ɵted((null as any),['\n      ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co.cssClasses.sortAscending;
    const currVal_3:any = co.cssClasses.sortDescending;
    const currVal_4:any = co.scrollbarH;
    const currVal_5:any = co.innerWidth;
    const currVal_6:any = co.offsetX;
    const currVal_7:any = co.sorts;
    const currVal_8:any = co.sortType;
    const currVal_9:any = co.allRowsSelected;
    const currVal_10:any = co.selectionType;
    const currVal_11:any = co.reorderable;
    const currVal_12:any = co.headerHeight;
    const currVal_13:any = co.columns;
    ck(v,1,1,[
      currVal_2,
      currVal_3,
      currVal_4,
      currVal_5,
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10,
      currVal_11,
      currVal_12,
      currVal_13
    ]
    );
  },(ck,v) => {
    const currVal_0:any = import1.ɵnov(v,1).headerHeight;
    const currVal_1:any = import1.ɵnov(v,1).headerWidth;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_DatatableComponent_2(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
      (l()(),import1.ɵeld(0,(null as any),(null as any),2,'datatable-footer',[[
        'class',
        'datatable-footer'
      ]
      ],(null as any),[[
        (null as any),
        'page'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('page' === en)) {
        const pd_0:any = ((<any>co.onFooterPage($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import4.View_DataTableFooterComponent_0,import4.RenderType_DataTableFooterComponent)),
    import1.ɵdid(24576,(null as any),0,import5.DataTableFooterComponent,([] as any[]),{
      footerHeight: [
        0,
        'footerHeight'
      ]
      ,
      rowCount: [
        1,
        'rowCount'
      ]
      ,
      pageSize: [
        2,
        'pageSize'
      ]
      ,
      offset: [
        3,
        'offset'
      ]
      ,
      pagerLeftArrowIcon: [
        4,
        'pagerLeftArrowIcon'
      ]
      ,
      pagerRightArrowIcon: [
        5,
        'pagerRightArrowIcon'
      ]
      ,
      pagerPreviousIcon: [
        6,
        'pagerPreviousIcon'
      ]
      ,
      pagerNextIcon: [
        7,
        'pagerNextIcon'
      ]
      ,
      totalMessage: [
        8,
        'totalMessage'
      ]
      ,
      footerTemplate: [
        9,
        'footerTemplate'
      ]
      ,
      selectedCount: [
        10,
        'selectedCount'
      ]
      ,
      selectedMessage: [
        11,
        'selectedMessage'
      ]

    }
    ,{page: 'page'}),
    (l()(),import1.ɵted((null as any),['\n      ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.footerHeight;
    const currVal_1:any = co.rowCount;
    const currVal_2:any = co.pageSize;
    const currVal_3:any = co.offset;
    const currVal_4:any = co.cssClasses.pagerLeftArrow;
    const currVal_5:any = co.cssClasses.pagerRightArrow;
    const currVal_6:any = co.cssClasses.pagerPrevious;
    const currVal_7:any = co.cssClasses.pagerNext;
    const currVal_8:any = co.messages.totalMessage;
    const currVal_9:any = co.footer;
    const currVal_10:any = co.selected.length;
    const currVal_11:boolean = (!!co.selectionType && co.messages.selectedMessage);
    ck(v,1,1,[
      currVal_0,
      currVal_1,
      currVal_2,
      currVal_3,
      currVal_4,
      currVal_5,
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10,
      currVal_11
    ]
    );
  },(null as any));
}
export function View_DatatableComponent_0(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
    import1.ɵqud(201326592,1,{bodyComponent: 0}),
    (l()(),import1.ɵted((null as any),['\n    '])),
      (l()(),import1.ɵeld(0,(null as any),(null as any),12,'div',[[
        'visibilityObserver',
        ''
      ]
      ],[[
        2,
        'visible',
        (null as any)
      ]
      ],[[
        (null as any),
        'visible'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import6.DatatableComponent = v.component;
      if (('visible' === en)) {
        const pd_0:any = ((<any>co.recalculate()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import1.ɵdid(106496,(null as any),0,import7.VisibilityDirective,[
      import1.ElementRef,
      import1.NgZone
    ]
    ,(null as any),{visible: 'visible'}),
    (l()(),import1.ɵted((null as any),['\n      '])),
    (l()(),import1.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DatatableComponent_1)),
    import1.ɵdid(8192,(null as any),0,import8.NgIf,[
      import1.ViewContainerRef,
      import1.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import1.ɵted((null as any),['\n      '])),
      (l()(),import1.ɵeld(0,(null as any),(null as any),2,'datatable-body',[[
        'class',
        'datatable-body'
      ]
    ],[
      [
        4,
        'width',
        (null as any)
      ]
      ,
      [
        4,
        'height',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'page'
      ]
      ,
      [
        (null as any),
        'activate'
      ]
      ,
      [
        (null as any),
        'rowContextmenu'
      ]
      ,
      [
        (null as any),
        'select'
      ]
      ,
      [
        (null as any),
        'scroll'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import6.DatatableComponent = v.component;
      if (('page' === en)) {
        const pd_0:any = ((<any>co.onBodyPage($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('activate' === en)) {
        const pd_1:any = ((<any>co.activate.emit($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('rowContextmenu' === en)) {
        const pd_2:any = ((<any>co.onRowContextmenu($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('select' === en)) {
        const pd_3:any = ((<any>co.onBodySelect($event)) !== false);
        ad = (pd_3 && ad);
      }
      if (('scroll' === en)) {
        const pd_4:any = ((<any>co.onBodyScroll($event)) !== false);
        ad = (pd_4 && ad);
      }
      return ad;
    },import9.View_DataTableBodyComponent_0,import9.RenderType_DataTableBodyComponent)),
      import1.ɵdid(122880,[[
        1,
        4
      ]
    ],0,import10.DataTableBodyComponent,([] as any[]),{
      scrollbarV: [
        0,
        'scrollbarV'
      ]
      ,
      scrollbarH: [
        1,
        'scrollbarH'
      ]
      ,
      loadingIndicator: [
        2,
        'loadingIndicator'
      ]
      ,
      externalPaging: [
        3,
        'externalPaging'
      ]
      ,
      rowHeight: [
        4,
        'rowHeight'
      ]
      ,
      offsetX: [
        5,
        'offsetX'
      ]
      ,
      emptyMessage: [
        6,
        'emptyMessage'
      ]
      ,
      selectionType: [
        7,
        'selectionType'
      ]
      ,
      selected: [
        8,
        'selected'
      ]
      ,
      rowIdentity: [
        9,
        'rowIdentity'
      ]
      ,
      rowDetail: [
        10,
        'rowDetail'
      ]
      ,
      selectCheck: [
        11,
        'selectCheck'
      ]
      ,
      trackByProp: [
        12,
        'trackByProp'
      ]
      ,
      rowClass: [
        13,
        'rowClass'
      ]
      ,
      pageSize: [
        14,
        'pageSize'
      ]
      ,
      rows: [
        15,
        'rows'
      ]
      ,
      columns: [
        16,
        'columns'
      ]
      ,
      offset: [
        17,
        'offset'
      ]
      ,
      rowCount: [
        18,
        'rowCount'
      ]
      ,
      innerWidth: [
        19,
        'innerWidth'
      ]
      ,
      bodyHeight: [
        20,
        'bodyHeight'
      ]

    }
    ,{
      scroll: 'scroll',
      page: 'page',
      activate: 'activate',
      select: 'select',
      rowContextmenu: 'rowContextmenu'
    }
    ),
    (l()(),import1.ɵted((null as any),['\n      '])),
    (l()(),import1.ɵted((null as any),['\n      '])),
    (l()(),import1.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DatatableComponent_2)),
    import1.ɵdid(8192,(null as any),0,import8.NgIf,[
      import1.ViewContainerRef,
      import1.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import1.ɵted((null as any),['\n    '])),
    (l()(),import1.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import6.DatatableComponent = v.component;
    ck(v,3,0);
    const currVal_1:any = co.headerHeight;
    ck(v,6,0,currVal_1);
    const currVal_4:any = co.scrollbarV;
    const currVal_5:any = co.scrollbarH;
    const currVal_6:any = co.loadingIndicator;
    const currVal_7:any = co.externalPaging;
    const currVal_8:any = co.rowHeight;
    const currVal_9:any = co.offsetX;
    const currVal_10:any = co.messages.emptyMessage;
    const currVal_11:any = co.selectionType;
    const currVal_12:any = co.selected;
    const currVal_13:any = co.rowIdentity;
    const currVal_14:any = co.rowDetail;
    const currVal_15:any = co.selectCheck;
    const currVal_16:any = co.trackByProp;
    const currVal_17:any = co.rowClass;
    const currVal_18:any = co.pageSize;
    const currVal_19:any = co.rows;
    const currVal_20:any = co.columns;
    const currVal_21:any = co.offset;
    const currVal_22:any = co.rowCount;
    const currVal_23:any = co.innerWidth;
    const currVal_24:any = co.bodyHeight;
    ck(v,9,1,[
      currVal_4,
      currVal_5,
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10,
      currVal_11,
      currVal_12,
      currVal_13,
      currVal_14,
      currVal_15,
      currVal_16,
      currVal_17,
      currVal_18,
      currVal_19,
      currVal_20,
      currVal_21,
      currVal_22,
      currVal_23,
      currVal_24
    ]
    );
    const currVal_25:any = co.footerHeight;
    ck(v,13,0,currVal_25);
  },(ck,v) => {
    const currVal_0:any = import1.ɵnov(v,3).isVisible;
    ck(v,2,0,currVal_0);
    const currVal_2:any = import1.ɵnov(v,9).bodyWidth;
    const currVal_3:any = import1.ɵnov(v,9).bodyHeight;
    ck(v,8,0,currVal_2,currVal_3);
  });
}
function View_DatatableComponent_Host_0(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
      (l()(),import1.ɵeld(0,(null as any),(null as any),4,'ngx-datatable',[[
        'class',
        'ngx-datatable'
      ]
    ],[
      [
        2,
        'fixed-header',
        (null as any)
      ]
      ,
      [
        2,
        'fixed-row',
        (null as any)
      ]
      ,
      [
        2,
        'scroll-vertical',
        (null as any)
      ]
      ,
      [
        2,
        'scroll-horz',
        (null as any)
      ]
      ,
      [
        2,
        'selectable',
        (null as any)
      ]
      ,
      [
        2,
        'checkbox-selection',
        (null as any)
      ]
      ,
      [
        2,
        'cell-selection',
        (null as any)
      ]
      ,
      [
        2,
        'single-selection',
        (null as any)
      ]
      ,
      [
        2,
        'multi-selection',
        (null as any)
      ]
      ,
      [
        2,
        'multi-click-selection',
        (null as any)
      ]

    ]
      ,[[
        'window',
        'resize'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('window:resize' === en)) {
        const pd_0:any = ((<any>import1.ɵnov(v,1).onWindowResize()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_DatatableComponent_0,RenderType_DatatableComponent)),
    import1.ɵdid(2285568,(null as any),3,import6.DatatableComponent,[
      import11.ScrollbarHelper,
      import1.ElementRef,
      import1.KeyValueDiffers
    ]
    ,(null as any),(null as any)),
    import1.ɵqud(301989888,1,{columnTemplates: 1}),
    import1.ɵqud(167772160,2,{rowDetail: 0}),
    import1.ɵqud(167772160,3,{footer: 0})
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import1.ɵnov(v,1).isFixedHeader;
    const currVal_1:any = import1.ɵnov(v,1).isFixedRow;
    const currVal_2:any = import1.ɵnov(v,1).isVertScroll;
    const currVal_3:any = import1.ɵnov(v,1).isHorScroll;
    const currVal_4:any = import1.ɵnov(v,1).isSelectable;
    const currVal_5:any = import1.ɵnov(v,1).isCheckboxSelection;
    const currVal_6:any = import1.ɵnov(v,1).isCellSelection;
    const currVal_7:any = import1.ɵnov(v,1).isSingleSelection;
    const currVal_8:any = import1.ɵnov(v,1).isMultiSelection;
    const currVal_9:any = import1.ɵnov(v,1).isMultiClickSelection;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9);
  });
}
export const DatatableComponentNgFactory:import1.ComponentFactory<import6.DatatableComponent> = import1.ɵccf('ngx-datatable',import6.DatatableComponent,View_DatatableComponent_Host_0,{
  rows: 'rows',
  columns: 'columns',
  selected: 'selected',
  scrollbarV: 'scrollbarV',
  scrollbarH: 'scrollbarH',
  rowHeight: 'rowHeight',
  columnMode: 'columnMode',
  headerHeight: 'headerHeight',
  footerHeight: 'footerHeight',
  externalPaging: 'externalPaging',
  externalSorting: 'externalSorting',
  limit: 'limit',
  count: 'count',
  offset: 'offset',
  loadingIndicator: 'loadingIndicator',
  selectionType: 'selectionType',
  reorderable: 'reorderable',
  sortType: 'sortType',
  sorts: 'sorts',
  cssClasses: 'cssClasses',
  messages: 'messages',
  rowIdentity: 'rowIdentity',
  rowClass: 'rowClass',
  selectCheck: 'selectCheck',
  trackByProp: 'trackByProp'
}
,{
  scroll: 'scroll',
  activate: 'activate',
  select: 'select',
  sort: 'sort',
  page: 'page',
  reorder: 'reorder',
  resize: 'resize',
  tableContextmenu: 'tableContextmenu'
}
,([] as any[]));
