// primeng/components/accordion/accordion.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/accordion/accordion';
import * as import2 from '@angular/common';
class AccordionModuleInjector extends import0.ɵNgModuleInjector<import1.AccordionModule> {
  _CommonModule_0:import2.CommonModule;
  _AccordionModule_1:import1.AccordionModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.AccordionModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._AccordionModule_1 = new import1.AccordionModule();
    return this._AccordionModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.AccordionModule)) { return this._AccordionModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const AccordionModuleNgFactory:import0.NgModuleFactory<import1.AccordionModule> = new import0.NgModuleFactory<any>(AccordionModuleInjector,import1.AccordionModule);
const styles_Accordion:any[] = ([] as any[]);
export const RenderType_Accordion:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Accordion,
  data: {}
}
);
export function View_Accordion_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Accordion = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-accordion ui-widget ui-helper-reset';
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
  },(null as any));
}
function View_Accordion_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-accordion',([] as any[]),(null as any),(null as any),(null as any),View_Accordion_0,RenderType_Accordion)),
    import0.ɵdid(24576,(null as any),0,import1.Accordion,[import0.ElementRef],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const AccordionNgFactory:import0.ComponentFactory<import1.Accordion> = import0.ɵccf('p-accordion',import1.Accordion,View_Accordion_Host_0,{
  multiple: 'multiple',
  style: 'style',
  styleClass: 'styleClass',
  lazy: 'lazy',
  activeIndex: 'activeIndex'
}
,{
  onClose: 'onClose',
  onOpen: 'onOpen'
}
,['*']);
const styles_AccordionTab:any[] = ([] as any[]);
export const RenderType_AccordionTab:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_AccordionTab,
      data: {'animation': [{
        name: 'tabContent',
        definitions: [
          {
            type: 0,
            name: 'hidden',
            styles: {
              type: 6,
              styles: {height: '0'}
            }

          }
          ,
          {
            type: 0,
            name: 'visible',
            styles: {
              type: 6,
              styles: {height: '*'}
            }

          }
          ,
          {
            type: 1,
            expr: 'visible <=> hidden',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms cubic-bezier(0.86, 0, 0.07, 1)'
            }

          }

        ]

      }
  ]}
}
);
function View_AccordionTab_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'a',[
      [
        'href',
        '#'
      ]
      ,
      [
        'role',
        'tab'
      ]

    ]
    ,[
      [
        1,
        'aria-expanded',
        0
      ]
      ,
      [
        1,
        'aria-selected',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.selected;
    const currVal_1:any = co.selected;
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_2:any = co.header;
    ck(v,1,0,currVal_2);
  });
}
function View_AccordionTab_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'a',[
      [
        'href',
        '#'
      ]
      ,
      [
        'role',
        'tab'
      ]

    ]
    ,[
      [
        1,
        'aria-expanded',
        0
      ]
      ,
      [
        1,
        'aria-selected',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.selected;
    const currVal_1:any = co.selected;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_AccordionTab_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-accordion-content ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),1),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(null as any));
}
export function View_AccordionTab_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),13,'div',[[
        'class',
        'ui-accordion-header ui-state-default ui-corner-all'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.AccordionTab = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggle($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-state-active',
      'ui-state-disabled'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'fa fa-fw'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'fa-caret-down',
      'fa-caret-right'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AccordionTab_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AccordionTab_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',[
      [
        'class',
        'ui-accordion-content-wrapper'
      ]
      ,
      [
        'role',
        'tabpanel'
      ]

    ]
    ,[
      [
        24,
        '@tabContent',
        0
      ]
      ,
      [
        1,
        'aria-hidden',
        0
      ]

    ]
      ,[[
        (null as any),
        '@tabContent.done'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.AccordionTab = v.component;
      if (('@tabContent.done' === en)) {
        const pd_0:any = ((<any>co.onToggleDone($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-accordion-content-wrapper-overflown']),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AccordionTab_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.AccordionTab = v.component;
    const currVal_0:any = 'ui-accordion-header ui-state-default ui-corner-all';
    const currVal_1:any = ck(v,3,0,co.selected,co.disabled);
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = 'fa fa-fw';
    const currVal_3:any = ck(v,7,0,co.selected,!co.selected);
    ck(v,6,0,currVal_2,currVal_3);
    const currVal_4:boolean = !co.hasHeaderFacet;
    ck(v,10,0,currVal_4);
    const currVal_5:any = co.hasHeaderFacet;
    ck(v,13,0,currVal_5);
    const currVal_8:any = 'ui-accordion-content-wrapper';
    const currVal_9:any = ck(v,18,0,(!co.selected || co.animating));
    ck(v,17,0,currVal_8,currVal_9);
    const currVal_10:any = (co.lazy? co.selected: true);
    ck(v,21,0,currVal_10);
  },(ck,v) => {
    var co:import1.AccordionTab = v.component;
    const currVal_6:any = (co.selected? 'visible': 'hidden');
    const currVal_7:boolean = !co.selected;
    ck(v,16,0,currVal_6,currVal_7);
  });
}
function View_AccordionTab_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-accordionTab',([] as any[]),(null as any),(null as any),(null as any),View_AccordionTab_0,RenderType_AccordionTab)),
    import0.ɵdid(90112,(null as any),1,import1.AccordionTab,[import1.Accordion],(null as any),(null as any)),
    import0.ɵqud(301989888,1,{headerFacet: 1})
  ]
  ,(null as any),(null as any));
}
export const AccordionTabNgFactory:import0.ComponentFactory<import1.AccordionTab> = import0.ɵccf('p-accordionTab',import1.AccordionTab,View_AccordionTab_Host_0,{
  header: 'header',
  selected: 'selected',
  disabled: 'disabled'
}
,{selectedChange: 'selectedChange'},[
  'p-header',
  '*'
]
);
