// primeng/components/checkbox/checkbox.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/checkbox/checkbox';
import * as import2 from '@angular/common';
import * as import3 from '@angular/forms';
class CheckboxModuleInjector extends import0.ɵNgModuleInjector<import1.CheckboxModule> {
  _CommonModule_0:import2.CommonModule;
  _CheckboxModule_1:import1.CheckboxModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.CheckboxModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._CheckboxModule_1 = new import1.CheckboxModule();
    return this._CheckboxModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.CheckboxModule)) { return this._CheckboxModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const CheckboxModuleNgFactory:import0.NgModuleFactory<import1.CheckboxModule> = new import0.NgModuleFactory<any>(CheckboxModuleInjector,import1.CheckboxModule);
const styles_Checkbox:any[] = ([] as any[]);
export const RenderType_Checkbox:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Checkbox,
  data: {}
}
);
function View_Checkbox_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'label',[[
        'class',
        'ui-chkbox-label'
      ]
      ],[[
        1,
        'for',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onClick($event,import0.ɵnov((<any>v.parent),7),true)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.inputId;
    ck(v,0,0,currVal_0);
    const currVal_1:any = co.label;
    ck(v,1,0,currVal_1);
  });
}
export function View_Checkbox_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),19,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',[[
        'class',
        'ui-helper-hidden-accessible'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,[[
        'cb',
        1
      ]
      ],(null as any),2,'input',[[
        'type',
        'checkbox'
      ]
    ],[
      [
        1,
        'id',
        0
      ]
      ,
      [
        8,
        'name',
        0
      ]
      ,
      [
        8,
        'value',
        0
      ]
      ,
      [
        8,
        'checked',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'change'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Checkbox = v.component;
      if (('focus' === en)) {
        const pd_0:any = ((<any>co.onFocus($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>co.onBlur($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('change' === en)) {
        const pd_2:any = ((<any>co.handleChange($event)) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['ui-state-focus']),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'ui-chkbox-box ui-widget ui-corner-all ui-state-default'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Checkbox = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onClick($event,import0.ɵnov(v,7),true)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-state-active',
      'ui-state-disabled',
      'ui-state-focus'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-chkbox-icon ui-clickable'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['fa fa-check']),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Checkbox_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Checkbox = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-chkbox ui-widget';
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_9:any = ck(v,9,0,co.focused);
    ck(v,8,0,currVal_9);
    const currVal_10:any = 'ui-chkbox-box ui-widget ui-corner-all ui-state-default';
    const currVal_11:any = ck(v,14,0,co.checked,co.disabled,co.focused);
    ck(v,13,0,currVal_10,currVal_11);
    const currVal_12:any = 'ui-chkbox-icon ui-clickable';
    const currVal_13:any = ck(v,18,0,co.checked);
    ck(v,17,0,currVal_12,currVal_13);
    const currVal_14:any = co.label;
    ck(v,23,0,currVal_14);
  },(ck,v) => {
    var co:import1.Checkbox = v.component;
    const currVal_3:any = co.inputId;
    const currVal_4:any = co.name;
    const currVal_5:any = co.value;
    const currVal_6:any = co.checked;
    const currVal_7:any = co.disabled;
    const currVal_8:any = co.tabindex;
    ck(v,7,0,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8);
  });
}
function View_Checkbox_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-checkbox',([] as any[]),(null as any),(null as any),(null as any),View_Checkbox_0,RenderType_Checkbox)),
    import0.ɵdid(24576,(null as any),0,import1.Checkbox,[import0.ChangeDetectorRef],(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import3.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.Checkbox])
  ]
  ,(null as any),(null as any));
}
export const CheckboxNgFactory:import0.ComponentFactory<import1.Checkbox> = import0.ɵccf('p-checkbox',import1.Checkbox,View_Checkbox_Host_0,{
  value: 'value',
  name: 'name',
  disabled: 'disabled',
  binary: 'binary',
  label: 'label',
  tabindex: 'tabindex',
  inputId: 'inputId',
  style: 'style',
  styleClass: 'styleClass'
}
,{onChange: 'onChange'},([] as any[]));
