// primeng/components/picklist/picklist.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/picklist/picklist';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/button/button';
import * as import4 from 'primeng/components/common/shared';
import * as import5 from 'primeng/components/dom/domhandler';
import * as import6 from 'primeng/components/utils/objectutils';
class PickListModuleInjector extends import0.ɵNgModuleInjector<import1.PickListModule> {
  _CommonModule_0:import2.CommonModule;
  _ButtonModule_1:import3.ButtonModule;
  _SharedModule_2:import4.SharedModule;
  _PickListModule_3:import1.PickListModule;
  __NgLocalization_4:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_4():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_4 == null)) { (this.__NgLocalization_4 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_4;
  }
  createInternal():import1.PickListModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ButtonModule_1 = new import3.ButtonModule();
    this._SharedModule_2 = new import4.SharedModule();
    this._PickListModule_3 = new import1.PickListModule();
    return this._PickListModule_3;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.ButtonModule)) { return this._ButtonModule_1; }
    if ((token === import4.SharedModule)) { return this._SharedModule_2; }
    if ((token === import1.PickListModule)) { return this._PickListModule_3; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_4; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const PickListModuleNgFactory:import0.NgModuleFactory<import1.PickListModule> = new import0.NgModuleFactory<any>(PickListModuleInjector,import1.PickListModule);
const styles_PickList:any[] = ([] as any[]);
export const RenderType_PickList:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_PickList,
  data: {}
}
);
function View_PickList_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),20,'div',[[
        'class',
        'ui-picklist-source-controls ui-picklist-buttons'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),17,'div',[[
        'class',
        'ui-picklist-buttons-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-up'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveUp(import0.ɵnov((<any>v.parent),21),co.source,co.selectedItemsSource,co.onSourceReorder)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-double-up'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveTop(import0.ɵnov((<any>v.parent),21),co.source,co.selectedItemsSource,co.onSourceReorder)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-down'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveDown(import0.ɵnov((<any>v.parent),21),co.source,co.selectedItemsSource,co.onSourceReorder)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-double-down'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveBottom(import0.ɵnov((<any>v.parent),21),co.source,co.selectedItemsSource,co.onSourceReorder)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'fa-angle-up';
    ck(v,6,0,currVal_0);
    const currVal_1:any = 'fa-angle-double-up';
    ck(v,10,0,currVal_1);
    const currVal_2:any = 'fa-angle-down';
    ck(v,14,0,currVal_2);
    const currVal_3:any = 'fa-angle-double-down';
    ck(v,18,0,currVal_3);
  },(null as any));
}
function View_PickList_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'ui-picklist-caption ui-widget-header ui-corner-tl ui-corner-tr'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.sourceHeader;
    ck(v,1,0,currVal_0);
  });
}
function View_PickList_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',[[
        'class',
        'ui-picklist-filter-container ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'input',[
      [
        'class',
        'ui-picklist-filter ui-inputtext ui-widget ui-state-default ui-corner-all'
      ]
      ,
      [
        'role',
        'textbox'
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
    ,[
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        1,
        'placeholder',
        0
      ]

    ]
      ,[[
        (null as any),
        'keyup'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('keyup' === en)) {
        const pd_0:any = ((<any>co.onFilter($event,co.source,(0 - 1))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-search'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.disabled;
    const currVal_1:any = co.sourceFilterPlaceholder;
    ck(v,2,0,currVal_0,currVal_1);
  });
}
function View_PickList_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'li',[[
        'class',
        'ui-picklist-droppoint'
      ]
      ],[[
        4,
        'display',
        (null as any)
      ]
    ],[
      [
        (null as any),
        'dragover'
      ]
      ,
      [
        (null as any),
        'drop'
      ]
      ,
      [
        (null as any),
        'dragleave'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('dragover' === en)) {
        const pd_0:any = ((<any>co.onDragOver($event,(<any>v.parent).context.index,(0 - 1))) !== false);
        ad = (pd_0 && ad);
      }
      if (('drop' === en)) {
        const pd_1:any = ((<any>co.onDrop($event,(<any>v.parent).context.index,(0 - 1))) !== false);
        ad = (pd_1 && ad);
      }
      if (('dragleave' === en)) {
        const pd_2:any = ((<any>co.onDragLeave($event,(0 - 1))) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-picklist-droppoint-highlight'])
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = 'ui-picklist-droppoint';
    const currVal_2:any = ck(v,2,0,((<any>v.parent).context.index === co.dragOverItemIndexSource));
    ck(v,1,0,currVal_1,currVal_2);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.isItemVisible((<any>v.parent).context.$implicit,(0 - 1))? 'block': 'none');
    ck(v,0,0,currVal_0);
  });
}
function View_PickList_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_PickList_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'li',[[
        'class',
        'ui-picklist-droppoint'
      ]
    ],(null as any),[
      [
        (null as any),
        'dragover'
      ]
      ,
      [
        (null as any),
        'drop'
      ]
      ,
      [
        (null as any),
        'dragleave'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('dragover' === en)) {
        const pd_0:any = ((<any>co.onDragOver($event,((<any>v.parent).context.index + 1),(0 - 1))) !== false);
        ad = (pd_0 && ad);
      }
      if (('drop' === en)) {
        const pd_1:any = ((<any>co.onDrop($event,((<any>v.parent).context.index + 1),(0 - 1))) !== false);
        ad = (pd_1 && ad);
      }
      if (('dragleave' === en)) {
        const pd_2:any = ((<any>co.onDragLeave($event,(0 - 1))) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-picklist-droppoint-highlight'])
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-picklist-droppoint';
    const currVal_1:any = ck(v,2,0,(((<any>v.parent).context.index + 1) === co.dragOverItemIndexSource));
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_PickList_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PickList_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'li',([] as any[]),[
      [
        4,
        'display',
        (null as any)
      ]
      ,
      [
        8,
        'draggable',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'dblclick'
      ]
      ,
      [
        (null as any),
        'touchend'
      ]
      ,
      [
        (null as any),
        'dragstart'
      ]
      ,
      [
        (null as any),
        'dragend'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onItemClick($event,v.context.$implicit,co.selectedItemsSource)) !== false);
        ad = (pd_0 && ad);
      }
      if (('dblclick' === en)) {
        const pd_1:any = ((<any>co.onSourceItemDblClick()) !== false);
        ad = (pd_1 && ad);
      }
      if (('touchend' === en)) {
        const pd_2:any = ((<any>co.onItemTouchEnd($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('dragstart' === en)) {
        const pd_3:any = ((<any>co.onDragStart($event,v.context.index,(0 - 1))) !== false);
        ad = (pd_3 && ad);
      }
      if (('dragend' === en)) {
        const pd_4:any = ((<any>co.onDragEnd($event)) !== false);
        ad = (pd_4 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-picklist-item',
      'ui-state-highlight'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PickList_6)),
    import0.ɵdid(106496,(null as any),0,import4.TemplateWrapper,[import0.ViewContainerRef],{
      item: [
        0,
        'item'
      ]
      ,
      templateRef: [
        1,
        'templateRef'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PickList_7)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.dragdrop;
    ck(v,2,0,currVal_0);
    const currVal_3:any = ck(v,6,0,true,co.isSelected(v.context.$implicit,co.selectedItemsSource));
    ck(v,5,0,currVal_3);
    const currVal_4:any = v.context.$implicit;
    const currVal_5:any = co.itemTemplate;
    ck(v,9,0,currVal_4,currVal_5);
    const currVal_6:any = (co.dragdrop && v.context.last);
    ck(v,13,0,currVal_6);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = (co.isItemVisible(v.context.$implicit,(0 - 1))? 'block': 'none');
    const currVal_2:any = co.dragdrop;
    ck(v,4,0,currVal_1,currVal_2);
  });
}
function View_PickList_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'ui-picklist-caption ui-widget-header ui-corner-tl ui-corner-tr'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.targetHeader;
    ck(v,1,0,currVal_0);
  });
}
function View_PickList_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',[[
        'class',
        'ui-picklist-filter-container ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'input',[
      [
        'class',
        'ui-picklist-filter ui-inputtext ui-widget ui-state-default ui-corner-all'
      ]
      ,
      [
        'role',
        'textbox'
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
    ,[
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        1,
        'placeholder',
        0
      ]

    ]
      ,[[
        (null as any),
        'keyup'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('keyup' === en)) {
        const pd_0:any = ((<any>co.onFilter($event,co.target,1)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-search'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.disabled;
    const currVal_1:any = co.targetFilterPlaceholder;
    ck(v,2,0,currVal_0,currVal_1);
  });
}
function View_PickList_11(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'li',[[
        'class',
        'ui-picklist-droppoint'
      ]
      ],[[
        4,
        'display',
        (null as any)
      ]
    ],[
      [
        (null as any),
        'dragover'
      ]
      ,
      [
        (null as any),
        'drop'
      ]
      ,
      [
        (null as any),
        'dragleave'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('dragover' === en)) {
        const pd_0:any = ((<any>co.onDragOver($event,(<any>v.parent).context.index,1)) !== false);
        ad = (pd_0 && ad);
      }
      if (('drop' === en)) {
        const pd_1:any = ((<any>co.onDrop($event,(<any>v.parent).context.index,1)) !== false);
        ad = (pd_1 && ad);
      }
      if (('dragleave' === en)) {
        const pd_2:any = ((<any>co.onDragLeave($event,1)) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-picklist-droppoint-highlight'])
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = 'ui-picklist-droppoint';
    const currVal_2:any = ck(v,2,0,((<any>v.parent).context.index === co.dragOverItemIndexTarget));
    ck(v,1,0,currVal_1,currVal_2);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.isItemVisible((<any>v.parent).context.$implicit,1)? 'block': 'none');
    ck(v,0,0,currVal_0);
  });
}
function View_PickList_12(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_PickList_13(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'li',[[
        'class',
        'ui-picklist-droppoint'
      ]
    ],(null as any),[
      [
        (null as any),
        'dragover'
      ]
      ,
      [
        (null as any),
        'drop'
      ]
      ,
      [
        (null as any),
        'dragleave'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('dragover' === en)) {
        const pd_0:any = ((<any>co.onDragOver($event,((<any>v.parent).context.index + 1),1)) !== false);
        ad = (pd_0 && ad);
      }
      if (('drop' === en)) {
        const pd_1:any = ((<any>co.onDrop($event,((<any>v.parent).context.index + 1),1)) !== false);
        ad = (pd_1 && ad);
      }
      if (('dragleave' === en)) {
        const pd_2:any = ((<any>co.onDragLeave($event,1)) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-picklist-droppoint-highlight'])
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-picklist-droppoint';
    const currVal_1:any = ck(v,2,0,(((<any>v.parent).context.index + 1) === co.dragOverItemIndexTarget));
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_PickList_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PickList_11)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'li',([] as any[]),[
      [
        4,
        'display',
        (null as any)
      ]
      ,
      [
        8,
        'draggable',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'dblclick'
      ]
      ,
      [
        (null as any),
        'touchend'
      ]
      ,
      [
        (null as any),
        'dragstart'
      ]
      ,
      [
        (null as any),
        'dragend'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onItemClick($event,v.context.$implicit,co.selectedItemsTarget)) !== false);
        ad = (pd_0 && ad);
      }
      if (('dblclick' === en)) {
        const pd_1:any = ((<any>co.onTargetItemDblClick()) !== false);
        ad = (pd_1 && ad);
      }
      if (('touchend' === en)) {
        const pd_2:any = ((<any>co.onItemTouchEnd($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('dragstart' === en)) {
        const pd_3:any = ((<any>co.onDragStart($event,v.context.index,1)) !== false);
        ad = (pd_3 && ad);
      }
      if (('dragend' === en)) {
        const pd_4:any = ((<any>co.onDragEnd($event)) !== false);
        ad = (pd_4 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-picklist-item',
      'ui-state-highlight'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PickList_12)),
    import0.ɵdid(106496,(null as any),0,import4.TemplateWrapper,[import0.ViewContainerRef],{
      item: [
        0,
        'item'
      ]
      ,
      templateRef: [
        1,
        'templateRef'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PickList_13)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.dragdrop;
    ck(v,2,0,currVal_0);
    const currVal_3:any = ck(v,6,0,true,co.isSelected(v.context.$implicit,co.selectedItemsTarget));
    ck(v,5,0,currVal_3);
    const currVal_4:any = v.context.$implicit;
    const currVal_5:any = co.itemTemplate;
    ck(v,9,0,currVal_4,currVal_5);
    const currVal_6:any = (co.dragdrop && v.context.last);
    ck(v,13,0,currVal_6);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = (co.isItemVisible(v.context.$implicit,1)? 'block': 'none');
    const currVal_2:any = co.dragdrop;
    ck(v,4,0,currVal_1,currVal_2);
  });
}
function View_PickList_14(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),20,'div',[[
        'class',
        'ui-picklist-target-controls ui-picklist-buttons'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),17,'div',[[
        'class',
        'ui-picklist-buttons-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-up'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveUp(import0.ɵnov((<any>v.parent),63),co.target,co.selectedItemsTarget,co.onTargetReorder)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-double-up'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveTop(import0.ɵnov((<any>v.parent),63),co.target,co.selectedItemsTarget,co.onTargetReorder)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-down'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveDown(import0.ɵnov((<any>v.parent),63),co.target,co.selectedItemsTarget,co.onTargetReorder)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-double-down'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveBottom(import0.ɵnov((<any>v.parent),63),co.target,co.selectedItemsTarget,co.onTargetReorder)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'fa-angle-up';
    ck(v,6,0,currVal_0);
    const currVal_1:any = 'fa-angle-double-up';
    ck(v,10,0,currVal_1);
    const currVal_2:any = 'fa-angle-down';
    ck(v,14,0,currVal_2);
    const currVal_3:any = 'fa-angle-double-down';
    ck(v,18,0,currVal_3);
  },(null as any));
}
export function View_PickList_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{listViewSourceChild: 0}),
    import0.ɵqud(201326592,2,{listViewTargetChild: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),72,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-picklist ui-widget ui-helper-clearfix',
      'ui-picklist-responsive'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PickList_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),18,'div',[[
        'class',
        'ui-picklist-listwrapper ui-picklist-source-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-picklist-listwrapper-nocontrols']),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PickList_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PickList_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'sourcelist',
        1
      ]

    ]
      ,(null as any),7,'ul',[[
        'class',
        'ui-widget-content ui-picklist-list ui-picklist-source ui-corner-bottom'
      ]
    ],(null as any),[
      [
        (null as any),
        'dragover'
      ]
      ,
      [
        (null as any),
        'dragleave'
      ]
      ,
      [
        (null as any),
        'drop'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.PickList = v.component;
      if (('dragover' === en)) {
        const pd_0:any = ((<any>co.onListMouseMove($event,(0 - 1))) !== false);
        ad = (pd_0 && ad);
      }
      if (('dragleave' === en)) {
        const pd_1:any = ((<any>co.onListDragLeave()) !== false);
        ad = (pd_1 && ad);
      }
      if (('drop' === en)) {
        const pd_2:any = ((<any>co.onListDrop($event,(0 - 1))) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-picklist-highlight']),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PickList_4)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),20,'div',[[
        'class',
        'ui-picklist-buttons'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),17,'div',[[
        'class',
        'ui-picklist-buttons-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-right'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.PickList = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveRight()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-double-right'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.PickList = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveAllRight()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-left'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.PickList = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveLeft()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'icon',
        'fa-angle-double-left'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.PickList = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.moveAllLeft()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
      ,{icon: [
        0,
        'icon'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),18,'div',[[
        'class',
        'ui-picklist-listwrapper ui-picklist-target-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-picklist-listwrapper-nocontrols']),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PickList_8)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PickList_9)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'targetlist',
        1
      ]

    ]
      ,(null as any),7,'ul',[[
        'class',
        'ui-widget-content ui-picklist-list ui-picklist-target ui-corner-bottom'
      ]
    ],(null as any),[
      [
        (null as any),
        'dragover'
      ]
      ,
      [
        (null as any),
        'dragleave'
      ]
      ,
      [
        (null as any),
        'drop'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.PickList = v.component;
      if (('dragover' === en)) {
        const pd_0:any = ((<any>co.onListMouseMove($event,1)) !== false);
        ad = (pd_0 && ad);
      }
      if (('dragleave' === en)) {
        const pd_1:any = ((<any>co.onListDragLeave()) !== false);
        ad = (pd_1 && ad);
      }
      if (('drop' === en)) {
        const pd_2:any = ((<any>co.onListDrop($event,1)) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-picklist-highlight']),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PickList_10)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PickList_14)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.PickList = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,5,0,true,co.responsive);
    ck(v,4,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,6,0,currVal_2);
    const currVal_3:any = co.showSourceControls;
    ck(v,9,0,currVal_3);
    const currVal_4:any = 'ui-picklist-listwrapper ui-picklist-source-wrapper';
    const currVal_5:any = ck(v,13,0,!co.showSourceControls);
    ck(v,12,0,currVal_4,currVal_5);
    const currVal_6:any = co.sourceHeader;
    ck(v,16,0,currVal_6);
    const currVal_7:any = co.filterBy;
    ck(v,19,0,currVal_7);
    const currVal_8:any = 'ui-widget-content ui-picklist-list ui-picklist-source ui-corner-bottom';
    const currVal_9:any = ck(v,23,0,co.listHighlightSource);
    ck(v,22,0,currVal_8,currVal_9);
    const currVal_10:any = co.sourceStyle;
    ck(v,24,0,currVal_10);
    const currVal_11:any = co.source;
    ck(v,27,0,currVal_11);
    const currVal_12:any = 'fa-angle-right';
    ck(v,37,0,currVal_12);
    const currVal_13:any = 'fa-angle-double-right';
    ck(v,41,0,currVal_13);
    const currVal_14:any = 'fa-angle-left';
    ck(v,45,0,currVal_14);
    const currVal_15:any = 'fa-angle-double-left';
    ck(v,49,0,currVal_15);
    const currVal_16:any = 'ui-picklist-listwrapper ui-picklist-target-wrapper';
    const currVal_17:any = ck(v,55,0,!co.showTargetControls);
    ck(v,54,0,currVal_16,currVal_17);
    const currVal_18:any = co.targetHeader;
    ck(v,58,0,currVal_18);
    const currVal_19:any = co.filterBy;
    ck(v,61,0,currVal_19);
    const currVal_20:any = 'ui-widget-content ui-picklist-list ui-picklist-target ui-corner-bottom';
    const currVal_21:any = ck(v,65,0,co.listHighlightTarget);
    ck(v,64,0,currVal_20,currVal_21);
    const currVal_22:any = co.targetStyle;
    ck(v,66,0,currVal_22);
    const currVal_23:any = co.target;
    ck(v,69,0,currVal_23);
    const currVal_24:any = co.showTargetControls;
    ck(v,74,0,currVal_24);
  },(null as any));
}
function View_PickList_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'p-pickList',([] as any[]),(null as any),(null as any),(null as any),View_PickList_0,RenderType_PickList)),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵprd(256,(null as any),import6.ObjectUtils,import6.ObjectUtils,([] as any[])),
    import0.ɵdid(4743168,(null as any),1,import1.PickList,[
      import0.ElementRef,
      import5.DomHandler,
      import6.ObjectUtils
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(301989888,1,{templates: 1})
  ]
  ,(null as any),(null as any));
}
export const PickListNgFactory:import0.ComponentFactory<import1.PickList> = import0.ɵccf('p-pickList',import1.PickList,View_PickList_Host_0,{
  source: 'source',
  target: 'target',
  sourceHeader: 'sourceHeader',
  targetHeader: 'targetHeader',
  responsive: 'responsive',
  filterBy: 'filterBy',
  metaKeySelection: 'metaKeySelection',
  dragdrop: 'dragdrop',
  dragdropScope: 'dragdropScope',
  style: 'style',
  styleClass: 'styleClass',
  sourceStyle: 'sourceStyle',
  targetStyle: 'targetStyle',
  showSourceControls: 'showSourceControls',
  showTargetControls: 'showTargetControls',
  sourceFilterPlaceholder: 'sourceFilterPlaceholder',
  targetFilterPlaceholder: 'targetFilterPlaceholder'
}
,{
  onMoveToSource: 'onMoveToSource',
  onMoveAllToSource: 'onMoveAllToSource',
  onMoveAllToTarget: 'onMoveAllToTarget',
  onMoveToTarget: 'onMoveToTarget',
  onSourceReorder: 'onSourceReorder',
  onTargetReorder: 'onTargetReorder'
}
,([] as any[]));
