// primeng/components/dropdown/dropdown.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/dropdown/dropdown';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from 'primeng/components/dom/domhandler';
import * as import5 from 'primeng/components/utils/objectutils';
import * as import6 from '@angular/forms';
class DropdownModuleInjector extends import0.ɵNgModuleInjector<import1.DropdownModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _DropdownModule_2:import1.DropdownModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.DropdownModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._DropdownModule_2 = new import1.DropdownModule();
    return this._DropdownModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import1.DropdownModule)) { return this._DropdownModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const DropdownModuleNgFactory:import0.NgModuleFactory<import1.DropdownModule> = new import0.NgModuleFactory<any>(DropdownModuleInjector,import1.DropdownModule);
const styles_Dropdown:any[] = ([] as any[]);
export const RenderType_Dropdown:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Dropdown,
      data: {'animation': [{
        name: 'panelState',
        definitions: [
          {
            type: 0,
            name: 'hidden',
            styles: {
              type: 6,
              styles: {opacity: 0}
            }

          }
          ,
          {
            type: 0,
            name: 'visible',
            styles: {
              type: 6,
              styles: {opacity: 1}
            }

          }
          ,
          {
            type: 1,
            expr: 'visible => hidden',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms ease-in'
            }

          }
          ,
          {
            type: 1,
            expr: 'hidden => visible',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms ease-out'
            }

          }

        ]

      }
  ]}
}
);
function View_Dropdown_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'option',([] as any[]),[
      [
        8,
        'value',
        0
      ]
      ,
      [
        8,
        'selected',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = v.context.$implicit.value;
    const currVal_1:any = (co.selectedOption == v.context.$implicit);
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_2:any = v.context.$implicit.label;
    ck(v,1,0,currVal_2);
  });
}
function View_Dropdown_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'ui-helper-hidden-accessible'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'select',[
      [
        'aria-hidden',
        'true'
      ]
      ,
      [
        'name',
        'options'
      ]
      ,
      [
        'tabindex',
        '-1'
      ]

    ]
    ,[
      [
        8,
        'required',
        0
      ]
      ,
      [
        1,
        'aria-label',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dropdown_2)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co.options;
    ck(v,5,0,currVal_2);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.required;
    const currVal_1:any = (co.selectedOption? co.selectedOption.label: ' ');
    ck(v,2,0,currVal_0,currVal_1);
  });
}
function View_Dropdown_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'label',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-dropdown-label ui-inputtext ui-corner-all',
      'ui-dropdown-label-empty'
    ]
    ),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,true,!co.label);
    ck(v,1,0,currVal_0);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = (co.label || 'empty');
    ck(v,3,0,currVal_1);
  });
}
function View_Dropdown_4(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵeld(0,[
      [
        6,
        0
      ]
      ,
      [
        'editableInput',
        1
      ]

    ]
    ,(null as any),0,'input',[
      [
        'class',
        'ui-dropdown-label ui-inputtext ui-corner-all'
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
    ,[
      [
        1,
        'aria-label',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        1,
        'placeholder',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onEditableInputClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('input' === en)) {
        const pd_1:any = ((<any>co.onEditableInputChange($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('focus' === en)) {
        const pd_2:any = ((<any>co.onEditableInputFocus($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('blur' === en)) {
        const pd_3:any = ((<any>co.onInputBlur($event)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
  },(null as any),(null as any)))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.selectedOption? co.selectedOption.label: ' ');
    const currVal_1:any = co.disabled;
    const currVal_2:any = co.placeholder;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
function View_Dropdown_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',[[
        'class',
        'ui-dropdown-filter-container'
      ]
    ],(null as any),[
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'click'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('input' === en)) {
        const pd_0:any = ((<any>co.onFilter($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>$event.stopPropagation()) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,[
      [
        4,
        0
      ]
      ,
      [
        'filter',
        1
      ]

    ]
    ,(null as any),0,'input',[
      [
        'autocomplete',
        'off'
      ]
      ,
      [
        'class',
        'ui-dropdown-filter ui-inputtext ui-widget ui-state-default ui-corner-all'
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
      ,[[
        1,
        'placeholder',
        0
      ]
    ],[
      [
        (null as any),
        'keydown.enter'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('keydown.enter' === en)) {
        const pd_0:any = ((<any>$event.preventDefault()) !== false);
        ad = (pd_0 && ad);
      }
      if (('keydown' === en)) {
        const pd_1:any = ((<any>co.onKeydown($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-search'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.filterPlaceholder;
    ck(v,2,0,currVal_0);
  });
}
function View_Dropdown_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = ((<any>v.parent).context.$implicit.label || 'empty');
    ck(v,1,0,currVal_0);
  });
}
function View_Dropdown_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_Dropdown_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dropdown_10)),
    import0.ɵdid(106496,(null as any),0,import3.TemplateWrapper,[import0.ViewContainerRef],{
      item: [
        0,
        'item'
      ]
      ,
      templateRef: [
        1,
        'templateRef'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (<any>v.parent).context.$implicit;
    const currVal_1:any = co.itemTemplate;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_Dropdown_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),9,'li',([] as any[]),(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onItemClick($event,v.context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-dropdown-item ui-corner-all',
      'ui-state-highlight',
      'ui-dropdown-item-empty'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dropdown_8)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dropdown_9)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,true,(co.selectedOption == v.context.$implicit),(!v.context.$implicit.label || (v.context.$implicit.label.length === 0)));
    ck(v,1,0,currVal_0);
    const currVal_1:boolean = !co.itemTemplate;
    ck(v,5,0,currVal_1);
    const currVal_2:any = co.itemTemplate;
    ck(v,8,0,currVal_2);
  },(null as any));
}
function View_Dropdown_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ul',[[
        'class',
        'ui-dropdown-items ui-dropdown-list ui-widget-content ui-widget ui-corner-all ui-helper-reset'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dropdown_7)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.optionsToDisplay;
    ck(v,3,0,currVal_0);
  },(null as any));
}
export function View_Dropdown_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{containerViewChild: 0}),
    import0.ɵqud(201326592,2,{panelViewChild: 0}),
    import0.ɵqud(201326592,3,{itemsWrapperViewChild: 0}),
    import0.ɵqud(335544320,4,{filterViewChild: 0}),
    import0.ɵqud(201326592,5,{focusViewChild: 0}),
    import0.ɵqud(335544320,6,{editableInputViewChild: 0}),
    (l()(),import0.ɵted((null as any),['\n         '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'container',
        1
      ]

    ]
      ,(null as any),37,'div',([] as any[]),(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Dropdown = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onMouseclick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-dropdown ui-widget ui-state-default ui-corner-all ui-helper-clearfix',
      'ui-state-disabled',
      'ui-dropdown-open',
      'ui-state-focus'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dropdown_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-helper-hidden-accessible'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,[
      [
        5,
        0
      ]
      ,
      [
        'in',
        1
      ]

    ]
    ,(null as any),0,'input',[
      [
        'readonly',
        ''
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
    ,[
      [
        1,
        'id',
        0
      ]
      ,
      [
        1,
        'aria-label',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        1,
        'autofocus',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Dropdown = v.component;
      if (('focus' === en)) {
        const pd_0:any = ((<any>co.onInputFocus($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>co.onInputBlur($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('keydown' === en)) {
        const pd_2:any = ((<any>co.onKeydown($event)) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dropdown_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dropdown_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-dropdown-trigger ui-state-default ui-corner-right'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-fw fa-caret-down ui-clickable'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'panel',
        1
      ]

    ]
    ,(null as any),12,'div',([] as any[]),[
      [
        24,
        '@panelState',
        0
      ]
      ,
      [
        4,
        'display',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dropdown_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,[
      [
        3,
        0
      ]
      ,
      [
        'itemswrapper',
        1
      ]

    ]
      ,(null as any),4,'div',[[
        'class',
        'ui-dropdown-items-wrapper'
      ]
      ],[[
        4,
        'max-height',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Dropdown_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Dropdown = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,9,0,true,co.disabled,co.panelVisible,co.focus);
    ck(v,8,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,10,0,currVal_2);
    const currVal_3:any = co.autoWidth;
    ck(v,13,0,currVal_3);
    const currVal_9:boolean = !co.editable;
    ck(v,21,0,currVal_9);
    const currVal_10:any = co.editable;
    ck(v,24,0,currVal_10);
    const currVal_13:any = co.panelStyleClass;
    const currVal_14:any = 'ui-dropdown-panel ui-widget-content ui-corner-all ui-helper-hidden ui-shadow';
    ck(v,32,0,currVal_13,currVal_14);
    const currVal_15:any = co.panelStyle;
    ck(v,33,0,currVal_15);
    const currVal_16:any = co.filter;
    ck(v,36,0,currVal_16);
    const currVal_18:any = (co.lazy? co.panelVisible: true);
    ck(v,41,0,currVal_18);
  },(ck,v) => {
    var co:import1.Dropdown = v.component;
    const currVal_4:any = co.inputId;
    const currVal_5:any = (co.selectedOption? co.selectedOption.label: ' ');
    const currVal_6:any = co.disabled;
    const currVal_7:any = co.tabindex;
    const currVal_8:any = co.autofocus;
    ck(v,17,0,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8);
    const currVal_11:any = (co.panelVisible? 'visible': 'hidden');
    const currVal_12:any = (co.panelVisible? 'block': 'none');
    ck(v,31,0,currVal_11,currVal_12);
    const currVal_17:any = (co.scrollHeight || 'auto');
    ck(v,38,0,currVal_17);
  });
}
function View_Dropdown_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'p-dropdown',([] as any[]),(null as any),(null as any),(null as any),View_Dropdown_0,RenderType_Dropdown)),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵprd(256,(null as any),import5.ObjectUtils,import5.ObjectUtils,([] as any[])),
    import0.ɵdid(6938624,(null as any),1,import1.Dropdown,[
      import0.ElementRef,
      import4.DomHandler,
      import0.Renderer2,
      import0.ChangeDetectorRef,
      import5.ObjectUtils
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(301989888,1,{templates: 1}),
    import0.ɵprd(2560,(null as any),import6.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.Dropdown])
  ]
  ,(ck,v) => {
    ck(v,3,0);
  },(null as any));
}
export const DropdownNgFactory:import0.ComponentFactory<import1.Dropdown> = import0.ɵccf('p-dropdown',import1.Dropdown,View_Dropdown_Host_0,{
  scrollHeight: 'scrollHeight',
  filter: 'filter',
  style: 'style',
  panelStyle: 'panelStyle',
  styleClass: 'styleClass',
  panelStyleClass: 'panelStyleClass',
  disabled: 'disabled',
  readonly: 'readonly',
  autoWidth: 'autoWidth',
  required: 'required',
  editable: 'editable',
  appendTo: 'appendTo',
  tabindex: 'tabindex',
  placeholder: 'placeholder',
  filterPlaceholder: 'filterPlaceholder',
  inputId: 'inputId',
  dataKey: 'dataKey',
  filterBy: 'filterBy',
  lazy: 'lazy',
  autofocus: 'autofocus',
  options: 'options'
}
,{
  onChange: 'onChange',
  onFocus: 'onFocus',
  onBlur: 'onBlur'
}
,([] as any[]));
