// @swimlane/ngx-datatable/release/components/body/body-row.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './body-cell.component.ngfactory';
import * as import2 from '@swimlane/ngx-datatable/release/components/body/body-cell.component';
import * as import3 from '@angular/common';
import * as import4 from '@swimlane/ngx-datatable/release/components/body/body-row.component';
import * as import5 from '@swimlane/ngx-datatable/release/services/scrollbar-helper.service';
const styles_DataTableBodyRowComponent:any[] = ([] as any[]);
export const RenderType_DataTableBodyRowComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DataTableBodyRowComponent,
  data: {}
}
);
function View_DataTableBodyRowComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'datatable-body-cell',[
      [
        'class',
        'datatable-body-cell'
      ]
      ,
      [
        'tabindex',
        '-1'
      ]

    ]
    ,[
      [
        8,
        'className',
        0
      ]
      ,
      [
        4,
        'width',
        'px'
      ]
      ,
      [
        4,
        'height',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'activate'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'dblclick'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('focus' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).onFocus()) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,1).onBlur()) !== false);
        ad = (pd_1 && ad);
      }
      if (('click' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,1).onClick($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('dblclick' === en)) {
        const pd_3:any = ((<any>import0.ɵnov(v,1).onDblClick($event)) !== false);
        ad = (pd_3 && ad);
      }
      if (('keydown' === en)) {
        const pd_4:any = ((<any>import0.ɵnov(v,1).onKeyDown($event)) !== false);
        ad = (pd_4 && ad);
      }
      if (('activate' === en)) {
        const pd_5:any = ((<any>co.onActivate($event,v.context.index)) !== false);
        ad = (pd_5 && ad);
      }
      return ad;
    },import1.View_DataTableBodyCellComponent_0,import1.RenderType_DataTableBodyCellComponent)),
    import0.ɵdid(90112,(null as any),0,import2.DataTableBodyCellComponent,[import0.ElementRef],{
      row: [
        0,
        'row'
      ]
      ,
      column: [
        1,
        'column'
      ]
      ,
      rowHeight: [
        2,
        'rowHeight'
      ]
      ,
      isSelected: [
        3,
        'isSelected'
      ]

    }
    ,{activate: 'activate'}),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = co.row;
    const currVal_4:any = v.context.$implicit;
    const currVal_5:any = co.rowHeight;
    const currVal_6:any = co.isSelected;
    ck(v,1,0,currVal_3,currVal_4,currVal_5,currVal_6);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).columnCssClasses;
    const currVal_1:any = import0.ɵnov(v,1).width;
    const currVal_2:any = import0.ɵnov(v,1).height;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
function View_DataTableBodyRowComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import3.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataTableBodyRowComponent_2)),
    import0.ɵdid(401408,(null as any),0,import3.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
    ,{
      ngForOf: [
        0,
        'ngForOf'
      ]
      ,
      ngForTrackBy: [
        1,
        'ngForTrackBy'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.stylesByGroup(v.context.$implicit.type);
    ck(v,1,0,currVal_1);
    const currVal_2:any = v.context.$implicit.columns;
    const currVal_3:any = co.columnTrackingFn;
    ck(v,4,0,currVal_2,currVal_3);
  },(ck,v) => {
    const currVal_0:any = import0.ɵinlineInterpolate(1,'datatable-row-',v.context.$implicit.type,' datatable-row-group');
    ck(v,0,0,currVal_0);
  });
}
export function View_DataTableBodyRowComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataTableBodyRowComponent_1)),
    import0.ɵdid(401408,(null as any),0,import3.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
    ,{
      ngForOf: [
        0,
        'ngForOf'
      ]
      ,
      ngForTrackBy: [
        1,
        'ngForTrackBy'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import4.DataTableBodyRowComponent = v.component;
    const currVal_0:any = co.columnsByPin;
    const currVal_1:any = co.trackByGroups;
    ck(v,2,0,currVal_0,currVal_1);
  },(null as any));
}
function View_DataTableBodyRowComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'datatable-body-row',([] as any[]),[
      [
        8,
        'className',
        0
      ]
      ,
      [
        4,
        'height',
        'px'
      ]
      ,
      [
        4,
        'width',
        'px'
      ]

    ]
      ,[[
        (null as any),
        'keydown'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('keydown' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).onKeyDown($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_DataTableBodyRowComponent_0,RenderType_DataTableBodyRowComponent)),
    import0.ɵdid(24576,(null as any),0,import4.DataTableBodyRowComponent,[
      import5.ScrollbarHelper,
      import0.ElementRef
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).cssClass;
    const currVal_1:any = import0.ɵnov(v,1).rowHeight;
    const currVal_2:any = import0.ɵnov(v,1).columnsTotalWidths;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export const DataTableBodyRowComponentNgFactory:import0.ComponentFactory<import4.DataTableBodyRowComponent> = import0.ɵccf('datatable-body-row',import4.DataTableBodyRowComponent,View_DataTableBodyRowComponent_Host_0,{
  columns: 'columns',
  innerWidth: 'innerWidth',
  rowClass: 'rowClass',
  row: 'row',
  offsetX: 'offsetX',
  isSelected: 'isSelected',
  rowHeight: 'rowHeight'
}
,{activate: 'activate'},([] as any[]));
