// @swimlane/ngx-datatable/release/components/body/body-cell.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@swimlane/ngx-datatable/release/components/body/body-cell.component';
const styles_DataTableBodyCellComponent:any[] = ([] as any[]);
export const RenderType_DataTableBodyCellComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DataTableBodyCellComponent,
  data: {}
}
);
function View_DataTableBodyCellComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'label',[[
        'class',
        'datatable-checkbox'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'input',[[
        'type',
        'checkbox'
      ]
      ],[[
        8,
        'checked',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onCheckboxChange($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.isSelected;
    ck(v,2,0,currVal_0);
  });
}
function View_DataTableBodyCellComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),[
      [
        8,
        'title',
        0
      ]
      ,
      [
        8,
        'innerHTML',
        1
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.value;
    const currVal_1:any = co.value;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_DataTableBodyCellComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n      ']))],(null as any),(null as any));
}
function View_DataTableBodyCellComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,[
      [
        1,
        3
      ]
      ,
      [
        'cellTemplate',
        2
      ]

    ]
    ,(null as any),2,(null as any),View_DataTableBodyCellComponent_4)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'value',
      'row',
      'column',
      'isSelected',
      'onCheckboxChangeFn',
      'activateFn'
    ]
    ),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.column.cellTemplate;
    const currVal_1:any = ck(v,2,0,co.value,co.row,co.column,co.isSelected,co.onCheckboxChangeFn,co.activateFn);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
export function View_DataTableBodyCellComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(335544320,1,{cellTemplate: 0}),
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'div',[[
        'class',
        'datatable-body-cell-label'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataTableBodyCellComponent_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataTableBodyCellComponent_2)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataTableBodyCellComponent_3)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import2.DataTableBodyCellComponent = v.component;
    const currVal_0:any = co.column.checkboxable;
    ck(v,5,0,currVal_0);
    const currVal_1:boolean = !co.column.cellTemplate;
    ck(v,8,0,currVal_1);
    const currVal_2:any = co.column.cellTemplate;
    ck(v,11,0,currVal_2);
  },(null as any));
}
function View_DataTableBodyCellComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'datatable-body-cell',[[
        'class',
        'datatable-body-cell'
      ]
    ],[
      [
        8,
        'className',
        0
      ]
      ,
      [
        4,
        'width',
        'px'
      ]
      ,
      [
        4,
        'height',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'dblclick'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('focus' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).onFocus()) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,1).onBlur()) !== false);
        ad = (pd_1 && ad);
      }
      if (('click' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,1).onClick($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('dblclick' === en)) {
        const pd_3:any = ((<any>import0.ɵnov(v,1).onDblClick($event)) !== false);
        ad = (pd_3 && ad);
      }
      if (('keydown' === en)) {
        const pd_4:any = ((<any>import0.ɵnov(v,1).onKeyDown($event)) !== false);
        ad = (pd_4 && ad);
      }
      return ad;
    },View_DataTableBodyCellComponent_0,RenderType_DataTableBodyCellComponent)),
    import0.ɵdid(90112,(null as any),0,import2.DataTableBodyCellComponent,[import0.ElementRef],(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).columnCssClasses;
    const currVal_1:any = import0.ɵnov(v,1).width;
    const currVal_2:any = import0.ɵnov(v,1).height;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export const DataTableBodyCellComponentNgFactory:import0.ComponentFactory<import2.DataTableBodyCellComponent> = import0.ɵccf('datatable-body-cell',import2.DataTableBodyCellComponent,View_DataTableBodyCellComponent_Host_0,{
  row: 'row',
  column: 'column',
  rowHeight: 'rowHeight',
  isSelected: 'isSelected',
  sorts: 'sorts'
}
,{activate: 'activate'},([] as any[]));
