// @swimlane/ngx-datatable/release/components/body/body.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './progress-bar.component.ngfactory';
import * as import2 from '@swimlane/ngx-datatable/release/components/body/progress-bar.component';
import * as import3 from './body-row-wrapper.component.ngfactory';
import * as import4 from '@angular/common';
import * as import5 from '@swimlane/ngx-datatable/release/components/body/body-row-wrapper.component';
import * as import6 from './body-row.component.ngfactory';
import * as import7 from '@swimlane/ngx-datatable/release/components/body/body-row.component';
import * as import8 from '@swimlane/ngx-datatable/release/services/scrollbar-helper.service';
import * as import9 from './scroller.component.ngfactory';
import * as import10 from '@swimlane/ngx-datatable/release/components/body/scroller.component';
import * as import11 from '@swimlane/ngx-datatable/release/components/body/body.component';
import * as import12 from './selection.component.ngfactory';
import * as import13 from '@swimlane/ngx-datatable/release/components/body/selection.component';
const styles_DataTableBodyComponent:any[] = ([] as any[]);
export const RenderType_DataTableBodyComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DataTableBodyComponent,
  data: {}
}
);
function View_DataTableBodyComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'datatable-progress',([] as any[]),(null as any),(null as any),(null as any),import1.View_ProgressBarComponent_0,import1.RenderType_ProgressBarComponent)),
    import0.ɵdid(24576,(null as any),0,import2.ProgressBarComponent,([] as any[]),(null as any),(null as any)),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(null as any),(null as any));
}
function View_DataTableBodyComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'datatable-row-wrapper',[[
        'class',
        'datatable-row-wrapper'
      ]
    ],(null as any),[
      [
        (null as any),
        'rowContextmenu'
      ]
      ,
      [
        (null as any),
        'contextmenu'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('contextmenu' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,2).onContextmenu($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('rowContextmenu' === en)) {
        const pd_1:any = ((<any>co.rowContextmenu.emit($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },import3.View_DataTableRowWrapperComponent_0,import3.RenderType_DataTableRowWrapperComponent)),
    import0.ɵdid(139264,(null as any),0,import4.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵdid(24576,(null as any),0,import5.DataTableRowWrapperComponent,([] as any[]),{
      rowDetail: [
        0,
        'rowDetail'
      ]
      ,
      detailRowHeight: [
        1,
        'detailRowHeight'
      ]
      ,
      expanded: [
        2,
        'expanded'
      ]
      ,
      row: [
        3,
        'row'
      ]

    }
    ,{rowContextmenu: 'rowContextmenu'}),
    (l()(),import0.ɵted(0,['\n          '])),
      (l()(),import0.ɵeld(0,(null as any),0,2,'datatable-body-row',[[
        'tabindex',
        '-1'
      ]
    ],[
      [
        8,
        'className',
        0
      ]
      ,
      [
        4,
        'height',
        'px'
      ]
      ,
      [
        4,
        'width',
        'px'
      ]

    ]
    ,[
      [
        (null as any),
        'activate'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('keydown' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,5).onKeyDown($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('activate' === en)) {
        const pd_1:any = ((<any>import0.ɵnov((<any>(<any>v.parent).parent),3).onActivate($event,v.context.index)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },import6.View_DataTableBodyRowComponent_0,import6.RenderType_DataTableBodyRowComponent)),
    import0.ɵdid(24576,(null as any),0,import7.DataTableBodyRowComponent,[
      import8.ScrollbarHelper,
      import0.ElementRef
    ]
    ,{
      columns: [
        0,
        'columns'
      ]
      ,
      innerWidth: [
        1,
        'innerWidth'
      ]
      ,
      rowClass: [
        2,
        'rowClass'
      ]
      ,
      row: [
        3,
        'row'
      ]
      ,
      offsetX: [
        4,
        'offsetX'
      ]
      ,
      isSelected: [
        5,
        'isSelected'
      ]
      ,
      rowHeight: [
        6,
        'rowHeight'
      ]

    }
    ,{activate: 'activate'}),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵted(0,['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.getRowsStyles(v.context.$implicit);
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.rowDetail;
    const currVal_2:any = co.getDetailRowHeight(v.context.$implicit,v.context.index);
    const currVal_3:any = (v.context.$implicit.$$expanded === 1);
    const currVal_4:any = v.context.$implicit;
    ck(v,2,0,currVal_1,currVal_2,currVal_3,currVal_4);
    const currVal_8:any = co.columns;
    const currVal_9:any = co.innerWidth;
    const currVal_10:any = co.rowClass;
    const currVal_11:any = v.context.$implicit;
    const currVal_12:any = co.offsetX;
    const currVal_13:any = import0.ɵnov((<any>(<any>v.parent).parent),3).getRowSelected(v.context.$implicit);
    const currVal_14:any = co.getRowHeight(v.context.$implicit);
    ck(v,5,0,currVal_8,currVal_9,currVal_10,currVal_11,currVal_12,currVal_13,currVal_14);
  },(ck,v) => {
    const currVal_5:any = import0.ɵnov(v,5).cssClass;
    const currVal_6:any = import0.ɵnov(v,5).rowHeight;
    const currVal_7:any = import0.ɵnov(v,5).columnsTotalWidths;
    ck(v,4,0,currVal_5,currVal_6,currVal_7);
  });
}
function View_DataTableBodyComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'datatable-scroller',[[
        'class',
        'datatable-scroll'
      ]
    ],[
      [
        4,
        'height',
        'px'
      ]
      ,
      [
        4,
        'width',
        'px'
      ]

    ]
      ,[[
        (null as any),
        'scroll'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('scroll' === en)) {
        const pd_0:any = ((<any>co.onBodyScroll($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import9.View_ScrollerComponent_0,import9.RenderType_ScrollerComponent)),
      import0.ɵdid(122880,[[
        1,
        4
      ]
    ],0,import10.ScrollerComponent,[
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      scrollbarV: [
        0,
        'scrollbarV'
      ]
      ,
      scrollbarH: [
        1,
        'scrollbarH'
      ]
      ,
      scrollHeight: [
        2,
        'scrollHeight'
      ]
      ,
      scrollWidth: [
        3,
        'scrollWidth'
      ]

    }
    ,{scroll: 'scroll'}),
    (l()(),import0.ɵted(0,['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DataTableBodyComponent_3)),
    import0.ɵdid(401408,(null as any),0,import4.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
    ,{
      ngForOf: [
        0,
        'ngForOf'
      ]
      ,
      ngForTrackBy: [
        1,
        'ngForTrackBy'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted(0,['\n      ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co.scrollbarV;
    const currVal_3:any = co.scrollbarH;
    const currVal_4:any = co.scrollHeight;
    const currVal_5:any = co.columnGroupWidths.total;
    ck(v,1,0,currVal_2,currVal_3,currVal_4,currVal_5);
    const currVal_6:any = co.temp;
    const currVal_7:any = co.rowTrackingFn;
    ck(v,4,0,currVal_6,currVal_7);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).scrollHeight;
    const currVal_1:any = import0.ɵnov(v,1).scrollWidth;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_DataTableBodyComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'empty-row'
      ]
      ],[[
        8,
        'innerHTML',
        1
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.emptyMessage;
    ck(v,0,0,currVal_0);
  });
}
export function View_DataTableBodyComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(335544320,1,{scroller: 0}),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),11,'datatable-selection',([] as any[]),(null as any),[
      [
        (null as any),
        'select'
      ]
      ,
      [
        (null as any),
        'activate'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import11.DataTableBodyComponent = v.component;
      if (('select' === en)) {
        const pd_0:any = ((<any>co.select.emit($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('activate' === en)) {
        const pd_1:any = ((<any>co.activate.emit($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },import12.View_DataTableSelectionComponent_0,import12.RenderType_DataTableSelectionComponent)),
      import0.ɵdid(24576,[[
        'selector',
        4
      ]
    ],0,import13.DataTableSelectionComponent,([] as any[]),{
      rows: [
        0,
        'rows'
      ]
      ,
      selected: [
        1,
        'selected'
      ]
      ,
      selectEnabled: [
        2,
        'selectEnabled'
      ]
      ,
      selectionType: [
        3,
        'selectionType'
      ]
      ,
      rowIdentity: [
        4,
        'rowIdentity'
      ]
      ,
      selectCheck: [
        5,
        'selectCheck'
      ]

    }
    ,{
      activate: 'activate',
      select: 'select'
    }
    ),
    (l()(),import0.ɵted(0,['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DataTableBodyComponent_1)),
    import0.ɵdid(8192,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DataTableBodyComponent_2)),
    import0.ɵdid(8192,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DataTableBodyComponent_4)),
    import0.ɵdid(8192,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import11.DataTableBodyComponent = v.component;
    const currVal_0:any = co.temp;
    const currVal_1:any = co.selected;
    const currVal_2:any = co.selectEnabled;
    const currVal_3:any = co.selectionType;
    const currVal_4:any = co.rowIdentity;
    const currVal_5:any = co.selectCheck;
    ck(v,3,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5);
    const currVal_6:any = co.loadingIndicator;
    ck(v,6,0,currVal_6);
    const currVal_7:any = ((co.rows == null)? (null as any): co.rows.length);
    ck(v,9,0,currVal_7);
    const currVal_8:boolean = !((co.rows == null)? (null as any): co.rows.length);
    ck(v,12,0,currVal_8);
  },(null as any));
}
function View_DataTableBodyComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'datatable-body',[[
        'class',
        'datatable-body'
      ]
    ],[
      [
        4,
        'width',
        (null as any)
      ]
      ,
      [
        4,
        'height',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_DataTableBodyComponent_0,RenderType_DataTableBodyComponent)),
    import0.ɵdid(122880,(null as any),0,import11.DataTableBodyComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).bodyWidth;
    const currVal_1:any = import0.ɵnov(v,1).bodyHeight;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const DataTableBodyComponentNgFactory:import0.ComponentFactory<import11.DataTableBodyComponent> = import0.ɵccf('datatable-body',import11.DataTableBodyComponent,View_DataTableBodyComponent_Host_0,{
  scrollbarV: 'scrollbarV',
  scrollbarH: 'scrollbarH',
  loadingIndicator: 'loadingIndicator',
  externalPaging: 'externalPaging',
  rowHeight: 'rowHeight',
  offsetX: 'offsetX',
  emptyMessage: 'emptyMessage',
  selectionType: 'selectionType',
  selected: 'selected',
  rowIdentity: 'rowIdentity',
  rowDetail: 'rowDetail',
  selectCheck: 'selectCheck',
  trackByProp: 'trackByProp',
  rowClass: 'rowClass',
  pageSize: 'pageSize',
  rows: 'rows',
  columns: 'columns',
  offset: 'offset',
  rowCount: 'rowCount',
  innerWidth: 'innerWidth',
  bodyHeight: 'bodyHeight'
}
,{
  scroll: 'scroll',
  page: 'page',
  activate: 'activate',
  select: 'select',
  detailToggle: 'detailToggle',
  rowContextmenu: 'rowContextmenu'
}
,([] as any[]));
