// @swimlane/ngx-datatable/release/components/body/progress-bar.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@swimlane/ngx-datatable/release/components/body/progress-bar.component';
const styles_ProgressBarComponent:any[] = ([] as any[]);
export const RenderType_ProgressBarComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ProgressBarComponent,
  data: {}
}
);
export function View_ProgressBarComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',[
      [
        'class',
        'progress-linear'
      ]
      ,
      [
        'role',
        'progressbar'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'container'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'bar'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(null as any),(null as any));
}
function View_ProgressBarComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'datatable-progress',([] as any[]),(null as any),(null as any),(null as any),View_ProgressBarComponent_0,RenderType_ProgressBarComponent)),
    import0.ɵdid(24576,(null as any),0,import1.ProgressBarComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ProgressBarComponentNgFactory:import0.ComponentFactory<import1.ProgressBarComponent> = import0.ɵccf('datatable-progress',import1.ProgressBarComponent,View_ProgressBarComponent_Host_0,{},{},([] as any[]));
