// @swimlane/ngx-datatable/release/components/body/scroller.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@swimlane/ngx-datatable/release/components/body/scroller.component';
const styles_ScrollerComponent:any[] = ([] as any[]);
export const RenderType_ScrollerComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ScrollerComponent,
  data: {}
}
);
export function View_ScrollerComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(null as any),(null as any));
}
function View_ScrollerComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'datatable-scroller',[[
        'class',
        'datatable-scroll'
      ]
    ],[
      [
        4,
        'height',
        'px'
      ]
      ,
      [
        4,
        'width',
        'px'
      ]

    ]
    ,(null as any),(null as any),View_ScrollerComponent_0,RenderType_ScrollerComponent)),
    import0.ɵdid(122880,(null as any),0,import1.ScrollerComponent,[
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).scrollHeight;
    const currVal_1:any = import0.ɵnov(v,1).scrollWidth;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const ScrollerComponentNgFactory:import0.ComponentFactory<import1.ScrollerComponent> = import0.ɵccf('datatable-scroller',import1.ScrollerComponent,View_ScrollerComponent_Host_0,{
  scrollbarV: 'scrollbarV',
  scrollbarH: 'scrollbarH',
  scrollHeight: 'scrollHeight',
  scrollWidth: 'scrollWidth'
}
,{scroll: 'scroll'},['*']);
