// primeng/components/multiselect/multiselect.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/multiselect/multiselect';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/dom/domhandler';
import * as import4 from 'primeng/components/utils/objectutils';
import * as import5 from '@angular/forms';
class MultiSelectModuleInjector extends import0.ɵNgModuleInjector<import1.MultiSelectModule> {
  _CommonModule_0:import2.CommonModule;
  _MultiSelectModule_1:import1.MultiSelectModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.MultiSelectModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._MultiSelectModule_1 = new import1.MultiSelectModule();
    return this._MultiSelectModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.MultiSelectModule)) { return this._MultiSelectModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const MultiSelectModuleNgFactory:import0.NgModuleFactory<import1.MultiSelectModule> = new import0.NgModuleFactory<any>(MultiSelectModuleInjector,import1.MultiSelectModule);
const styles_MultiSelect:any[] = ([] as any[]);
export const RenderType_MultiSelect:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_MultiSelect,
  data: {}
}
);
function View_MultiSelect_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',[[
        'class',
        'ui-multiselect-filter-container'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'input',[
      [
        'class',
        'ui-inputtext ui-widget ui-state-default ui-corner-all'
      ]
      ,
      [
        'role',
        'textbox'
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'input'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('input' === en)) {
        const pd_0:any = ((<any>co.onFilter($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-fw fa-search'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(null as any),(null as any));
}
function View_MultiSelect_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),23,'li',[[
        'class',
        'ui-multiselect-item ui-corner-all'
      ]
      ],[[
        4,
        'display',
        (null as any)
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onItemClick($event,v.context.$implicit.value)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-highlight']),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),15,'div',[[
        'class',
        'ui-chkbox ui-widget'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-helper-hidden-accessible'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'input',[
      [
        'readonly',
        'readonly'
      ]
      ,
      [
        'type',
        'checkbox'
      ]

    ]
      ,[[
        8,
        'checked',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'ui-chkbox-box ui-widget ui-corner-all ui-state-default'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-active']),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-chkbox-icon ui-clickable'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['fa fa-check']),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'label',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = 'ui-multiselect-item ui-corner-all';
    const currVal_2:any = ck(v,2,0,co.isSelected(v.context.$implicit.value));
    ck(v,1,0,currVal_1,currVal_2);
    const currVal_4:any = 'ui-chkbox-box ui-widget ui-corner-all ui-state-default';
    const currVal_5:any = ck(v,13,0,co.isSelected(v.context.$implicit.value));
    ck(v,12,0,currVal_4,currVal_5);
    const currVal_6:any = 'ui-chkbox-icon ui-clickable';
    const currVal_7:any = ck(v,17,0,co.isSelected(v.context.$implicit.value));
    ck(v,16,0,currVal_6,currVal_7);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.isItemVisible(v.context.$implicit)? 'block': 'none');
    ck(v,0,0,currVal_0);
    const currVal_3:any = co.isSelected(v.context.$implicit.value);
    ck(v,8,0,currVal_3);
    const currVal_8:any = v.context.$implicit.label;
    ck(v,22,0,currVal_8);
  });
}
export function View_MultiSelect_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{containerViewChild: 0}),
    import0.ɵqud(201326592,2,{panelViewChild: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'container',
        1
      ]

    ]
      ,(null as any),62,'div',([] as any[]),(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.MultiSelect = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onMouseclick($event,import0.ɵnov(v,10))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-multiselect ui-widget ui-state-default ui-corner-all',
      'ui-state-focus',
      'ui-state-disabled'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-helper-hidden-accessible'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,[[
        'in',
        1
      ]
    ],(null as any),0,'input',[
      [
        'readonly',
        'readonly'
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
    ,[
      [
        1,
        'id',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.MultiSelect = v.component;
      if (('focus' === en)) {
        const pd_0:any = ((<any>co.onFocus($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>co.onInputBlur($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'ui-multiselect-label-container'
      ]
      ],[[
        8,
        'title',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'label',[[
        'class',
        'ui-multiselect-label ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['ui-multiselect-trigger ui-state-default ui-corner-right']),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-fw fa-caret-down ui-clickable'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'panel',
        1
      ]

    ]
      ,(null as any),38,'div',[[
        'class',
        'ui-multiselect-panel ui-widget ui-widget-content ui-corner-all ui-shadow'
      ]
      ],[[
        4,
        'display',
        (null as any)
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.MultiSelect = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>(co.panelClick = true)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),26,'div',[[
        'class',
        'ui-widget-header ui-corner-all ui-multiselect-header ui-helper-clearfix'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),15,'div',[[
        'class',
        'ui-chkbox ui-widget'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-helper-hidden-accessible'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,[[
        'cb',
        1
      ]
    ],(null as any),0,'input',[
      [
        'readonly',
        'readonly'
      ]
      ,
      [
        'type',
        'checkbox'
      ]

    ]
      ,[[
        8,
        'checked',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'ui-chkbox-box ui-widget ui-corner-all ui-state-default'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.MultiSelect = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggleAll($event,import0.ɵnov(v,34))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-active']),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-chkbox-icon ui-clickable'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['fa fa-check']),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiSelect_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'a',[
      [
        'class',
        'ui-multiselect-close ui-corner-all'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.MultiSelect = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.close($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-close'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'ui-multiselect-items-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ul',[[
        'class',
        'ui-multiselect-items ui-multiselect-list ui-widget-content ui-widget ui-corner-all ui-helper-reset'
      ]
      ],[[
        4,
        'max-height',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiSelect_2)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.MultiSelect = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,5,0,true,co.focus,co.disabled);
    ck(v,4,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,6,0,currVal_2);
    const currVal_8:any = ck(v,21,0,true);
    ck(v,20,0,currVal_8);
    const currVal_11:any = 'ui-chkbox-box ui-widget ui-corner-all ui-state-default';
    const currVal_12:any = ck(v,39,0,co.isAllChecked());
    ck(v,38,0,currVal_11,currVal_12);
    const currVal_13:any = 'ui-chkbox-icon ui-clickable';
    const currVal_14:any = ck(v,43,0,co.isAllChecked());
    ck(v,42,0,currVal_13,currVal_14);
    const currVal_15:any = co.filter;
    ck(v,48,0,currVal_15);
    const currVal_17:any = co.options;
    ck(v,61,0,currVal_17);
  },(ck,v) => {
    var co:import1.MultiSelect = v.component;
    const currVal_3:any = co.inputId;
    const currVal_4:any = co.disabled;
    const currVal_5:any = co.tabindex;
    ck(v,10,0,currVal_3,currVal_4,currVal_5);
    const currVal_6:any = co.valuesAsString;
    ck(v,13,0,currVal_6);
    const currVal_7:any = co.valuesAsString;
    ck(v,16,0,currVal_7);
    const currVal_9:any = (co.overlayVisible? 'block': 'none');
    ck(v,26,0,currVal_9);
    const currVal_10:any = co.isAllChecked();
    ck(v,34,0,currVal_10);
    const currVal_16:any = (co.scrollHeight || 'auto');
    ck(v,58,0,currVal_16);
  });
}
function View_MultiSelect_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'p-multiSelect',([] as any[]),(null as any),(null as any),(null as any),View_MultiSelect_0,RenderType_MultiSelect)),
    import0.ɵprd(256,(null as any),import3.DomHandler,import3.DomHandler,([] as any[])),
    import0.ɵprd(256,(null as any),import4.ObjectUtils,import4.ObjectUtils,([] as any[])),
    import0.ɵdid(6545408,(null as any),0,import1.MultiSelect,[
      import0.ElementRef,
      import3.DomHandler,
      import0.Renderer2,
      import0.IterableDiffers,
      import4.ObjectUtils,
      import0.ChangeDetectorRef
    ]
    ,(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import5.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.MultiSelect])
  ]
  ,(ck,v) => {
    ck(v,3,0);
  },(null as any));
}
export const MultiSelectNgFactory:import0.ComponentFactory<import1.MultiSelect> = import0.ɵccf('p-multiSelect',import1.MultiSelect,View_MultiSelect_Host_0,{
  options: 'options',
  scrollHeight: 'scrollHeight',
  defaultLabel: 'defaultLabel',
  style: 'style',
  styleClass: 'styleClass',
  inputId: 'inputId',
  disabled: 'disabled',
  filter: 'filter',
  overlayVisible: 'overlayVisible',
  tabindex: 'tabindex',
  appendTo: 'appendTo',
  dataKey: 'dataKey',
  displaySelectedLabel: 'displaySelectedLabel',
  maxSelectedLabels: 'maxSelectedLabels',
  selectedItemsLabel: 'selectedItemsLabel'
}
,{
  onChange: 'onChange',
  onBlur: 'onBlur'
}
,([] as any[]));
