// primeng/components/panelmenu/panelmenu.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/panelmenu/panelmenu';
import * as import2 from '@angular/common';
import * as import3 from '@angular/router';
class PanelMenuModuleInjector extends import0.ɵNgModuleInjector<import1.PanelMenuModule> {
  _CommonModule_0:import2.CommonModule;
  _RouterModule_1:import3.RouterModule;
  _PanelMenuModule_2:import1.PanelMenuModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.PanelMenuModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._RouterModule_1 = new import3.RouterModule(this.parent.get(import3.ɵa,(null as any)),this.parent.get(import3.Router,(null as any)));
    this._PanelMenuModule_2 = new import1.PanelMenuModule();
    return this._PanelMenuModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.RouterModule)) { return this._RouterModule_1; }
    if ((token === import1.PanelMenuModule)) { return this._PanelMenuModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const PanelMenuModuleNgFactory:import0.NgModuleFactory<import1.PanelMenuModule> = new import0.NgModuleFactory<any>(PanelMenuModuleInjector,import1.PanelMenuModule);
const styles_PanelMenu:any[] = ([] as any[]);
export const RenderType_PanelMenu:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_PanelMenu,
      data: {'animation': [{
        name: 'rootItem',
        definitions: [
          {
            type: 0,
            name: 'hidden',
            styles: {
              type: 6,
              styles: {height: '0px'}
            }

          }
          ,
          {
            type: 0,
            name: 'visible',
            styles: {
              type: 6,
              styles: {height: '*'}
            }

          }
          ,
          {
            type: 1,
            expr: 'visible => hidden',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms cubic-bezier(0.86, 0, 0.07, 1)'
            }

          }
          ,
          {
            type: 1,
            expr: 'hidden => visible',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms cubic-bezier(0.86, 0, 0.07, 1)'
            }

          }

        ]

      }
  ]}
}
);
function View_PanelMenu_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-panelmenu-icon fa'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'fa-caret-right',
      'fa-caret-down'
    ]
    )
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-panelmenu-icon fa';
    const currVal_1:any = ck(v,2,0,!(<any>(<any>v.parent).parent).context.$implicit.expanded,(<any>(<any>v.parent).parent).context.$implicit.expanded);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_PanelMenu_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-icon fa'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-menuitem-icon fa';
    const currVal_1:any = (<any>(<any>v.parent).parent).context.$implicit.icon;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_PanelMenu_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'a',([] as any[]),[
      [
        8,
        'href',
        4
      ]
      ,
      [
        1,
        'target',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.handleClick($event,(<any>v.parent).context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['ui-panelmenu-headerlink-hasicon']),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenu_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenu_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    const currVal_2:any = ck(v,2,0,(<any>v.parent).context.$implicit.icon);
    ck(v,1,0,currVal_2);
    const currVal_3:any = (<any>v.parent).context.$implicit.items;
    ck(v,5,0,currVal_3);
    const currVal_4:any = (<any>v.parent).context.$implicit.icon;
    ck(v,7,0,currVal_4);
  },(ck,v) => {
    const currVal_0:any = ((<any>v.parent).context.$implicit.url || '#');
    const currVal_1:any = (<any>v.parent).context.$implicit.target;
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_5:any = (<any>v.parent).context.$implicit.label;
    ck(v,9,0,currVal_5);
  });
}
function View_PanelMenu_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-panelmenu-icon fa'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'fa-caret-right',
      'fa-caret-down'
    ]
    )
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-panelmenu-icon fa';
    const currVal_1:any = ck(v,2,0,!(<any>(<any>v.parent).parent).context.$implicit.expanded,(<any>(<any>v.parent).parent).context.$implicit.expanded);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_PanelMenu_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-icon fa'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-menuitem-icon fa';
    const currVal_1:any = (<any>(<any>v.parent).parent).context.$implicit.icon;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_PanelMenu_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),15,'a',([] as any[]),[
      [
        1,
        'target',
        0
      ]
      ,
      [
        1,
        'target',
        0
      ]
      ,
      [
        8,
        'href',
        4
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,3).onClick($event.button,$event.ctrlKey,$event.metaKey)) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>co.handleClick($event,(<any>v.parent).context.$implicit)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['ui-panelmenu-headerlink-hasicon']),
      import0.ɵdid(335872,[[
        2,
        4
      ]
    ],0,import3.RouterLinkWithHref,[
      import3.Router,
      import3.ActivatedRoute,
      import2.LocationStrategy
    ]
      ,{routerLink: [
        0,
        'routerLink'
      ]
    },(null as any)),
    import0.ɵdid(860160,(null as any),2,import3.RouterLinkActive,[
      import3.Router,
      import0.ElementRef,
      import0.Renderer,
      import0.ChangeDetectorRef
    ]
    ,{
      routerLinkActiveOptions: [
        0,
        'routerLinkActiveOptions'
      ]
      ,
      routerLinkActive: [
        1,
        'routerLinkActive'
      ]

    }
    ,(null as any)),
    import0.ɵqud(301989888,1,{links: 1}),
    import0.ɵqud(301989888,2,{linksWithHrefs: 1}),
    import0.ɵpod(['exact']),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenu_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenu_7)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    const currVal_3:any = ck(v,2,0,(<any>v.parent).context.$implicit.icon);
    ck(v,1,0,currVal_3);
    const currVal_4:any = (<any>v.parent).context.$implicit.routerLink;
    ck(v,3,0,currVal_4);
    const currVal_5:any = ((<any>v.parent).context.$implicit.routerLinkActiveOptions || ck(v,7,0,false));
    const currVal_6:any = 'ui-state-active';
    ck(v,4,0,currVal_5,currVal_6);
    const currVal_7:any = (<any>v.parent).context.$implicit.items;
    ck(v,10,0,currVal_7);
    const currVal_8:any = (<any>v.parent).context.$implicit.icon;
    ck(v,12,0,currVal_8);
  },(ck,v) => {
    const currVal_0:any = (<any>v.parent).context.$implicit.target;
    const currVal_1:any = import0.ɵnov(v,3).target;
    const currVal_2:any = import0.ɵnov(v,3).href;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
    const currVal_9:any = (<any>v.parent).context.$implicit.label;
    ck(v,14,0,currVal_9);
  });
}
function View_PanelMenu_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),9,'div',[[
        'class',
        'ui-panelmenu-content-wrapper'
      ]
      ],[[
        24,
        '@rootItem',
        0
      ]
      ],[[
        (null as any),
        '@rootItem.done'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('@rootItem.done' === en)) {
        const pd_0:any = ((<any>co.onToggleDone($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-panelmenu-content-wrapper-overflown']),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'ui-panelmenu-content ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-panelMenuSub',([] as any[]),(null as any),(null as any),(null as any),View_PanelMenuSub_0,RenderType_PanelMenuSub)),
    import0.ɵdid(24576,(null as any),0,import1.PanelMenuSub,([] as any[]),{
      item: [
        0,
        'item'
      ]
      ,
      expanded: [
        1,
        'expanded'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = 'ui-panelmenu-content-wrapper';
    const currVal_2:any = ck(v,2,0,(!(<any>v.parent).context.$implicit.expanded || co.animating));
    ck(v,1,0,currVal_1,currVal_2);
    const currVal_3:any = (<any>v.parent).context.$implicit;
    const currVal_4:any = true;
    ck(v,7,0,currVal_3,currVal_4);
  },(ck,v) => {
    const currVal_0:any = ((<any>v.parent).context.$implicit.expanded? 'visible': 'hidden');
    ck(v,0,0,currVal_0);
  });
}
function View_PanelMenu_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),16,'div',[[
        'class',
        'ui-panelmenu-panel'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-widget ui-panelmenu-header ui-state-default',
      'ui-corner-top',
      'ui-corner-bottom',
      'ui-state-active',
      'ui-state-disabled'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenu_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenu_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenu_8)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.styleClass;
    const currVal_1:any = ck(v,4,0,true,v.context.first,(v.context.last && !v.context.$implicit.expanded),v.context.$implicit.expanded,v.context.$implicit.disabled);
    ck(v,3,0,currVal_0,currVal_1);
    const currVal_2:any = v.context.$implicit.style;
    ck(v,5,0,currVal_2);
    const currVal_3:boolean = !v.context.$implicit.routerLink;
    ck(v,8,0,currVal_3);
    const currVal_4:any = v.context.$implicit.routerLink;
    ck(v,11,0,currVal_4);
    const currVal_5:any = v.context.$implicit.items;
    ck(v,15,0,currVal_5);
  },(null as any));
}
export function View_PanelMenu_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenu_1)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.PanelMenu = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-panelmenu ui-widget';
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_3:any = co.model;
    ck(v,6,0,currVal_3);
  },(null as any));
}
function View_PanelMenu_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-panelMenu',([] as any[]),(null as any),(null as any),(null as any),View_PanelMenu_0,RenderType_PanelMenu)),
    import0.ɵdid(24576,(null as any),0,import1.PanelMenu,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const PanelMenuNgFactory:import0.ComponentFactory<import1.PanelMenu> = import0.ɵccf('p-panelMenu',import1.PanelMenu,View_PanelMenu_Host_0,{
  model: 'model',
  style: 'style',
  styleClass: 'styleClass'
}
,{},([] as any[]));
const styles_PanelMenuSub:any[] = ([] as any[]);
export const RenderType_PanelMenuSub:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_PanelMenuSub,
      data: {'animation': [{
        name: 'submenu',
        definitions: [
          {
            type: 0,
            name: 'hidden',
            styles: {
              type: 6,
              styles: {height: '0px'}
            }

          }
          ,
          {
            type: 0,
            name: 'visible',
            styles: {
              type: 6,
              styles: {height: '*'}
            }

          }
          ,
          {
            type: 1,
            expr: 'visible => hidden',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms cubic-bezier(0.86, 0, 0.07, 1)'
            }

          }
          ,
          {
            type: 1,
            expr: 'hidden => visible',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms cubic-bezier(0.86, 0, 0.07, 1)'
            }

          }

        ]

      }
  ]}
}
);
function View_PanelMenuSub_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'li',[[
        'class',
        'ui-menu-separator ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(null as any),(null as any));
}
function View_PanelMenuSub_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-panelmenu-icon fa fa-fw'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'fa-caret-right',
      'fa-caret-down'
    ]
    )
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-panelmenu-icon fa fa-fw';
    const currVal_1:any = ck(v,2,0,!(<any>(<any>(<any>v.parent).parent).parent).context.$implicit.expanded,(<any>(<any>(<any>v.parent).parent).parent).context.$implicit.expanded);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_PanelMenuSub_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-icon fa fa-fw'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-menuitem-icon fa fa-fw';
    const currVal_1:any = (<any>(<any>(<any>v.parent).parent).parent).context.$implicit.icon;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_PanelMenuSub_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'a',[[
        'class',
        'ui-menuitem-link ui-corner-all'
      ]
    ],[
      [
        8,
        'href',
        4
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        1,
        'target',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.handleClick($event,(<any>(<any>v.parent).parent).context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-menuitem-link-hasicon',
      'ui-state-disabled'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenuSub_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenuSub_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    const currVal_3:any = 'ui-menuitem-link ui-corner-all';
    const currVal_4:any = ck(v,2,0,((<any>(<any>v.parent).parent).context.$implicit.icon && (<any>(<any>v.parent).parent).context.$implicit.items),(<any>(<any>v.parent).parent).context.$implicit.disabled);
    ck(v,1,0,currVal_3,currVal_4);
    const currVal_5:any = (<any>(<any>v.parent).parent).context.$implicit.items;
    ck(v,5,0,currVal_5);
    const currVal_6:any = (<any>(<any>v.parent).parent).context.$implicit.icon;
    ck(v,7,0,currVal_6);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ((<any>(<any>v.parent).parent).context.$implicit.url || '#');
    const currVal_1:any = (co.item.expanded? (null as any): '-1');
    const currVal_2:any = (<any>(<any>v.parent).parent).context.$implicit.target;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
    const currVal_7:any = (<any>(<any>v.parent).parent).context.$implicit.label;
    ck(v,9,0,currVal_7);
  });
}
function View_PanelMenuSub_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-panelmenu-icon fa fa-fw'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'fa-caret-right',
      'fa-caret-down'
    ]
    )
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-panelmenu-icon fa fa-fw';
    const currVal_1:any = ck(v,2,0,!(<any>(<any>(<any>v.parent).parent).parent).context.$implicit.expanded,(<any>(<any>(<any>v.parent).parent).parent).context.$implicit.expanded);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_PanelMenuSub_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-icon fa fa-fw'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-menuitem-icon fa fa-fw';
    const currVal_1:any = (<any>(<any>(<any>v.parent).parent).parent).context.$implicit.icon;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_PanelMenuSub_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),15,'a',[[
        'class',
        'ui-menuitem-link ui-corner-all'
      ]
    ],[
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        1,
        'target',
        0
      ]
      ,
      [
        1,
        'target',
        0
      ]
      ,
      [
        8,
        'href',
        4
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,3).onClick($event.button,$event.ctrlKey,$event.metaKey)) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>co.handleClick($event,(<any>(<any>v.parent).parent).context.$implicit)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-menuitem-link-hasicon',
      'ui-state-disabled'
    ]
    ),
      import0.ɵdid(335872,[[
        2,
        4
      ]
    ],0,import3.RouterLinkWithHref,[
      import3.Router,
      import3.ActivatedRoute,
      import2.LocationStrategy
    ]
      ,{routerLink: [
        0,
        'routerLink'
      ]
    },(null as any)),
    import0.ɵdid(860160,(null as any),2,import3.RouterLinkActive,[
      import3.Router,
      import0.ElementRef,
      import0.Renderer,
      import0.ChangeDetectorRef
    ]
    ,{
      routerLinkActiveOptions: [
        0,
        'routerLinkActiveOptions'
      ]
      ,
      routerLinkActive: [
        1,
        'routerLinkActive'
      ]

    }
    ,(null as any)),
    import0.ɵqud(301989888,1,{links: 1}),
    import0.ɵqud(301989888,2,{linksWithHrefs: 1}),
    import0.ɵpod(['exact']),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenuSub_8)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenuSub_9)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    const currVal_4:any = 'ui-menuitem-link ui-corner-all';
    const currVal_5:any = ck(v,2,0,((<any>(<any>v.parent).parent).context.$implicit.icon && (<any>(<any>v.parent).parent).context.$implicit.items),(<any>(<any>v.parent).parent).context.$implicit.disabled);
    ck(v,1,0,currVal_4,currVal_5);
    const currVal_6:any = (<any>(<any>v.parent).parent).context.$implicit.routerLink;
    ck(v,3,0,currVal_6);
    const currVal_7:any = ((<any>(<any>v.parent).parent).context.$implicit.routerLinkActiveOptions || ck(v,7,0,false));
    const currVal_8:any = 'ui-state-active';
    ck(v,4,0,currVal_7,currVal_8);
    const currVal_9:any = (<any>(<any>v.parent).parent).context.$implicit.items;
    ck(v,10,0,currVal_9);
    const currVal_10:any = (<any>(<any>v.parent).parent).context.$implicit.icon;
    ck(v,12,0,currVal_10);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.item.expanded? (null as any): '-1');
    const currVal_1:any = (<any>(<any>v.parent).parent).context.$implicit.target;
    const currVal_2:any = import0.ɵnov(v,3).target;
    const currVal_3:any = import0.ɵnov(v,3).href;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
    const currVal_11:any = (<any>(<any>v.parent).parent).context.$implicit.label;
    ck(v,14,0,currVal_11);
  });
}
function View_PanelMenuSub_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-panelMenuSub',([] as any[]),(null as any),(null as any),(null as any),View_PanelMenuSub_0,RenderType_PanelMenuSub)),
    import0.ɵdid(24576,(null as any),0,import1.PanelMenuSub,([] as any[]),{
      item: [
        0,
        'item'
      ]
      ,
      expanded: [
        1,
        'expanded'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = (<any>(<any>v.parent).parent).context.$implicit;
    const currVal_1:any = (<any>(<any>v.parent).parent).context.$implicit.expanded;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_PanelMenuSub_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),13,'li',[[
        'class',
        'ui-menuitem ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-menu-parent']),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenuSub_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenuSub_7)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenuSub_10)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = (<any>v.parent).context.$implicit.styleClass;
    const currVal_1:any = ck(v,2,0,(<any>v.parent).context.$implicit.items);
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = (<any>v.parent).context.$implicit.style;
    ck(v,3,0,currVal_2);
    const currVal_3:boolean = !(<any>v.parent).context.$implicit.routerLink;
    ck(v,6,0,currVal_3);
    const currVal_4:any = (<any>v.parent).context.$implicit.routerLink;
    ck(v,9,0,currVal_4);
    const currVal_5:any = (<any>v.parent).context.$implicit.items;
    ck(v,12,0,currVal_5);
  },(null as any));
}
function View_PanelMenuSub_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenuSub_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenuSub_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.separator;
    ck(v,2,0,currVal_0);
    const currVal_1:boolean = !v.context.$implicit.separator;
    ck(v,4,0,currVal_1);
  },(null as any));
}
export function View_PanelMenuSub_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ul',[[
        'class',
        'ui-menu-list ui-helper-reset'
      ]
      ],[[
        24,
        '@submenu',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PanelMenuSub_1)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.PanelMenuSub = v.component;
    const currVal_1:any = co.item.items;
    ck(v,4,0,currVal_1);
  },(ck,v) => {
    var co:import1.PanelMenuSub = v.component;
    const currVal_0:any = (co.expanded? 'visible': 'hidden');
    ck(v,1,0,currVal_0);
  });
}
function View_PanelMenuSub_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-panelMenuSub',([] as any[]),(null as any),(null as any),(null as any),View_PanelMenuSub_0,RenderType_PanelMenuSub)),
    import0.ɵdid(24576,(null as any),0,import1.PanelMenuSub,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const PanelMenuSubNgFactory:import0.ComponentFactory<import1.PanelMenuSub> = import0.ɵccf('p-panelMenuSub',import1.PanelMenuSub,View_PanelMenuSub_Host_0,{
  item: 'item',
  expanded: 'expanded'
}
,{},([] as any[]));
