// primeng/components/overlaypanel/overlaypanel.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/overlaypanel/overlaypanel';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/dom/domhandler';
class OverlayPanelModuleInjector extends import0.ɵNgModuleInjector<import1.OverlayPanelModule> {
  _CommonModule_0:import2.CommonModule;
  _OverlayPanelModule_1:import1.OverlayPanelModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.OverlayPanelModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._OverlayPanelModule_1 = new import1.OverlayPanelModule();
    return this._OverlayPanelModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.OverlayPanelModule)) { return this._OverlayPanelModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const OverlayPanelModuleNgFactory:import0.NgModuleFactory<import1.OverlayPanelModule> = new import0.NgModuleFactory<any>(OverlayPanelModuleInjector,import1.OverlayPanelModule);
const styles_OverlayPanel:any[] = ([] as any[]);
export const RenderType_OverlayPanel:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_OverlayPanel,
  data: {}
}
);
function View_OverlayPanel_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'a',[
      [
        'class',
        'ui-overlaypanel-close ui-state-default'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onCloseClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-fw fa-close'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(null as any));
}
export function View_OverlayPanel_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'div',([] as any[]),[[
        4,
        'display',
        (null as any)
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.OverlayPanel = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onPanelClick()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-overlaypanel-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_OverlayPanel_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.OverlayPanel = v.component;
    const currVal_1:any = co.styleClass;
    const currVal_2:any = 'ui-overlaypanel ui-widget ui-widget-content ui-corner-all ui-shadow';
    ck(v,2,0,currVal_1,currVal_2);
    const currVal_3:any = co.style;
    ck(v,3,0,currVal_3);
    const currVal_4:any = co.showCloseIcon;
    ck(v,11,0,currVal_4);
  },(ck,v) => {
    var co:import1.OverlayPanel = v.component;
    const currVal_0:any = (co.visible? 'block': 'none');
    ck(v,1,0,currVal_0);
  });
}
function View_OverlayPanel_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-overlayPanel',([] as any[]),(null as any),(null as any),(null as any),View_OverlayPanel_0,RenderType_OverlayPanel)),
    import0.ɵprd(256,(null as any),import3.DomHandler,import3.DomHandler,([] as any[])),
    import0.ɵdid(2220032,(null as any),0,import1.OverlayPanel,[
      import0.ElementRef,
      import3.DomHandler,
      import0.Renderer2,
      import0.ChangeDetectorRef
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,2,0);
  },(null as any));
}
export const OverlayPanelNgFactory:import0.ComponentFactory<import1.OverlayPanel> = import0.ɵccf('p-overlayPanel',import1.OverlayPanel,View_OverlayPanel_Host_0,{
  dismissable: 'dismissable',
  showCloseIcon: 'showCloseIcon',
  style: 'style',
  styleClass: 'styleClass',
  appendTo: 'appendTo'
}
,{
  onBeforeShow: 'onBeforeShow',
  onAfterShow: 'onAfterShow',
  onBeforeHide: 'onBeforeHide',
  onAfterHide: 'onAfterHide'
}
,['*']);
