// primeng/components/messages/messages.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/messages/messages';
import * as import2 from '@angular/common';
class MessagesModuleInjector extends import0.ɵNgModuleInjector<import1.MessagesModule> {
  _CommonModule_0:import2.CommonModule;
  _MessagesModule_1:import1.MessagesModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.MessagesModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._MessagesModule_1 = new import1.MessagesModule();
    return this._MessagesModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.MessagesModule)) { return this._MessagesModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const MessagesModuleNgFactory:import0.NgModuleFactory<import1.MessagesModule> = new import0.NgModuleFactory<any>(MessagesModuleInjector,import1.MessagesModule);
const styles_Messages:any[] = ([] as any[]);
export const RenderType_Messages:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Messages,
  data: {}
}
);
function View_Messages_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'a',[
      [
        'class',
        'ui-messages-close'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.clear($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'i',[[
        'class',
        'fa fa-close'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(null as any));
}
function View_Messages_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'li',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-messages-summary'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-messages-detail'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = v.context.$implicit.summary;
    ck(v,3,0,currVal_0);
    const currVal_1:any = v.context.$implicit.detail;
    ck(v,6,0,currVal_1);
  });
}
function View_Messages_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),15,'div',[
      [
        'class',
        'ui-messages ui-widget ui-corner-all'
      ]
      ,
      [
        'style',
        'display:block'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-messages-info',
      'ui-messages-warn',
      'ui-messages-error',
      'ui-messages-success'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Messages_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-messages-icon fa fa-fw fa-2x'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ul',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Messages_3)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-messages ui-widget ui-corner-all';
    const currVal_1:any = ck(v,2,0,(co.value[0].severity === 'info'),(co.value[0].severity === 'warn'),(co.value[0].severity === 'error'),(co.value[0].severity === 'success'));
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = co.closable;
    ck(v,5,0,currVal_2);
    const currVal_3:any = 'ui-messages-icon fa fa-fw fa-2x';
    const currVal_4:any = co.icon;
    ck(v,8,0,currVal_3,currVal_4);
    const currVal_5:any = co.value;
    ck(v,13,0,currVal_5);
  },(null as any));
}
export function View_Messages_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Messages_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Messages = v.component;
    const currVal_0:any = co.hasMessages();
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_Messages_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-messages',([] as any[]),(null as any),(null as any),(null as any),View_Messages_0,RenderType_Messages)),
    import0.ɵdid(24576,(null as any),0,import1.Messages,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const MessagesNgFactory:import0.ComponentFactory<import1.Messages> = import0.ɵccf('p-messages',import1.Messages,View_Messages_Host_0,{
  value: 'value',
  closable: 'closable'
}
,{valueChange: 'valueChange'},([] as any[]));
