// primeng/components/datascroller/datascroller.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/datascroller/datascroller';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from 'primeng/components/dom/domhandler';
class DataScrollerModuleInjector extends import0.ɵNgModuleInjector<import1.DataScrollerModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _DataScrollerModule_2:import1.DataScrollerModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.DataScrollerModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._DataScrollerModule_2 = new import1.DataScrollerModule();
    return this._DataScrollerModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import1.DataScrollerModule)) { return this._DataScrollerModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const DataScrollerModuleNgFactory:import0.NgModuleFactory<import1.DataScrollerModule> = new import0.NgModuleFactory<any>(DataScrollerModuleInjector,import1.DataScrollerModule);
const styles_DataScroller:any[] = ([] as any[]);
export const RenderType_DataScroller:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DataScroller,
  data: {}
}
);
function View_DataScroller_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-datascroller-header ui-widget-header ui-corner-top'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(null as any),(null as any));
}
function View_DataScroller_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_DataScroller_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'li',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataScroller_3)),
    import0.ɵdid(106496,(null as any),0,import3.TemplateWrapper,[import0.ViewContainerRef],{
      item: [
        0,
        'item'
      ]
      ,
      templateRef: [
        1,
        'templateRef'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = v.context.$implicit;
    const currVal_1:any = co.itemTemplate;
    ck(v,3,0,currVal_0,currVal_1);
  },(null as any));
}
function View_DataScroller_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-datascroller-footer ui-widget-header ui-corner-bottom'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    import0.ɵncd((null as any),1),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(null as any),(null as any));
}
export function View_DataScroller_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{contentViewChild: 0}),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),21,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-datascroller ui-widget',
      'ui-datascroller-inline'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataScroller_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'content',
        1
      ]

    ]
      ,(null as any),9,'div',[[
        'class',
        'ui-datascroller-content ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod(['max-height']),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ul',[[
        'class',
        'ui-datascroller-list'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataScroller_2)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataScroller_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.DataScroller = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,4,0,true,co.inline);
    ck(v,3,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,5,0,currVal_2);
    const currVal_3:any = co.header;
    ck(v,8,0,currVal_3);
    const currVal_4:any = ck(v,12,0,co.scrollHeight);
    ck(v,11,0,currVal_4);
    const currVal_5:any = co.dataToRender;
    ck(v,17,0,currVal_5);
    const currVal_6:any = co.footer;
    ck(v,22,0,currVal_6);
  },(null as any));
}
function View_DataScroller_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'p-dataScroller',([] as any[]),(null as any),(null as any),(null as any),View_DataScroller_0,RenderType_DataScroller)),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵdid(2711552,(null as any),3,import1.DataScroller,[
      import0.ElementRef,
      import0.Renderer2,
      import4.DomHandler
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(167772160,1,{header: 0}),
    import0.ɵqud(167772160,2,{footer: 0}),
    import0.ɵqud(301989888,3,{templates: 1})
  ]
  ,(null as any),(null as any));
}
export const DataScrollerNgFactory:import0.ComponentFactory<import1.DataScroller> = import0.ɵccf('p-dataScroller',import1.DataScroller,View_DataScroller_Host_0,{
  rows: 'rows',
  lazy: 'lazy',
  style: 'style',
  styleClass: 'styleClass',
  buffer: 'buffer',
  inline: 'inline',
  scrollHeight: 'scrollHeight',
  loader: 'loader',
  value: 'value'
}
,{onLazyLoad: 'onLazyLoad'},[
  'p-header',
  'p-footer'
]
);
