// primeng/components/galleria/galleria.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/galleria/galleria';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/dom/domhandler';
class GalleriaModuleInjector extends import0.ɵNgModuleInjector<import1.GalleriaModule> {
  _CommonModule_0:import2.CommonModule;
  _GalleriaModule_1:import1.GalleriaModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.GalleriaModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._GalleriaModule_1 = new import1.GalleriaModule();
    return this._GalleriaModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.GalleriaModule)) { return this._GalleriaModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const GalleriaModuleNgFactory:import0.NgModuleFactory<import1.GalleriaModule> = new import0.NgModuleFactory<any>(GalleriaModuleInjector,import1.GalleriaModule);
const styles_Galleria:any[] = ([] as any[]);
export const RenderType_Galleria:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Galleria,
  data: {}
}
);
function View_Galleria_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'li',[[
        'class',
        'ui-galleria-panel'
      ]
    ],[
      [
        4,
        'width',
        'px'
      ]
      ,
      [
        4,
        'height',
        'px'
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.clickImage($event,v.context.$implicit,v.context.index)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-helper-hidden']),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'img',[[
        'class',
        'ui-panel-images'
      ]
    ],[
      [
        8,
        'src',
        4
      ]
      ,
      [
        8,
        'alt',
        0
      ]
      ,
      [
        8,
        'title',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = 'ui-galleria-panel';
    const currVal_3:any = ck(v,2,0,(v.context.index != co.activeIndex));
    ck(v,1,0,currVal_2,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.panelWidth;
    const currVal_1:any = co.panelHeight;
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_4:any = v.context.$implicit.source;
    const currVal_5:any = v.context.$implicit.alt;
    const currVal_6:any = v.context.$implicit.title;
    ck(v,4,0,currVal_4,currVal_5,currVal_6);
  });
}
function View_Galleria_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,[[
        'frame',
        1
      ]
      ],(null as any),8,'li',[[
        'class',
        'ui-galleria-frame'
      ]
    ],[
      [
        4,
        'width',
        'px'
      ]
      ,
      [
        4,
        'height',
        'px'
      ]
      ,
      [
        4,
        'transition',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.frameClick(import0.ɵnov(v,0))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-galleria-frame-active']),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-galleria-frame-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'img',[[
        'class',
        'ui-galleria-frame-image'
      ]
    ],[
      [
        8,
        'src',
        4
      ]
      ,
      [
        8,
        'alt',
        0
      ]
      ,
      [
        8,
        'title',
        0
      ]
      ,
      [
        4,
        'width',
        'px'
      ]
      ,
      [
        4,
        'height',
        'px'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = 'ui-galleria-frame';
    const currVal_4:any = ck(v,2,0,(v.context.index == co.activeIndex));
    ck(v,1,0,currVal_3,currVal_4);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.frameWidth;
    const currVal_1:any = co.frameHeight;
    const currVal_2:any = 'opacity 0.75s ease';
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
    const currVal_5:any = v.context.$implicit.source;
    const currVal_6:any = v.context.$implicit.alt;
    const currVal_7:any = v.context.$implicit.title;
    const currVal_8:any = co.frameWidth;
    const currVal_9:any = co.frameHeight;
    ck(v,6,0,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9);
  });
}
function View_Galleria_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),9,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['ui-galleria-filmstrip-wrapper']),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ul',[
      [
        'class',
        'ui-galleria-filmstrip'
      ]
      ,
      [
        'style',
        'transition:left 1s'
      ]

    ]
      ,[[
        4,
        'left',
        'px'
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Galleria_3)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,true);
    ck(v,1,0,currVal_0);
    const currVal_2:any = co.images;
    ck(v,7,0,currVal_2);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.stripLeft;
    ck(v,4,0,currVal_1);
  });
}
function View_Galleria_4(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'ui-galleria-nav-prev fa fa-fw fa-chevron-circle-left'
      ]
      ],[[
        4,
        'bottom',
        'px'
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.clickNavLeft()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
  },(null as any),(null as any)))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.frameHeight / 2);
    ck(v,0,0,currVal_0);
  });
}
function View_Galleria_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',[
      [
        'class',
        'ui-galleria-caption'
      ]
      ,
      [
        'style',
        'display:block'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'h4',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ((co.images[co.activeIndex] == null)? (null as any): co.images[co.activeIndex].title);
    ck(v,3,0,currVal_0);
    const currVal_1:any = ((co.images[co.activeIndex] == null)? (null as any): co.images[co.activeIndex].alt);
    ck(v,5,0,currVal_1);
  });
}
export function View_Galleria_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),21,'div',([] as any[]),[[
        4,
        'width',
        'px'
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-galleria ui-widget ui-widget-content ui-corner-all']),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ul',[[
        'class',
        'ui-galleria-panel-wrapper'
      ]
    ],[
      [
        4,
        'width',
        'px'
      ]
      ,
      [
        4,
        'height',
        'px'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Galleria_1)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Galleria_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Galleria_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'ui-galleria-nav-next fa fa-fw fa-chevron-circle-right'
      ]
      ],[[
        4,
        'bottom',
        'px'
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Galleria = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.clickNavRight()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Galleria_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Galleria = v.component;
    const currVal_1:any = co.styleClass;
    const currVal_2:any = ck(v,3,0,true);
    ck(v,2,0,currVal_1,currVal_2);
    const currVal_3:any = co.style;
    ck(v,4,0,currVal_3);
    const currVal_6:any = co.images;
    ck(v,9,0,currVal_6);
    const currVal_7:any = co.showFilmstrip;
    ck(v,13,0,currVal_7);
    const currVal_8:any = (co.activeIndex !== 0);
    ck(v,16,0,currVal_8);
    const currVal_10:any = (co.showCaption && co.images);
    ck(v,21,0,currVal_10);
  },(ck,v) => {
    var co:import1.Galleria = v.component;
    const currVal_0:any = co.panelWidth;
    ck(v,1,0,currVal_0);
    const currVal_4:any = co.panelWidth;
    const currVal_5:any = co.panelHeight;
    ck(v,6,0,currVal_4,currVal_5);
    const currVal_9:any = (co.frameHeight / 2);
    ck(v,18,0,currVal_9);
  });
}
function View_Galleria_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-galleria',([] as any[]),(null as any),(null as any),(null as any),View_Galleria_0,RenderType_Galleria)),
    import0.ɵprd(256,(null as any),import3.DomHandler,import3.DomHandler,([] as any[])),
    import0.ɵdid(6381568,(null as any),0,import1.Galleria,[
      import0.ElementRef,
      import3.DomHandler
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const GalleriaNgFactory:import0.ComponentFactory<import1.Galleria> = import0.ɵccf('p-galleria',import1.Galleria,View_Galleria_Host_0,{
  style: 'style',
  styleClass: 'styleClass',
  panelWidth: 'panelWidth',
  panelHeight: 'panelHeight',
  frameWidth: 'frameWidth',
  frameHeight: 'frameHeight',
  activeIndex: 'activeIndex',
  showFilmstrip: 'showFilmstrip',
  autoPlay: 'autoPlay',
  transitionInterval: 'transitionInterval',
  showCaption: 'showCaption',
  images: 'images'
}
,{onImageClicked: 'onImageClicked'},([] as any[]));
