// primeng/components/splitbutton/splitbutton.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/splitbutton/splitbutton';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/button/button';
import * as import4 from '@angular/router';
import * as import5 from 'primeng/components/dom/domhandler';
class SplitButtonModuleInjector extends import0.ɵNgModuleInjector<import1.SplitButtonModule> {
  _CommonModule_0:import2.CommonModule;
  _ButtonModule_1:import3.ButtonModule;
  _RouterModule_2:import4.RouterModule;
  _SplitButtonModule_3:import1.SplitButtonModule;
  __NgLocalization_4:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_4():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_4 == null)) { (this.__NgLocalization_4 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_4;
  }
  createInternal():import1.SplitButtonModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ButtonModule_1 = new import3.ButtonModule();
    this._RouterModule_2 = new import4.RouterModule(this.parent.get(import4.ɵa,(null as any)),this.parent.get(import4.Router,(null as any)));
    this._SplitButtonModule_3 = new import1.SplitButtonModule();
    return this._SplitButtonModule_3;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.ButtonModule)) { return this._ButtonModule_1; }
    if ((token === import4.RouterModule)) { return this._RouterModule_2; }
    if ((token === import1.SplitButtonModule)) { return this._SplitButtonModule_3; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_4; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const SplitButtonModuleNgFactory:import0.NgModuleFactory<import1.SplitButtonModule> = new import0.NgModuleFactory<any>(SplitButtonModuleInjector,import1.SplitButtonModule);
const styles_SplitButton:any[] = ([] as any[]);
export const RenderType_SplitButton:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_SplitButton,
      data: {'animation': [{
        name: 'overlayState',
        definitions: [
          {
            type: 0,
            name: 'hidden',
            styles: {
              type: 6,
              styles: {opacity: 0}
            }

          }
          ,
          {
            type: 0,
            name: 'visible',
            styles: {
              type: 6,
              styles: {opacity: 1}
            }

          }
          ,
          {
            type: 1,
            expr: 'visible => hidden',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms ease-in'
            }

          }
          ,
          {
            type: 1,
            expr: 'hidden => visible',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms ease-out'
            }

          }

        ]

      }
  ]}
}
);
function View_SplitButton_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = (<any>(<any>v.parent).parent).context.$implicit.icon;
    const currVal_1:any = 'ui-menuitem-icon fa fa-fw';
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_SplitButton_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),9,'a',[[
        'class',
        'ui-menuitem-link ui-corner-all'
      ]
    ],[
      [
        8,
        'href',
        4
      ]
      ,
      [
        1,
        'target',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.itemClick($event,(<any>v.parent).context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled']),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SplitButton_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                        ']))
  ]
  ,(ck,v) => {
    const currVal_2:any = 'ui-menuitem-link ui-corner-all';
    const currVal_3:any = ck(v,2,0,(<any>v.parent).context.$implicit.disabled);
    ck(v,1,0,currVal_2,currVal_3);
    const currVal_4:any = (<any>v.parent).context.$implicit.icon;
    ck(v,5,0,currVal_4);
  },(ck,v) => {
    const currVal_0:any = ((<any>v.parent).context.$implicit.url || '#');
    const currVal_1:any = (<any>v.parent).context.$implicit.target;
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_5:any = (<any>v.parent).context.$implicit.label;
    ck(v,8,0,currVal_5);
  });
}
function View_SplitButton_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = (<any>(<any>v.parent).parent).context.$implicit.icon;
    const currVal_1:any = 'ui-menuitem-icon fa fa-fw';
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_SplitButton_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'a',[[
        'class',
        'ui-menuitem-link ui-corner-all'
      ]
    ],[
      [
        1,
        'target',
        0
      ]
      ,
      [
        1,
        'target',
        0
      ]
      ,
      [
        8,
        'href',
        4
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,3).onClick($event.button,$event.ctrlKey,$event.metaKey)) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>co.itemClick($event,(<any>v.parent).context.$implicit)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled']),
    import0.ɵdid(335872,(null as any),0,import4.RouterLinkWithHref,[
      import4.Router,
      import4.ActivatedRoute,
      import2.LocationStrategy
    ]
      ,{routerLink: [
        0,
        'routerLink'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SplitButton_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                        ']))
  ]
  ,(ck,v) => {
    const currVal_3:any = 'ui-menuitem-link ui-corner-all';
    const currVal_4:any = ck(v,2,0,(<any>v.parent).context.$implicit.disabled);
    ck(v,1,0,currVal_3,currVal_4);
    const currVal_5:any = (<any>v.parent).context.$implicit.routerLink;
    ck(v,3,0,currVal_5);
    const currVal_6:any = (<any>v.parent).context.$implicit.icon;
    ck(v,6,0,currVal_6);
  },(ck,v) => {
    const currVal_0:any = (<any>v.parent).context.$implicit.target;
    const currVal_1:any = import0.ɵnov(v,3).target;
    const currVal_2:any = import0.ɵnov(v,3).href;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
    const currVal_7:any = (<any>v.parent).context.$implicit.label;
    ck(v,9,0,currVal_7);
  });
}
function View_SplitButton_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'li',[
      [
        'class',
        'ui-menuitem ui-widget ui-corner-all'
      ]
      ,
      [
        'role',
        'menuitem'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SplitButton_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SplitButton_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    const currVal_0:boolean = !v.context.$implicit.routerLink;
    ck(v,3,0,currVal_0);
    const currVal_1:any = v.context.$implicit.routerLink;
    ck(v,6,0,currVal_1);
  },(null as any));
}
export function View_SplitButton_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{buttonViewChild: 0}),
    import0.ɵqud(201326592,2,{overlayViewChild: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,[[
        'container',
        1
      ]
    ],(null as any),23,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-splitbutton ui-buttonset ui-widget',
      'ui-state-disabled'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'defaultbtn',
        1
      ]

    ]
    ,(null as any),3,'button',[
      [
        'cornerStyleClass',
        'ui-corner-left'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,[
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.SplitButton = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onDefaultButtonClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
    ,{
      iconPos: [
        0,
        'iconPos'
      ]
      ,
      cornerStyleClass: [
        1,
        'cornerStyleClass'
      ]
      ,
      label: [
        2,
        'label'
      ]
      ,
      icon: [
        3,
        'icon'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'class',
        'ui-splitbutton-menubutton'
      ]
      ,
      [
        'cornerStyleClass',
        'ui-corner-right'
      ]
      ,
      [
        'icon',
        'fa-caret-down'
      ]
      ,
      [
        'pButton',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,[[
        8,
        'disabled',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.SplitButton = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onDropdownButtonClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(2170880,(null as any),0,import3.Button,[
      import0.ElementRef,
      import5.DomHandler
    ]
    ,{
      cornerStyleClass: [
        0,
        'cornerStyleClass'
      ]
      ,
      icon: [
        1,
        'icon'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'overlay',
        1
      ]

    ]
    ,(null as any),9,'div',([] as any[]),[
      [
        4,
        'display',
        (null as any)
      ]
      ,
      [
        24,
        '@overlayState',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ul',[[
        'class',
        'ui-menu-list ui-helper-reset'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SplitButton_1)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.SplitButton = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,5,0,true,co.disabled);
    ck(v,4,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,6,0,currVal_2);
    const currVal_5:any = co.iconPos;
    const currVal_6:any = 'ui-corner-left';
    const currVal_7:any = co.label;
    const currVal_8:any = co.icon;
    ck(v,10,0,currVal_5,currVal_6,currVal_7,currVal_8);
    const currVal_10:any = 'ui-corner-right';
    const currVal_11:any = 'fa-caret-down';
    ck(v,14,0,currVal_10,currVal_11);
    const currVal_14:any = co.menuStyleClass;
    const currVal_15:any = 'ui-menu ui-menu-dynamic ui-widget ui-widget-content ui-corner-all ui-helper-clearfix ui-shadow';
    ck(v,17,0,currVal_14,currVal_15);
    const currVal_16:any = co.menuStyle;
    ck(v,18,0,currVal_16);
    const currVal_17:any = co.model;
    ck(v,23,0,currVal_17);
  },(ck,v) => {
    var co:import1.SplitButton = v.component;
    const currVal_3:any = co.disabled;
    const currVal_4:any = co.tabindex;
    ck(v,8,0,currVal_3,currVal_4);
    const currVal_9:any = co.disabled;
    ck(v,12,0,currVal_9);
    const currVal_12:any = (co.menuVisible? 'block': 'none');
    const currVal_13:any = (co.menuVisible? 'visible': 'hidden');
    ck(v,16,0,currVal_12,currVal_13);
  });
}
function View_SplitButton_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-splitButton',([] as any[]),(null as any),(null as any),(null as any),View_SplitButton_0,RenderType_SplitButton)),
    import0.ɵprd(256,(null as any),import5.DomHandler,import5.DomHandler,([] as any[])),
    import0.ɵdid(6381568,(null as any),0,import1.SplitButton,[
      import0.ElementRef,
      import5.DomHandler,
      import0.Renderer2,
      import4.Router,
      import0.ChangeDetectorRef
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const SplitButtonNgFactory:import0.ComponentFactory<import1.SplitButton> = import0.ɵccf('p-splitButton',import1.SplitButton,View_SplitButton_Host_0,{
  model: 'model',
  icon: 'icon',
  iconPos: 'iconPos',
  label: 'label',
  style: 'style',
  styleClass: 'styleClass',
  menuStyle: 'menuStyle',
  menuStyleClass: 'menuStyleClass',
  disabled: 'disabled',
  tabindex: 'tabindex',
  appendTo: 'appendTo'
}
,{
  onClick: 'onClick',
  onDropdownClick: 'onDropdownClick'
}
,([] as any[]));
