// primeng/components/editor/editor.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/editor/editor';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from 'primeng/components/dom/domhandler';
import * as import5 from '@angular/forms';
class EditorModuleInjector extends import0.ɵNgModuleInjector<import1.EditorModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _EditorModule_2:import1.EditorModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.EditorModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._EditorModule_2 = new import1.EditorModule();
    return this._EditorModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import1.EditorModule)) { return this._EditorModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const EditorModuleNgFactory:import0.NgModuleFactory<import1.EditorModule> = new import0.NgModuleFactory<any>(EditorModuleInjector,import1.EditorModule);
const styles_Editor:any[] = ([] as any[]);
export const RenderType_Editor:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Editor,
  data: {}
}
);
function View_Editor_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-editor-toolbar ui-widget-header ui-corner-top'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(null as any));
}
function View_Editor_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),76,'div',[[
        'class',
        'ui-editor-toolbar ui-widget-header ui-corner-top'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),25,'span',[[
        'class',
        'ql-formats'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'select',[[
        'class',
        'ql-header'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'option',[[
        'value',
        '1'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Heading'])),
    (l()(),import0.ɵted((null as any),['\n                      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'option',[[
        'value',
        '2'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Subheading'])),
    (l()(),import0.ɵted((null as any),['\n                      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'option',[[
        'selected',
        ''
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Normal'])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'select',[[
        'class',
        'ql-font'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'option',[[
        'selected',
        ''
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Sans Serif'])),
    (l()(),import0.ɵted((null as any),['\n                      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'option',[[
        'value',
        'serif'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Serif'])),
    (l()(),import0.ɵted((null as any),['\n                      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'option',[[
        'value',
        'monospace'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Monospace'])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'span',[[
        'class',
        'ql-formats'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'button',[
      [
        'aria-label',
        'Bold'
      ]
      ,
      [
        'class',
        'ql-bold'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'button',[
      [
        'aria-label',
        'Italic'
      ]
      ,
      [
        'class',
        'ql-italic'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'button',[
      [
        'aria-label',
        'Underline'
      ]
      ,
      [
        'class',
        'ql-underline'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'span',[[
        'class',
        'ql-formats'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'select',[[
        'class',
        'ql-color'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'select',[[
        'class',
        'ql-background'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),16,'span',[[
        'class',
        'ql-formats'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'button',[
      [
        'aria-label',
        'Ordered List'
      ]
      ,
      [
        'class',
        'ql-list'
      ]
      ,
      [
        'value',
        'ordered'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'button',[
      [
        'aria-label',
        'Unordered List'
      ]
      ,
      [
        'class',
        'ql-list'
      ]
      ,
      [
        'value',
        'bullet'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),9,'select',[[
        'class',
        'ql-align'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'option',[[
        'selected',
        ''
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'option',[[
        'value',
        'center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'option',[[
        'value',
        'right'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'option',[[
        'value',
        'justify'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'span',[[
        'class',
        'ql-formats'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'button',[
      [
        'aria-label',
        'Insert Link'
      ]
      ,
      [
        'class',
        'ql-link'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'button',[
      [
        'aria-label',
        'Insert Image'
      ]
      ,
      [
        'class',
        'ql-image'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'button',[
      [
        'aria-label',
        'Insert Code Block'
      ]
      ,
      [
        'class',
        'ql-code-block'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'span',[[
        'class',
        'ql-formats'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'button',[
      [
        'aria-label',
        'Remove Styles'
      ]
      ,
      [
        'class',
        'ql-clean'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(null as any));
}
export function View_Editor_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),11,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Editor_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Editor_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'ui-editor-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Editor = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-widget ui-editor-container ui-corner-all';
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.toolbar;
    ck(v,5,0,currVal_2);
    const currVal_3:boolean = !co.toolbar;
    ck(v,8,0,currVal_3);
    const currVal_4:any = co.style;
    ck(v,11,0,currVal_4);
  },(null as any));
}
function View_Editor_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'p-editor',([] as any[]),(null as any),(null as any),(null as any),View_Editor_0,RenderType_Editor)),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵdid(2121728,(null as any),1,import1.Editor,[
      import0.ElementRef,
      import4.DomHandler
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(167772160,1,{toolbar: 0}),
    import0.ɵprd(2560,(null as any),import5.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.Editor])
  ]
  ,(null as any),(null as any));
}
export const EditorNgFactory:import0.ComponentFactory<import1.Editor> = import0.ɵccf('p-editor',import1.Editor,View_Editor_Host_0,{
  style: 'style',
  styleClass: 'styleClass',
  placeholder: 'placeholder',
  readonly: 'readonly',
  formats: 'formats'
}
,{
  onTextChange: 'onTextChange',
  onSelectionChange: 'onSelectionChange',
  onInit: 'onInit'
}
,['p-header']);
