// primeng/components/contextmenu/contextmenu.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/contextmenu/contextmenu';
import * as import2 from '@angular/common';
import * as import3 from '@angular/router';
import * as import4 from 'primeng/components/dom/domhandler';
class ContextMenuModuleInjector extends import0.ɵNgModuleInjector<import1.ContextMenuModule> {
  _CommonModule_0:import2.CommonModule;
  _RouterModule_1:import3.RouterModule;
  _ContextMenuModule_2:import1.ContextMenuModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.ContextMenuModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._RouterModule_1 = new import3.RouterModule(this.parent.get(import3.ɵa,(null as any)),this.parent.get(import3.Router,(null as any)));
    this._ContextMenuModule_2 = new import1.ContextMenuModule();
    return this._ContextMenuModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.RouterModule)) { return this._RouterModule_1; }
    if ((token === import1.ContextMenuModule)) { return this._ContextMenuModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const ContextMenuModuleNgFactory:import0.NgModuleFactory<import1.ContextMenuModule> = new import0.NgModuleFactory<any>(ContextMenuModuleInjector,import1.ContextMenuModule);
const styles_ContextMenu:any[] = ([] as any[]);
export const RenderType_ContextMenu:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ContextMenu,
  data: {}
}
);
export function View_ContextMenu_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{containerViewChild: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'container',
        1
      ]

    ]
      ,(null as any),7,'div',([] as any[]),[[
        4,
        'display',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-contextMenuSub',[[
        'root',
        'root'
      ]
    ],(null as any),(null as any),(null as any),View_ContextMenuSub_0,RenderType_ContextMenuSub)),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵdid(24576,(null as any),0,import1.ContextMenuSub,[
      import4.DomHandler,
      import1.ContextMenu
    ]
    ,{
      item: [
        0,
        'item'
      ]
      ,
      root: [
        1,
        'root'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.ContextMenu = v.component;
    const currVal_1:any = co.styleClass;
    const currVal_2:any = 'ui-contextmenu ui-menu ui-widget ui-widget-content ui-corner-all ui-helper-clearfix ui-menu-dynamic ui-shadow';
    ck(v,3,0,currVal_1,currVal_2);
    const currVal_3:any = co.style;
    ck(v,4,0,currVal_3);
    const currVal_4:any = co.model;
    const currVal_5:any = 'root';
    ck(v,8,0,currVal_4,currVal_5);
  },(ck,v) => {
    var co:import1.ContextMenu = v.component;
    const currVal_0:any = (co.visible? 'block': 'none');
    ck(v,2,0,currVal_0);
  });
}
function View_ContextMenu_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-contextMenu',([] as any[]),(null as any),(null as any),(null as any),View_ContextMenu_0,RenderType_ContextMenu)),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵdid(2187264,(null as any),0,import1.ContextMenu,[
      import0.ElementRef,
      import4.DomHandler,
      import0.Renderer2
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ContextMenuNgFactory:import0.ComponentFactory<import1.ContextMenu> = import0.ɵccf('p-contextMenu',import1.ContextMenu,View_ContextMenu_Host_0,{
  model: 'model',
  global: 'global',
  target: 'target',
  style: 'style',
  styleClass: 'styleClass',
  appendTo: 'appendTo'
}
,{},([] as any[]));
const styles_ContextMenuSub:any[] = ([] as any[]);
export const RenderType_ContextMenuSub:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ContextMenuSub,
  data: {}
}
);
function View_ContextMenuSub_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'li',[[
        'class',
        'ui-menu-separator ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(null as any),(null as any));
}
function View_ContextMenuSub_5(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'ui-submenu-icon fa fa-fw fa-caret-right'
      ]
  ],(null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),(null as any));
}
function View_ContextMenuSub_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-icon fa fa-fw'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-menuitem-icon fa fa-fw';
    const currVal_1:any = (<any>(<any>(<any>v.parent).parent).parent).context.$implicit.icon;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_ContextMenuSub_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),13,'a',([] as any[]),[
      [
        8,
        'href',
        4
      ]
      ,
      [
        1,
        'target',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.itemClick($event,(<any>(<any>v.parent).parent).context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-menuitem-link ui-corner-all',
      'ui-state-disabled'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ContextMenuSub_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ContextMenuSub_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    const currVal_2:any = (<any>(<any>v.parent).parent).context.$implicit.styleClass;
    const currVal_3:any = ck(v,2,0,true,(<any>(<any>v.parent).parent).context.$implicit.disabled);
    ck(v,1,0,currVal_2,currVal_3);
    const currVal_4:any = (<any>(<any>v.parent).parent).context.$implicit.style;
    ck(v,3,0,currVal_4);
    const currVal_5:any = (<any>(<any>v.parent).parent).context.$implicit.items;
    ck(v,6,0,currVal_5);
    const currVal_6:any = (<any>(<any>v.parent).parent).context.$implicit.icon;
    ck(v,9,0,currVal_6);
  },(ck,v) => {
    const currVal_0:any = ((<any>(<any>v.parent).parent).context.$implicit.url || '#');
    const currVal_1:any = (<any>(<any>v.parent).parent).context.$implicit.target;
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_7:any = (<any>(<any>v.parent).parent).context.$implicit.label;
    ck(v,12,0,currVal_7);
  });
}
function View_ContextMenuSub_8(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'ui-submenu-icon fa fa-fw fa-caret-right'
      ]
  ],(null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),(null as any));
}
function View_ContextMenuSub_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-icon fa fa-fw'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'ui-menuitem-icon fa fa-fw';
    const currVal_1:any = (<any>(<any>(<any>v.parent).parent).parent).context.$implicit.icon;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_ContextMenuSub_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),18,'a',([] as any[]),[
      [
        1,
        'target',
        0
      ]
      ,
      [
        1,
        'target',
        0
      ]
      ,
      [
        8,
        'href',
        4
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,4).onClick($event.button,$event.ctrlKey,$event.metaKey)) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>co.itemClick($event,(<any>(<any>v.parent).parent).context.$implicit)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-menuitem-link ui-corner-all',
      'ui-state-disabled'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
      import0.ɵdid(335872,[[
        2,
        4
      ]
    ],0,import3.RouterLinkWithHref,[
      import3.Router,
      import3.ActivatedRoute,
      import2.LocationStrategy
    ]
      ,{routerLink: [
        0,
        'routerLink'
      ]
    },(null as any)),
    import0.ɵdid(860160,(null as any),2,import3.RouterLinkActive,[
      import3.Router,
      import0.ElementRef,
      import0.Renderer,
      import0.ChangeDetectorRef
    ]
    ,{
      routerLinkActiveOptions: [
        0,
        'routerLinkActiveOptions'
      ]
      ,
      routerLinkActive: [
        1,
        'routerLinkActive'
      ]

    }
    ,(null as any)),
    import0.ɵqud(301989888,1,{links: 1}),
    import0.ɵqud(301989888,2,{linksWithHrefs: 1}),
    import0.ɵpod(['exact']),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ContextMenuSub_8)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ContextMenuSub_9)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    const currVal_3:any = (<any>(<any>v.parent).parent).context.$implicit.styleClass;
    const currVal_4:any = ck(v,2,0,true,(<any>(<any>v.parent).parent).context.$implicit.disabled);
    ck(v,1,0,currVal_3,currVal_4);
    const currVal_5:any = (<any>(<any>v.parent).parent).context.$implicit.style;
    ck(v,3,0,currVal_5);
    const currVal_6:any = (<any>(<any>v.parent).parent).context.$implicit.routerLink;
    ck(v,4,0,currVal_6);
    const currVal_7:any = ((<any>(<any>v.parent).parent).context.$implicit.routerLinkActiveOptions || ck(v,8,0,false));
    const currVal_8:any = 'ui-state-active';
    ck(v,5,0,currVal_7,currVal_8);
    const currVal_9:any = (<any>(<any>v.parent).parent).context.$implicit.items;
    ck(v,11,0,currVal_9);
    const currVal_10:any = (<any>(<any>v.parent).parent).context.$implicit.icon;
    ck(v,14,0,currVal_10);
  },(ck,v) => {
    const currVal_0:any = (<any>(<any>v.parent).parent).context.$implicit.target;
    const currVal_1:any = import0.ɵnov(v,4).target;
    const currVal_2:any = import0.ɵnov(v,4).href;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
    const currVal_11:any = (<any>(<any>v.parent).parent).context.$implicit.label;
    ck(v,17,0,currVal_11);
  });
}
function View_ContextMenuSub_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-contextMenuSub',[[
        'class',
        'ui-submenu'
      ]
    ],(null as any),(null as any),(null as any),View_ContextMenuSub_0,RenderType_ContextMenuSub)),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵdid(24576,(null as any),0,import1.ContextMenuSub,[
      import4.DomHandler,
      import1.ContextMenu
    ]
      ,{item: [
        0,
        'item'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = (<any>(<any>v.parent).parent).context.$implicit;
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_ContextMenuSub_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,[[
        'item',
        1
      ]
      ],(null as any),12,'li',([] as any[]),[[
        4,
        'display',
        (null as any)
      ]
    ],[
      [
        (null as any),
        'mouseenter'
      ]
      ,
      [
        (null as any),
        'mouseleave'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('mouseenter' === en)) {
        const pd_0:any = ((<any>co.onItemMouseEnter($event,import0.ɵnov(v,0),(<any>v.parent).context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mouseleave' === en)) {
        const pd_1:any = ((<any>co.onItemMouseLeave($event,import0.ɵnov(v,0))) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-menuitem ui-widget ui-corner-all',
      'ui-menu-parent',
      'ui-menuitem-active'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ContextMenuSub_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ContextMenuSub_7)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ContextMenuSub_10)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = ck(v,2,0,true,(<any>v.parent).context.$implicit.items,(import0.ɵnov(v,0) == co.activeItem));
    ck(v,1,0,currVal_1);
    const currVal_2:boolean = !(<any>v.parent).context.$implicit.routerLink;
    ck(v,5,0,currVal_2);
    const currVal_3:any = (<any>v.parent).context.$implicit.routerLink;
    ck(v,8,0,currVal_3);
    const currVal_4:any = (<any>v.parent).context.$implicit.items;
    ck(v,11,0,currVal_4);
  },(ck,v) => {
    const currVal_0:any = (((<any>v.parent).context.$implicit.visible === false)? 'none': 'block');
    ck(v,0,0,currVal_0);
  });
}
function View_ContextMenuSub_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ContextMenuSub_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ContextMenuSub_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.separator;
    ck(v,2,0,currVal_0);
    const currVal_1:boolean = !v.context.$implicit.separator;
    ck(v,4,0,currVal_1);
  },(null as any));
}
export function View_ContextMenuSub_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'ul',[[
        'class',
        'ui-menu-list'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.ContextMenuSub = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.listClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-helper-reset',
      'ui-widget-content ui-corner-all ui-helper-clearfix ui-menu-child ui-shadow'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ContextMenuSub_1)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.ContextMenuSub = v.component;
    const currVal_0:any = 'ui-menu-list';
    const currVal_1:any = ck(v,3,0,co.root,!co.root);
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = (co.root? co.item: co.item.items);
    ck(v,6,0,currVal_2);
  },(null as any));
}
function View_ContextMenuSub_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-contextMenuSub',([] as any[]),(null as any),(null as any),(null as any),View_ContextMenuSub_0,RenderType_ContextMenuSub)),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵdid(24576,(null as any),0,import1.ContextMenuSub,[
      import4.DomHandler,
      import1.ContextMenu
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ContextMenuSubNgFactory:import0.ComponentFactory<import1.ContextMenuSub> = import0.ɵccf('p-contextMenuSub',import1.ContextMenuSub,View_ContextMenuSub_Host_0,{
  item: 'item',
  root: 'root'
}
,{},([] as any[]));
