// primeng/components/carousel/carousel.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/carousel/carousel';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from 'primeng/components/dom/domhandler';
class CarouselModuleInjector extends import0.ɵNgModuleInjector<import1.CarouselModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _CarouselModule_2:import1.CarouselModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.CarouselModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._CarouselModule_2 = new import1.CarouselModule();
    return this._CarouselModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import1.CarouselModule)) { return this._CarouselModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const CarouselModuleNgFactory:import0.NgModuleFactory<import1.CarouselModule> = new import0.NgModuleFactory<any>(CarouselModuleInjector,import1.CarouselModule);
const styles_Carousel:any[] = ([] as any[]);
export const RenderType_Carousel:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Carousel,
  data: {}
}
);
function View_Carousel_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-carousel-button ui-carousel-next-button fa fa-arrow-circle-right'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onNextNav()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled'])
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-carousel-button ui-carousel-next-button fa fa-arrow-circle-right';
    const currVal_1:any = ck(v,2,0,((co.page === (co.totalPages - 1)) && !co.circular));
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_Carousel_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-carousel-button ui-carousel-prev-button fa fa-arrow-circle-left'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onPrevNav()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled'])
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-carousel-button ui-carousel-prev-button fa fa-arrow-circle-left';
    const currVal_1:any = ck(v,2,0,((co.page === 0) && !co.circular));
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_Carousel_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'a',[
      [
        'class',
        'ui-carousel-page-link fa fa-circle-o'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.setPageWithLink($event,v.context.index)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['fa-dot-circle-o'])
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-carousel-page-link fa fa-circle-o';
    const currVal_1:any = ck(v,2,0,(co.page === v.context.index));
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_Carousel_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'ui-carousel-page-links'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Carousel_4)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.anchorPageLinks;
    ck(v,3,0,currVal_0);
  },(null as any));
}
function View_Carousel_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'option',([] as any[]),[
      [
        8,
        'value',
        0
      ]
      ,
      [
        8,
        'selected',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = v.context.$implicit;
    const currVal_1:any = (co.value == v.context.$implicit);
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_2:any = (v.context.$implicit + 1);
    ck(v,1,0,currVal_2);
  });
}
function View_Carousel_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'select',[[
        'class',
        'ui-carousel-dropdown ui-widget ui-state-default ui-corner-left'
      ]
      ],[[
        8,
        'value',
        0
      ]
      ],[[
        (null as any),
        'change'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.onDropdownChange($event.target.value)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Carousel_6)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.selectDropdownOptions;
    ck(v,3,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.page;
    ck(v,0,0,currVal_0);
  });
}
function View_Carousel_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'option',([] as any[]),[
      [
        8,
        'value',
        0
      ]
      ,
      [
        8,
        'selected',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = v.context.$implicit;
    const currVal_1:any = (co.value == v.context.$implicit);
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_2:any = (v.context.$implicit + 1);
    ck(v,1,0,currVal_2);
  });
}
function View_Carousel_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'select',[[
        'class',
        'ui-carousel-mobiledropdown ui-widget ui-state-default ui-corner-left'
      ]
    ],[
      [
        8,
        'value',
        0
      ]
      ,
      [
        4,
        'display',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'change'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.onDropdownChange($event.target.value)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Carousel_8)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co.mobileDropdownOptions;
    ck(v,3,0,currVal_2);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.page;
    const currVal_1:any = (co.shrinked? 'block': 'none');
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_Carousel_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_Carousel_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'li',[[
        'class',
        'ui-carousel-item ui-widget-content ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Carousel_10)),
    import0.ɵdid(106496,(null as any),0,import3.TemplateWrapper,[import0.ViewContainerRef],{
      item: [
        0,
        'item'
      ]
      ,
      templateRef: [
        1,
        'templateRef'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = v.context.$implicit;
    const currVal_1:any = co.itemTemplate;
    ck(v,3,0,currVal_0,currVal_1);
  },(null as any));
}
export function View_Carousel_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{containerViewChild: 0}),
    import0.ɵqud(201326592,2,{viewportViewChild: 0}),
    import0.ɵqud(201326592,3,{itemsViewChild: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'container',
        1
      ]

    ]
    ,(null as any),34,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-carousel ui-widget ui-widget-content ui-corner-all']),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),19,'div',[[
        'class',
        'ui-carousel-header ui-widget-header ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-carousel-header-title'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Carousel_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Carousel_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Carousel_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Carousel_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Carousel_7)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'viewport',
        1
      ]

    ]
      ,(null as any),7,'div',[[
        'class',
        'ui-carousel-viewport'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,[
      [
        3,
        0
      ]
      ,
      [
        'items',
        1
      ]

    ]
      ,(null as any),4,'ul',[[
        'class',
        'ui-carousel-items'
      ]
    ],[
      [
        4,
        'left',
        'px'
      ]
      ,
      [
        4,
        'transitionProperty',
        (null as any)
      ]
      ,
      [
        4,
        'transitionDuration',
        (null as any)
      ]
      ,
      [
        4,
        'transitionTimingFunction',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Carousel_9)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Carousel = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,6,0,true);
    ck(v,5,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,7,0,currVal_2);
    const currVal_4:any = (co.value && co.value.length);
    ck(v,15,0,currVal_4);
    const currVal_5:any = (co.value && co.value.length);
    ck(v,18,0,currVal_5);
    const currVal_6:any = co.displayPageLinks;
    ck(v,21,0,currVal_6);
    const currVal_7:any = co.displayPageDropdown;
    ck(v,24,0,currVal_7);
    const currVal_8:any = ((co.responsive && co.value) && co.value.length);
    ck(v,27,0,currVal_8);
    const currVal_13:any = co.value;
    ck(v,35,0,currVal_13);
  },(ck,v) => {
    var co:import1.Carousel = v.component;
    const currVal_3:any = co.headerText;
    ck(v,12,0,currVal_3);
    const currVal_9:any = co.left;
    const currVal_10:any = 'left';
    const currVal_11:any = co.effectDuration;
    const currVal_12:any = co.easing;
    ck(v,32,0,currVal_9,currVal_10,currVal_11,currVal_12);
  });
}
function View_Carousel_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'p-carousel',([] as any[]),(null as any),(null as any),(null as any),View_Carousel_0,RenderType_Carousel)),
    import0.ɵprd(256,(null as any),import4.DomHandler,import4.DomHandler,([] as any[])),
    import0.ɵdid(6905856,(null as any),1,import1.Carousel,[
      import0.ElementRef,
      import4.DomHandler,
      import0.Renderer2,
      import0.ChangeDetectorRef
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(301989888,1,{templates: 1})
  ]
  ,(null as any),(null as any));
}
export const CarouselNgFactory:import0.ComponentFactory<import1.Carousel> = import0.ɵccf('p-carousel',import1.Carousel,View_Carousel_Host_0,{
  numVisible: 'numVisible',
  firstVisible: 'firstVisible',
  headerText: 'headerText',
  circular: 'circular',
  breakpoint: 'breakpoint',
  responsive: 'responsive',
  autoplayInterval: 'autoplayInterval',
  effectDuration: 'effectDuration',
  easing: 'easing',
  pageLinks: 'pageLinks',
  style: 'style',
  styleClass: 'styleClass',
  value: 'value'
}
,{onPage: 'onPage'},([] as any[]));
