// primeng/components/datagrid/datagrid.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/datagrid/datagrid';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/common/shared';
import * as import4 from 'primeng/components/paginator/paginator';
import * as import5 from '../paginator/paginator.ngfactory';
class DataGridModuleInjector extends import0.ɵNgModuleInjector<import1.DataGridModule> {
  _CommonModule_0:import2.CommonModule;
  _SharedModule_1:import3.SharedModule;
  _PaginatorModule_2:import4.PaginatorModule;
  _DataGridModule_3:import1.DataGridModule;
  __NgLocalization_4:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_4():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_4 == null)) { (this.__NgLocalization_4 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_4;
  }
  createInternal():import1.DataGridModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SharedModule_1 = new import3.SharedModule();
    this._PaginatorModule_2 = new import4.PaginatorModule();
    this._DataGridModule_3 = new import1.DataGridModule();
    return this._DataGridModule_3;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.SharedModule)) { return this._SharedModule_1; }
    if ((token === import4.PaginatorModule)) { return this._PaginatorModule_2; }
    if ((token === import1.DataGridModule)) { return this._DataGridModule_3; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_4; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const DataGridModuleNgFactory:import0.NgModuleFactory<import1.DataGridModule> = new import0.NgModuleFactory<any>(DataGridModuleInjector,import1.DataGridModule);
const styles_DataGrid:any[] = ([] as any[]);
export const RenderType_DataGrid:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DataGrid,
  data: {}
}
);
function View_DataGrid_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-datagrid-header ui-widget-header ui-corner-top'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(null as any));
}
function View_DataGrid_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-paginator',[[
        'styleClass',
        'ui-paginator-bottom'
      ]
      ],(null as any),[[
        (null as any),
        'onPageChange'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('onPageChange' === en)) {
        const pd_0:any = ((<any>co.paginate($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import5.View_Paginator_0,import5.RenderType_Paginator)),
    import0.ɵdid(24576,(null as any),0,import4.Paginator,([] as any[]),{
      pageLinkSize: [
        0,
        'pageLinkSize'
      ]
      ,
      styleClass: [
        1,
        'styleClass'
      ]
      ,
      rowsPerPageOptions: [
        2,
        'rowsPerPageOptions'
      ]
      ,
      alwaysShow: [
        3,
        'alwaysShow'
      ]
      ,
      totalRecords: [
        4,
        'totalRecords'
      ]
      ,
      first: [
        5,
        'first'
      ]
      ,
      rows: [
        6,
        'rows'
      ]

    }
    ,{onPageChange: 'onPageChange'})
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.pageLinks;
    const currVal_1:any = 'ui-paginator-bottom';
    const currVal_2:any = co.rowsPerPageOptions;
    const currVal_3:any = co.alwaysShowPaginator;
    const currVal_4:any = co.totalRecords;
    const currVal_5:any = co.first;
    const currVal_6:any = co.rows;
    ck(v,1,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6);
  },(null as any));
}
function View_DataGrid_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_DataGrid_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'ui-widget-content ui-g-12'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.emptyMessage;
    ck(v,1,0,currVal_0);
  });
}
function View_DataGrid_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-paginator',[[
        'styleClass',
        'ui-paginator-bottom'
      ]
      ],(null as any),[[
        (null as any),
        'onPageChange'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('onPageChange' === en)) {
        const pd_0:any = ((<any>co.paginate($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import5.View_Paginator_0,import5.RenderType_Paginator)),
    import0.ɵdid(24576,(null as any),0,import4.Paginator,([] as any[]),{
      pageLinkSize: [
        0,
        'pageLinkSize'
      ]
      ,
      styleClass: [
        1,
        'styleClass'
      ]
      ,
      rowsPerPageOptions: [
        2,
        'rowsPerPageOptions'
      ]
      ,
      alwaysShow: [
        3,
        'alwaysShow'
      ]
      ,
      totalRecords: [
        4,
        'totalRecords'
      ]
      ,
      first: [
        5,
        'first'
      ]
      ,
      rows: [
        6,
        'rows'
      ]

    }
    ,{onPageChange: 'onPageChange'})
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.pageLinks;
    const currVal_1:any = 'ui-paginator-bottom';
    const currVal_2:any = co.rowsPerPageOptions;
    const currVal_3:any = co.alwaysShowPaginator;
    const currVal_4:any = co.totalRecords;
    const currVal_5:any = co.first;
    const currVal_6:any = co.rows;
    ck(v,1,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6);
  },(null as any));
}
function View_DataGrid_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-datagrid-footer ui-widget-header ui-corner-top'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    import0.ɵncd((null as any),1),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(null as any));
}
export function View_DataGrid_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),27,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataGrid_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataGrid_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'div',[[
        'class',
        'ui-datagrid-content ui-widget-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'ui-g'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataGrid_3)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
    ,{
      ngForOf: [
        0,
        'ngForOf'
      ]
      ,
      ngForTrackBy: [
        1,
        'ngForTrackBy'
      ]
      ,
      ngForTemplate: [
        2,
        'ngForTemplate'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataGrid_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataGrid_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataGrid_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.DataGrid = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-datagrid ui-widget';
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_3:any = co.header;
    ck(v,6,0,currVal_3);
    const currVal_4:any = ((co.paginator && (co.paginatorPosition != 'bottom')) || (co.paginatorPosition == 'both'));
    ck(v,9,0,currVal_4);
    const currVal_5:any = co.dataToRender;
    const currVal_6:any = co.trackBy;
    const currVal_7:any = co.itemTemplate;
    ck(v,16,0,currVal_5,currVal_6,currVal_7);
    const currVal_8:any = co.isEmpty();
    ck(v,19,0,currVal_8);
    const currVal_9:any = ((co.paginator && (co.paginatorPosition != 'top')) || (co.paginatorPosition == 'both'));
    ck(v,24,0,currVal_9);
    const currVal_10:any = co.footer;
    ck(v,27,0,currVal_10);
  },(null as any));
}
function View_DataGrid_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'p-dataGrid',([] as any[]),(null as any),(null as any),(null as any),View_DataGrid_0,RenderType_DataGrid)),
    import0.ɵdid(2777088,(null as any),3,import1.DataGrid,[
      import0.ElementRef,
      import0.IterableDiffers
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(167772160,1,{header: 0}),
    import0.ɵqud(167772160,2,{footer: 0}),
    import0.ɵqud(301989888,3,{templates: 1})
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const DataGridNgFactory:import0.ComponentFactory<import1.DataGrid> = import0.ɵccf('p-dataGrid',import1.DataGrid,View_DataGrid_Host_0,{
  paginator: 'paginator',
  rows: 'rows',
  totalRecords: 'totalRecords',
  pageLinks: 'pageLinks',
  rowsPerPageOptions: 'rowsPerPageOptions',
  lazy: 'lazy',
  emptyMessage: 'emptyMessage',
  style: 'style',
  styleClass: 'styleClass',
  paginatorPosition: 'paginatorPosition',
  alwaysShowPaginator: 'alwaysShowPaginator',
  trackBy: 'trackBy',
  immutable: 'immutable',
  value: 'value'
}
,{
  onLazyLoad: 'onLazyLoad',
  onPage: 'onPage'
}
,[
  'p-header',
  'p-footer'
]
);
