// @ng-bootstrap/ng-bootstrap/datepicker/datepicker-navigation-select.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/forms';
import * as import2 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker-navigation-select';
import * as import3 from '@angular/common';
import * as import4 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker-i18n';
import * as import5 from '@ng-bootstrap/ng-bootstrap/datepicker/ngb-calendar';
const styles_NgbDatepickerNavigationSelect:any[] = ['select[_ngcontent-%COMP%] {\n      \n      padding: 0.25rem 0.5rem;\n      font-size: 0.875rem;      \n      line-height: 1.25;\n      \n      height: inherit;\n      width: 50%;\n    }'];
export const RenderType_NgbDatepickerNavigationSelect:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 0,
  styles: styles_NgbDatepickerNavigationSelect,
  data: {}
}
);
function View_NgbDatepickerNavigationSelect_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'option',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(73728,(null as any),0,import1.NgSelectOption,[
      import0.ElementRef,
      import0.Renderer,
      [
        8,
        (null as any)
      ]

    ]
      ,{value: [
        0,
        'value'
      ]
    },(null as any)),
    import0.ɵdid(73728,(null as any),0,import1.ɵq,[
      import0.ElementRef,
      import0.Renderer,
      [
        8,
        (null as any)
      ]

    ]
      ,{value: [
        0,
        'value'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit;
    ck(v,1,0,currVal_0);
    const currVal_1:any = v.context.$implicit;
    ck(v,2,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co.i18n.getMonthShortName(v.context.$implicit);
    ck(v,3,0,currVal_2);
  });
}
function View_NgbDatepickerNavigationSelect_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'option',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(73728,(null as any),0,import1.NgSelectOption,[
      import0.ElementRef,
      import0.Renderer,
      [
        8,
        (null as any)
      ]

    ]
      ,{value: [
        0,
        'value'
      ]
    },(null as any)),
    import0.ɵdid(73728,(null as any),0,import1.ɵq,[
      import0.ElementRef,
      import0.Renderer,
      [
        8,
        (null as any)
      ]

    ]
      ,{value: [
        0,
        'value'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit;
    ck(v,1,0,currVal_0);
    const currVal_1:any = v.context.$implicit;
    ck(v,2,0,currVal_1);
  },(ck,v) => {
    const currVal_2:any = v.context.$implicit;
    ck(v,3,0,currVal_2);
  });
}
export function View_NgbDatepickerNavigationSelect_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'select',[
      [
        'class',
        'custom-select d-inline-block'
      ]
      ,
      [
        'tabindex',
        '-1'
      ]

    ]
    ,[
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        8,
        'value',
        0
      ]

    ]
      ,[[
        (null as any),
        'change'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import2.NgbDatepickerNavigationSelect = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.changeMonth($event.target.value)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbDatepickerNavigationSelect_1)),
    import0.ɵdid(401408,(null as any),0,import3.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'select',[
      [
        'class',
        'custom-select d-inline-block'
      ]
      ,
      [
        'tabindex',
        '-1'
      ]

    ]
    ,[
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        8,
        'value',
        0
      ]

    ]
      ,[[
        (null as any),
        'change'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import2.NgbDatepickerNavigationSelect = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.changeYear($event.target.value)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbDatepickerNavigationSelect_2)),
    import0.ɵdid(401408,(null as any),0,import3.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),[' \n  ']))
  ]
  ,(ck,v) => {
    var co:import2.NgbDatepickerNavigationSelect = v.component;
    const currVal_2:any = co.months;
    ck(v,4,0,currVal_2);
    const currVal_5:any = co.years;
    ck(v,9,0,currVal_5);
  },(ck,v) => {
    var co:import2.NgbDatepickerNavigationSelect = v.component;
    const currVal_0:any = co.disabled;
    const currVal_1:any = ((co.date == null)? (null as any): co.date.month);
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_3:any = co.disabled;
    const currVal_4:any = ((co.date == null)? (null as any): co.date.year);
    ck(v,6,0,currVal_3,currVal_4);
  });
}
function View_NgbDatepickerNavigationSelect_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ngb-datepicker-navigation-select',([] as any[]),(null as any),(null as any),(null as any),View_NgbDatepickerNavigationSelect_0,RenderType_NgbDatepickerNavigationSelect)),
    import0.ɵdid(286720,(null as any),0,import2.NgbDatepickerNavigationSelect,[
      import4.NgbDatepickerI18n,
      import5.NgbCalendar
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const NgbDatepickerNavigationSelectNgFactory:import0.ComponentFactory<import2.NgbDatepickerNavigationSelect> = import0.ɵccf('ngb-datepicker-navigation-select',import2.NgbDatepickerNavigationSelect,View_NgbDatepickerNavigationSelect_Host_0,{
  date: 'date',
  disabled: 'disabled',
  maxDate: 'maxDate',
  minDate: 'minDate'
}
,{select: 'select'},([] as any[]));
