// @ng-bootstrap/ng-bootstrap/rating/rating.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@ng-bootstrap/ng-bootstrap/rating/rating';
import * as import3 from '@ng-bootstrap/ng-bootstrap/rating/rating-config';
import * as import4 from '@angular/forms';
const styles_NgbRating:any[] = ([] as any[]);
export const RenderType_NgbRating:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NgbRating,
  data: {}
}
);
function View_NgbRating_1(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
  ))],(null as any),(ck,v) => {
    const currVal_0:any = ((v.context.fill === 100)? '★': '☆');
    ck(v,0,0,currVal_0);
  });
}
function View_NgbRating_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_NgbRating_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'sr-only'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '(',
      ')'
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'span',([] as any[]),[[
        4,
        'cursor',
        (null as any)
      ]
    ],[
      [
        (null as any),
        'mouseenter'
      ]
      ,
      [
        (null as any),
        'click'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('mouseenter' === en)) {
        const pd_0:any = ((<any>co.enter((v.context.index + 1))) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>co.handleClick((v.context.index + 1))) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbRating_3)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutletContext: [
        0,
        'ngTemplateOutletContext'
      ]
      ,
      ngTemplateOutlet: [
        1,
        'ngTemplateOutlet'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co.contexts[v.context.index];
    const currVal_3:any = (co.starTemplate || import0.ɵnov((<any>v.parent),1));
    ck(v,7,0,currVal_2,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ((v.context.index < co.nextRate)? '*': ' ');
    ck(v,2,0,currVal_0);
    const currVal_1:any = ((co.readonly || co.disabled)? 'default': 'pointer');
    ck(v,4,0,currVal_1);
  });
}
export function View_NgbRating_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵand(0,[[
        't',
        2
      ]
    ],(null as any),0,(null as any),View_NgbRating_1)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbRating_2)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import2.NgbRating = v.component;
    const currVal_0:any = co.contexts;
    ck(v,4,0,currVal_0);
  },(null as any));
}
function View_NgbRating_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'ngb-rating',[
      [
        'aria-valuemin',
        '0'
      ]
      ,
      [
        'class',
        'd-inline-flex'
      ]
      ,
      [
        'role',
        'slider'
      ]
      ,
      [
        'tabindex',
        '0'
      ]

    ]
    ,[
      [
        1,
        'aria-valuemax',
        0
      ]
      ,
      [
        1,
        'aria-valuenow',
        0
      ]
      ,
      [
        1,
        'aria-valuetext',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]
      ,
      [
        (null as any),
        'mouseleave'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('blur' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).handleBlur()) !== false);
        ad = (pd_0 && ad);
      }
      if (('keydown' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,1).handleKeyDown($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('mouseleave' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,1).reset()) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },View_NgbRating_0,RenderType_NgbRating)),
    import0.ɵdid(319488,(null as any),1,import2.NgbRating,[
      import3.NgbRatingConfig,
      import0.ChangeDetectorRef
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(167772160,1,{starTemplate: 0}),
    import0.ɵprd(2560,(null as any),import4.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import2.NgbRating])
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).max;
    const currVal_1:any = import0.ɵnov(v,1).nextRate;
    const currVal_2:any = import0.ɵnov(v,1).ariaValueText();
    const currVal_3:any = (import0.ɵnov(v,1).readonly? true: (null as any));
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
export const NgbRatingNgFactory:import0.ComponentFactory<import2.NgbRating> = import0.ɵccf('ngb-rating',import2.NgbRating,View_NgbRating_Host_0,{
  max: 'max',
  rate: 'rate',
  readonly: 'readonly',
  resettable: 'resettable',
  starTemplate: 'starTemplate'
}
,{
  hover: 'hover',
  leave: 'leave',
  rateChange: 'rateChange'
}
,([] as any[]));
