// primeng/components/selectbutton/selectbutton.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/selectbutton/selectbutton';
import * as import2 from '@angular/common';
import * as import3 from '@angular/forms';
class SelectButtonModuleInjector extends import0.ɵNgModuleInjector<import1.SelectButtonModule> {
  _CommonModule_0:import2.CommonModule;
  _SelectButtonModule_1:import1.SelectButtonModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.SelectButtonModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SelectButtonModule_1 = new import1.SelectButtonModule();
    return this._SelectButtonModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.SelectButtonModule)) { return this._SelectButtonModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const SelectButtonModuleNgFactory:import0.NgModuleFactory<import1.SelectButtonModule> = new import0.NgModuleFactory<any>(SelectButtonModuleInjector,import1.SelectButtonModule);
const styles_SelectButton:any[] = ([] as any[]);
export const RenderType_SelectButton:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_SelectButton,
  data: {}
}
);
function View_SelectButton_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'div',[[
        'class',
        'ui-button ui-widget ui-state-default ui-button-text-only'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onItemClick($event,v.context.$implicit,import0.ɵnov(v,9))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-state-active',
      'ui-state-disabled',
      'ui-state-focus'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-button-text ui-clickable'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-helper-hidden-accessible'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,[[
        'cbox',
        1
      ]
      ],(null as any),0,'input',[[
        'type',
        'checkbox'
      ]
    ],[
      [
        8,
        'checked',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        1,
        'disabled',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('focus' === en)) {
        const pd_0:any = ((<any>co.onFocus($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>co.onBlur($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'ui-button ui-widget ui-state-default ui-button-text-only';
    const currVal_1:any = ck(v,2,0,co.isSelected(v.context.$implicit),co.disabled,(import0.ɵnov(v,9) == co.focusedItem));
    ck(v,1,0,currVal_0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = v.context.$implicit.label;
    ck(v,5,0,currVal_2);
    const currVal_3:any = co.isSelected(v.context.$implicit);
    const currVal_4:any = co.tabindex;
    const currVal_5:any = co.disabled;
    ck(v,9,0,currVal_3,currVal_4,currVal_5);
  });
}
export function View_SelectButton_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SelectButton_1)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.SelectButton = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ('ui-selectbutton ui-buttonset ui-widget ui-corner-all ui-buttonset-' + co.options.length);
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_3:any = co.options;
    ck(v,6,0,currVal_3);
  },(null as any));
}
function View_SelectButton_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-selectButton',([] as any[]),(null as any),(null as any),(null as any),View_SelectButton_0,RenderType_SelectButton)),
    import0.ɵdid(24576,(null as any),0,import1.SelectButton,([] as any[]),(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import3.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.SelectButton])
  ]
  ,(null as any),(null as any));
}
export const SelectButtonNgFactory:import0.ComponentFactory<import1.SelectButton> = import0.ɵccf('p-selectButton',import1.SelectButton,View_SelectButton_Host_0,{
  options: 'options',
  tabindex: 'tabindex',
  multiple: 'multiple',
  style: 'style',
  styleClass: 'styleClass',
  disabled: 'disabled'
}
,{onChange: 'onChange'},([] as any[]));
