// primeng/components/blockui/blockui.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/blockui/blockui';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/dom/domhandler';
class BlockUIModuleInjector extends import0.ɵNgModuleInjector<import1.BlockUIModule> {
  _CommonModule_0:import2.CommonModule;
  _BlockUIModule_1:import1.BlockUIModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.BlockUIModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._BlockUIModule_1 = new import1.BlockUIModule();
    return this._BlockUIModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.BlockUIModule)) { return this._BlockUIModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const BlockUIModuleNgFactory:import0.NgModuleFactory<import1.BlockUIModule> = new import0.NgModuleFactory<any>(BlockUIModuleInjector,import1.BlockUIModule);
const styles_BlockUI:any[] = ([] as any[]);
export const RenderType_BlockUI:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_BlockUI,
  data: {}
}
);
export function View_BlockUI_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{mask: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'mask',
        1
      ]

    ]
      ,(null as any),7,'div',[[
        'class',
        'ui-blockui ui-widget-overlay'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-blockui-document']),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod(['display']),
    (l()(),import0.ɵted((null as any),['\n            '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.BlockUI = v.component;
    const currVal_0:any = 'ui-blockui ui-widget-overlay';
    const currVal_1:any = ck(v,4,0,!co.target);
    ck(v,3,0,currVal_0,currVal_1);
    const currVal_2:any = ck(v,6,0,(co.blocked? 'block': 'none'));
    ck(v,5,0,currVal_2);
  },(null as any));
}
function View_BlockUI_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'p-blockUI',([] as any[]),(null as any),(null as any),(null as any),View_BlockUI_0,RenderType_BlockUI)),
    import0.ɵprd(256,(null as any),import3.DomHandler,import3.DomHandler,([] as any[])),
    import0.ɵdid(2187264,(null as any),0,import1.BlockUI,[
      import0.ElementRef,
      import3.DomHandler
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const BlockUINgFactory:import0.ComponentFactory<import1.BlockUI> = import0.ɵccf('p-blockUI',import1.BlockUI,View_BlockUI_Host_0,{
  target: 'target',
  blocked: 'blocked'
}
,{},['*']);
