// primeng/components/slider/slider.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/slider/slider';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/dom/domhandler';
import * as import4 from '@angular/forms';
class SliderModuleInjector extends import0.ɵNgModuleInjector<import1.SliderModule> {
  _CommonModule_0:import2.CommonModule;
  _SliderModule_1:import1.SliderModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.SliderModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._SliderModule_1 = new import1.SliderModule();
    return this._SliderModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.SliderModule)) { return this._SliderModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const SliderModuleNgFactory:import0.NgModuleFactory<import1.SliderModule> = new import0.NgModuleFactory<any>(SliderModuleInjector,import1.SliderModule);
const styles_Slider:any[] = ([] as any[]);
export const RenderType_Slider:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Slider,
  data: {}
}
);
function View_Slider_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-slider-range ui-widget-header ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'left',
      'width'
    ]
    )
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,(co.handleValues[0] + '%'),((co.handleValues[1] - co.handleValues[0]) + '%'));
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_Slider_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-slider-range ui-widget-header ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'bottom',
      'height'
    ]
    )
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,(co.handleValues[0] + '%'),((co.handleValues[1] - co.handleValues[0]) + '%'));
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_Slider_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-slider-range ui-slider-range-min ui-widget-header ui-corner-all'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod(['height'])
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,(co.handleValue + '%'));
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_Slider_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'ui-slider-handle ui-state-default ui-corner-all ui-clickable'
      ]
      ],[[
        4,
        'transition',
        (null as any)
      ]
    ],[
      [
        (null as any),
        'mousedown'
      ]
      ,
      [
        (null as any),
        'touchstart'
      ]
      ,
      [
        (null as any),
        'touchmove'
      ]
      ,
      [
        (null as any),
        'touchend'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('mousedown' === en)) {
        const pd_0:any = ((<any>co.onMouseDown($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('touchstart' === en)) {
        const pd_1:any = ((<any>co.onTouchStart($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('touchmove' === en)) {
        const pd_2:any = ((<any>co.onTouchMove($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('touchend' === en)) {
        const pd_3:any = ((<any>(co.dragging = false)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'left',
      'bottom'
    ]
    )
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = ck(v,2,0,((co.orientation == 'horizontal')? (co.handleValue + '%'): (null as any)),((co.orientation == 'vertical')? (co.handleValue + '%'): (null as any)));
    ck(v,1,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.dragging? 'none': (null as any));
    ck(v,0,0,currVal_0);
  });
}
function View_Slider_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'span',[[
        'class',
        'ui-slider-handle ui-state-default ui-corner-all ui-clickable'
      ]
      ],[[
        4,
        'transition',
        (null as any)
      ]
    ],[
      [
        (null as any),
        'mousedown'
      ]
      ,
      [
        (null as any),
        'touchstart'
      ]
      ,
      [
        (null as any),
        'touchmove'
      ]
      ,
      [
        (null as any),
        'touchend'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('mousedown' === en)) {
        const pd_0:any = ((<any>co.onMouseDown($event,0)) !== false);
        ad = (pd_0 && ad);
      }
      if (('touchstart' === en)) {
        const pd_1:any = ((<any>co.onTouchStart($event,0)) !== false);
        ad = (pd_1 && ad);
      }
      if (('touchmove' === en)) {
        const pd_2:any = ((<any>co.onTouchMove($event,0)) !== false);
        ad = (pd_2 && ad);
      }
      if (('touchend' === en)) {
        const pd_3:any = ((<any>(co.dragging = false)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-slider-handle-active']),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'left',
      'bottom'
    ]
    )
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = 'ui-slider-handle ui-state-default ui-corner-all ui-clickable';
    const currVal_2:any = ck(v,2,0,(co.handleIndex == 0));
    ck(v,1,0,currVal_1,currVal_2);
    const currVal_3:any = ck(v,4,0,co.rangeStartLeft,co.rangeStartBottom);
    ck(v,3,0,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.dragging? 'none': (null as any));
    ck(v,0,0,currVal_0);
  });
}
function View_Slider_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'span',[[
        'class',
        'ui-slider-handle ui-state-default ui-corner-all ui-clickable'
      ]
      ],[[
        4,
        'transition',
        (null as any)
      ]
    ],[
      [
        (null as any),
        'mousedown'
      ]
      ,
      [
        (null as any),
        'touchstart'
      ]
      ,
      [
        (null as any),
        'touchmove'
      ]
      ,
      [
        (null as any),
        'touchend'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('mousedown' === en)) {
        const pd_0:any = ((<any>co.onMouseDown($event,1)) !== false);
        ad = (pd_0 && ad);
      }
      if (('touchstart' === en)) {
        const pd_1:any = ((<any>co.onTouchStart($event,1)) !== false);
        ad = (pd_1 && ad);
      }
      if (('touchmove' === en)) {
        const pd_2:any = ((<any>co.onTouchMove($event,1)) !== false);
        ad = (pd_2 && ad);
      }
      if (('touchend' === en)) {
        const pd_3:any = ((<any>(co.dragging = false)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-slider-handle-active']),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'left',
      'bottom'
    ]
    )
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = 'ui-slider-handle ui-state-default ui-corner-all ui-clickable';
    const currVal_2:any = ck(v,2,0,(co.handleIndex == 1));
    ck(v,1,0,currVal_1,currVal_2);
    const currVal_3:any = ck(v,4,0,co.rangeEndLeft,co.rangeEndBottom);
    ck(v,3,0,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.dragging? 'none': (null as any));
    ck(v,0,0,currVal_0);
  });
}
export function View_Slider_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),22,'div',([] as any[]),(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Slider = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onBarClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-slider ui-widget ui-widget-content ui-corner-all',
      'ui-state-disabled',
      'ui-slider-horizontal',
      'ui-slider-vertical',
      'ui-slider-animate'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Slider_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Slider_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Slider_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Slider_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Slider_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Slider_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Slider = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,3,0,true,co.disabled,(co.orientation == 'horizontal'),(co.orientation == 'vertical'),co.animate);
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,4,0,currVal_2);
    const currVal_3:any = (co.range && (co.orientation == 'horizontal'));
    ck(v,7,0,currVal_3);
    const currVal_4:any = (co.range && (co.orientation == 'vertical'));
    ck(v,10,0,currVal_4);
    const currVal_5:boolean = (!co.range && (co.orientation == 'vertical'));
    ck(v,13,0,currVal_5);
    const currVal_6:boolean = !co.range;
    ck(v,16,0,currVal_6);
    const currVal_7:any = co.range;
    ck(v,19,0,currVal_7);
    const currVal_8:any = co.range;
    ck(v,22,0,currVal_8);
  },(null as any));
}
function View_Slider_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'p-slider',([] as any[]),(null as any),(null as any),(null as any),View_Slider_0,RenderType_Slider)),
    import0.ɵprd(256,(null as any),import3.DomHandler,import3.DomHandler,([] as any[])),
    import0.ɵdid(2187264,(null as any),0,import1.Slider,[
      import0.ElementRef,
      import3.DomHandler,
      import0.Renderer2
    ]
    ,(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import4.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.Slider])
  ]
  ,(null as any),(null as any));
}
export const SliderNgFactory:import0.ComponentFactory<import1.Slider> = import0.ɵccf('p-slider',import1.Slider,View_Slider_Host_0,{
  animate: 'animate',
  disabled: 'disabled',
  min: 'min',
  max: 'max',
  orientation: 'orientation',
  step: 'step',
  range: 'range',
  style: 'style',
  styleClass: 'styleClass'
}
,{
  onChange: 'onChange',
  onSlideEnd: 'onSlideEnd'
}
,([] as any[]));
