// primeng/components/breadcrumb/breadcrumb.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/breadcrumb/breadcrumb';
import * as import2 from '@angular/common';
import * as import3 from '@angular/router';
class BreadcrumbModuleInjector extends import0.ɵNgModuleInjector<import1.BreadcrumbModule> {
  _CommonModule_0:import2.CommonModule;
  _RouterModule_1:import3.RouterModule;
  _BreadcrumbModule_2:import1.BreadcrumbModule;
  __NgLocalization_3:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_3():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == null)) { (this.__NgLocalization_3 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  createInternal():import1.BreadcrumbModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._RouterModule_1 = new import3.RouterModule(this.parent.get(import3.ɵa,(null as any)),this.parent.get(import3.Router,(null as any)));
    this._BreadcrumbModule_2 = new import1.BreadcrumbModule();
    return this._BreadcrumbModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.RouterModule)) { return this._RouterModule_1; }
    if ((token === import1.BreadcrumbModule)) { return this._BreadcrumbModule_2; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_3; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const BreadcrumbModuleNgFactory:import0.NgModuleFactory<import1.BreadcrumbModule> = new import0.NgModuleFactory<any>(BreadcrumbModuleInjector,import1.BreadcrumbModule);
const styles_Breadcrumb:any[] = ([] as any[]);
export const RenderType_Breadcrumb:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Breadcrumb,
  data: {}
}
);
function View_Breadcrumb_1(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'li',[[
        'class',
        'ui-breadcrumb-home fa fa-home'
      ]
  ],(null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),(null as any));
}
function View_Breadcrumb_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'a',[[
        'class',
        'ui-menuitem-link'
      ]
    ],[
      [
        8,
        'href',
        4
      ]
      ,
      [
        1,
        'target',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.itemClick($event,co.home)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled']),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-home'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = 'ui-menuitem-link';
    const currVal_3:any = ck(v,2,0,co.home.disabled);
    ck(v,1,0,currVal_2,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.home.url || '#');
    const currVal_1:any = co.home.target;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_Breadcrumb_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'a',[[
        'class',
        'ui-menuitem-link'
      ]
    ],[
      [
        1,
        'target',
        0
      ]
      ,
      [
        1,
        'target',
        0
      ]
      ,
      [
        8,
        'href',
        4
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,3).onClick($event.button,$event.ctrlKey,$event.metaKey)) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>co.itemClick($event,co.home)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled']),
      import0.ɵdid(335872,[[
        2,
        4
      ]
    ],0,import3.RouterLinkWithHref,[
      import3.Router,
      import3.ActivatedRoute,
      import2.LocationStrategy
    ]
      ,{routerLink: [
        0,
        'routerLink'
      ]
    },(null as any)),
    import0.ɵdid(860160,(null as any),2,import3.RouterLinkActive,[
      import3.Router,
      import0.ElementRef,
      import0.Renderer,
      import0.ChangeDetectorRef
    ]
    ,{
      routerLinkActiveOptions: [
        0,
        'routerLinkActiveOptions'
      ]
      ,
      routerLinkActive: [
        1,
        'routerLinkActive'
      ]

    }
    ,(null as any)),
    import0.ɵqud(301989888,1,{links: 1}),
    import0.ɵqud(301989888,2,{linksWithHrefs: 1}),
    import0.ɵpod(['exact']),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'fa fa-home'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = 'ui-menuitem-link';
    const currVal_4:any = ck(v,2,0,co.home.disabled);
    ck(v,1,0,currVal_3,currVal_4);
    const currVal_5:any = co.home.routerLink;
    ck(v,3,0,currVal_5);
    const currVal_6:any = (co.home.routerLinkActiveOptions || ck(v,7,0,false));
    const currVal_7:any = 'ui-state-active';
    ck(v,4,0,currVal_6,currVal_7);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.home.target;
    const currVal_1:any = import0.ɵnov(v,3).target;
    const currVal_2:any = import0.ɵnov(v,3).href;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
function View_Breadcrumb_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'li',[[
        'class',
        'ui-breadcrumb-home'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Breadcrumb_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Breadcrumb_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:boolean = !co.home.routerLink;
    ck(v,3,0,currVal_0);
    const currVal_1:any = co.home.routerLink;
    ck(v,6,0,currVal_1);
  },(null as any));
}
function View_Breadcrumb_5(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'li',[[
        'class',
        'ui-breadcrumb-chevron fa fa-chevron-right'
      ]
  ],(null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),(null as any));
}
function View_Breadcrumb_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'a',[[
        'class',
        'ui-menuitem-link'
      ]
    ],[
      [
        8,
        'href',
        4
      ]
      ,
      [
        1,
        'target',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.itemClick($event,(<any>v.parent).context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled']),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                        ']))
  ]
  ,(ck,v) => {
    const currVal_2:any = 'ui-menuitem-link';
    const currVal_3:any = ck(v,2,0,(<any>v.parent).context.$implicit.disabled);
    ck(v,1,0,currVal_2,currVal_3);
  },(ck,v) => {
    const currVal_0:any = ((<any>v.parent).context.$implicit.url || '#');
    const currVal_1:any = (<any>v.parent).context.$implicit.target;
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_4:any = (<any>v.parent).context.$implicit.label;
    ck(v,5,0,currVal_4);
  });
}
function View_Breadcrumb_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'a',[[
        'class',
        'ui-menuitem-link'
      ]
    ],[
      [
        1,
        'target',
        0
      ]
      ,
      [
        1,
        'target',
        0
      ]
      ,
      [
        8,
        'href',
        4
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,3).onClick($event.button,$event.ctrlKey,$event.metaKey)) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>co.itemClick($event,(<any>v.parent).context.$implicit)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['ui-state-disabled']),
      import0.ɵdid(335872,[[
        4,
        4
      ]
    ],0,import3.RouterLinkWithHref,[
      import3.Router,
      import3.ActivatedRoute,
      import2.LocationStrategy
    ]
      ,{routerLink: [
        0,
        'routerLink'
      ]
    },(null as any)),
    import0.ɵdid(860160,(null as any),2,import3.RouterLinkActive,[
      import3.Router,
      import0.ElementRef,
      import0.Renderer,
      import0.ChangeDetectorRef
    ]
    ,{
      routerLinkActiveOptions: [
        0,
        'routerLinkActiveOptions'
      ]
      ,
      routerLinkActive: [
        1,
        'routerLinkActive'
      ]

    }
    ,(null as any)),
    import0.ɵqud(301989888,3,{links: 1}),
    import0.ɵqud(301989888,4,{linksWithHrefs: 1}),
    import0.ɵpod(['exact']),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-menuitem-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                        ']))
  ]
  ,(ck,v) => {
    const currVal_3:any = 'ui-menuitem-link';
    const currVal_4:any = ck(v,2,0,(<any>v.parent).context.$implicit.disabled);
    ck(v,1,0,currVal_3,currVal_4);
    const currVal_5:any = (<any>v.parent).context.$implicit.routerLink;
    ck(v,3,0,currVal_5);
    const currVal_6:any = ((<any>v.parent).context.$implicit.routerLinkActiveOptions || ck(v,7,0,false));
    const currVal_7:any = 'ui-state-active';
    ck(v,4,0,currVal_6,currVal_7);
  },(ck,v) => {
    const currVal_0:any = (<any>v.parent).context.$implicit.target;
    const currVal_1:any = import0.ɵnov(v,3).target;
    const currVal_2:any = import0.ɵnov(v,3).href;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
    const currVal_8:any = (<any>v.parent).context.$implicit.label;
    ck(v,10,0,currVal_8);
  });
}
function View_Breadcrumb_9(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'li',[[
        'class',
        'ui-breadcrumb-chevron fa fa-chevron-right'
      ]
  ],(null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),(null as any));
}
function View_Breadcrumb_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'li',[[
        'role',
        'menuitem'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Breadcrumb_7)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Breadcrumb_8)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Breadcrumb_9)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    const currVal_0:boolean = !v.context.$implicit.routerLink;
    ck(v,4,0,currVal_0);
    const currVal_1:any = v.context.$implicit.routerLink;
    ck(v,7,0,currVal_1);
    const currVal_2:boolean = !v.context.last;
    ck(v,11,0,currVal_2);
  },(null as any));
}
export function View_Breadcrumb_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),18,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),13,'ul',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Breadcrumb_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Breadcrumb_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Breadcrumb_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_Breadcrumb_6)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.Breadcrumb = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = 'ui-breadcrumb ui-widget ui-widget-header ui-helper-clearfix ui-corner-all';
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_3:boolean = !co.home;
    ck(v,8,0,currVal_3);
    const currVal_4:any = co.home;
    ck(v,11,0,currVal_4);
    const currVal_5:any = co.model;
    ck(v,14,0,currVal_5);
    const currVal_6:any = co.model;
    ck(v,17,0,currVal_6);
  },(null as any));
}
function View_Breadcrumb_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'p-breadcrumb',([] as any[]),(null as any),(null as any),(null as any),View_Breadcrumb_0,RenderType_Breadcrumb)),
    import0.ɵdid(24576,(null as any),0,import1.Breadcrumb,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const BreadcrumbNgFactory:import0.ComponentFactory<import1.Breadcrumb> = import0.ɵccf('p-breadcrumb',import1.Breadcrumb,View_Breadcrumb_Host_0,{
  model: 'model',
  style: 'style',
  styleClass: 'styleClass',
  home: 'home'
}
,{},([] as any[]));
