// @ng-bootstrap/ng-bootstrap/timepicker/timepicker.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@ng-bootstrap/ng-bootstrap/timepicker/timepicker';
import * as import3 from '@ng-bootstrap/ng-bootstrap/timepicker/timepicker-config';
import * as import4 from '@angular/forms';
const styles_NgbTimepicker:any[] = ['.ngb-tp[_ngcontent-%COMP%] {\n      display: flex;\n      align-items: center;\n    }\n\n    .ngb-tp-hour[_ngcontent-%COMP%], .ngb-tp-minute[_ngcontent-%COMP%], .ngb-tp-second[_ngcontent-%COMP%], .ngb-tp-meridian[_ngcontent-%COMP%] {\n      display: flex;\n      flex-direction: column;\n      align-items: center;\n      justify-content: space-around;\n    }\n\n    .ngb-tp-spacer[_ngcontent-%COMP%] {\n      width: 1em;\n      text-align: center;\n    }\n\n    .chevron[_ngcontent-%COMP%]::before {\n      border-style: solid;\n      border-width: 0.29em 0.29em 0 0;\n      content: \'\';\n      display: inline-block;\n      height: 0.69em;\n      left: 0.05em;\n      position: relative;\n      top: 0.15em;\n      transform: rotate(-45deg);\n      -webkit-transform: rotate(-45deg);\n      -ms-transform: rotate(-45deg);\n      vertical-align: middle;\n      width: 0.71em;\n    }\n\n    .chevron.bottom[_ngcontent-%COMP%]:before {\n      top: -.3em;\n      -webkit-transform: rotate(135deg);\n      -ms-transform: rotate(135deg);\n      transform: rotate(135deg);\n    }\n\n    .btn-link[_ngcontent-%COMP%] {\n      outline: 0;\n    }\n\n    .btn-link.disabled[_ngcontent-%COMP%] {\n      cursor: not-allowed;\n      opacity: .65;\n    }\n\n    input[_ngcontent-%COMP%] {\n      text-align: center;\n      display: inline-block;\n      width: auto;\n    }'];
export const RenderType_NgbTimepicker:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 0,
  styles: styles_NgbTimepicker,
  data: {}
}
);
function View_NgbTimepicker_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'button',[
      [
        'class',
        'btn-link'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,[
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        2,
        'disabled',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.changeHour(co.hourStep)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'chevron'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'sr-only'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Increment hours'])),
    (l()(),import0.ɵted((null as any),['\n          ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = 'btn-link';
    const currVal_3:any = co.setButtonSize();
    ck(v,1,0,currVal_2,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.disabled;
    const currVal_1:any = co.disabled;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_NgbTimepicker_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'button',[
      [
        'class',
        'btn-link'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,[
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        2,
        'disabled',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.changeHour((0 - co.hourStep))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'chevron bottom'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'sr-only'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Decrement hours'])),
    (l()(),import0.ɵted((null as any),['\n          ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = 'btn-link';
    const currVal_3:any = co.setButtonSize();
    ck(v,1,0,currVal_2,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.disabled;
    const currVal_1:any = co.disabled;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_NgbTimepicker_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'button',[
      [
        'class',
        'btn-link'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,[
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        2,
        'disabled',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.changeMinute(co.minuteStep)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'chevron'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'sr-only'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Increment minutes'])),
    (l()(),import0.ɵted((null as any),['\n          ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = 'btn-link';
    const currVal_3:any = co.setButtonSize();
    ck(v,1,0,currVal_2,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.disabled;
    const currVal_1:any = co.disabled;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_NgbTimepicker_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'button',[
      [
        'class',
        'btn-link'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,[
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        2,
        'disabled',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.changeMinute((0 - co.minuteStep))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'chevron bottom'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'sr-only'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Decrement minutes'])),
    (l()(),import0.ɵted((null as any),['\n          ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = 'btn-link';
    const currVal_3:any = co.setButtonSize();
    ck(v,1,0,currVal_2,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.disabled;
    const currVal_1:any = co.disabled;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_NgbTimepicker_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'ngb-tp-spacer'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[':']))
  ]
  ,(null as any),(null as any));
}
function View_NgbTimepicker_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'button',[
      [
        'class',
        'btn-link'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,[
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        2,
        'disabled',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.changeSecond(co.secondStep)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'chevron'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'sr-only'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Increment seconds'])),
    (l()(),import0.ɵted((null as any),['\n          ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = 'btn-link';
    const currVal_3:any = co.setButtonSize();
    ck(v,1,0,currVal_2,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.disabled;
    const currVal_1:any = co.disabled;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_NgbTimepicker_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'button',[
      [
        'class',
        'btn-link'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,[
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        2,
        'disabled',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.changeSecond((0 - co.secondStep))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'chevron bottom'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'sr-only'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Decrement seconds'])),
    (l()(),import0.ɵted((null as any),['\n          ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = 'btn-link';
    const currVal_3:any = co.setButtonSize();
    ck(v,1,0,currVal_2,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.disabled;
    const currVal_1:any = co.disabled;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_NgbTimepicker_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'div',[[
        'class',
        'ngb-tp-second'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTimepicker_7)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'input',[
      [
        'aria-label',
        'Seconds'
      ]
      ,
      [
        'class',
        'form-control'
      ]
      ,
      [
        'maxlength',
        '2'
      ]
      ,
      [
        'placeholder',
        'SS'
      ]
      ,
      [
        'size',
        '2'
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
    ,[
      [
        8,
        'value',
        0
      ]
      ,
      [
        8,
        'readOnly',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]

    ]
      ,[[
        (null as any),
        'change'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.updateSecond($event.target.value)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTimepicker_8)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.spinners;
    ck(v,3,0,currVal_0);
    const currVal_4:any = 'form-control';
    const currVal_5:any = co.setFormControlSize();
    ck(v,6,0,currVal_4,currVal_5);
    const currVal_6:any = co.spinners;
    ck(v,9,0,currVal_6);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.formatMinSec(((co.model == null)? (null as any): co.model.second));
    const currVal_2:any = co.readonlyInputs;
    const currVal_3:any = co.disabled;
    ck(v,5,0,currVal_1,currVal_2,currVal_3);
  });
}
function View_NgbTimepicker_9(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'ngb-tp-spacer'
      ]
  ],(null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),(null as any));
}
function View_NgbTimepicker_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',[[
        'class',
        'ngb-tp-meridian'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'class',
        'btn btn-outline-primary'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,[
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        2,
        'disabled',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggleMeridian()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = 'btn btn-outline-primary';
    const currVal_3:any = co.setButtonSize();
    ck(v,3,0,currVal_2,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.disabled;
    const currVal_1:any = co.disabled;
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_4:any = ((((co.model == null)? (null as any): co.model.hour) >= 12)? 'PM': 'AM');
    ck(v,4,0,currVal_4);
  });
}
export function View_NgbTimepicker_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),43,'fieldset',([] as any[]),[
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        2,
        'disabled',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),40,'div',[[
        'class',
        'ngb-tp'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'div',[[
        'class',
        'ngb-tp-hour'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTimepicker_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'input',[
      [
        'aria-label',
        'Hours'
      ]
      ,
      [
        'class',
        'form-control'
      ]
      ,
      [
        'maxlength',
        '2'
      ]
      ,
      [
        'placeholder',
        'HH'
      ]
      ,
      [
        'size',
        '2'
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
    ,[
      [
        8,
        'value',
        0
      ]
      ,
      [
        8,
        'readOnly',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]

    ]
      ,[[
        (null as any),
        'change'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import2.NgbTimepicker = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.updateHour($event.target.value)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTimepicker_2)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'ngb-tp-spacer'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[':'])),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'div',[[
        'class',
        'ngb-tp-minute'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTimepicker_3)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'input',[
      [
        'aria-label',
        'Minutes'
      ]
      ,
      [
        'class',
        'form-control'
      ]
      ,
      [
        'maxlength',
        '2'
      ]
      ,
      [
        'placeholder',
        'MM'
      ]
      ,
      [
        'size',
        '2'
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
    ,[
      [
        8,
        'value',
        0
      ]
      ,
      [
        8,
        'readOnly',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]

    ]
      ,[[
        (null as any),
        'change'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import2.NgbTimepicker = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.updateMinute($event.target.value)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTimepicker_4)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTimepicker_5)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTimepicker_6)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTimepicker_9)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbTimepicker_10)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import2.NgbTimepicker = v.component;
    const currVal_2:any = co.spinners;
    ck(v,8,0,currVal_2);
    const currVal_6:any = 'form-control';
    const currVal_7:any = co.setFormControlSize();
    ck(v,11,0,currVal_6,currVal_7);
    const currVal_8:any = co.spinners;
    ck(v,14,0,currVal_8);
    const currVal_9:any = co.spinners;
    ck(v,23,0,currVal_9);
    const currVal_13:any = 'form-control';
    const currVal_14:any = co.setFormControlSize();
    ck(v,26,0,currVal_13,currVal_14);
    const currVal_15:any = co.spinners;
    ck(v,29,0,currVal_15);
    const currVal_16:any = co.seconds;
    ck(v,33,0,currVal_16);
    const currVal_17:any = co.seconds;
    ck(v,36,0,currVal_17);
    const currVal_18:any = co.meridian;
    ck(v,39,0,currVal_18);
    const currVal_19:any = co.meridian;
    ck(v,42,0,currVal_19);
  },(ck,v) => {
    var co:import2.NgbTimepicker = v.component;
    const currVal_0:any = co.disabled;
    const currVal_1:any = co.disabled;
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_3:any = co.formatHour(((co.model == null)? (null as any): co.model.hour));
    const currVal_4:any = co.readonlyInputs;
    const currVal_5:any = co.disabled;
    ck(v,10,0,currVal_3,currVal_4,currVal_5);
    const currVal_10:any = co.formatMinSec(((co.model == null)? (null as any): co.model.minute));
    const currVal_11:any = co.readonlyInputs;
    const currVal_12:any = co.disabled;
    ck(v,25,0,currVal_10,currVal_11,currVal_12);
  });
}
function View_NgbTimepicker_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'ngb-timepicker',([] as any[]),(null as any),(null as any),(null as any),View_NgbTimepicker_0,RenderType_NgbTimepicker)),
    import0.ɵdid(286720,(null as any),0,import2.NgbTimepicker,[import3.NgbTimepickerConfig],(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import4.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import2.NgbTimepicker])
  ]
  ,(null as any),(null as any));
}
export const NgbTimepickerNgFactory:import0.ComponentFactory<import2.NgbTimepicker> = import0.ɵccf('ngb-timepicker',import2.NgbTimepicker,View_NgbTimepicker_Host_0,{
  meridian: 'meridian',
  spinners: 'spinners',
  seconds: 'seconds',
  hourStep: 'hourStep',
  minuteStep: 'minuteStep',
  secondStep: 'secondStep',
  readonlyInputs: 'readonlyInputs',
  size: 'size'
}
,{},([] as any[]));
