// @swimlane/ngx-datatable/release/components/header/header.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './header-cell.component.ngfactory';
import * as import2 from '@swimlane/ngx-datatable/release/directives/draggable.directive';
import * as import3 from '@swimlane/ngx-datatable/release/directives/resizeable.directive';
import * as import4 from '@swimlane/ngx-datatable/release/directives/long-press.directive';
import * as import5 from '@swimlane/ngx-datatable/release/components/header/header-cell.component';
import * as import6 from '@angular/common';
import * as import7 from '@swimlane/ngx-datatable/release/components/header/header.component';
import * as import8 from '@swimlane/ngx-datatable/release/directives/orderable.directive';
import * as import9 from '@angular/platform-browser';
const styles_DataTableHeaderComponent:any[] = ([] as any[]);
export const RenderType_DataTableHeaderComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DataTableHeaderComponent,
  data: {}
}
);
function View_DataTableHeaderComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'datatable-header-cell',[
      [
        'class',
        'datatable-header-cell'
      ]
      ,
      [
        'draggable',
        ''
      ]
      ,
      [
        'long-press',
        ''
      ]
      ,
      [
        'resizeable',
        ''
      ]

    ]
    ,[
      [
        2,
        'resizeable',
        (null as any)
      ]
      ,
      [
        2,
        'press',
        (null as any)
      ]
      ,
      [
        2,
        'longpress',
        (null as any)
      ]
      ,
      [
        4,
        'height',
        'px'
      ]
      ,
      [
        8,
        'className',
        0
      ]
      ,
      [
        1,
        'title',
        0
      ]
      ,
      [
        4,
        'minWidth',
        'px'
      ]
      ,
      [
        4,
        'maxWidth',
        'px'
      ]
      ,
      [
        4,
        'width',
        'px'
      ]

    ]
    ,[
      [
        (null as any),
        'resize'
      ]
      ,
      [
        (null as any),
        'longPressStart'
      ]
      ,
      [
        (null as any),
        'longPressEnd'
      ]
      ,
      [
        (null as any),
        'sort'
      ]
      ,
      [
        (null as any),
        'select'
      ]
      ,
      [
        (null as any),
        'columnContextmenu'
      ]
      ,
      [
        (null as any),
        'mousedown'
      ]
      ,
      [
        (null as any),
        'contextmenu'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('mousedown' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,2).onMousedown($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mousedown' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,3).onMouseDown($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('contextmenu' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,4).onContextmenu($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('resize' === en)) {
        const pd_3:any = ((<any>co.onColumnResized($event,v.context.$implicit)) !== false);
        ad = (pd_3 && ad);
      }
      if (('longPressStart' === en)) {
        const pd_4:any = ((<any>co.onLongPressStart($event)) !== false);
        ad = (pd_4 && ad);
      }
      if (('longPressEnd' === en)) {
        const pd_5:any = ((<any>co.onLongPressEnd($event)) !== false);
        ad = (pd_5 && ad);
      }
      if (('sort' === en)) {
        const pd_6:any = ((<any>co.onSort($event)) !== false);
        ad = (pd_6 && ad);
      }
      if (('select' === en)) {
        const pd_7:any = ((<any>co.select.emit($event)) !== false);
        ad = (pd_7 && ad);
      }
      if (('columnContextmenu' === en)) {
        const pd_8:any = ((<any>co.columnContextmenu.emit($event)) !== false);
        ad = (pd_8 && ad);
      }
      return ad;
    },import1.View_DataTableHeaderCellComponent_0,import1.RenderType_DataTableHeaderCellComponent)),
      import0.ɵdid(335872,[[
        1,
        4
      ]
    ],0,import2.DraggableDirective,[import0.ElementRef],{
      dragEventTarget: [
        0,
        'dragEventTarget'
      ]
      ,
      dragModel: [
        1,
        'dragModel'
      ]
      ,
      dragX: [
        2,
        'dragX'
      ]
      ,
      dragY: [
        3,
        'dragY'
      ]

    }
    ,(null as any)),
      import0.ɵdid(2170880,(null as any),0,import3.ResizeableDirective,[import0.ElementRef],{resizeEnabled: [
        0,
        'resizeEnabled'
      ]
    },{resize: 'resize'}),
    import0.ɵdid(73728,(null as any),0,import4.LongPressDirective,([] as any[]),{
      pressEnabled: [
        0,
        'pressEnabled'
      ]
      ,
      pressModel: [
        1,
        'pressModel'
      ]

    }
    ,{
      longPressStart: 'longPressStart',
      longPressEnd: 'longPressEnd'
    }
    ),
    import0.ɵdid(24576,(null as any),0,import5.DataTableHeaderCellComponent,([] as any[]),{
      sortType: [
        0,
        'sortType'
      ]
      ,
      column: [
        1,
        'column'
      ]
      ,
      sortAscendingIcon: [
        2,
        'sortAscendingIcon'
      ]
      ,
      sortDescendingIcon: [
        3,
        'sortDescendingIcon'
      ]
      ,
      allRowsSelected: [
        4,
        'allRowsSelected'
      ]
      ,
      selectionType: [
        5,
        'selectionType'
      ]
      ,
      headerHeight: [
        6,
        'headerHeight'
      ]
      ,
      sorts: [
        7,
        'sorts'
      ]

    }
    ,{
      sort: 'sort',
      select: 'select',
      columnContextmenu: 'columnContextmenu'
    }
    ),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_9:any = co.dragEventTarget;
    const currVal_10:any = v.context.$implicit;
    const currVal_11:any = ((co.reorderable && v.context.$implicit.draggable) && v.context.$implicit.dragging);
    const currVal_12:any = false;
    ck(v,1,0,currVal_9,currVal_10,currVal_11,currVal_12);
    const currVal_13:any = v.context.$implicit.resizeable;
    ck(v,2,0,currVal_13);
    const currVal_14:any = (co.reorderable && v.context.$implicit.draggable);
    const currVal_15:any = v.context.$implicit;
    ck(v,3,0,currVal_14,currVal_15);
    const currVal_16:any = co.sortType;
    const currVal_17:any = v.context.$implicit;
    const currVal_18:any = co.sortAscendingIcon;
    const currVal_19:any = co.sortDescendingIcon;
    const currVal_20:any = co.allRowsSelected;
    const currVal_21:any = co.selectionType;
    const currVal_22:any = co.headerHeight;
    const currVal_23:any = co.sorts;
    ck(v,4,0,currVal_16,currVal_17,currVal_18,currVal_19,currVal_20,currVal_21,currVal_22,currVal_23);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2).resizeEnabled;
    const currVal_1:any = import0.ɵnov(v,3).press;
    const currVal_2:any = import0.ɵnov(v,3).isLongPress;
    const currVal_3:any = import0.ɵnov(v,4).headerHeight;
    const currVal_4:any = import0.ɵnov(v,4).columnCssClasses;
    const currVal_5:any = import0.ɵnov(v,4).name;
    const currVal_6:any = import0.ɵnov(v,4).minWidth;
    const currVal_7:any = import0.ɵnov(v,4).maxWidth;
    const currVal_8:any = import0.ɵnov(v,4).width;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8);
  });
}
function View_DataTableHeaderComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import6.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataTableHeaderComponent_2)),
    import0.ɵdid(401408,(null as any),0,import6.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
    ,{
      ngForOf: [
        0,
        'ngForOf'
      ]
      ,
      ngForTrackBy: [
        1,
        'ngForTrackBy'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.stylesByGroup(v.context.$implicit.type);
    ck(v,1,0,currVal_1);
    const currVal_2:any = v.context.$implicit.columns;
    const currVal_3:any = co.columnTrackingFn;
    ck(v,4,0,currVal_2,currVal_3);
  },(ck,v) => {
    const currVal_0:any = ('datatable-row-' + v.context.$implicit.type);
    ck(v,0,0,currVal_0);
  });
}
export function View_DataTableHeaderComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',[
      [
        'class',
        'datatable-header-inner'
      ]
      ,
      [
        'orderable',
        ''
      ]

    ]
      ,[[
        4,
        'width',
        'px'
      ]
      ],[[
        (null as any),
        'reorder'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import7.DataTableHeaderComponent = v.component;
      if (('reorder' === en)) {
        const pd_0:any = ((<any>co.onColumnReordered($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(598016,(null as any),1,import8.OrderableDirective,[
      import0.KeyValueDiffers,
      import9.DOCUMENT
    ]
    ,(null as any),{reorder: 'reorder'}),
    import0.ɵqud(301989888,1,{draggables: 1}),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataTableHeaderComponent_1)),
    import0.ɵdid(401408,(null as any),0,import6.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
    ,{
      ngForOf: [
        0,
        'ngForOf'
      ]
      ,
      ngForTrackBy: [
        1,
        'ngForTrackBy'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import7.DataTableHeaderComponent = v.component;
    const currVal_1:any = co.columnsByPin;
    const currVal_2:any = co.trackByGroups;
    ck(v,6,0,currVal_1,currVal_2);
  },(ck,v) => {
    var co:import7.DataTableHeaderComponent = v.component;
    const currVal_0:any = co.columnGroupWidths.total;
    ck(v,1,0,currVal_0);
  });
}
function View_DataTableHeaderComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'datatable-header',[[
        'class',
        'datatable-header'
      ]
    ],[
      [
        4,
        'height',
        (null as any)
      ]
      ,
      [
        4,
        'width',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_DataTableHeaderComponent_0,RenderType_DataTableHeaderComponent)),
    import0.ɵdid(24576,(null as any),0,import7.DataTableHeaderComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).headerHeight;
    const currVal_1:any = import0.ɵnov(v,1).headerWidth;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const DataTableHeaderComponentNgFactory:import0.ComponentFactory<import7.DataTableHeaderComponent> = import0.ɵccf('datatable-header',import7.DataTableHeaderComponent,View_DataTableHeaderComponent_Host_0,{
  sortAscendingIcon: 'sortAscendingIcon',
  sortDescendingIcon: 'sortDescendingIcon',
  scrollbarH: 'scrollbarH',
  innerWidth: 'innerWidth',
  offsetX: 'offsetX',
  sorts: 'sorts',
  sortType: 'sortType',
  allRowsSelected: 'allRowsSelected',
  selectionType: 'selectionType',
  reorderable: 'reorderable',
  headerHeight: 'headerHeight',
  columns: 'columns'
}
,{
  sort: 'sort',
  reorder: 'reorder',
  resize: 'resize',
  select: 'select',
  columnContextmenu: 'columnContextmenu'
}
,([] as any[]));
