// @swimlane/ngx-datatable/release/components/header/header-cell.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@swimlane/ngx-datatable/release/components/header/header-cell.component';
const styles_DataTableHeaderCellComponent:any[] = ([] as any[]);
export const RenderType_DataTableHeaderCellComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DataTableHeaderCellComponent,
  data: {}
}
);
function View_DataTableHeaderCellComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'label',[[
        'class',
        'datatable-checkbox'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'input',[[
        'type',
        'checkbox'
      ]
      ],[[
        8,
        'checked',
        0
      ]
      ],[[
        (null as any),
        'change'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.select.emit(!co.allRowsSelected)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.allRowsSelected;
    ck(v,2,0,currVal_0);
  });
}
function View_DataTableHeaderCellComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'span',[[
        'class',
        'datatable-header-cell-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'datatable-header-cell-label draggable'
      ]
      ],[[
        8,
        'innerHTML',
        1
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onSort()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.name;
    ck(v,2,0,currVal_0);
  });
}
function View_DataTableHeaderCellComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n      ']))],(null as any),(null as any));
}
function View_DataTableHeaderCellComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_DataTableHeaderCellComponent_4)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'column',
      'sortDir',
      'sortFn',
      'allRowsSelected',
      'selectFn'
    ]
    ),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.column.headerTemplate;
    const currVal_1:any = ck(v,2,0,co.column,co.sortDir,co.sortFn,co.allRowsSelected,co.selectFn);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
export function View_DataTableHeaderCellComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),13,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataTableHeaderCellComponent_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataTableHeaderCellComponent_2)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DataTableHeaderCellComponent_3)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),[[
        8,
        'className',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import2.DataTableHeaderCellComponent = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onSort()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import2.DataTableHeaderCellComponent = v.component;
    const currVal_0:any = co.isCheckboxable;
    ck(v,4,0,currVal_0);
    const currVal_1:boolean = !co.column.headerTemplate;
    ck(v,7,0,currVal_1);
    const currVal_2:any = co.column.headerTemplate;
    ck(v,10,0,currVal_2);
  },(ck,v) => {
    var co:import2.DataTableHeaderCellComponent = v.component;
    const currVal_3:any = co.sortClass;
    ck(v,12,0,currVal_3);
  });
}
function View_DataTableHeaderCellComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'datatable-header-cell',[[
        'class',
        'datatable-header-cell'
      ]
    ],[
      [
        4,
        'height',
        'px'
      ]
      ,
      [
        8,
        'className',
        0
      ]
      ,
      [
        1,
        'title',
        0
      ]
      ,
      [
        4,
        'minWidth',
        'px'
      ]
      ,
      [
        4,
        'maxWidth',
        'px'
      ]
      ,
      [
        4,
        'width',
        'px'
      ]

    ]
      ,[[
        (null as any),
        'contextmenu'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('contextmenu' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).onContextmenu($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_DataTableHeaderCellComponent_0,RenderType_DataTableHeaderCellComponent)),
    import0.ɵdid(24576,(null as any),0,import2.DataTableHeaderCellComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).headerHeight;
    const currVal_1:any = import0.ɵnov(v,1).columnCssClasses;
    const currVal_2:any = import0.ɵnov(v,1).name;
    const currVal_3:any = import0.ɵnov(v,1).minWidth;
    const currVal_4:any = import0.ɵnov(v,1).maxWidth;
    const currVal_5:any = import0.ɵnov(v,1).width;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5);
  });
}
export const DataTableHeaderCellComponentNgFactory:import0.ComponentFactory<import2.DataTableHeaderCellComponent> = import0.ɵccf('datatable-header-cell',import2.DataTableHeaderCellComponent,View_DataTableHeaderCellComponent_Host_0,{
  sortType: 'sortType',
  column: 'column',
  sortAscendingIcon: 'sortAscendingIcon',
  sortDescendingIcon: 'sortDescendingIcon',
  allRowsSelected: 'allRowsSelected',
  selectionType: 'selectionType',
  headerHeight: 'headerHeight',
  sorts: 'sorts'
}
,{
  sort: 'sort',
  select: 'select',
  columnContextmenu: 'columnContextmenu'
}
,([] as any[]));
