// angular-2-dropdown-multiselect/dropdown/dropdown.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/forms';
import * as import2 from '@angular/common';
import * as import3 from 'angular-2-dropdown-multiselect/dropdown/dropdown.component';
import * as import4 from 'angular-2-dropdown-multiselect/dropdown/search-filter.pipe';
const styles_MultiselectDropdown:any[] = ['a[_ngcontent-%COMP%] {  outline: none !important;}.dropdown-inline[_ngcontent-%COMP%] {  display: inline-block;}.dropdown-toggle[_ngcontent-%COMP%]   .caret[_ngcontent-%COMP%] {  margin-left: 4px;  white-space: nowrap;  display: inline-block;}'];
export const RenderType_MultiselectDropdown:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 0,
  styles: styles_MultiselectDropdown,
  data: {}
}
);
function View_MultiselectDropdown_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'input-group-btn'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'button',[
      [
        'class',
        'btn btn-default btn-secondary'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.clearSearch($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'i',[[
        'class',
        'fa fa-times'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(null as any),(null as any));
}
function View_MultiselectDropdown_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),13,'li',[[
        'class',
        'dropdown-item search'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),12,'div',[[
        'class',
        'input-group input-group-sm'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[
      [
        'class',
        'input-group-addon'
      ]
      ,
      [
        'id',
        'sizing-addon3'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'i',[[
        'class',
        'fa fa-search'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[' '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'input',[
      [
        'aria-describedby',
        'sizing-addon3'
      ]
      ,
      [
        'autofocus',
        ''
      ]
      ,
      [
        'class',
        'form-control'
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
    ,[
      [
        8,
        'placeholder',
        0
      ]
      ,
      [
        2,
        'ng-untouched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-touched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pristine',
        (null as any)
      ]
      ,
      [
        2,
        'ng-dirty',
        (null as any)
      ]
      ,
      [
        2,
        'ng-valid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-invalid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pending',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'compositionstart'
      ]
      ,
      [
        (null as any),
        'compositionend'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('input' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,6)._handleInput($event.target.value)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,6).onTouched()) !== false);
        ad = (pd_1 && ad);
      }
      if (('compositionstart' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,6)._compositionStart()) !== false);
        ad = (pd_2 && ad);
      }
      if (('compositionend' === en)) {
        const pd_3:any = ((<any>import0.ɵnov(v,6)._compositionEnd($event.target.value)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(8192,(null as any),0,import1.DefaultValueAccessor,[
      import0.Renderer,
      import0.ElementRef,
      [
        2,
        import1.COMPOSITION_BUFFER_MODE
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵprd(512,(null as any),import1.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.DefaultValueAccessor]),
    import0.ɵdid(270336,(null as any),0,import1.FormControlDirective,[
      [
        8,
        (null as any)
      ]
      ,
      [
        8,
        (null as any)
      ]
      ,
      [
        2,
        import1.NG_VALUE_ACCESSOR
      ]

    ]
      ,{form: [
        0,
        'form'
      ]
    },(null as any)),
    import0.ɵprd(1024,(null as any),import1.NgControl,(null as any),[import1.FormControlDirective]),
    import0.ɵdid(8192,(null as any),0,import1.NgControlStatus,[import1.NgControl],(null as any),(null as any)),
    (l()(),import0.ɵted((null as any),[' '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiselectDropdown_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_8:any = co.filterControl;
    ck(v,8,0,currVal_8);
    const currVal_9:any = (co.filterControl.value.length > 0);
    ck(v,13,0,currVal_9);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = import0.ɵinlineInterpolate(1,'',co.texts.searchPlaceholder,'');
    const currVal_1:any = import0.ɵnov(v,10).ngClassUntouched;
    const currVal_2:any = import0.ɵnov(v,10).ngClassTouched;
    const currVal_3:any = import0.ɵnov(v,10).ngClassPristine;
    const currVal_4:any = import0.ɵnov(v,10).ngClassDirty;
    const currVal_5:any = import0.ɵnov(v,10).ngClassValid;
    const currVal_6:any = import0.ɵnov(v,10).ngClassInvalid;
    const currVal_7:any = import0.ɵnov(v,10).ngClassPending;
    ck(v,5,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7);
  });
}
function View_MultiselectDropdown_4(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'li',[[
        'class',
        'dropdown-divider divider'
      ]
  ],(null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),(null as any));
}
function View_MultiselectDropdown_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'li',[[
        'class',
        'dropdown-item check-control check-control-check'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'a',[
      [
        'href',
        'javascript:;'
      ]
      ,
      [
        'role',
        'menuitem'
      ]
      ,
      [
        'tabindex',
        '-1'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.checkAll()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'style',
        'width: 16px'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'glyphicon glyphicon-ok',
      'fa fa-check'
    ]
    ),
    (l()(),import0.ɵted((null as any),[
      ' ',
      ''
    ]
    ))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,4,0,(co.settings.checkedStyle !== 'fontawesome'),(co.settings.checkedStyle === 'fontawesome'));
    ck(v,3,0,currVal_0);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.texts.checkAll;
    ck(v,5,0,currVal_1);
  });
}
function View_MultiselectDropdown_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'li',[[
        'class',
        'dropdown-item check-control check-control-uncheck'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'a',[
      [
        'href',
        'javascript:;'
      ]
      ,
      [
        'role',
        'menuitem'
      ]
      ,
      [
        'tabindex',
        '-1'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.uncheckAll()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'style',
        'width: 16px'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'glyphicon glyphicon-remove',
      'fa fa-times'
    ]
    ),
    (l()(),import0.ɵted((null as any),[
      ' ',
      ''
    ]
    ))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,4,0,(co.settings.checkedStyle !== 'fontawesome'),(co.settings.checkedStyle === 'fontawesome'));
    ck(v,3,0,currVal_0);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.texts.uncheckAll;
    ck(v,5,0,currVal_1);
  });
}
function View_MultiselectDropdown_7(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'li',[[
        'class',
        'dropdown-divider divider'
      ]
  ],(null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),(null as any));
}
function View_MultiselectDropdown_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'li',[[
        'class',
        'dropdown-item empty'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.texts.searchNoRenderText;
    ck(v,1,0,currVal_0);
  });
}
function View_MultiselectDropdown_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'li',[[
        'class',
        'dropdown-item empty'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.texts.searchEmptyResult;
    ck(v,1,0,currVal_0);
  });
}
function View_MultiselectDropdown_12(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'input',[[
        'type',
        'checkbox'
      ]
    ],[
      [
        8,
        'checked',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.preventCheckboxCheck($event,(<any>(<any>v.parent).parent).context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co.getItemStyleSelectionDisabled();
    ck(v,1,0,currVal_2);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.isSelected((<any>(<any>v.parent).parent).context.$implicit);
    const currVal_1:any = co.isCheckboxDisabled();
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_MultiselectDropdown_13(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[
      [
        'class',
        'glyphicon'
      ]
      ,
      [
        'style',
        'width: 16px'
      ]

    ]
      ,[[
        2,
        'glyphicon-ok',
        (null as any)
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.isSelected((<any>(<any>v.parent).parent).context.$implicit);
    ck(v,0,0,currVal_0);
  });
}
function View_MultiselectDropdown_15(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'i',[
      [
        'aria-hidden',
        'true'
      ]
      ,
      [
        'class',
        'fa fa-check'
      ]

    ]
  ,(null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),(null as any));
}
function View_MultiselectDropdown_14(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'style',
        'width: 16px;display: inline-block'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiselectDropdown_15)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.isSelected((<any>(<any>v.parent).parent).context.$implicit);
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_MultiselectDropdown_11(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),14,'a',[
      [
        'href',
        'javascript:;'
      ]
      ,
      [
        'role',
        'menuitem'
      ]
      ,
      [
        'tabindex',
        '-1'
      ]

    ]
      ,[[
        4,
        'padding-left',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵeld(0,(null as any),(null as any),9,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      import0.ɵdid(8192,(null as any),0,import2.NgSwitch,([] as any[]),{ngSwitch: [
        0,
        'ngSwitch'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiselectDropdown_12)),
    import0.ɵdid(139264,(null as any),0,import2.NgSwitchCase,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import2.NgSwitch
    ]
      ,{ngSwitchCase: [
        0,
        'ngSwitchCase'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[' '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiselectDropdown_13)),
    import0.ɵdid(139264,(null as any),0,import2.NgSwitchCase,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import2.NgSwitch
    ]
      ,{ngSwitchCase: [
        0,
        'ngSwitchCase'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[' '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiselectDropdown_14)),
    import0.ɵdid(139264,(null as any),0,import2.NgSwitchCase,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import2.NgSwitch
    ]
      ,{ngSwitchCase: [
        0,
        'ngSwitchCase'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',([] as any[]),[[
        4,
        'font-weight',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.getItemStyleSelectionDisabled();
    ck(v,1,0,currVal_1);
    const currVal_2:any = co.settings.checkedStyle;
    ck(v,3,0,currVal_2);
    const currVal_3:any = 'checkboxes';
    ck(v,5,0,currVal_3);
    const currVal_4:any = 'glyphicon';
    ck(v,8,0,currVal_4);
    const currVal_5:any = 'fontawesome';
    ck(v,11,0,currVal_5);
    const currVal_7:any = co.settings.itemClasses;
    ck(v,13,0,currVal_7);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (((co.parents.length > 0) && (co.parents.indexOf((<any>v.parent).context.$implicit.id) < 0)) && '30px');
    ck(v,0,0,currVal_0);
    const currVal_6:any = ((co.parents.indexOf((<any>v.parent).context.$implicit.id) >= 0)? 'bold': 'normal');
    ck(v,12,0,currVal_6);
    const currVal_8:any = (<any>v.parent).context.$implicit.name;
    ck(v,14,0,currVal_8);
  });
}
function View_MultiselectDropdown_16(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
  ))],(null as any),(ck,v) => {
    const currVal_0:any = (<any>v.parent).context.$implicit.name;
    ck(v,0,0,currVal_0);
  });
}
function View_MultiselectDropdown_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'li',[[
        'class',
        'dropdown-item'
      ]
      ],[[
        2,
        'dropdown-header',
        (null as any)
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.setSelected($event,v.context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiselectDropdown_11)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
    ,{
      ngIf: [
        0,
        'ngIf'
      ]
      ,
      ngIfElse: [
        1,
        'ngIfElse'
      ]

    }
    ,(null as any)),
      (l()(),import0.ɵand(0,[[
        'label',
        2
      ]
    ],(null as any),0,(null as any),View_MultiselectDropdown_16))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = 'dropdown-item';
    const currVal_2:any = v.context.$implicit.classes;
    ck(v,1,0,currVal_1,currVal_2);
    const currVal_3:any = co.getItemStyle(v.context.$implicit);
    ck(v,2,0,currVal_3);
    const currVal_4:boolean = !v.context.$implicit.isLabel;
    const currVal_5:any = import0.ɵnov(v,5);
    ck(v,4,0,currVal_4,currVal_5);
  },(ck,v) => {
    const currVal_0:any = v.context.$implicit.isLabel;
    ck(v,0,0,currVal_0);
  });
}
function View_MultiselectDropdown_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,[[
        'scroller',
        1
      ]
    ],(null as any),16,'ul',[
      [
        'class',
        'dropdown-menu'
      ]
      ,
      [
        'style',
        'display: block; height: auto; overflow-y: auto'
      ]

    ]
    ,[
      [
        2,
        'pull-right',
        (null as any)
      ]
      ,
      [
        2,
        'dropdown-menu-right',
        (null as any)
      ]
      ,
      [
        4,
        'max-height',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'scroll'
      ]
      ,
      [
        (null as any),
        'wheel'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('scroll' === en)) {
        const pd_0:any = ((<any>(co.settings.isLazyLoad? co.checkScrollPosition($event): (null as any))) !== false);
        ad = (pd_0 && ad);
      }
      if (('wheel' === en)) {
        const pd_1:any = ((<any>(co.settings.stopScrollPropagation? co.checkScrollPropagation($event,import0.ɵnov(v,0)): (null as any))) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiselectDropdown_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiselectDropdown_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiselectDropdown_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiselectDropdown_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiselectDropdown_7)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiselectDropdown_8)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiselectDropdown_9)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiselectDropdown_10)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = co.settings.enableSearch;
    ck(v,2,0,currVal_3);
    const currVal_4:any = co.settings.enableSearch;
    ck(v,4,0,currVal_4);
    const currVal_5:any = (co.settings.showCheckAll && !co.disabledSelection);
    ck(v,6,0,currVal_5);
    const currVal_6:any = (co.settings.showUncheckAll && !co.disabledSelection);
    ck(v,8,0,currVal_6);
    const currVal_7:any = (co.settings.showCheckAll || co.settings.showUncheckAll);
    ck(v,10,0,currVal_7);
    const currVal_8:boolean = !co.renderItems;
    ck(v,12,0,currVal_8);
    const currVal_9:any = (co.renderItems && !co.renderFilteredOptions.length);
    ck(v,14,0,currVal_9);
    const currVal_10:any = co.renderFilteredOptions;
    ck(v,16,0,currVal_10);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.settings.pullRight;
    const currVal_1:any = co.settings.pullRight;
    const currVal_2:any = co.settings.maxHeight;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export function View_MultiselectDropdown_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'dropdown'
      ]
      ],[[
        2,
        'open',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'button',[
      [
        'class',
        'dropdown-toggle'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,[[
        8,
        'disabled',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import3.MultiselectDropdown = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggleDropdown()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'caret'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MultiselectDropdown_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:import3.MultiselectDropdown = v.component;
    const currVal_1:any = 'dropdown';
    const currVal_2:any = co.settings.containerClasses;
    ck(v,1,0,currVal_1,currVal_2);
    const currVal_4:any = 'dropdown-toggle';
    const currVal_5:any = co.settings.buttonClasses;
    ck(v,3,0,currVal_4,currVal_5);
    const currVal_7:any = co.isVisible;
    ck(v,7,0,currVal_7);
  },(ck,v) => {
    var co:import3.MultiselectDropdown = v.component;
    const currVal_0:any = co.isVisible;
    ck(v,0,0,currVal_0);
    const currVal_3:any = co.disabled;
    ck(v,2,0,currVal_3);
    const currVal_6:any = co.title;
    ck(v,4,0,currVal_6);
  });
}
function View_MultiselectDropdown_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'ss-multiselect-dropdown',([] as any[]),(null as any),[[
        'document',
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('document:click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,2).onClick($event.target)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_MultiselectDropdown_0,RenderType_MultiselectDropdown)),
    import0.ɵprd(256,(null as any),import4.MultiSelectSearchFilter,import4.MultiSelectSearchFilter,([] as any[])),
    import0.ɵdid(516096,(null as any),0,import3.MultiselectDropdown,[
      import0.ElementRef,
      import1.FormBuilder,
      import4.MultiSelectSearchFilter,
      import0.IterableDiffers
    ]
    ,(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import1.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import3.MultiselectDropdown])
  ]
  ,(ck,v) => {
    ck(v,2,0);
  },(null as any));
}
export const MultiselectDropdownNgFactory:import0.ComponentFactory<import3.MultiselectDropdown> = import0.ɵccf('ss-multiselect-dropdown',import3.MultiselectDropdown,View_MultiselectDropdown_Host_0,{
  options: 'options',
  settings: 'settings',
  texts: 'texts',
  disabled: 'disabled',
  disabledSelection: 'disabledSelection'
}
,{
  selectionLimitReached: 'selectionLimitReached',
  dropdownClosed: 'dropdownClosed',
  dropdownOpened: 'dropdownOpened',
  onAdded: 'onAdded',
  onRemoved: 'onRemoved',
  onLazyLoad: 'onLazyLoad',
  onFilter: 'onFilter'
}
,([] as any[]));
