// primeng/components/inputswitch/inputswitch.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'primeng/components/inputswitch/inputswitch';
import * as import2 from '@angular/common';
import * as import3 from 'primeng/components/dom/domhandler';
import * as import4 from '@angular/forms';
class InputSwitchModuleInjector extends import0.ɵNgModuleInjector<import1.InputSwitchModule> {
  _CommonModule_0:import2.CommonModule;
  _InputSwitchModule_1:import1.InputSwitchModule;
  __NgLocalization_2:import2.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _NgLocalization_2():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_2 == null)) { (this.__NgLocalization_2 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_2;
  }
  createInternal():import1.InputSwitchModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._InputSwitchModule_1 = new import1.InputSwitchModule();
    return this._InputSwitchModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import1.InputSwitchModule)) { return this._InputSwitchModule_1; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_2; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const InputSwitchModuleNgFactory:import0.NgModuleFactory<import1.InputSwitchModule> = new import0.NgModuleFactory<any>(InputSwitchModuleInjector,import1.InputSwitchModule);
const styles_InputSwitch:any[] = ([] as any[]);
export const RenderType_InputSwitch:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_InputSwitch,
  data: {}
}
);
export function View_InputSwitch_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),25,'div',([] as any[]),(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.InputSwitch = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggle($event,import0.ɵnov(v,24))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'ui-inputswitch ui-widget ui-widget-content ui-corner-all',
      'ui-state-disabled',
      'ui-inputswitch-checked',
      'ui-state-focus'
    ]
    ),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'ui-inputswitch-off'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-inputswitch-offlabel'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'ui-inputswitch-on'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'ui-inputswitch-onlabel'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'ui-inputswitch-handle ui-state-default',
      'ui-state-focus'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'ui-helper-hidden-accessible'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,[[
        'in',
        1
      ]
    ],(null as any),0,'input',[
      [
        'aria-live',
        'polite'
      ]
      ,
      [
        'readonly',
        'readonly'
      ]
      ,
      [
        'type',
        'checkbox'
      ]

    ]
    ,[
      [
        1,
        'aria-label',
        0
      ]
      ,
      [
        1,
        'aria-labelledby',
        0
      ]
      ,
      [
        1,
        'id',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.InputSwitch = v.component;
      if (('focus' === en)) {
        const pd_0:any = ((<any>co.onFocus($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>co.onBlur($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.InputSwitch = v.component;
    const currVal_0:any = co.styleClass;
    const currVal_1:any = ck(v,3,0,true,co.disabled,co.checked,co.focused);
    ck(v,2,0,currVal_0,currVal_1);
    const currVal_2:any = co.style;
    ck(v,4,0,currVal_2);
    const currVal_5:any = ck(v,20,0,true,co.focused);
    ck(v,19,0,currVal_5);
  },(ck,v) => {
    var co:import1.InputSwitch = v.component;
    const currVal_3:any = co.offLabel;
    ck(v,9,0,currVal_3);
    const currVal_4:any = co.onLabel;
    ck(v,15,0,currVal_4);
    const currVal_6:any = co.ariaLabel;
    const currVal_7:any = co.ariaLabelledBy;
    const currVal_8:any = co.inputId;
    const currVal_9:any = co.tabindex;
    ck(v,24,0,currVal_6,currVal_7,currVal_8,currVal_9);
  });
}
function View_InputSwitch_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'p-inputSwitch',([] as any[]),(null as any),(null as any),(null as any),View_InputSwitch_0,RenderType_InputSwitch)),
    import0.ɵprd(256,(null as any),import3.DomHandler,import3.DomHandler,([] as any[])),
    import0.ɵdid(6316032,(null as any),0,import1.InputSwitch,[
      import0.ElementRef,
      import3.DomHandler
    ]
    ,(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import4.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.InputSwitch])
  ]
  ,(null as any),(null as any));
}
export const InputSwitchNgFactory:import0.ComponentFactory<import1.InputSwitch> = import0.ɵccf('p-inputSwitch',import1.InputSwitch,View_InputSwitch_Host_0,{
  onLabel: 'onLabel',
  offLabel: 'offLabel',
  disabled: 'disabled',
  style: 'style',
  styleClass: 'styleClass',
  tabindex: 'tabindex',
  inputId: 'inputId',
  ariaLabelTemplate: 'ariaLabelTemplate'
}
,{onChange: 'onChange'},([] as any[]));
